{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# *GMOnotebook*\n",
    "#### **Notebook template for applying routine hyperspectral/segmentation cross-analysis phenomics workflow over new datasets**"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "###### Notebook template v0.2.01 (Oct 14, 2020)\n",
    "###### v0.2.0 onward applies the Python `GMOdetector` package instead of the deprecated R `GMOdetectoR` package, for much faster runtime\n",
    "###### v0.2.01 has calls to `conda` cleaned up with `conda activate` and `conda deactivate` in the same code blocks to avoid any problems with warnings/errors preventing automatic execution of subsequent blocks"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<img src=\"WorkflowFlowchart.png\">"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "In this workflow, images taken with the macroPhor Array dual RGB/hyperspectral imaging platform are analyzed by a workflow in which regression quantifies fluorescent signals in hyperspectral images, deep learning segments RGB images into different tissues, and these datasets are cross-referenced to produce statistics on growth of transgenic callus and shoot."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Instructions for use\n",
    "1.  Enter information for the experiment below\n",
    "2. Set <font color=blue>variables</font> for data paths and parameters, as instructed by colored boxes.\n",
    "3. \"Save as\" with filename describing experiment and anything special about this analysis (e.g. T18_OD_TAO_wk7_automation_test_attempt2.ipynb)\n",
    "4. Run notebook from console, using the below command with the notebook filename inserted<br>\n",
    "```jupyter nbconvert --to HTML --ExecutePreprocessor.timeout=-1 --allow-errors --execute insert_filename_here```\n",
    "5. Wait for email"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Experiment ID and quick description:"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<div class=\"alert alert-block alert-success\">\n",
    "Provide a short description of the experiment in the below box. This should include unique identifier codes for the experiment, along with a short description of genotypes and treatments studied. The timepoint should also be included. </div>"
   ]
  },
  {
   "cell_type": "raw",
   "metadata": {},
   "source": [
    "GFPA: Cottonwoods transformed with experimental plasmids harboring GFP plasmids that may be used for GWAS of transformation<br>\n",
    "Week 3 timepoint – Conclusion of callus induction treatment"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Parameters for analysis:"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<div class=\"alert alert-block alert-success\">\n",
    "The below variables must be modified appropriately every time this workflow is run over new images.\n",
    "</div>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### Data location\n",
    "The `data` variable below provides the **complete** path to the folder containing data to be analyzed. This should include all folders and subfolders in which the data of interest is organized by. For the organizational system used for our lab's data, this should follow the format \"/Experiment/Subexperiment/Timepoint/\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "data=\"/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### Sample information\n",
    "Every experiment has a randomization datasheet, which was used to organize treatment and genotype information for each plate, prepare labels, and randomize plates. The path to this file is provided through the `randomization_datasheets` variable below. This workflow requires this datasheet in order to know which plates have which genotype/treatment. At a later date, we will integrate an ability to read this data directly from labels."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "randomization_datasheet=\"/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/GFPA_randomized list_MNrevised.xlsx\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### Exclusion of missing/contaminated explants from analysis"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Set the `missing_explants` variable to `\"Automatic\"` if using model to automatically detect missing and contaminated explants. Note that this model is only supported for plates with 12 explants. Otherwise, provide an appropriately formatted `.csv` spreadsheet (see example)  of manually scored contamination / missing explant data."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "missing_explants=\"Automatic\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Enter your email where results will be sent"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "email=michael.nagle@oregonstate.edu"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<div class=\"alert alert-block alert-warning\">\n",
    "All variables below should be modified only as needed to indicate the fluorescent proteins in samples and the grid layout of explants. </div>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### Provide information pertinent to the fluorophores in the sample"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below should be all known fluorescent components contained in the sample. This includes each fluorescent protein, as well as a \"noise\" or \"diffraction\" term if applicable. All of these components must exist in the user's spectral library. `GMOdetector` currently comes with a spectral library that includes, by default:\n",
    "- DsRed\n",
    "- ZsYellow\n",
    "- GFP\n",
    "- Chl\n",
    "- ChlA\n",
    "- ChlB\n",
    "- Diffraction"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "fluorophores=(GFP Chl Diffraction) # An explanation of array variables in bash is here: https://tldp.org/LDP/Bash-Beginners-Guide/html/sect_10_02.html"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "spectral_library_path=\"/scratch2/NSF_GWAS/notebooks/gmodetector_py/spectral_library/\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The user has the option of limiting loading of hyperspectral data and subsequent regression to a specific range of wavelengths, using the `desired_wavelength_range` array variable. This range should cover all fluorophores provided in the `fluorophores` array variable able. Aside from runtime, there is no disadvantage to including a wider range than is needed."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "desired_wavelength_range=(500 900)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### To assist user inspection of regression results, false color plots will be produced by `GMOdetector` to show results of regression over whole samples. \n",
    "###### Note: In v0.2.0 of the workflow, these parameters are independent of those used later by `GMOlabeler` to produce by-explant plots including false color plots. These will be made the same in a later update.\n",
    "The `FalseColor_channels` array variable indicates the components to be plotted as red, green and blue, in that order."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "FalseColor_channels=(Chl GFP Diffraction)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The `FalseColor_caps` array variable indicates an upper limit of signal for each of these component. Any signal at or above these values will appear with maximum brightness; thus, these variables are comparable to exposure on a camera. If caps are too high, not much signal at lower ranges will be seen. If cap for a given component is too low, the false color images will appear overexposed with respect to the component."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "FalseColor_caps=(200 400 50)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### `GMOlabeler` will use the below parameters to classify individual explants on plates as transgenic or not."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "reporter=GFP\n",
    "#reporter=DsRed"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Parameters for reporter signal threshold and pixel threshold must be provided by the user. Our grid search yielded several noted below. These were most recently calculated from statistics produced with Python `GMOdetector`."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<img src=\"GMOlabeler_parameters.png\">"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "reporter_threshold=38\n",
    "pixel_threshold=3"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### Grid information"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Currently, `GMOlabeler` distinguishes explants by their expected position on a plate. This is supported for plates with 12 or 20 explants, arranged in specific positions on a grid. False positives may result in cases in which one explant grows to such an extent that it intrudes into the grid square meant for an adjacent explant. <br>In the future, an additional neural network will be used to segment individual explants on a per-pixel basis, avoiding the need for crude cropping, and the outputs from the resulting model will be integrated with `GMOlabeler`."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Cropping to grid squares is supported for standard grids with 12 or 20 explants, as indicated by the `grid` variable. An image of each grid square (with a file path provided by the `grid_file` variable) will be superimposed over each image during `GMOlabeler` analysis to allow the user to easily inspect and verify grid positions."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "#grid=20\n",
    "grid=12\n",
    "\n",
    "#grid_file=\"/scratch2/NSF_GWAS/macroPhor_Array/grids/grid20_post_processed.png\"\n",
    "grid_file=\"/scratch2/NSF_GWAS/macroPhor_Array/grids/grids_left_facing_125208_1_0_1_rgb_processed.jpg\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<div class=\"alert alert-block alert-danger\">\n",
    "The below variables do not need to be modified during any routine use of the workflow.\n",
    "</div>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "For the following variables, use 0 (False) or 1 (True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "composite=1"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Set dimensions for plot outputs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "#width=15\n",
    "width=9\n",
    "height=5"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<div class=\"alert alert-block alert-info\">\n",
    "With all above variables set, please \"Save as...\" with a filename referencing this specific dataset. <br>Finally, execute the workflow by running the below command in a console, replacing insert_filename_here with the filename for the saved notebook.\n",
    "</div>"
   ]
  },
  {
   "cell_type": "raw",
   "metadata": {},
   "source": [
    "jupyter nbconvert --to HTML --ExecutePreprocessor.timeout=-1 --allow-errors --execute EA_v0.1.50_labdemo.ipynb"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Automated workflow to be deployed"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "See the below code for a walkthrough of how GMOnotebook works, or view the outputs after running the workflow for help troubleshooting errors in specific steps of analysis."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<div class=\"alert alert-block alert-danger\"> <b>Danger:</b> Do not modify any below code without creating a new version of the template notebook. During routine usage, this workflow should be customized only by modifying variables above, while leaving the below code unmodified. </div>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### Time analysis begins:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Tue Oct 27 20:08:52 PDT 2020\n"
     ]
    }
   ],
   "source": [
    "echo $(date)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "These internal variables are set automatically."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "datestamp=$(date +”%Y-%m-%d”)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    }
   ],
   "source": [
    "echo $data_folder"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "data_folder=$(echo $data | cut -d/ -f5-)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "timepoint=\"$(basename -- $data_folder)\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "#data=\"/scratch2/NSF_GWAS/macroPhor_Array${data_folder}\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "gmodetector_wd=\"/scratch2/NSF_GWAS/notebooks/gmodetector_py/\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "output_directory_prefix=\"/scratch2/NSF_GWAS/gmodetector_out/\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "output_directory_full=\"$output_directory_prefix$data_folder\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "dataset_name=$(echo $subdirectory | sed -e 's/\\//-/g')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 1. Quantification of fluorescent proteins by regression"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The Python package `GMOdetector` is used to quantify fluorescent proteins in each pixel of hyperspectral images via linear regression. Hyperspectral images are regressed over spectra of known components, and pixelwise maps of test-statistics are constructed for each component in the sample. This approach to quantifying components of hyperspectral images is described in-depth in the Methods section from <a href=\"https://link.springer.com/article/10.1007/s40789-019-0252-7\" target=\"_blank\">Böhme, et al. 2019</a>. Code and documentation for `GMOdetector` is on <a href=\"https://github.com/naglemi/GMOdetector_py\" target=\"_blank\">Github</a>."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "cd $gmodetector_wd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "job_list_name=\"$dataset_name.jobs\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "rm -rf $job_list_name"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "for file in $data/*.hdr\n",
    "do\n",
    " if [[ \"$file\" != *'hroma'* ]]; then\n",
    "  echo \"python wrappers/analyze_sample.py \\\n",
    "--file_path $file \\\n",
    "--fluorophores ${fluorophores[*]} \\\n",
    "--min_desired_wavelength ${desired_wavelength_range[0]} \\\n",
    "--max_desired_wavelength ${desired_wavelength_range[1]} \\\n",
    "--red_channel ${FalseColor_channels[0]} \\\n",
    "--green_channel ${FalseColor_channels[1]} \\\n",
    "--blue_channel ${FalseColor_channels[2]} \\\n",
    "--red_cap ${FalseColor_caps[0]} \\\n",
    "--green_cap ${FalseColor_caps[1]} \\\n",
    "--blue_cap ${FalseColor_caps[2]} \\\n",
    "--spectral_library_path /scratch2/NSF_GWAS/notebooks/gmodetector_py/spectral_library/ \\\n",
    "--output_dir $output_directory_full\" >> $job_list_name\n",
    " fi\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(test-environment) Academic tradition requires you to cite works you base your article on.\n",
      "If you use programs that use GNU Parallel to process data for an article in a\n",
      "scientific publication, please cite:\n",
      "\n",
      "  Tange, O. (2020, September 22). GNU Parallel 20200922 ('Ginsburg').\n",
      "  Zenodo. https://doi.org/10.5281/zenodo.4045386\n",
      "\n",
      "This helps funding further development; AND IT WON'T COST YOU A CENT.\n",
      "If you pay 10000 EUR you should feel free to use GNU Parallel without citing.\n",
      "\n",
      "More about funding GNU Parallel and the citation notice:\n",
      "https://www.gnu.org/software/parallel/parallel_design.html#Citation-notice\n",
      "\n",
      "To silence this citation notice: run 'parallel --citation' once.\n",
      "\n",
      "Come on: You have run parallel 25 times. Isn't it about time \n",
      "you run 'parallel --citation' once to silence the citation notice?\n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA1_I5.0_F1.9_L100_162853_13_2_0_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA1_I5.0_F1.9_L100_162853_13_2_0.hdr in 46.11705369595438s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA1_I5.0_F1.9_L100_161651_0_0_0_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA1_I5.0_F1.9_L100_161651_0_0_0.hdr in 48.32055210787803s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA3_I5.0_F1.9_L100_162457_4_0_4_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA3_I5.0_F1.9_L100_162457_4_0_4.hdr in 48.58410762436688s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA2_I5.0_F1.9_L100_164955_8_1_2_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA2_I5.0_F1.9_L100_164955_8_1_2.hdr in 49.261040409095585s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA1_I5.0_F1.9_L100_161923_3_0_3_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA1_I5.0_F1.9_L100_161923_3_0_3.hdr in 49.425096871796995s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA2_I5.0_F1.9_L100_164528_3_0_6_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA2_I5.0_F1.9_L100_164528_3_0_6.hdr in 50.12657819176093s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA1_I5.0_F1.9_L100_162212_6_0_6_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA1_I5.0_F1.9_L100_162212_6_0_6.hdr in 51.431097326334566s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA1_I5.0_F1.9_L100_162450_9_1_4_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA1_I5.0_F1.9_L100_162450_9_1_4.hdr in 51.39889712911099s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA3_I5.0_F1.9_L100_162648_6_0_6_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA3_I5.0_F1.9_L100_162648_6_0_6.hdr in 51.37907062890008s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA2_I5.0_F1.9_L100_164338_1_0_4_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA2_I5.0_F1.9_L100_164338_1_0_4.hdr in 54.16653602384031s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA2_I5.0_F1.9_L100_164901_7_1_3_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA2_I5.0_F1.9_L100_164901_7_1_3.hdr in 54.254501913208514s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA1_I5.0_F1.9_L100_161732_1_0_1_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA1_I5.0_F1.9_L100_161732_1_0_1.hdr in 55.08281432278454s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA3_I5.0_F1.9_L100_162131_0_0_0_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA3_I5.0_F1.9_L100_162131_0_0_0.hdr in 56.7550732428208s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA1_I5.0_F1.9_L100_162757_12_1_0_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA1_I5.0_F1.9_L100_162757_12_1_0.hdr in 57.664813708048314s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA3_I5.0_F1.9_L100_162553_5_0_5_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA3_I5.0_F1.9_L100_162553_5_0_5.hdr in 58.405687517952174s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA2_I5.0_F1.9_L100_164708_5_1_5_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA2_I5.0_F1.9_L100_164708_5_1_5.hdr in 58.971767224837095s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA2_I5.0_F1.9_L100_164807_6_1_4_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA2_I5.0_F1.9_L100_164807_6_1_4.hdr in 60.1358782579191s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA1_I5.0_F1.9_L100_162939_14_2_1_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA1_I5.0_F1.9_L100_162939_14_2_1.hdr in 60.671850277110934s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA2_I5.0_F1.9_L100_165144_10_1_0_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA2_I5.0_F1.9_L100_165144_10_1_0.hdr in 61.135499643627554s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA2_I5.0_F1.9_L100_165610_15_2_4_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA2_I5.0_F1.9_L100_165610_15_2_4.hdr in 61.589447875041515s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA1_I5.0_F1.9_L100_162354_8_1_5_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA1_I5.0_F1.9_L100_162354_8_1_5.hdr in 61.901298542972654s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA2_I5.0_F1.9_L100_165239_11_2_0_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA2_I5.0_F1.9_L100_165239_11_2_0.hdr in 61.6665623690933s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA2_I5.0_F1.9_L100_165705_16_2_5_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA2_I5.0_F1.9_L100_165705_16_2_5.hdr in 61.85304874414578s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA1_I5.0_F1.9_L100_163039_15_2_2_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA1_I5.0_F1.9_L100_163039_15_2_2.hdr in 62.384412460029125s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA1_I5.0_F1.9_L100_162649_11_1_2_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA1_I5.0_F1.9_L100_162649_11_1_2.hdr in 63.01543249608949s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA2_I5.0_F1.9_L100_164623_4_1_6_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA2_I5.0_F1.9_L100_164623_4_1_6.hdr in 63.12372067570686s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA3_I5.0_F1.9_L100_162828_8_1_5_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA3_I5.0_F1.9_L100_162828_8_1_5.hdr in 63.47557700285688s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA2_I5.0_F1.9_L100_164229_0_0_3_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA2_I5.0_F1.9_L100_164229_0_0_3.hdr in 63.674491078127176s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA1_I5.0_F1.9_L100_162117_5_0_5_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA1_I5.0_F1.9_L100_162117_5_0_5.hdr in 63.70739058405161s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA1_I5.0_F1.9_L100_163133_16_2_3_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA1_I5.0_F1.9_L100_163133_16_2_3.hdr in 63.79497406305745s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA1_I5.0_F1.9_L100_162020_4_0_4_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA1_I5.0_F1.9_L100_162020_4_0_4.hdr in 63.99224318610504s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA1_I5.0_F1.9_L100_162555_10_1_3_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA1_I5.0_F1.9_L100_162555_10_1_3.hdr in 64.10446687787771s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA1_I5.0_F1.9_L100_163432_19_2_6_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA1_I5.0_F1.9_L100_163432_19_2_6.hdr in 64.138363244012s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA1_I5.0_F1.9_L100_163228_17_2_4_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA1_I5.0_F1.9_L100_163228_17_2_4.hdr in 64.122289002873s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA1_I5.0_F1.9_L100_163322_18_2_5_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA1_I5.0_F1.9_L100_163322_18_2_5.hdr in 64.1284919269383s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA3_I5.0_F1.9_L100_162308_2_0_2_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA3_I5.0_F1.9_L100_162308_2_0_2.hdr in 64.17821446899325s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA2_I5.0_F1.9_L100_165325_12_2_1_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA2_I5.0_F1.9_L100_165325_12_2_1.hdr in 63.96995387179777s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA2_I5.0_F1.9_L100_165513_14_2_3_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA2_I5.0_F1.9_L100_165513_14_2_3.hdr in 64.13324237521738s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA3_I5.0_F1.9_L100_162743_7_1_6_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA3_I5.0_F1.9_L100_162743_7_1_6.hdr in 64.32014117576182s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA2_I5.0_F1.9_L100_165800_17_2_6_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA2_I5.0_F1.9_L100_165800_17_2_6.hdr in 64.46508658397943s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA3_I5.0_F1.9_L100_162212_1_0_1_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA3_I5.0_F1.9_L100_162212_1_0_1.hdr in 64.0148523086682s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA3_I5.0_F1.9_L100_162924_9_1_4_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA3_I5.0_F1.9_L100_162924_9_1_4.hdr in 64.48438400914893s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA2_I5.0_F1.9_L100_164433_2_0_5_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA2_I5.0_F1.9_L100_164433_2_0_5.hdr in 64.38594599580392s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA1_I5.0_F1.9_L100_162308_7_1_6_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA1_I5.0_F1.9_L100_162308_7_1_6.hdr in 64.2914644270204s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA2_I5.0_F1.9_L100_165049_9_1_1_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA2_I5.0_F1.9_L100_165049_9_1_1.hdr in 64.38487957790494s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA3_I5.0_F1.9_L100_162402_3_0_3_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA3_I5.0_F1.9_L100_162402_3_0_3.hdr in 64.49943609675393s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA1_I5.0_F1.9_L100_161827_2_0_2_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA1_I5.0_F1.9_L100_161827_2_0_2.hdr in 64.44931759592146s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA2_I5.0_F1.9_L100_165418_13_2_2_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA2_I5.0_F1.9_L100_165418_13_2_2.hdr in 64.44759909436107s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA5_I5.0_F1.9_L100_174625_8_1_3_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA5_I5.0_F1.9_L100_174625_8_1_3.hdr in 29.831400705035776s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA5_I5.0_F1.9_L100_174241_4_0_4_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA5_I5.0_F1.9_L100_174241_4_0_4.hdr in 35.85298613412306s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA5_I5.0_F1.9_L100_174528_7_1_4_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA5_I5.0_F1.9_L100_174528_7_1_4.hdr in 40.10856280801818s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA5_I5.0_F1.9_L100_174910_11_1_0_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA5_I5.0_F1.9_L100_174910_11_1_0.hdr in 42.05964320199564s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA5_I5.0_F1.9_L100_174815_10_1_1_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA5_I5.0_F1.9_L100_174815_10_1_1.hdr in 43.68281300412491s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA5_I5.0_F1.9_L100_174432_6_0_6_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA5_I5.0_F1.9_L100_174432_6_0_6.hdr in 47.61109986901283s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA4_I5.0_F1.9_L100_171419_7_1_5_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA4_I5.0_F1.9_L100_171419_7_1_5.hdr in 85.05234843492508s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA4_I5.0_F1.9_L100_175704_3_2_4_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA4_I5.0_F1.9_L100_175704_3_2_4.hdr in 85.23396565904841s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA4_I5.0_F1.9_L100_171330_6_1_6_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA4_I5.0_F1.9_L100_171330_6_1_6.hdr in 90.47830157028511s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA3_I5.0_F1.9_L100_163544_16_2_2_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA3_I5.0_F1.9_L100_163544_16_2_2.hdr in 90.3161319210194s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA4_I5.0_F1.9_L100_175759_4_2_5_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA4_I5.0_F1.9_L100_175759_4_2_5.hdr in 92.14921843027696s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA4_I5.0_F1.9_L100_171759_11_1_1_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA4_I5.0_F1.9_L100_171759_11_1_1.hdr in 95.7137783463113s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA4_I5.0_F1.9_L100_175854_5_2_6_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA4_I5.0_F1.9_L100_175854_5_2_6.hdr in 97.70162876415998s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA4_I5.0_F1.9_L100_171855_12_1_0_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA4_I5.0_F1.9_L100_171855_12_1_0.hdr in 101.69941684696823s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA3_I5.0_F1.9_L100_163306_13_1_0_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA3_I5.0_F1.9_L100_163306_13_1_0.hdr in 103.31079571321607s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA4_I5.0_F1.9_L100_171610_9_1_3_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA4_I5.0_F1.9_L100_171610_9_1_3.hdr in 103.14430606784299s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA4_I5.0_F1.9_L100_171140_4_0_4_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA4_I5.0_F1.9_L100_171140_4_0_4.hdr in 104.05121723469347s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA3_I5.0_F1.9_L100_163211_12_1_1_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA3_I5.0_F1.9_L100_163211_12_1_1.hdr in 107.08299922803417s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA4_I5.0_F1.9_L100_175417_0_2_1_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA4_I5.0_F1.9_L100_175417_0_2_1.hdr in 106.79691107105464s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA4_I5.0_F1.9_L100_171705_10_1_2_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA4_I5.0_F1.9_L100_171705_10_1_2.hdr in 111.04444393608719s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA4_I5.0_F1.9_L100_171234_5_0_5_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA4_I5.0_F1.9_L100_171234_5_0_5.hdr in 111.27795373415574s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA4_I5.0_F1.9_L100_170813_0_0_0_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA4_I5.0_F1.9_L100_170813_0_0_0.hdr in 111.64734542788938s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA3_I5.0_F1.9_L100_163739_18_2_5_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA3_I5.0_F1.9_L100_163739_18_2_5.hdr in 113.24519831361249s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA4_I5.0_F1.9_L100_171045_3_0_3_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA4_I5.0_F1.9_L100_171045_3_0_3.hdr in 113.20882059819996s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA4_I5.0_F1.9_L100_175610_2_2_3_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA4_I5.0_F1.9_L100_175610_2_2_3.hdr in 113.06420804094523s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA5_I5.0_F1.9_L100_174146_3_0_3_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA5_I5.0_F1.9_L100_174146_3_0_3.hdr in 117.25941875297576s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA4_I5.0_F1.9_L100_175515_1_2_2_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA4_I5.0_F1.9_L100_175515_1_2_2.hdr in 118.84628379624337s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA3_I5.0_F1.9_L100_163401_14_2_0_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA3_I5.0_F1.9_L100_163401_14_2_0.hdr in 119.63428592588753s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA4_I5.0_F1.9_L100_170854_1_0_1_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA4_I5.0_F1.9_L100_170854_1_0_1.hdr in 119.8866879940033s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA3_I5.0_F1.9_L100_163639_17_2_4_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA3_I5.0_F1.9_L100_163639_17_2_4.hdr in 120.66191446315497s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA3_I5.0_F1.9_L100_163834_19_2_6_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA3_I5.0_F1.9_L100_163834_19_2_6.hdr in 132.0557501539588s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA3_I5.0_F1.9_L100_163450_15_2_1_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA3_I5.0_F1.9_L100_163450_15_2_1.hdr in 133.73523959796876s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA5_I5.0_F1.9_L100_173952_1_0_1_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA5_I5.0_F1.9_L100_173952_1_0_1.hdr in 136.35405896790326s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA3_I5.0_F1.9_L100_163114_11_1_2_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA3_I5.0_F1.9_L100_163114_11_1_2.hdr in 142.56033006682992s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA5_I5.0_F1.9_L100_174335_5_0_5_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA5_I5.0_F1.9_L100_174335_5_0_5.hdr in 141.34087013779208s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA4_I5.0_F1.9_L100_171951_13_2_0_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA4_I5.0_F1.9_L100_171951_13_2_0.hdr in 143.53028885321692s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA5_I5.0_F1.9_L100_173911_0_0_0_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA5_I5.0_F1.9_L100_173911_0_0_0.hdr in 148.4033411378041s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA5_I5.0_F1.9_L100_174720_9_1_2_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA5_I5.0_F1.9_L100_174720_9_1_2.hdr in 148.00722620403394s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA5_I5.0_F1.9_L100_174051_2_0_2_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA5_I5.0_F1.9_L100_174051_2_0_2.hdr in 148.851427747868s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA4_I5.0_F1.9_L100_171515_8_1_4_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA4_I5.0_F1.9_L100_171515_8_1_4.hdr in 150.96092656021938s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA3_I5.0_F1.9_L100_163019_10_1_3_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA3_I5.0_F1.9_L100_163019_10_1_3.hdr in 151.11710244603455s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "Running GMOdetector version 0.0.855\n",
      "load mode isload_full_then_crop\n",
      "Saving to /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/GFPA4_I5.0_F1.9_L100_170950_2_0_2_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//GFPA4_I5.0_F1.9_L100_170950_2_0_2.hdr in 151.70391030609608s\n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.3 \n",
      "2: package ‘datasets’ was built under R version 3.6.3 \n",
      "3: package ‘utils’ was built under R version 3.6.3 \n",
      "4: package ‘grDevices’ was built under R version 3.6.3 \n",
      "5: package ‘graphics’ was built under R version 3.6.3 \n",
      "6: package ‘stats’ was built under R version 3.6.3 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/test-environment/lib/python3.7/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.3 \n",
      "\n",
      "(test-environment) (base) "
     ]
    }
   ],
   "source": [
    "conda activate test-environment\n",
    "parallel -a $job_list_name\n",
    "conda deactivate\n",
    "conda deactivate"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### Time regression completes:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Tue Oct 27 20:12:58 PDT 2020\n"
     ]
    }
   ],
   "source": [
    "echo $(date)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 2. Neural networks to segment tissues, classify missing/contaminated explants"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 2.1. Semantic segmentation of tissues"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Images are segmented into specific plant tissues by a deep neural network of the state-of-the-art Deeplab v3 architecture <a href=\"https://arxiv.org/abs/1706.05587\" target=\"_blank\">Liang-Chieh et al., 2017</a>. The model has been trained using training sets generated with our annotation GUI Intelligent DEep Annotator for Segmentation (IDEAS, available on <a href=\"https://bitbucket.org/JialinYuan/image-annotator/src/master/\" target=\"_blank\">Bitbucket</a>, publication pending). Our branch of the Deeplab v3 repo, including a Jupyter walkthrough for training, can be found on Github."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Training is completed upstream of this notebook, which only entails analysis of test data using the latest model."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<img src=\"Figures/downsized/segmentation_composite2.png\">"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Figure: This example image was taken from an experiment on the effects of different CIMs on cottonwood regeneration. This composite image illustrates that for every sample, tissues are segmented into stem (red), callus (blue) and shoot (green). These composite images, useful for manual inspection of results, are produced when the 'composite' option is on."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### 2.1.1. Pre-processing\n",
    "##### 2.1.1.1. Crop to remove labels, and resize images to 900x600"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This script resizes images to 900x900 and then crops away top and bottom 150 pixels for a final image size of 900x600."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "cd /scratch2/NSF_GWAS/GMOdetectoR/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(base) (base) "
     ]
    }
   ],
   "source": [
    "conda activate base\n",
    "python crop.py $data\n",
    "conda deactivate"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/\n"
     ]
    }
   ],
   "source": [
    "echo $data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### 2.1.1.2. prepare test.csv file for inference"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Make a list of all our image files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "cd $data\n",
    "ls -d $PWD/* $data | grep -i \"rgb_cropped\" > test.csv"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Remove the chroma standard from list of RGB image data to be segmented"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [],
   "source": [
    "sed -i '/hroma/d' \"${data}/test.csv\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### 2.1.2. Inference"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The trained model is deployed to perform semantic segmentation of experimental images. A list of RGB images to be segmented by the trained model is passed through the --image-list option. For each of these images, we will obtain an output mask (.png) of labeled tissues"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Dependencies include `opencv`, `scipy`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(deeplab) /scratch2/NSF_GWAS/anaconda3/envs/deeplab/lib/python3.6/site-packages/tensorflow/python/framework/dtypes.py:516: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.\n",
      "  _np_qint8 = np.dtype([(\"qint8\", np.int8, 1)])\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/deeplab/lib/python3.6/site-packages/tensorflow/python/framework/dtypes.py:517: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.\n",
      "  _np_quint8 = np.dtype([(\"quint8\", np.uint8, 1)])\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/deeplab/lib/python3.6/site-packages/tensorflow/python/framework/dtypes.py:518: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.\n",
      "  _np_qint16 = np.dtype([(\"qint16\", np.int16, 1)])\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/deeplab/lib/python3.6/site-packages/tensorflow/python/framework/dtypes.py:519: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.\n",
      "  _np_quint16 = np.dtype([(\"quint16\", np.uint16, 1)])\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/deeplab/lib/python3.6/site-packages/tensorflow/python/framework/dtypes.py:520: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.\n",
      "  _np_qint32 = np.dtype([(\"qint32\", np.int32, 1)])\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/deeplab/lib/python3.6/site-packages/tensorflow/python/framework/dtypes.py:525: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.\n",
      "  np_resource = np.dtype([(\"resource\", np.ubyte, 1)])\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/deeplab/lib/python3.6/site-packages/tensorboard/compat/tensorflow_stub/dtypes.py:541: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.\n",
      "  _np_qint8 = np.dtype([(\"qint8\", np.int8, 1)])\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/deeplab/lib/python3.6/site-packages/tensorboard/compat/tensorflow_stub/dtypes.py:542: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.\n",
      "  _np_quint8 = np.dtype([(\"quint8\", np.uint8, 1)])\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/deeplab/lib/python3.6/site-packages/tensorboard/compat/tensorflow_stub/dtypes.py:543: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.\n",
      "  _np_qint16 = np.dtype([(\"qint16\", np.int16, 1)])\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/deeplab/lib/python3.6/site-packages/tensorboard/compat/tensorflow_stub/dtypes.py:544: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.\n",
      "  _np_quint16 = np.dtype([(\"quint16\", np.uint16, 1)])\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/deeplab/lib/python3.6/site-packages/tensorboard/compat/tensorflow_stub/dtypes.py:545: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.\n",
      "  _np_qint32 = np.dtype([(\"qint32\", np.int32, 1)])\n",
      "/scratch2/NSF_GWAS/anaconda3/envs/deeplab/lib/python3.6/site-packages/tensorboard/compat/tensorflow_stub/dtypes.py:550: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.\n",
      "  np_resource = np.dtype([(\"resource\", np.ubyte, 1)])\n",
      "WARNING:tensorflow:From /scratch2/NSF_GWAS/deeplab/inference.py:34: The name tf.gfile.GFile is deprecated. Please use tf.io.gfile.GFile instead.\n",
      "\n",
      "WARNING:tensorflow:From /scratch2/NSF_GWAS/deeplab/inference.py:35: The name tf.GraphDef is deprecated. Please use tf.compat.v1.GraphDef instead.\n",
      "\n",
      "WARNING:tensorflow:From /scratch2/NSF_GWAS/deeplab/inference.py:41: The name tf.Session is deprecated. Please use tf.compat.v1.Session instead.\n",
      "\n",
      "2020-10-27 20:14:00.882293: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcuda.so.1\n",
      "2020-10-27 20:14:06.293944: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1640] Found device 0 with properties: \n",
      "name: Tesla K80 major: 3 minor: 7 memoryClockRate(GHz): 0.8235\n",
      "pciBusID: 0000:05:00.0\n",
      "2020-10-27 20:14:06.295426: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1640] Found device 1 with properties: \n",
      "name: Tesla K80 major: 3 minor: 7 memoryClockRate(GHz): 0.8235\n",
      "pciBusID: 0000:06:00.0\n",
      "2020-10-27 20:14:06.296773: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1640] Found device 2 with properties: \n",
      "name: Tesla K80 major: 3 minor: 7 memoryClockRate(GHz): 0.8235\n",
      "pciBusID: 0000:84:00.0\n",
      "2020-10-27 20:14:06.298242: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1640] Found device 3 with properties: \n",
      "name: Tesla K80 major: 3 minor: 7 memoryClockRate(GHz): 0.8235\n",
      "pciBusID: 0000:85:00.0\n",
      "2020-10-27 20:14:06.298714: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Could not dlopen library 'libcudart.so.10.0'; dlerror: libcudart.so.10.0: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /usr/local/apps/cuda/cuda-8.0.61/lib64:/usr/local/apps/cuda/cuda-9.0/lib64/\n",
      "2020-10-27 20:14:06.298910: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Could not dlopen library 'libcublas.so.10.0'; dlerror: libcublas.so.10.0: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /usr/local/apps/cuda/cuda-8.0.61/lib64:/usr/local/apps/cuda/cuda-9.0/lib64/\n",
      "2020-10-27 20:14:06.299087: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Could not dlopen library 'libcufft.so.10.0'; dlerror: libcufft.so.10.0: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /usr/local/apps/cuda/cuda-8.0.61/lib64:/usr/local/apps/cuda/cuda-9.0/lib64/\n",
      "2020-10-27 20:14:06.299260: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Could not dlopen library 'libcurand.so.10.0'; dlerror: libcurand.so.10.0: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /usr/local/apps/cuda/cuda-8.0.61/lib64:/usr/local/apps/cuda/cuda-9.0/lib64/\n",
      "2020-10-27 20:14:06.299432: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Could not dlopen library 'libcusolver.so.10.0'; dlerror: libcusolver.so.10.0: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /usr/local/apps/cuda/cuda-8.0.61/lib64:/usr/local/apps/cuda/cuda-9.0/lib64/\n",
      "2020-10-27 20:14:06.299601: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Could not dlopen library 'libcusparse.so.10.0'; dlerror: libcusparse.so.10.0: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /usr/local/apps/cuda/cuda-8.0.61/lib64:/usr/local/apps/cuda/cuda-9.0/lib64/\n",
      "2020-10-27 20:14:06.310332: I tensorflow/stream_executor/platform/default/dso_loader.cc:42] Successfully opened dynamic library libcudnn.so.7\n",
      "2020-10-27 20:14:06.310400: W tensorflow/core/common_runtime/gpu/gpu_device.cc:1663] Cannot dlopen some GPU libraries. Skipping registering GPU devices...\n",
      "2020-10-27 20:14:06.311174: I tensorflow/core/platform/cpu_feature_guard.cc:142] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA\n",
      "2020-10-27 20:14:06.669033: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x557fc1d30280 executing computations on platform CUDA. Devices:\n",
      "2020-10-27 20:14:06.669105: I tensorflow/compiler/xla/service/service.cc:175]   StreamExecutor device (0): Tesla K80, Compute Capability 3.7\n",
      "2020-10-27 20:14:06.669123: I tensorflow/compiler/xla/service/service.cc:175]   StreamExecutor device (1): Tesla K80, Compute Capability 3.7\n",
      "2020-10-27 20:14:06.669136: I tensorflow/compiler/xla/service/service.cc:175]   StreamExecutor device (2): Tesla K80, Compute Capability 3.7\n",
      "2020-10-27 20:14:06.669149: I tensorflow/compiler/xla/service/service.cc:175]   StreamExecutor device (3): Tesla K80, Compute Capability 3.7\n",
      "2020-10-27 20:14:06.675933: I tensorflow/core/platform/profile_utils/cpu_utils.cc:94] CPU Frequency: 1800110000 Hz\n",
      "2020-10-27 20:14:06.679955: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x557fc4168220 executing computations on platform Host. Devices:\n",
      "2020-10-27 20:14:06.680013: I tensorflow/compiler/xla/service/service.cc:175]   StreamExecutor device (0): <undefined>, <undefined>\n",
      "2020-10-27 20:14:06.680162: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1181] Device interconnect StreamExecutor with strength 1 edge matrix:\n",
      "2020-10-27 20:14:06.680194: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1187]      \n",
      "Running inference for  90  images.\n",
      "Running deeplab on image: 1\n",
      "2020-10-27 20:14:08.799466: W tensorflow/compiler/jit/mark_for_compilation_pass.cc:1412] (One-time warning): Not using XLA:CPU for cluster because envvar TF_XLA_FLAGS=--tf_xla_cpu_global_jit was not set.  If you want XLA:CPU, either set that envvar, or use experimental_jit_scope to enable XLA:CPU.  To confirm that XLA is active, pass --vmodule=xla_compilation_cache=1 (as a proper command-line flag, not via TF_XLA_FLAGS) or set the envvar XLA_FLAGS=--xla_hlo_profile.\n",
      "Running deeplab on image: 2\n",
      "Running deeplab on image: 3\n",
      "Running deeplab on image: 4\n",
      "Running deeplab on image: 5\n",
      "Running deeplab on image: 6\n",
      "Running deeplab on image: 7\n",
      "Running deeplab on image: 8\n",
      "Running deeplab on image: 9\n",
      "Running deeplab on image: 10\n",
      "Running deeplab on image: 11\n",
      "Running deeplab on image: 12\n",
      "Running deeplab on image: 13\n",
      "Running deeplab on image: 14\n",
      "Running deeplab on image: 15\n",
      "Running deeplab on image: 16\n",
      "Running deeplab on image: 17\n",
      "Running deeplab on image: 18\n",
      "Running deeplab on image: 19\n",
      "Running deeplab on image: 20\n",
      "Running deeplab on image: 21\n",
      "Running deeplab on image: 22\n",
      "Running deeplab on image: 23\n",
      "Running deeplab on image: 24\n",
      "Running deeplab on image: 25\n",
      "Running deeplab on image: 26\n",
      "Running deeplab on image: 27\n",
      "Running deeplab on image: 28\n",
      "Running deeplab on image: 29\n",
      "Running deeplab on image: 30\n",
      "Running deeplab on image: 31\n",
      "Running deeplab on image: 32\n",
      "Running deeplab on image: 33\n",
      "Running deeplab on image: 34\n",
      "Running deeplab on image: 35\n",
      "Running deeplab on image: 36\n",
      "Running deeplab on image: 37\n",
      "Running deeplab on image: 38\n",
      "Running deeplab on image: 39\n",
      "Running deeplab on image: 40\n",
      "Running deeplab on image: 41\n",
      "Running deeplab on image: 42\n",
      "Running deeplab on image: 43\n",
      "Running deeplab on image: 44\n",
      "Running deeplab on image: 45\n",
      "Running deeplab on image: 46\n",
      "Running deeplab on image: 47\n",
      "Running deeplab on image: 48\n",
      "Running deeplab on image: 49\n",
      "Running deeplab on image: 50\n",
      "Running deeplab on image: 51\n",
      "Running deeplab on image: 52\n",
      "Running deeplab on image: 53\n",
      "Running deeplab on image: 54\n",
      "Running deeplab on image: 55\n",
      "Running deeplab on image: 56\n",
      "Running deeplab on image: 57\n",
      "Running deeplab on image: 58\n",
      "Running deeplab on image: 59\n",
      "Running deeplab on image: 60\n",
      "Running deeplab on image: 61\n",
      "Running deeplab on image: 62\n",
      "Running deeplab on image: 63\n",
      "Running deeplab on image: 64\n",
      "Running deeplab on image: 65\n",
      "Running deeplab on image: 66\n",
      "Running deeplab on image: 67\n",
      "Running deeplab on image: 68\n",
      "Running deeplab on image: 69\n",
      "Running deeplab on image: 70\n",
      "Running deeplab on image: 71\n",
      "Running deeplab on image: 72\n",
      "Running deeplab on image: 73\n",
      "Running deeplab on image: 74\n",
      "Running deeplab on image: 75\n",
      "Running deeplab on image: 76\n",
      "Running deeplab on image: 77\n",
      "Running deeplab on image: 78\n",
      "Running deeplab on image: 79\n",
      "Running deeplab on image: 80\n",
      "Running deeplab on image: 81\n",
      "Running deeplab on image: 82\n",
      "Running deeplab on image: 83\n",
      "Running deeplab on image: 84\n",
      "Running deeplab on image: 85\n",
      "Running deeplab on image: 86\n",
      "Running deeplab on image: 87\n",
      "Running deeplab on image: 88\n",
      "Running deeplab on image: 89\n",
      "Running deeplab on image: 90\n",
      "(deeplab) "
     ]
    }
   ],
   "source": [
    "cd /scratch2/NSF_GWAS/deeplab/\n",
    "conda activate deeplab\n",
    "python /scratch2/NSF_GWAS/deeplab/inference.py --image-list \"${data}/test.csv\"\n",
    "conda deactivate\n",
    "conda deactivate"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### 2.3. Post-processing"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Name outputs to reflect that they are segmentation results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [],
   "source": [
    "cd $data\n",
    "for file in *_rgb_cropped.png; do mv -f \"$file\" \"${file%_rgb_cropped.png}_segment_cropped.png\"; done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Re-expand segment outputs to same size as original RGB files"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Dependencies include `scikit-image`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(base) (base) Working in directory/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/\n",
      "Number of files: 90\n",
      "Reading file GFPA4_I5.0_F1.9_L100_170854_1_0_1_segment_cropped.png\n",
      "Writing file GFPA4_I5.0_F1.9_L100_170854_1_0_1_segment_uncropped.png\n",
      "Reading file GFPA4_I5.0_F1.9_L100_170950_2_0_2_segment_cropped.png\n",
      "Writing file GFPA4_I5.0_F1.9_L100_170950_2_0_2_segment_uncropped.png\n",
      "Reading file GFPA4_I5.0_F1.9_L100_175704_3_2_4_segment_cropped.png\n",
      "Writing file GFPA4_I5.0_F1.9_L100_175704_3_2_4_segment_uncropped.png\n",
      "Reading file GFPA2_I5.0_F1.9_L100_164338_1_0_4_segment_cropped.png\n",
      "Writing file GFPA2_I5.0_F1.9_L100_164338_1_0_4_segment_uncropped.png\n",
      "Reading file GFPA3_I5.0_F1.9_L100_162402_3_0_3_segment_cropped.png\n",
      "Writing file GFPA3_I5.0_F1.9_L100_162402_3_0_3_segment_uncropped.png\n",
      "Reading file GFPA3_I5.0_F1.9_L100_162553_5_0_5_segment_cropped.png\n",
      "Writing file GFPA3_I5.0_F1.9_L100_162553_5_0_5_segment_uncropped.png\n",
      "Reading file GFPA2_I5.0_F1.9_L100_164433_2_0_5_segment_cropped.png\n",
      "Writing file GFPA2_I5.0_F1.9_L100_164433_2_0_5_segment_uncropped.png\n",
      "Reading file GFPA4_I5.0_F1.9_L100_171515_8_1_4_segment_cropped.png\n",
      "Writing file GFPA4_I5.0_F1.9_L100_171515_8_1_4_segment_uncropped.png\n",
      "Reading file GFPA5_I5.0_F1.9_L100_174335_5_0_5_segment_cropped.png\n",
      "Writing file GFPA5_I5.0_F1.9_L100_174335_5_0_5_segment_uncropped.png\n",
      "Reading file GFPA4_I5.0_F1.9_L100_171045_3_0_3_segment_cropped.png\n",
      "Writing file GFPA4_I5.0_F1.9_L100_171045_3_0_3_segment_uncropped.png\n",
      "Reading file GFPA3_I5.0_F1.9_L100_162212_1_0_1_segment_cropped.png\n",
      "Writing file GFPA3_I5.0_F1.9_L100_162212_1_0_1_segment_uncropped.png\n",
      "Reading file GFPA4_I5.0_F1.9_L100_171419_7_1_5_segment_cropped.png\n",
      "Writing file GFPA4_I5.0_F1.9_L100_171419_7_1_5_segment_uncropped.png\n",
      "Reading file GFPA2_I5.0_F1.9_L100_164528_3_0_6_segment_cropped.png\n",
      "Writing file GFPA2_I5.0_F1.9_L100_164528_3_0_6_segment_uncropped.png\n",
      "Reading file GFPA5_I5.0_F1.9_L100_174146_3_0_3_segment_cropped.png\n",
      "Writing file GFPA5_I5.0_F1.9_L100_174146_3_0_3_segment_uncropped.png\n",
      "Reading file GFPA4_I5.0_F1.9_L100_171234_5_0_5_segment_cropped.png\n",
      "Writing file GFPA4_I5.0_F1.9_L100_171234_5_0_5_segment_uncropped.png\n",
      "Reading file GFPA3_I5.0_F1.9_L100_162131_0_0_0_segment_cropped.png\n",
      "Writing file GFPA3_I5.0_F1.9_L100_162131_0_0_0_segment_uncropped.png\n",
      "Reading file GFPA4_I5.0_F1.9_L100_170813_0_0_0_segment_cropped.png\n",
      "Writing file GFPA4_I5.0_F1.9_L100_170813_0_0_0_segment_uncropped.png\n",
      "Reading file GFPA5_I5.0_F1.9_L100_174432_6_0_6_segment_cropped.png\n",
      "Writing file GFPA5_I5.0_F1.9_L100_174432_6_0_6_segment_uncropped.png\n",
      "Reading file GFPA4_I5.0_F1.9_L100_171330_6_1_6_segment_cropped.png\n",
      "Writing file GFPA4_I5.0_F1.9_L100_171330_6_1_6_segment_uncropped.png\n",
      "Reading file GFPA3_I5.0_F1.9_L100_162308_2_0_2_segment_cropped.png\n",
      "Writing file GFPA3_I5.0_F1.9_L100_162308_2_0_2_segment_uncropped.png\n",
      "Reading file GFPA4_I5.0_F1.9_L100_171610_9_1_3_segment_cropped.png\n",
      "Writing file GFPA4_I5.0_F1.9_L100_171610_9_1_3_segment_uncropped.png\n",
      "Reading file GFPA3_I5.0_F1.9_L100_162648_6_0_6_segment_cropped.png\n",
      "Writing file GFPA3_I5.0_F1.9_L100_162648_6_0_6_segment_uncropped.png\n",
      "Reading file GFPA3_I5.0_F1.9_L100_162457_4_0_4_segment_cropped.png\n",
      "Writing file GFPA3_I5.0_F1.9_L100_162457_4_0_4_segment_uncropped.png\n",
      "Reading file GFPA4_I5.0_F1.9_L100_175515_1_2_2_segment_cropped.png\n",
      "Writing file GFPA4_I5.0_F1.9_L100_175515_1_2_2_segment_uncropped.png\n",
      "Reading file GFPA2_I5.0_F1.9_L100_164623_4_1_6_segment_cropped.png\n",
      "Writing file GFPA2_I5.0_F1.9_L100_164623_4_1_6_segment_uncropped.png\n",
      "Reading file GFPA4_I5.0_F1.9_L100_175417_0_2_1_segment_cropped.png\n",
      "Writing file GFPA4_I5.0_F1.9_L100_175417_0_2_1_segment_uncropped.png\n",
      "Reading file GFPA2_I5.0_F1.9_L100_164807_6_1_4_segment_cropped.png\n",
      "Writing file GFPA2_I5.0_F1.9_L100_164807_6_1_4_segment_uncropped.png\n",
      "Reading file GFPA4_I5.0_F1.9_L100_175610_2_2_3_segment_cropped.png\n",
      "Writing file GFPA4_I5.0_F1.9_L100_175610_2_2_3_segment_uncropped.png\n",
      "Reading file GFPA3_I5.0_F1.9_L100_162924_9_1_4_segment_cropped.png\n",
      "Writing file GFPA3_I5.0_F1.9_L100_162924_9_1_4_segment_uncropped.png\n",
      "Reading file GFPA1_I5.0_F1.9_L100_162450_9_1_4_segment_cropped.png\n",
      "Writing file GFPA1_I5.0_F1.9_L100_162450_9_1_4_segment_uncropped.png\n",
      "Reading file GFPA4_I5.0_F1.9_L100_175759_4_2_5_segment_cropped.png\n",
      "Writing file GFPA4_I5.0_F1.9_L100_175759_4_2_5_segment_uncropped.png\n",
      "Reading file GFPA2_I5.0_F1.9_L100_165049_9_1_1_segment_cropped.png\n",
      "Writing file GFPA2_I5.0_F1.9_L100_165049_9_1_1_segment_uncropped.png\n",
      "Reading file GFPA1_I5.0_F1.9_L100_162117_5_0_5_segment_cropped.png\n",
      "Writing file GFPA1_I5.0_F1.9_L100_162117_5_0_5_segment_uncropped.png\n",
      "Reading file GFPA4_I5.0_F1.9_L100_175854_5_2_6_segment_cropped.png\n",
      "Writing file GFPA4_I5.0_F1.9_L100_175854_5_2_6_segment_uncropped.png\n",
      "Reading file GFPA5_I5.0_F1.9_L100_173911_0_0_0_segment_cropped.png\n",
      "Writing file GFPA5_I5.0_F1.9_L100_173911_0_0_0_segment_uncropped.png\n",
      "Reading file GFPA3_I5.0_F1.9_L100_162743_7_1_6_segment_cropped.png\n",
      "Writing file GFPA3_I5.0_F1.9_L100_162743_7_1_6_segment_uncropped.png\n",
      "Reading file GFPA3_I5.0_F1.9_L100_162828_8_1_5_segment_cropped.png\n",
      "Writing file GFPA3_I5.0_F1.9_L100_162828_8_1_5_segment_uncropped.png\n",
      "Reading file GFPA2_I5.0_F1.9_L100_164708_5_1_5_segment_cropped.png\n",
      "Writing file GFPA2_I5.0_F1.9_L100_164708_5_1_5_segment_uncropped.png\n",
      "Reading file GFPA5_I5.0_F1.9_L100_173952_1_0_1_segment_cropped.png\n",
      "Writing file GFPA5_I5.0_F1.9_L100_173952_1_0_1_segment_uncropped.png\n",
      "Reading file GFPA2_I5.0_F1.9_L100_164955_8_1_2_segment_cropped.png\n",
      "Writing file GFPA2_I5.0_F1.9_L100_164955_8_1_2_segment_uncropped.png\n",
      "Reading file GFPA5_I5.0_F1.9_L100_174051_2_0_2_segment_cropped.png\n",
      "Writing file GFPA5_I5.0_F1.9_L100_174051_2_0_2_segment_uncropped.png\n",
      "Reading file GFPA2_I5.0_F1.9_L100_164901_7_1_3_segment_cropped.png\n",
      "Writing file GFPA2_I5.0_F1.9_L100_164901_7_1_3_segment_uncropped.png\n",
      "Reading file GFPA5_I5.0_F1.9_L100_174241_4_0_4_segment_cropped.png\n",
      "Writing file GFPA5_I5.0_F1.9_L100_174241_4_0_4_segment_uncropped.png\n",
      "Reading file GFPA5_I5.0_F1.9_L100_174528_7_1_4_segment_cropped.png\n",
      "Writing file GFPA5_I5.0_F1.9_L100_174528_7_1_4_segment_uncropped.png\n",
      "Reading file GFPA1_I5.0_F1.9_L100_161827_2_0_2_segment_cropped.png\n",
      "Writing file GFPA1_I5.0_F1.9_L100_161827_2_0_2_segment_uncropped.png\n",
      "Reading file GFPA1_I5.0_F1.9_L100_162020_4_0_4_segment_cropped.png\n",
      "Writing file GFPA1_I5.0_F1.9_L100_162020_4_0_4_segment_uncropped.png\n",
      "Reading file GFPA5_I5.0_F1.9_L100_174625_8_1_3_segment_cropped.png\n",
      "Writing file GFPA5_I5.0_F1.9_L100_174625_8_1_3_segment_uncropped.png\n",
      "Reading file GFPA1_I5.0_F1.9_L100_161923_3_0_3_segment_cropped.png\n",
      "Writing file GFPA1_I5.0_F1.9_L100_161923_3_0_3_segment_uncropped.png\n",
      "Reading file GFPA1_I5.0_F1.9_L100_162308_7_1_6_segment_cropped.png\n",
      "Writing file GFPA1_I5.0_F1.9_L100_162308_7_1_6_segment_uncropped.png\n",
      "Reading file GFPA1_I5.0_F1.9_L100_162212_6_0_6_segment_cropped.png\n",
      "Writing file GFPA1_I5.0_F1.9_L100_162212_6_0_6_segment_uncropped.png\n",
      "Reading file GFPA5_I5.0_F1.9_L100_174720_9_1_2_segment_cropped.png\n",
      "Writing file GFPA5_I5.0_F1.9_L100_174720_9_1_2_segment_uncropped.png\n",
      "Reading file GFPA2_I5.0_F1.9_L100_164229_0_0_3_segment_cropped.png\n",
      "Writing file GFPA2_I5.0_F1.9_L100_164229_0_0_3_segment_uncropped.png\n",
      "Reading file GFPA4_I5.0_F1.9_L100_171140_4_0_4_segment_cropped.png\n",
      "Writing file GFPA4_I5.0_F1.9_L100_171140_4_0_4_segment_uncropped.png\n",
      "Reading file GFPA1_I5.0_F1.9_L100_161732_1_0_1_segment_cropped.png\n",
      "Writing file GFPA1_I5.0_F1.9_L100_161732_1_0_1_segment_uncropped.png\n",
      "Reading file GFPA3_I5.0_F1.9_L100_163306_13_1_0_segment_cropped.png\n",
      "Writing file GFPA3_I5.0_F1.9_L100_163306_13_1_0_segment_uncropped.png\n",
      "Reading file GFPA3_I5.0_F1.9_L100_163739_18_2_5_segment_cropped.png\n",
      "Writing file GFPA3_I5.0_F1.9_L100_163739_18_2_5_segment_uncropped.png\n",
      "Reading file GFPA5_I5.0_F1.9_L100_174815_10_1_1_segment_cropped.png\n",
      "Writing file GFPA5_I5.0_F1.9_L100_174815_10_1_1_segment_uncropped.png\n",
      "Reading file GFPA5_I5.0_F1.9_L100_174910_11_1_0_segment_cropped.png\n",
      "Writing file GFPA5_I5.0_F1.9_L100_174910_11_1_0_segment_uncropped.png\n",
      "Reading file GFPA4_I5.0_F1.9_L100_171759_11_1_1_segment_cropped.png\n",
      "Writing file GFPA4_I5.0_F1.9_L100_171759_11_1_1_segment_uncropped.png\n",
      "Reading file GFPA4_I5.0_F1.9_L100_171855_12_1_0_segment_cropped.png\n",
      "Writing file GFPA4_I5.0_F1.9_L100_171855_12_1_0_segment_uncropped.png\n",
      "Reading file GFPA4_I5.0_F1.9_L100_171951_13_2_0_segment_cropped.png\n",
      "Writing file GFPA4_I5.0_F1.9_L100_171951_13_2_0_segment_uncropped.png\n",
      "Reading file GFPA3_I5.0_F1.9_L100_163639_17_2_4_segment_cropped.png\n",
      "Writing file GFPA3_I5.0_F1.9_L100_163639_17_2_4_segment_uncropped.png\n",
      "Reading file GFPA3_I5.0_F1.9_L100_163834_19_2_6_segment_cropped.png\n",
      "Writing file GFPA3_I5.0_F1.9_L100_163834_19_2_6_segment_uncropped.png\n",
      "Reading file GFPA1_I5.0_F1.9_L100_161651_0_0_0_segment_cropped.png\n",
      "Writing file GFPA1_I5.0_F1.9_L100_161651_0_0_0_segment_uncropped.png\n",
      "Reading file GFPA1_I5.0_F1.9_L100_162354_8_1_5_segment_cropped.png\n",
      "Writing file GFPA1_I5.0_F1.9_L100_162354_8_1_5_segment_uncropped.png\n",
      "Reading file GFPA1_I5.0_F1.9_L100_162555_10_1_3_segment_cropped.png\n",
      "Writing file GFPA1_I5.0_F1.9_L100_162555_10_1_3_segment_uncropped.png\n",
      "Reading file GFPA1_I5.0_F1.9_L100_162649_11_1_2_segment_cropped.png\n",
      "Writing file GFPA1_I5.0_F1.9_L100_162649_11_1_2_segment_uncropped.png\n",
      "Reading file GFPA1_I5.0_F1.9_L100_162757_12_1_0_segment_cropped.png\n",
      "Writing file GFPA1_I5.0_F1.9_L100_162757_12_1_0_segment_uncropped.png\n",
      "Reading file GFPA1_I5.0_F1.9_L100_162853_13_2_0_segment_cropped.png\n",
      "Writing file GFPA1_I5.0_F1.9_L100_162853_13_2_0_segment_uncropped.png\n",
      "Reading file GFPA1_I5.0_F1.9_L100_162939_14_2_1_segment_cropped.png\n",
      "Writing file GFPA1_I5.0_F1.9_L100_162939_14_2_1_segment_uncropped.png\n",
      "Reading file GFPA1_I5.0_F1.9_L100_163039_15_2_2_segment_cropped.png\n",
      "Writing file GFPA1_I5.0_F1.9_L100_163039_15_2_2_segment_uncropped.png\n",
      "Reading file GFPA1_I5.0_F1.9_L100_163133_16_2_3_segment_cropped.png\n",
      "Writing file GFPA1_I5.0_F1.9_L100_163133_16_2_3_segment_uncropped.png\n",
      "Reading file GFPA1_I5.0_F1.9_L100_163228_17_2_4_segment_cropped.png\n",
      "Writing file GFPA1_I5.0_F1.9_L100_163228_17_2_4_segment_uncropped.png\n",
      "Reading file GFPA1_I5.0_F1.9_L100_163322_18_2_5_segment_cropped.png\n",
      "Writing file GFPA1_I5.0_F1.9_L100_163322_18_2_5_segment_uncropped.png\n",
      "Reading file GFPA1_I5.0_F1.9_L100_163432_19_2_6_segment_cropped.png\n",
      "Writing file GFPA1_I5.0_F1.9_L100_163432_19_2_6_segment_uncropped.png\n",
      "Reading file GFPA2_I5.0_F1.9_L100_165144_10_1_0_segment_cropped.png\n",
      "Writing file GFPA2_I5.0_F1.9_L100_165144_10_1_0_segment_uncropped.png\n",
      "Reading file GFPA2_I5.0_F1.9_L100_165239_11_2_0_segment_cropped.png\n",
      "Writing file GFPA2_I5.0_F1.9_L100_165239_11_2_0_segment_uncropped.png\n",
      "Reading file GFPA2_I5.0_F1.9_L100_165325_12_2_1_segment_cropped.png\n",
      "Writing file GFPA2_I5.0_F1.9_L100_165325_12_2_1_segment_uncropped.png\n",
      "Reading file GFPA2_I5.0_F1.9_L100_165418_13_2_2_segment_cropped.png\n",
      "Writing file GFPA2_I5.0_F1.9_L100_165418_13_2_2_segment_uncropped.png\n",
      "Reading file GFPA2_I5.0_F1.9_L100_165513_14_2_3_segment_cropped.png\n",
      "Writing file GFPA2_I5.0_F1.9_L100_165513_14_2_3_segment_uncropped.png\n",
      "Reading file GFPA2_I5.0_F1.9_L100_165610_15_2_4_segment_cropped.png\n",
      "Writing file GFPA2_I5.0_F1.9_L100_165610_15_2_4_segment_uncropped.png\n",
      "Reading file GFPA2_I5.0_F1.9_L100_165705_16_2_5_segment_cropped.png\n",
      "Writing file GFPA2_I5.0_F1.9_L100_165705_16_2_5_segment_uncropped.png\n",
      "Reading file GFPA2_I5.0_F1.9_L100_165800_17_2_6_segment_cropped.png\n",
      "Writing file GFPA2_I5.0_F1.9_L100_165800_17_2_6_segment_uncropped.png\n",
      "Reading file GFPA3_I5.0_F1.9_L100_163019_10_1_3_segment_cropped.png\n",
      "Writing file GFPA3_I5.0_F1.9_L100_163019_10_1_3_segment_uncropped.png\n",
      "Reading file GFPA3_I5.0_F1.9_L100_163114_11_1_2_segment_cropped.png\n",
      "Writing file GFPA3_I5.0_F1.9_L100_163114_11_1_2_segment_uncropped.png\n",
      "Reading file GFPA3_I5.0_F1.9_L100_163211_12_1_1_segment_cropped.png\n",
      "Writing file GFPA3_I5.0_F1.9_L100_163211_12_1_1_segment_uncropped.png\n",
      "Reading file GFPA3_I5.0_F1.9_L100_163401_14_2_0_segment_cropped.png\n",
      "Writing file GFPA3_I5.0_F1.9_L100_163401_14_2_0_segment_uncropped.png\n",
      "Reading file GFPA3_I5.0_F1.9_L100_163450_15_2_1_segment_cropped.png\n",
      "Writing file GFPA3_I5.0_F1.9_L100_163450_15_2_1_segment_uncropped.png\n",
      "Reading file GFPA3_I5.0_F1.9_L100_163544_16_2_2_segment_cropped.png\n",
      "Writing file GFPA3_I5.0_F1.9_L100_163544_16_2_2_segment_uncropped.png\n",
      "Reading file GFPA4_I5.0_F1.9_L100_171705_10_1_2_segment_cropped.png\n",
      "Writing file GFPA4_I5.0_F1.9_L100_171705_10_1_2_segment_uncropped.png\n",
      "(base) "
     ]
    }
   ],
   "source": [
    "conda activate base\n",
    "cd /scratch2/NSF_GWAS/ImageAlignment/\n",
    "python expand.py $data\n",
    "conda deactivate"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Make composite images with side-by-side RGB, segmentation outputs and blended images"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(base) ['image_blender.py', '/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/', '0.75', 'both', '1', '180']\n",
      "Building composite for image GFPA1_I5.0_F1.9_L100_161732_1_0_1_rgb.jpg\n",
      "Building composite for image GFPA1_I5.0_F1.9_L100_161827_2_0_2_rgb.jpg\n",
      "Building composite for image GFPA1_I5.0_F1.9_L100_161923_3_0_3_rgb.jpg\n",
      "Building composite for image GFPA1_I5.0_F1.9_L100_162020_4_0_4_rgb.jpg\n",
      "Building composite for image GFPA1_I5.0_F1.9_L100_162117_5_0_5_rgb.jpg\n",
      "Building composite for image GFPA1_I5.0_F1.9_L100_162212_6_0_6_rgb.jpg\n",
      "Building composite for image GFPA1_I5.0_F1.9_L100_162308_7_1_6_rgb.jpg\n",
      "Building composite for image GFPA1_I5.0_F1.9_L100_162354_8_1_5_rgb.jpg\n",
      "Building composite for image GFPA1_I5.0_F1.9_L100_162450_9_1_4_rgb.jpg\n",
      "Building composite for image GFPA1_I5.0_F1.9_L100_162555_10_1_3_rgb.jpg\n",
      "Building composite for image GFPA1_I5.0_F1.9_L100_162649_11_1_2_rgb.jpg\n",
      "Building composite for image GFPA1_I5.0_F1.9_L100_162757_12_1_0_rgb.jpg\n",
      "Building composite for image GFPA1_I5.0_F1.9_L100_162853_13_2_0_rgb.jpg\n",
      "Building composite for image GFPA1_I5.0_F1.9_L100_162939_14_2_1_rgb.jpg\n",
      "Building composite for image GFPA1_I5.0_F1.9_L100_163039_15_2_2_rgb.jpg\n",
      "Building composite for image GFPA1_I5.0_F1.9_L100_163133_16_2_3_rgb.jpg\n",
      "Building composite for image GFPA1_I5.0_F1.9_L100_163228_17_2_4_rgb.jpg\n",
      "Building composite for image GFPA1_I5.0_F1.9_L100_163322_18_2_5_rgb.jpg\n",
      "Building composite for image GFPA1_I5.0_F1.9_L100_163432_19_2_6_rgb.jpg\n",
      "Building composite for image GFPA2_I5.0_F1.9_L100_164229_0_0_3_rgb.jpg\n",
      "Building composite for image GFPA2_I5.0_F1.9_L100_164338_1_0_4_rgb.jpg\n",
      "Building composite for image GFPA2_I5.0_F1.9_L100_164433_2_0_5_rgb.jpg\n",
      "Building composite for image GFPA2_I5.0_F1.9_L100_164528_3_0_6_rgb.jpg\n",
      "Building composite for image GFPA2_I5.0_F1.9_L100_164623_4_1_6_rgb.jpg\n",
      "Building composite for image GFPA2_I5.0_F1.9_L100_164708_5_1_5_rgb.jpg\n",
      "Building composite for image GFPA2_I5.0_F1.9_L100_164807_6_1_4_rgb.jpg\n",
      "Building composite for image GFPA2_I5.0_F1.9_L100_164901_7_1_3_rgb.jpg\n",
      "Building composite for image GFPA2_I5.0_F1.9_L100_164955_8_1_2_rgb.jpg\n",
      "Building composite for image GFPA2_I5.0_F1.9_L100_165049_9_1_1_rgb.jpg\n",
      "Building composite for image GFPA2_I5.0_F1.9_L100_165144_10_1_0_rgb.jpg\n",
      "Building composite for image GFPA2_I5.0_F1.9_L100_165239_11_2_0_rgb.jpg\n",
      "Building composite for image GFPA2_I5.0_F1.9_L100_165325_12_2_1_rgb.jpg\n",
      "Building composite for image GFPA2_I5.0_F1.9_L100_165418_13_2_2_rgb.jpg\n",
      "Building composite for image GFPA2_I5.0_F1.9_L100_165513_14_2_3_rgb.jpg\n",
      "Building composite for image GFPA2_I5.0_F1.9_L100_165610_15_2_4_rgb.jpg\n",
      "Building composite for image GFPA2_I5.0_F1.9_L100_165705_16_2_5_rgb.jpg\n",
      "Building composite for image GFPA2_I5.0_F1.9_L100_165800_17_2_6_rgb.jpg\n",
      "Building composite for image GFPA3_I5.0_F1.9_L100_162131_0_0_0_rgb.jpg\n",
      "Building composite for image GFPA3_I5.0_F1.9_L100_162212_1_0_1_rgb.jpg\n",
      "Building composite for image GFPA3_I5.0_F1.9_L100_162308_2_0_2_rgb.jpg\n",
      "Building composite for image GFPA3_I5.0_F1.9_L100_162402_3_0_3_rgb.jpg\n",
      "Building composite for image GFPA3_I5.0_F1.9_L100_162457_4_0_4_rgb.jpg\n",
      "Building composite for image GFPA3_I5.0_F1.9_L100_162553_5_0_5_rgb.jpg\n",
      "Building composite for image GFPA3_I5.0_F1.9_L100_162648_6_0_6_rgb.jpg\n",
      "Building composite for image GFPA3_I5.0_F1.9_L100_162743_7_1_6_rgb.jpg\n",
      "Building composite for image GFPA3_I5.0_F1.9_L100_162828_8_1_5_rgb.jpg\n",
      "Building composite for image GFPA3_I5.0_F1.9_L100_162924_9_1_4_rgb.jpg\n",
      "Building composite for image GFPA3_I5.0_F1.9_L100_163019_10_1_3_rgb.jpg\n",
      "Building composite for image GFPA3_I5.0_F1.9_L100_163114_11_1_2_rgb.jpg\n",
      "Building composite for image GFPA3_I5.0_F1.9_L100_163211_12_1_1_rgb.jpg\n",
      "Building composite for image GFPA3_I5.0_F1.9_L100_163306_13_1_0_rgb.jpg\n",
      "Building composite for image GFPA3_I5.0_F1.9_L100_163401_14_2_0_rgb.jpg\n",
      "Building composite for image GFPA3_I5.0_F1.9_L100_163450_15_2_1_rgb.jpg\n",
      "Building composite for image GFPA3_I5.0_F1.9_L100_163544_16_2_2_rgb.jpg\n",
      "Building composite for image GFPA3_I5.0_F1.9_L100_163639_17_2_4_rgb.jpg\n",
      "Building composite for image GFPA3_I5.0_F1.9_L100_163739_18_2_5_rgb.jpg\n",
      "Building composite for image GFPA3_I5.0_F1.9_L100_163834_19_2_6_rgb.jpg\n",
      "Building composite for image GFPA4_I5.0_F1.9_L100_170813_0_0_0_rgb.jpg\n",
      "Building composite for image GFPA4_I5.0_F1.9_L100_170854_1_0_1_rgb.jpg\n",
      "Building composite for image GFPA4_I5.0_F1.9_L100_170950_2_0_2_rgb.jpg\n",
      "Building composite for image GFPA4_I5.0_F1.9_L100_171045_3_0_3_rgb.jpg\n",
      "Building composite for image GFPA4_I5.0_F1.9_L100_171140_4_0_4_rgb.jpg\n",
      "Building composite for image GFPA4_I5.0_F1.9_L100_171234_5_0_5_rgb.jpg\n",
      "Building composite for image GFPA4_I5.0_F1.9_L100_171330_6_1_6_rgb.jpg\n",
      "Building composite for image GFPA4_I5.0_F1.9_L100_171419_7_1_5_rgb.jpg\n",
      "Building composite for image GFPA4_I5.0_F1.9_L100_171515_8_1_4_rgb.jpg\n",
      "Building composite for image GFPA4_I5.0_F1.9_L100_171610_9_1_3_rgb.jpg\n",
      "Building composite for image GFPA4_I5.0_F1.9_L100_171705_10_1_2_rgb.jpg\n",
      "Building composite for image GFPA4_I5.0_F1.9_L100_171759_11_1_1_rgb.jpg\n",
      "Building composite for image GFPA4_I5.0_F1.9_L100_171855_12_1_0_rgb.jpg\n",
      "Building composite for image GFPA4_I5.0_F1.9_L100_171951_13_2_0_rgb.jpg\n",
      "Building composite for image GFPA4_I5.0_F1.9_L100_175417_0_2_1_rgb.jpg\n",
      "Building composite for image GFPA4_I5.0_F1.9_L100_175515_1_2_2_rgb.jpg\n",
      "Building composite for image GFPA4_I5.0_F1.9_L100_175610_2_2_3_rgb.jpg\n",
      "Building composite for image GFPA4_I5.0_F1.9_L100_175704_3_2_4_rgb.jpg\n",
      "Building composite for image GFPA4_I5.0_F1.9_L100_175759_4_2_5_rgb.jpg\n",
      "Building composite for image GFPA4_I5.0_F1.9_L100_175854_5_2_6_rgb.jpg\n",
      "Building composite for image GFPA5_I5.0_F1.9_L100_173911_0_0_0_rgb.jpg\n",
      "Building composite for image GFPA5_I5.0_F1.9_L100_173952_1_0_1_rgb.jpg\n",
      "Building composite for image GFPA5_I5.0_F1.9_L100_174051_2_0_2_rgb.jpg\n",
      "Building composite for image GFPA5_I5.0_F1.9_L100_174146_3_0_3_rgb.jpg\n",
      "Building composite for image GFPA5_I5.0_F1.9_L100_174241_4_0_4_rgb.jpg\n",
      "Building composite for image GFPA5_I5.0_F1.9_L100_174335_5_0_5_rgb.jpg\n",
      "Building composite for image GFPA5_I5.0_F1.9_L100_174432_6_0_6_rgb.jpg\n",
      "Building composite for image GFPA5_I5.0_F1.9_L100_174528_7_1_4_rgb.jpg\n",
      "Building composite for image GFPA5_I5.0_F1.9_L100_174625_8_1_3_rgb.jpg\n",
      "Building composite for image GFPA5_I5.0_F1.9_L100_174720_9_1_2_rgb.jpg\n",
      "Building composite for image GFPA5_I5.0_F1.9_L100_174815_10_1_1_rgb.jpg\n",
      "Building composite for image GFPA5_I5.0_F1.9_L100_174910_11_1_0_rgb.jpg\n",
      "(base) "
     ]
    }
   ],
   "source": [
    "conda activate base\n",
    "if [ $composite -eq 1 ]\n",
    "then\n",
    "    cd /scratch2/NSF_GWAS/GMOlabeler/\n",
    "    python image_blender.py $data 0.75 'both' 1 180\n",
    "fi\n",
    "conda deactivate"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### 2.4. Classification of contaminated/missing explants"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Plates are cropped into sub-images for each explant and each is analyzed to determine if the explant position should be excluded from analysis due to being missing or contamination. Missing and contaminated explants are recognized using a trained Densenet model (<a href=\"https://github.com/Contamination-Classification/DenseNet\" target=\"_blank\">Huang, et al. 2018</a>). Our fork of the Densenet repository is available on <a href=\"https://arxiv.org/abs/1608.06993\" target=\"_blank\">GitHub</a>."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<img src=\"Figures/Densenet.png\">\n",
    "Figure: These are four examples of contaminated explants used in the training set for this pre-trained model"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "If the mode for missing explant data is automatic, prepare input file for script to detect missing explants and run this script."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [],
   "source": [
    "img_list_path=\"${data}/rgb_list.txt\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Dependencies include `keras-preprocessing`, `termcolor`,  `protobuf` and `absl-py`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//rgb_list.txt\n"
     ]
    }
   ],
   "source": [
    "echo $img_list_path"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Missing explants will be inferred.\n",
      "Using TensorFlow backend.\n",
      "Reading Arguments: \n",
      "Processing 90 images.\n",
      "Intializing the pretrained model\n",
      "2020-10-27 20:49:08.725479: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcuda.so.1\n",
      "2020-10-27 20:49:14.384983: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1561] Found device 0 with properties: \n",
      "pciBusID: 0000:05:00.0 name: Tesla K80 computeCapability: 3.7\n",
      "coreClock: 0.8235GHz coreCount: 13 deviceMemorySize: 11.17GiB deviceMemoryBandwidth: 223.96GiB/s\n",
      "2020-10-27 20:49:14.385421: W tensorflow/stream_executor/platform/default/dso_loader.cc:55] Could not load dynamic library 'libcudart.so.10.1'; dlerror: libcudart.so.10.1: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /usr/local/apps/cuda/cuda-8.0.61/lib64:/usr/local/apps/cuda/cuda-9.0/lib64/\n",
      "2020-10-27 20:49:14.385727: W tensorflow/stream_executor/platform/default/dso_loader.cc:55] Could not load dynamic library 'libcublas.so.10'; dlerror: libcublas.so.10: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /usr/local/apps/cuda/cuda-8.0.61/lib64:/usr/local/apps/cuda/cuda-9.0/lib64/\n",
      "2020-10-27 20:49:14.386000: W tensorflow/stream_executor/platform/default/dso_loader.cc:55] Could not load dynamic library 'libcufft.so.10'; dlerror: libcufft.so.10: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /usr/local/apps/cuda/cuda-8.0.61/lib64:/usr/local/apps/cuda/cuda-9.0/lib64/\n",
      "2020-10-27 20:49:14.386268: W tensorflow/stream_executor/platform/default/dso_loader.cc:55] Could not load dynamic library 'libcurand.so.10'; dlerror: libcurand.so.10: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /usr/local/apps/cuda/cuda-8.0.61/lib64:/usr/local/apps/cuda/cuda-9.0/lib64/\n",
      "2020-10-27 20:49:14.386532: W tensorflow/stream_executor/platform/default/dso_loader.cc:55] Could not load dynamic library 'libcusolver.so.10'; dlerror: libcusolver.so.10: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /usr/local/apps/cuda/cuda-8.0.61/lib64:/usr/local/apps/cuda/cuda-9.0/lib64/\n",
      "2020-10-27 20:49:14.386809: W tensorflow/stream_executor/platform/default/dso_loader.cc:55] Could not load dynamic library 'libcusparse.so.10'; dlerror: libcusparse.so.10: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /usr/local/apps/cuda/cuda-8.0.61/lib64:/usr/local/apps/cuda/cuda-9.0/lib64/\n",
      "2020-10-27 20:49:14.397930: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudnn.so.7\n",
      "2020-10-27 20:49:14.398017: W tensorflow/core/common_runtime/gpu/gpu_device.cc:1598] Cannot dlopen some GPU libraries. Please make sure the missing libraries mentioned above are installed properly if you would like to use GPU. Follow the guide at https://www.tensorflow.org/install/gpu for how to download and setup the required libraries for your platform.\n",
      "Skipping registering GPU devices...\n",
      "2020-10-27 20:49:14.399867: I tensorflow/core/platform/cpu_feature_guard.cc:143] Your CPU supports instructions that this TensorFlow binary was not compiled to use: AVX2 FMA\n",
      "2020-10-27 20:49:14.550959: I tensorflow/core/platform/profile_utils/cpu_utils.cc:102] CPU Frequency: 1800110000 Hz\n",
      "2020-10-27 20:49:14.575356: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x55afa0b52b50 initialized for platform Host (this does not guarantee that XLA will be used). Devices:\n",
      "2020-10-27 20:49:14.575470: I tensorflow/compiler/xla/service/service.cc:176]   StreamExecutor device (0): Host, Default Version\n",
      "2020-10-27 20:49:14.588816: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1102] Device interconnect StreamExecutor with strength 1 edge matrix:\n",
      "2020-10-27 20:49:14.588887: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1108]      \n",
      "[INFO]Weights loaded!\n",
      "Model loaded sucessfully!\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n",
      "Preprocess the image.\n"
     ]
    }
   ],
   "source": [
    "if [ $missing_explants = \"Automatic\" ]; then\n",
    "    conda activate densenet\n",
    "    echo \"Missing explants will be inferred.\"\n",
    "    cd $data\n",
    "    ls -d $PWD/* $data | grep -i \"rgb.jpg\" > rgb_list.txt\n",
    "    sed -i '/hroma/d' rgb_list.txt\n",
    "    #sed -i 's/.jpg//g' rgb_list.txt\n",
    "    #cat rgb_list.txt\n",
    "    cd /scratch2/NSF_GWAS/Contamination\n",
    "    img_list_path=\"${data}/rgb_list.txt\"\n",
    "    #echo $img_list_path\n",
    "    python inference.py --img-list=$img_list_path --output_file=output.csv\n",
    "    mv -f output.csv \"${data}/output.csv\"\n",
    "    missing_explants=\"${data}/output.csv\"\n",
    "    conda deactivate\n",
    "else\n",
    "    echo \"Missing explants input manually by user, in file: \"\n",
    "    echo $missing_explants\n",
    "fi"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 3. Alignment of RGB and hyperspectral layers"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "To match the frame and angle of RGB and hyperspectral image layers, we apply a scale-invariant feature transformation (<a href=\"https://github.com/NSF-Image-alignment/ImageAlignment\" target=\"_blank\">GitHub</a>). Using a pair of standard images, a homography matrix is calculated for the necessary transformation of RGB images to align with hyperspectral images. The transformation can then be applied to large batches of images rapidly, as long as the RGB and hyperspectral cameras remain in the same positions relative to one another (as they do in the macroPhor Array platform)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<img src=\"Figures/Alignment.png\">\n",
    "Figure: To enable precise calculation of a homography matrix for transformation of RGB images to match hyperspectral images, we used images of a piece of paper with grid marks. These images are provided by the user inputs to --hyper-img and --rgb-img in the below call to the alignment script. If using a phenotyping platform other than the macroPhor Array, or using updated camera settings, these variables will need to be replaced."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### 3.1. Prepare input file for alignment"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [],
   "source": [
    "cd $data\n",
    "ls | grep -i 'rgb\\.jpg' > file_list_part1.csv\n",
    "ls | grep -i 'segment_uncropped\\.png' > file_list_part2.csv\n",
    "cat file_list_part* > file_list.csv\n",
    "sed -i '/hroma/d' file_list.csv\n",
    "cwd=$(pwd)/\n",
    "awk -v prefix=\"$cwd\" '{print prefix $0}' file_list.csv > temp\n",
    "mv -f temp file_list.csv\n",
    "echo 'rgb_images' | cat - file_list.csv > temp && mv -f temp file_list.csv"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### 3.2 Run alignment"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(alignment) (alignment) (alignment) [  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  0  24  26  27  28  29  30  31  32  33  34  35  36  37  38  39  40  41\n",
      "  42  43  44  45  46  47  48  49  50  51  52  53  54  55  56  57  58  59\n",
      "  60  61  62  63  64  65  66  67  68  69  70  71  72  73  74  75  76  77\n",
      "  78  79  80  81  82  83  84  85  86  87  88  89  90  91  92  93  94  95\n",
      "  96  97  98  99 100 101 102 103 104 105 106 107 108 109 110 111 112 113\n",
      " 114 115 116 117 118 119 120 121 122 123 124 125 126 127 128 129 130 131\n",
      " 132 133 134 135 136 137 138 139 140 141 142 143 144 145 146 147 148 149\n",
      " 150 151 152 153 154 155 156 157 158 159 160 161 162 163 164 165 166 167\n",
      " 168 169 170 171 172 173 174 175 176 177 178 179 180 181 182 183 184 185\n",
      " 186 187 188 189 190 191 192 193 194 195 196 197 198 199 200 201 202 203\n",
      " 204 205 206 207 208 209 210 211 212 213 214 215 216 217 218 219 220 221\n",
      " 222 223 224 225 226 227 228 229 230 231 232 233 234 235 236 237 238 239\n",
      " 240 241 242 243 244 245 246 247 248 249 250 251 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17  18\n",
      "  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35  36\n",
      "  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53  54\n",
      "  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71  72\n",
      "  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89  90\n",
      "  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107 108\n",
      " 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125 126\n",
      " 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143 144\n",
      " 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161 162\n",
      " 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179 180\n",
      " 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197 198\n",
      " 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215 216\n",
      " 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233 234\n",
      " 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251 252\n",
      " 253 254 255]\n",
      "[  0  16  17  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32\n",
      "  33  34  35  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50\n",
      "  51  52  53  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68\n",
      "  69  70  71  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86\n",
      "  87  88  89  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104\n",
      " 105 106 107 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122\n",
      " 123 124 125 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140\n",
      " 141 142 143 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158\n",
      " 159 160 161 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176\n",
      " 177 178 179 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194\n",
      " 195 196 197 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212\n",
      " 213 214 215 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230\n",
      " 231 232 233 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248\n",
      " 249 250 251 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  0   7   8   9  10  11  12  13  14  15  16  17  18  19  20  21  22  23\n",
      "  24  25  26  27  28  29  30  31  32  33  34  35  36  37  38  39  40  41\n",
      "  42  43  44  45  46  47  48  49  50  51  52  53  54  55  56  57  58  59\n",
      "  60  61  62  63  64  65  66  67  68  69  70  71  72  73  74  75  76  77\n",
      "  78  79  80  81  82  83  84  85  86  87  88  89  90  91  92  93  94  95\n",
      "  96  97  98  99 100 101 102 103 104 105 106 107 108 109 110 111 112 113\n",
      " 114 115 116 117 118 119 120 121 122 123 124 125 126 127 128 129 130 131\n",
      " 132 133 134 135 136 137 138 139 140 141 142 143 144 145 146 147 148 149\n",
      " 150 151 152 153 154 155 156 157 158 159 160 161 162 163 164 165 166 167\n",
      " 168 169 170 171 172 173 174 175 176 177 178 179 180 181 182 183 184 185\n",
      " 186 187 188 189 190 191 192 193 194 195 196 197 198 199 200 201 202 203\n",
      " 204 205 206 207 208 209 210 211 212 213 214 215 216 217 218 219 220 221\n",
      " 222 223 224 225 226 227 228 229 230 231 232 233 234 235 236 237 238 239\n",
      " 240 241 242 243 244 245 246 247 248 249 250 251 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17  18  19\n",
      "  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35  36  37\n",
      "  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53  54  55\n",
      "  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71  72  73\n",
      "  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89  90  91\n",
      "  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107 108 109\n",
      " 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125 126 127\n",
      " 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143 144 145\n",
      " 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161 162 163\n",
      " 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179 180 181\n",
      " 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197 198 199\n",
      " 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215 216 217\n",
      " 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233 234 235\n",
      " 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251 252 253\n",
      " 254 255]\n",
      "[  0   7   8   9  10  11  12  13  14  15  16  17  18  19  20  21  22  23\n",
      "  24  25  26  27  28  29  30  31  32  33  34  35  36  37  38  39  40  41\n",
      "  42  43  44  45  46  47  48  49  50  51  52  53  54  55  56  57  58  59\n",
      "  60  61  62  63  64  65  66  67  68  69  70  71  72  73  74  75  76  77\n",
      "  78  79  80  81  82  83  84  85  86  87  88  89  90  91  92  93  94  95\n",
      "  96  97  98  99 100 101 102 103 104 105 106 107 108 109 110 111 112 113\n",
      " 114 115 116 117 118 119 120 121 122 123 124 125 126 127 128 129 130 131\n",
      " 132 133 134 135 136 137 138 139 140 141 142 143 144 145 146 147 148 149\n",
      " 150 151 152 153 154 155 156 157 158 159 160 161 162 163 164 165 166 167\n",
      " 168 169 170 171 172 173 174 175 176 177 178 179 180 181 182 183 184 185\n",
      " 186 187 188 189 190 191 192 193 194 195 196 197 198 199 200 201 202 203\n",
      " 204 205 206 207 208 209 210 211 212 213 214 215 216 217 218 219 220 221\n",
      " 222 223 224 225 226 227 228 229 230 231 232 233 234 235 236 237 238 239\n",
      " 240 241 242 243 244 245 246 247 248 249 250 251 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17  18  19\n",
      "  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35  36  37\n",
      "  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53  54  55\n",
      "  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71  72  73\n",
      "  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89  90  91\n",
      "  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107 108 109\n",
      " 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125 126 127\n",
      " 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143 144 145\n",
      " 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161 162 163\n",
      " 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179 180 181\n",
      " 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197 198 199\n",
      " 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215 216 217\n",
      " 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233 234 235\n",
      " 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251 252 253\n",
      " 254 255]\n",
      "[  0   9  12  13  14  15  16  17  18  19  20  21  22  23  24  25  26  27\n",
      "  28  29  30  31  32  33  34  35  36  37  38  39  40  41  42  43  44  45\n",
      "  46  47  48  49  50  51  52  53  54  55  56  57  58  59  60  61  62  63\n",
      "  64  65  66  67  68  69  70  71  72  73  74  75  76  77  78  79  80  81\n",
      "  82  83  84  85  86  87  88  89  90  91  92  93  94  95  96  97  98  99\n",
      " 100 101 102 103 104 105 106 107 108 109 110 111 112 113 114 115 116 117\n",
      " 118 119 120 121 122 123 124 125 126 127 128 129 130 131 132 133 134 135\n",
      " 136 137 138 139 140 141 142 143 144 145 146 147 148 149 150 151 152 153\n",
      " 154 155 156 157 158 159 160 161 162 163 164 165 166 167 168 169 170 171\n",
      " 172 173 174 175 176 177 178 179 180 181 182 183 184 185 186 187 188 189\n",
      " 190 191 192 193 194 195 196 197 198 199 200 201 202 203 204 205 206 207\n",
      " 208 209 210 211 212 213 214 215 216 217 218 219 220 221 222 223 224 225\n",
      " 226 227 228 229 230 231 232 233 234 235 236 237 238 239 240 241 242 243\n",
      " 244 245 246 247 248 249 250 251 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17  18\n",
      "  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35  36\n",
      "  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53  54\n",
      "  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71  72\n",
      "  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89  90\n",
      "  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107 108\n",
      " 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125 126\n",
      " 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143 144\n",
      " 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161 162\n",
      " 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179 180\n",
      " 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197 198\n",
      " 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215 216\n",
      " 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233 234\n",
      " 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251 252\n",
      " 253 254 255]\n",
      "[  0   8  15  17  18  19  20  21  22  23  24  25  26  27  28  29  30  31\n",
      "  32  33  34  35  36  37  38  39  40  41  42  43  44  45  46  47  48  49\n",
      "  50  51  52  53  54  55  56  57  58  59  60  61  62  63  64  65  66  67\n",
      "  68  69  70  71  72  73  74  75  76  77  78  79  80  81  82  83  84  85\n",
      "  86  87  88  89  90  91  92  93  94  95  96  97  98  99 100 101 102 103\n",
      " 104 105 106 107 108 109 110 111 112 113 114 115 116 117 118 119 120 121\n",
      " 122 123 124 125 126 127 128 129 130 131 132 133 134 135 136 137 138 139\n",
      " 140 141 142 143 144 145 146 147 148 149 150 151 152 153 154 155 156 157\n",
      " 158 159 160 161 162 163 164 165 166 167 168 169 170 171 172 173 174 175\n",
      " 176 177 178 179 180 181 182 183 184 185 186 187 188 189 190 191 192 193\n",
      " 194 195 196 197 198 199 200 201 202 203 204 205 206 207 208 209 210 211\n",
      " 212 213 214 215 216 217 218 219 220 221 222 223 224 225 226 227 228 229\n",
      " 230 231 232 233 234 235 236 237 238 239 240 241 242 243 244 245 246 247\n",
      " 248 249 250 251 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17  18\n",
      "  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35  36\n",
      "  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53  54\n",
      "  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71  72\n",
      "  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89  90\n",
      "  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107 108\n",
      " 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125 126\n",
      " 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143 144\n",
      " 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161 162\n",
      " 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179 180\n",
      " 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197 198\n",
      " 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215 216\n",
      " 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233 234\n",
      " 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251 252\n",
      " 253 254 255]\n",
      "[  0   1   6  10  11  12  13  14  15  16  17  18  19  20  21  22  23  24\n",
      "  25  26  27  28  29  30  31  32  33  34  35  36  37  38  39  40  41  42\n",
      "  43  44  45  46  47  48  49  50  51  52  53  54  55  56  57  58  59  60\n",
      "  61  62  63  64  65  66  67  68  69  70  71  72  73  74  75  76  77  78\n",
      "  79  80  81  82  83  84  85  86  87  88  89  90  91  92  93  94  95  96\n",
      "  97  98  99 100 101 102 103 104 105 106 107 108 109 110 111 112 113 114\n",
      " 115 116 117 118 119 120 121 122 123 124 125 126 127 128 129 130 131 132\n",
      " 133 134 135 136 137 138 139 140 141 142 143 144 145 146 147 148 149 150\n",
      " 151 152 153 154 155 156 157 158 159 160 161 162 163 164 165 166 167 168\n",
      " 169 170 171 172 173 174 175 176 177 178 179 180 181 182 183 184 185 186\n",
      " 187 188 189 190 191 192 193 194 195 196 197 198 199 200 201 202 203 204\n",
      " 205 206 207 208 209 210 211 212 213 214 215 216 217 218 219 220 221 222\n",
      " 223 224 225 226 227 228 229 230 231 232 233 234 235 236 237 238 239 240\n",
      " 241 242 243 244 245 246 247 248 249 250 251 252 253 254 255]\n",
      "[  1   5   7   8   9  10  11  12  13  14  15  16  17  18  19  20  21  22\n",
      "  23  24  25  26  27  28  29  30  31  32  33  34  35  36  37  38  39  40\n",
      "  41  42  43  44  45  46  47  48  49  50  51  52  53  54  55  56  57  58\n",
      "  59  60  61  62  63  64  65  66  67  68  69  70  71  72  73  74  75  76\n",
      "  77  78  79  80  81  82  83  84  85  86  87  88  89  90  91  92  93  94\n",
      "  95  96  97  98  99 100 101 102 103 104 105 106 107 108 109 110 111 112\n",
      " 113 114 115 116 117 118 119 120 121 122 123 124 125 126 127 128 129 130\n",
      " 131 132 133 134 135 136 137 138 139 140 141 142 143 144 145 146 147 148\n",
      " 149 150 151 152 153 154 155 156 157 158 159 160 161 162 163 164 165 166\n",
      " 167 168 169 170 171 172 173 174 175 176 177 178 179 180 181 182 183 184\n",
      " 185 186 187 188 189 190 191 192 193 194 195 196 197 198 199 200 201 202\n",
      " 203 204 205 206 207 208 209 210 211 212 213 214 215 216 217 218 219 220\n",
      " 221 222 223 224 225 226 227 228 229 230 231 232 233 234 235 236 237 238\n",
      " 239 240 241 242 243 244 245 246 247 248 249 250 251 252 253 254 255]\n",
      "[  7   8   9  10  11  12  13  14  15  16  17  18  19  20  21  22  23  24\n",
      "  25  26  27  28  29  30  31  32  33  34  35  36  37  38  39  40  41  42\n",
      "  43  44  45  46  47  48  49  50  51  52  53  54  55  56  57  58  59  60\n",
      "  61  62  63  64  65  66  67  68  69  70  71  72  73  74  75  76  77  78\n",
      "  79  80  81  82  83  84  85  86  87  88  89  90  91  92  93  94  95  96\n",
      "  97  98  99 100 101 102 103 104 105 106 107 108 109 110 111 112 113 114\n",
      " 115 116 117 118 119 120 121 122 123 124 125 126 127 128 129 130 131 132\n",
      " 133 134 135 136 137 138 139 140 141 142 143 144 145 146 147 148 149 150\n",
      " 151 152 153 154 155 156 157 158 159 160 161 162 163 164 165 166 167 168\n",
      " 169 170 171 172 173 174 175 176 177 178 179 180 181 182 183 184 185 186\n",
      " 187 188 189 190 191 192 193 194 195 196 197 198 199 200 201 202 203 204\n",
      " 205 206 207 208 209 210 211 212 213 214 215 216 217 218 219 220 221 222\n",
      " 223 224 225 226 227 228 229 230 231 232 233 234 235 236 237 238 239 240\n",
      " 241 242 243 244 245 246 247 248 249 250 251 252 253 254 255]\n",
      "[  0  18  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  1   2   3   4   5   6   8   9  10  11  12  13  14  15  16  17  18  19\n",
      "  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35  36  37\n",
      "  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53  54  55\n",
      "  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71  72  73\n",
      "  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89  90  91\n",
      "  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107 108 109\n",
      " 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125 126 127\n",
      " 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143 144 145\n",
      " 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161 162 163\n",
      " 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179 180 181\n",
      " 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197 198 199\n",
      " 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215 216 217\n",
      " 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233 234 235\n",
      " 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251 252 253\n",
      " 254 255]\n",
      "[  8  11  12  13  14  15  16  17  18  19  20  21  22  23  24  25  26  27\n",
      "  28  29  30  31  32  33  34  35  36  37  38  39  40  41  42  43  44  45\n",
      "  46  47  48  49  50  51  52  53  54  55  56  57  58  59  60  61  62  63\n",
      "  64  65  66  67  68  69  70  71  72  73  74  75  76  77  78  79  80  81\n",
      "  82  83  84  85  86  87  88  89  90  91  92  93  94  95  96  97  98  99\n",
      " 100 101 102 103 104 105 106 107 108 109 110 111 112 113 114 115 116 117\n",
      " 118 119 120 121 122 123 124 125 126 127 128 129 130 131 132 133 134 135\n",
      " 136 137 138 139 140 141 142 143 144 145 146 147 148 149 150 151 152 153\n",
      " 154 155 156 157 158 159 160 161 162 163 164 165 166 167 168 169 170 171\n",
      " 172 173 174 175 176 177 178 179 180 181 182 183 184 185 186 187 188 189\n",
      " 190 191 192 193 194 195 196 197 198 199 200 201 202 203 204 205 206 207\n",
      " 208 209 210 211 212 213 214 215 216 217 218 219 220 221 222 223 224 225\n",
      " 226 227 228 229 230 231 232 233 234 235 236 237 238 239 240 241 242 243\n",
      " 244 245 246 247 248 249 250 251 252 253 254 255]\n",
      "[  0  11  12  13  14  15  16  17  18  19  20  21  22  23  24  25  26  27\n",
      "  28  29  30  31  32  33  34  35  36  37  38  39  40  41  42  43  44  45\n",
      "  46  47  48  49  50  51  52  53  54  55  56  57  58  59  60  61  62  63\n",
      "  64  65  66  67  68  69  70  71  72  73  74  75  76  77  78  79  80  81\n",
      "  82  83  84  85  86  87  88  89  90  91  92  93  94  95  96  97  98  99\n",
      " 100 101 102 103 104 105 106 107 108 109 110 111 112 113 114 115 116 117\n",
      " 118 119 120 121 122 123 124 125 126 127 128 129 130 131 132 133 134 135\n",
      " 136 137 138 139 140 141 142 143 144 145 146 147 148 149 150 151 152 153\n",
      " 154 155 156 157 158 159 160 161 162 163 164 165 166 167 168 169 170 171\n",
      " 172 173 174 175 176 177 178 179 180 181 182 183 184 185 186 187 188 189\n",
      " 190 191 192 193 194 195 196 197 198 199 200 201 202 203 204 205 206 207\n",
      " 208 209 210 211 212 213 214 215 216 217 218 219 220 221 222 223 224 225\n",
      " 226 227 228 229 230 231 232 233 234 235 236 237 238 239 240 241 242 243\n",
      " 244 245 246 247 248 249 250 251 252 253 254 255]\n",
      "[  0   1   2   5   6   7   8   9  10  11  12  13  14  15  16  17  18  19\n",
      "  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35  36  37\n",
      "  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53  54  55\n",
      "  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71  72  73\n",
      "  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89  90  91\n",
      "  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107 108 109\n",
      " 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125 126 127\n",
      " 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143 144 145\n",
      " 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161 162 163\n",
      " 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179 180 181\n",
      " 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197 198 199\n",
      " 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215 216 217\n",
      " 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233 234 235\n",
      " 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251 252 253\n",
      " 254 255]\n",
      "[ 10  12  13  14  15  16  17  18  19  20  21  22  23  24  25  26  27  28\n",
      "  29  30  31  32  33  34  35  36  37  38  39  40  41  42  43  44  45  46\n",
      "  47  48  49  50  51  52  53  54  55  56  57  58  59  60  61  62  63  64\n",
      "  65  66  67  68  69  70  71  72  73  74  75  76  77  78  79  80  81  82\n",
      "  83  84  85  86  87  88  89  90  91  92  93  94  95  96  97  98  99 100\n",
      " 101 102 103 104 105 106 107 108 109 110 111 112 113 114 115 116 117 118\n",
      " 119 120 121 122 123 124 125 126 127 128 129 130 131 132 133 134 135 136\n",
      " 137 138 139 140 141 142 143 144 145 146 147 148 149 150 151 152 153 154\n",
      " 155 156 157 158 159 160 161 162 163 164 165 166 167 168 169 170 171 172\n",
      " 173 174 175 176 177 178 179 180 181 182 183 184 185 186 187 188 189 190\n",
      " 191 192 193 194 195 196 197 198 199 200 201 202 203 204 205 206 207 208\n",
      " 209 210 211 212 213 214 215 216 217 218 219 220 221 222 223 224 225 226\n",
      " 227 228 229 230 231 232 233 234 235 236 237 238 239 240 241 242 243 244\n",
      " 245 246 247 248 249 250 251 252 253 254 255]\n",
      "[  0  10  12  14  15  16  17  18  19  20  21  22  23  24  25  26  27  28\n",
      "  29  30  31  32  33  34  35  36  37  38  39  40  41  42  43  44  45  46\n",
      "  47  48  49  50  51  52  53  54  55  56  57  58  59  60  61  62  63  64\n",
      "  65  66  67  68  69  70  71  72  73  74  75  76  77  78  79  80  81  82\n",
      "  83  84  85  86  87  88  89  90  91  92  93  94  95  96  97  98  99 100\n",
      " 101 102 103 104 105 106 107 108 109 110 111 112 113 114 115 116 117 118\n",
      " 119 120 121 122 123 124 125 126 127 128 129 130 131 132 133 134 135 136\n",
      " 137 138 139 140 141 142 143 144 145 146 147 148 149 150 151 152 153 154\n",
      " 155 156 157 158 159 160 161 162 163 164 165 166 167 168 169 170 171 172\n",
      " 173 174 175 176 177 178 179 180 181 182 183 184 185 186 187 188 189 190\n",
      " 191 192 193 194 195 196 197 198 199 200 201 202 203 204 205 206 207 208\n",
      " 209 210 211 212 213 214 215 216 217 218 219 220 221 222 223 224 225 226\n",
      " 227 228 229 230 231 232 233 234 235 236 237 238 239 240 241 242 243 244\n",
      " 245 246 247 248 249 250 251 252 253 254 255]\n",
      "[  0   7   8   9  10  11  12  13  14  15  16  17  18  19  20  21  22  23\n",
      "  24  25  26  27  28  29  30  31  32  33  34  35  36  37  38  39  40  41\n",
      "  42  43  44  45  46  47  48  49  50  51  52  53  54  55  56  57  58  59\n",
      "  60  61  62  63  64  65  66  67  68  69  70  71  72  73  74  75  76  77\n",
      "  78  79  80  81  82  83  84  85  86  87  88  89  90  91  92  93  94  95\n",
      "  96  97  98  99 100 101 102 103 104 105 106 107 108 109 110 111 112 113\n",
      " 114 115 116 117 118 119 120 121 122 123 124 125 126 127 128 129 130 131\n",
      " 132 133 134 135 136 137 138 139 140 141 142 143 144 145 146 147 148 149\n",
      " 150 151 152 153 154 155 156 157 158 159 160 161 162 163 164 165 166 167\n",
      " 168 169 170 171 172 173 174 175 176 177 178 179 180 181 182 183 184 185\n",
      " 186 187 188 189 190 191 192 193 194 195 196 197 198 199 200 201 202 203\n",
      " 204 205 206 207 208 209 210 211 212 213 214 215 216 217 218 219 220 221\n",
      " 222 223 224 225 226 227 228 229 230 231 232 233 234 235 236 237 238 239\n",
      " 240 241 242 243 244 245 246 247 248 249 250 251 252 253 254 255]\n",
      "[  8   9  10  11  12  13  14  15  16  17  18  19  20  21  22  23  24  25\n",
      "  26  27  28  29  30  31  32  33  34  35  36  37  38  39  40  41  42  43\n",
      "  44  45  46  47  48  49  50  51  52  53  54  55  56  57  58  59  60  61\n",
      "  62  63  64  65  66  67  68  69  70  71  72  73  74  75  76  77  78  79\n",
      "  80  81  82  83  84  85  86  87  88  89  90  91  92  93  94  95  96  97\n",
      "  98  99 100 101 102 103 104 105 106 107 108 109 110 111 112 113 114 115\n",
      " 116 117 118 119 120 121 122 123 124 125 126 127 128 129 130 131 132 133\n",
      " 134 135 136 137 138 139 140 141 142 143 144 145 146 147 148 149 150 151\n",
      " 152 153 154 155 156 157 158 159 160 161 162 163 164 165 166 167 168 169\n",
      " 170 171 172 173 174 175 176 177 178 179 180 181 182 183 184 185 186 187\n",
      " 188 189 190 191 192 193 194 195 196 197 198 199 200 201 202 203 204 205\n",
      " 206 207 208 209 210 211 212 213 214 215 216 217 218 219 220 221 222 223\n",
      " 224 225 226 227 228 229 230 231 232 233 234 235 236 237 238 239 240 241\n",
      " 242 243 244 245 246 247 248 249 250 251 252 253 254 255]\n",
      "[  0   8   9  10  11  12  13  14  15  16  17  18  19  20  21  22  23  24\n",
      "  25  26  27  28  29  30  31  32  33  34  35  36  37  38  39  40  41  42\n",
      "  43  44  45  46  47  48  49  50  51  52  53  54  55  56  57  58  59  60\n",
      "  61  62  63  64  65  66  67  68  69  70  71  72  73  74  75  76  77  78\n",
      "  79  80  81  82  83  84  85  86  87  88  89  90  91  92  93  94  95  96\n",
      "  97  98  99 100 101 102 103 104 105 106 107 108 109 110 111 112 113 114\n",
      " 115 116 117 118 119 120 121 122 123 124 125 126 127 128 129 130 131 132\n",
      " 133 134 135 136 137 138 139 140 141 142 143 144 145 146 147 148 149 150\n",
      " 151 152 153 154 155 156 157 158 159 160 161 162 163 164 165 166 167 168\n",
      " 169 170 171 172 173 174 175 176 177 178 179 180 181 182 183 184 185 186\n",
      " 187 188 189 190 191 192 193 194 195 196 197 198 199 200 201 202 203 204\n",
      " 205 206 207 208 209 210 211 212 213 214 215 216 217 218 219 220 221 222\n",
      " 223 224 225 226 227 228 229 230 231 232 233 234 235 236 237 238 239 240\n",
      " 241 242 243 244 245 246 247 248 249 250 251 252 253 254 255]\n",
      "[ 11  14  15  16  17  18  19  20  21  22  23  24  25  26  27  28  29  30\n",
      "  31  32  33  34  35  36  37  38  39  40  41  42  43  44  45  46  47  48\n",
      "  49  50  51  52  53  54  55  56  57  58  59  60  61  62  63  64  65  66\n",
      "  67  68  69  70  71  72  73  74  75  76  77  78  79  80  81  82  83  84\n",
      "  85  86  87  88  89  90  91  92  93  94  95  96  97  98  99 100 101 102\n",
      " 103 104 105 106 107 108 109 110 111 112 113 114 115 116 117 118 119 120\n",
      " 121 122 123 124 125 126 127 128 129 130 131 132 133 134 135 136 137 138\n",
      " 139 140 141 142 143 144 145 146 147 148 149 150 151 152 153 154 155 156\n",
      " 157 158 159 160 161 162 163 164 165 166 167 168 169 170 171 172 173 174\n",
      " 175 176 177 178 179 180 181 182 183 184 185 186 187 188 189 190 191 192\n",
      " 193 194 195 196 197 198 199 200 201 202 203 204 205 206 207 208 209 210\n",
      " 211 212 213 214 215 216 217 218 219 220 221 222 223 224 225 226 227 228\n",
      " 229 230 231 232 233 234 235 236 237 238 239 240 241 242 243 244 245 246\n",
      " 247 248 249 250 251 252 253 254 255]\n",
      "[ 17  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34\n",
      "  35  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52\n",
      "  53  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70\n",
      "  71  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88\n",
      "  89  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106\n",
      " 107 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124\n",
      " 125 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142\n",
      " 143 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160\n",
      " 161 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178\n",
      " 179 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196\n",
      " 197 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214\n",
      " 215 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232\n",
      " 233 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250\n",
      " 251 252 253 254 255]\n",
      "[  0  17  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33\n",
      "  34  35  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51\n",
      "  52  53  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69\n",
      "  70  71  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87\n",
      "  88  89  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105\n",
      " 106 107 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123\n",
      " 124 125 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141\n",
      " 142 143 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159\n",
      " 160 161 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177\n",
      " 178 179 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195\n",
      " 196 197 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213\n",
      " 214 215 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231\n",
      " 232 233 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249\n",
      " 250 251 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  2   5   6  11  13  14  15  16  17  18  19  20  21  22  23  24  25  26\n",
      "  27  28  29  30  31  32  33  34  35  36  37  38  39  40  41  42  43  44\n",
      "  45  46  47  48  49  50  51  52  53  54  55  56  57  58  59  60  61  62\n",
      "  63  64  65  66  67  68  69  70  71  72  73  74  75  76  77  78  79  80\n",
      "  81  82  83  84  85  86  87  88  89  90  91  92  93  94  95  96  97  98\n",
      "  99 100 101 102 103 104 105 106 107 108 109 110 111 112 113 114 115 116\n",
      " 117 118 119 120 121 122 123 124 125 126 127 128 129 130 131 132 133 134\n",
      " 135 136 137 138 139 140 141 142 143 144 145 146 147 148 149 150 151 152\n",
      " 153 154 155 156 157 158 159 160 161 162 163 164 165 166 167 168 169 170\n",
      " 171 172 173 174 175 176 177 178 179 180 181 182 183 184 185 186 187 188\n",
      " 189 190 191 192 193 194 195 196 197 198 199 200 201 202 203 204 205 206\n",
      " 207 208 209 210 211 212 213 214 215 216 217 218 219 220 221 222 223 224\n",
      " 225 226 227 228 229 230 231 232 233 234 235 236 237 238 239 240 241 242\n",
      " 243 244 245 246 247 248 249 250 251 252 253 254 255]\n",
      "[  0   2   5   6  11  13  14  15  16  17  18  19  20  21  22  23  24  25\n",
      "  26  27  28  29  30  31  32  33  34  35  36  37  38  39  40  41  42  43\n",
      "  44  45  46  47  48  49  50  51  52  53  54  55  56  57  58  59  60  61\n",
      "  62  63  64  65  66  67  68  69  70  71  72  73  74  75  76  77  78  79\n",
      "  80  81  82  83  84  85  86  87  88  89  90  91  92  93  94  95  96  97\n",
      "  98  99 100 101 102 103 104 105 106 107 108 109 110 111 112 113 114 115\n",
      " 116 117 118 119 120 121 122 123 124 125 126 127 128 129 130 131 132 133\n",
      " 134 135 136 137 138 139 140 141 142 143 144 145 146 147 148 149 150 151\n",
      " 152 153 154 155 156 157 158 159 160 161 162 163 164 165 166 167 168 169\n",
      " 170 171 172 173 174 175 176 177 178 179 180 181 182 183 184 185 186 187\n",
      " 188 189 190 191 192 193 194 195 196 197 198 199 200 201 202 203 204 205\n",
      " 206 207 208 209 210 211 212 213 214 215 216 217 218 219 220 221 222 223\n",
      " 224 225 226 227 228 229 230 231 232 233 234 235 236 237 238 239 240 241\n",
      " 242 243 244 245 246 247 248 249 250 251 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17  18  19\n",
      "  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35  36  37\n",
      "  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53  54  55\n",
      "  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71  72  73\n",
      "  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89  90  91\n",
      "  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107 108 109\n",
      " 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125 126 127\n",
      " 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143 144 145\n",
      " 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161 162 163\n",
      " 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179 180 181\n",
      " 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197 198 199\n",
      " 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215 216 217\n",
      " 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233 234 235\n",
      " 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251 252 253\n",
      " 254 255]\n",
      "[  0   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17  18\n",
      "  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35  36\n",
      "  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53  54\n",
      "  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71  72\n",
      "  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89  90\n",
      "  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107 108\n",
      " 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125 126\n",
      " 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143 144\n",
      " 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161 162\n",
      " 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179 180\n",
      " 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197 198\n",
      " 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215 216\n",
      " 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233 234\n",
      " 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251 252\n",
      " 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17  18\n",
      "  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35  36\n",
      "  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53  54\n",
      "  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71  72\n",
      "  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89  90\n",
      "  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107 108\n",
      " 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125 126\n",
      " 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143 144\n",
      " 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161 162\n",
      " 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179 180\n",
      " 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197 198\n",
      " 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215 216\n",
      " 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233 234\n",
      " 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251 252\n",
      " 253 254 255]\n",
      "[  0   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17  18\n",
      "  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35  36\n",
      "  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53  54\n",
      "  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71  72\n",
      "  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89  90\n",
      "  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107 108\n",
      " 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125 126\n",
      " 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143 144\n",
      " 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161 162\n",
      " 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179 180\n",
      " 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197 198\n",
      " 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215 216\n",
      " 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233 234\n",
      " 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251 252\n",
      " 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  4   5   6   7   8   9  10  11  12  13  14  15  16  17  18  19  20  21\n",
      "  22  23  24  25  26  27  28  29  30  31  32  33  34  35  36  37  38  39\n",
      "  40  41  42  43  44  45  46  47  48  49  50  51  52  53  54  55  56  57\n",
      "  58  59  60  61  62  63  64  65  66  67  68  69  70  71  72  73  74  75\n",
      "  76  77  78  79  80  81  82  83  84  85  86  87  88  89  90  91  92  93\n",
      "  94  95  96  97  98  99 100 101 102 103 104 105 106 107 108 109 110 111\n",
      " 112 113 114 115 116 117 118 119 120 121 122 123 124 125 126 127 128 129\n",
      " 130 131 132 133 134 135 136 137 138 139 140 141 142 143 144 145 146 147\n",
      " 148 149 150 151 152 153 154 155 156 157 158 159 160 161 162 163 164 165\n",
      " 166 167 168 169 170 171 172 173 174 175 176 177 178 179 180 181 182 183\n",
      " 184 185 186 187 188 189 190 191 192 193 194 195 196 197 198 199 200 201\n",
      " 202 203 204 205 206 207 208 209 210 211 212 213 214 215 216 217 218 219\n",
      " 220 221 222 223 224 225 226 227 228 229 230 231 232 233 234 235 236 237\n",
      " 238 239 240 241 242 243 244 245 246 247 248 249 250 251 252 253 254 255]\n",
      "[  0   4   5   6   7   8   9  10  11  12  13  14  15  16  17  18  19  20\n",
      "  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35  36  37  38\n",
      "  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53  54  55  56\n",
      "  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71  72  73  74\n",
      "  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89  90  91  92\n",
      "  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107 108 109 110\n",
      " 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125 126 127 128\n",
      " 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143 144 145 146\n",
      " 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161 162 163 164\n",
      " 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179 180 181 182\n",
      " 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197 198 199 200\n",
      " 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215 216 217 218\n",
      " 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233 234 235 236\n",
      " 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251 252 253 254\n",
      " 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  2   4   8   9  10  11  12  13  14  15  16  17  18  19  20  21  22  23\n",
      "  24  25  26  27  28  29  30  31  32  33  34  35  36  37  38  39  40  41\n",
      "  42  43  44  45  46  47  48  49  50  51  52  53  54  55  56  57  58  59\n",
      "  60  61  62  63  64  65  66  67  68  69  70  71  72  73  74  75  76  77\n",
      "  78  79  80  81  82  83  84  85  86  87  88  89  90  91  92  93  94  95\n",
      "  96  97  98  99 100 101 102 103 104 105 106 107 108 109 110 111 112 113\n",
      " 114 115 116 117 118 119 120 121 122 123 124 125 126 127 128 129 130 131\n",
      " 132 133 134 135 136 137 138 139 140 141 142 143 144 145 146 147 148 149\n",
      " 150 151 152 153 154 155 156 157 158 159 160 161 162 163 164 165 166 167\n",
      " 168 169 170 171 172 173 174 175 176 177 178 179 180 181 182 183 184 185\n",
      " 186 187 188 189 190 191 192 193 194 195 196 197 198 199 200 201 202 203\n",
      " 204 205 206 207 208 209 210 211 212 213 214 215 216 217 218 219 220 221\n",
      " 222 223 224 225 226 227 228 229 230 231 232 233 234 235 236 237 238 239\n",
      " 240 241 242 243 244 245 246 247 248 249 250 251 252 253 254 255]\n",
      "[  0   2   4   8   9  10  11  12  13  14  15  16  17  18  19  20  21  22\n",
      "  23  24  25  26  27  28  29  30  31  32  33  34  35  36  37  38  39  40\n",
      "  41  42  43  44  45  46  47  48  49  50  51  52  53  54  55  56  57  58\n",
      "  59  60  61  62  63  64  65  66  67  68  69  70  71  72  73  74  75  76\n",
      "  77  78  79  80  81  82  83  84  85  86  87  88  89  90  91  92  93  94\n",
      "  95  96  97  98  99 100 101 102 103 104 105 106 107 108 109 110 111 112\n",
      " 113 114 115 116 117 118 119 120 121 122 123 124 125 126 127 128 129 130\n",
      " 131 132 133 134 135 136 137 138 139 140 141 142 143 144 145 146 147 148\n",
      " 149 150 151 152 153 154 155 156 157 158 159 160 161 162 163 164 165 166\n",
      " 167 168 169 170 171 172 173 174 175 176 177 178 179 180 181 182 183 184\n",
      " 185 186 187 188 189 190 191 192 193 194 195 196 197 198 199 200 201 202\n",
      " 203 204 205 206 207 208 209 210 211 212 213 214 215 216 217 218 219 220\n",
      " 221 222 223 224 225 226 227 228 229 230 231 232 233 234 235 236 237 238\n",
      " 239 240 241 242 243 244 245 246 247 248 249 250 251 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  5   6   7   8   9  10  11  12  13  14  15  16  17  18  19  20  21  22\n",
      "  23  24  25  26  27  28  29  30  31  32  33  34  35  36  37  38  39  40\n",
      "  41  42  43  44  45  46  47  48  49  50  51  52  53  54  55  56  57  58\n",
      "  59  60  61  62  63  64  65  66  67  68  69  70  71  72  73  74  75  76\n",
      "  77  78  79  80  81  82  83  84  85  86  87  88  89  90  91  92  93  94\n",
      "  95  96  97  98  99 100 101 102 103 104 105 106 107 108 109 110 111 112\n",
      " 113 114 115 116 117 118 119 120 121 122 123 124 125 126 127 128 129 130\n",
      " 131 132 133 134 135 136 137 138 139 140 141 142 143 144 145 146 147 148\n",
      " 149 150 151 152 153 154 155 156 157 158 159 160 161 162 163 164 165 166\n",
      " 167 168 169 170 171 172 173 174 175 176 177 178 179 180 181 182 183 184\n",
      " 185 186 187 188 189 190 191 192 193 194 195 196 197 198 199 200 201 202\n",
      " 203 204 205 206 207 208 209 210 211 212 213 214 215 216 217 218 219 220\n",
      " 221 222 223 224 225 226 227 228 229 230 231 232 233 234 235 236 237 238\n",
      " 239 240 241 242 243 244 245 246 247 248 249 250 251 252 253 254 255]\n",
      "[  0   5   6   7   8   9  10  11  12  13  14  15  16  17  18  19  20  21\n",
      "  22  23  24  25  26  27  28  29  30  31  32  33  34  35  36  37  38  39\n",
      "  40  41  42  43  44  45  46  47  48  49  50  51  52  53  54  55  56  57\n",
      "  58  59  60  61  62  63  64  65  66  67  68  69  70  71  72  73  74  75\n",
      "  76  77  78  79  80  81  82  83  84  85  86  87  88  89  90  91  92  93\n",
      "  94  95  96  97  98  99 100 101 102 103 104 105 106 107 108 109 110 111\n",
      " 112 113 114 115 116 117 118 119 120 121 122 123 124 125 126 127 128 129\n",
      " 130 131 132 133 134 135 136 137 138 139 140 141 142 143 144 145 146 147\n",
      " 148 149 150 151 152 153 154 155 156 157 158 159 160 161 162 163 164 165\n",
      " 166 167 168 169 170 171 172 173 174 175 176 177 178 179 180 181 182 183\n",
      " 184 185 186 187 188 189 190 191 192 193 194 195 196 197 198 199 200 201\n",
      " 202 203 204 205 206 207 208 209 210 211 212 213 214 215 216 217 218 219\n",
      " 220 221 222 223 224 225 226 227 228 229 230 231 232 233 234 235 236 237\n",
      " 238 239 240 241 242 243 244 245 246 247 248 249 250 251 252 253 254 255]\n",
      "[  0   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17  18  19\n",
      "  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35  36  37\n",
      "  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53  54  55\n",
      "  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71  72  73\n",
      "  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89  90  91\n",
      "  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107 108 109\n",
      " 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125 126 127\n",
      " 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143 144 145\n",
      " 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161 162 163\n",
      " 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179 180 181\n",
      " 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197 198 199\n",
      " 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215 216 217\n",
      " 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233 234 235\n",
      " 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251 252 253\n",
      " 254 255]\n",
      "[  6   8  10  11  14  15  16  17  18  19  20  21  22  23  24  25  26  27\n",
      "  28  29  30  31  32  33  34  35  36  37  38  39  40  41  42  43  44  45\n",
      "  46  47  48  49  50  51  52  53  54  55  56  57  58  59  60  61  62  63\n",
      "  64  65  66  67  68  69  70  71  72  73  74  75  76  77  78  79  80  81\n",
      "  82  83  84  85  86  87  88  89  90  91  92  93  94  95  96  97  98  99\n",
      " 100 101 102 103 104 105 106 107 108 109 110 111 112 113 114 115 116 117\n",
      " 118 119 120 121 122 123 124 125 126 127 128 129 130 131 132 133 134 135\n",
      " 136 137 138 139 140 141 142 143 144 145 146 147 148 149 150 151 152 153\n",
      " 154 155 156 157 158 159 160 161 162 163 164 165 166 167 168 169 170 171\n",
      " 172 173 174 175 176 177 178 179 180 181 182 183 184 185 186 187 188 189\n",
      " 190 191 192 193 194 195 196 197 198 199 200 201 202 203 204 205 206 207\n",
      " 208 209 210 211 212 213 214 215 216 217 218 219 220 221 222 223 224 225\n",
      " 226 227 228 229 230 231 232 233 234 235 236 237 238 239 240 241 242 243\n",
      " 244 245 246 247 248 249 250 251 252 253 254 255]\n",
      "[  0  14  15  16  17  18  19  20  21  22  23  24  25  26  27  28  29  30\n",
      "  31  32  33  34  35  36  37  38  39  40  41  42  43  44  45  46  47  48\n",
      "  49  50  51  52  53  54  55  56  57  58  59  60  61  62  63  64  65  66\n",
      "  67  68  69  70  71  72  73  74  75  76  77  78  79  80  81  82  83  84\n",
      "  85  86  87  88  89  90  91  92  93  94  95  96  97  98  99 100 101 102\n",
      " 103 104 105 106 107 108 109 110 111 112 113 114 115 116 117 118 119 120\n",
      " 121 122 123 124 125 126 127 128 129 130 131 132 133 134 135 136 137 138\n",
      " 139 140 141 142 143 144 145 146 147 148 149 150 151 152 153 154 155 156\n",
      " 157 158 159 160 161 162 163 164 165 166 167 168 169 170 171 172 173 174\n",
      " 175 176 177 178 179 180 181 182 183 184 185 186 187 188 189 190 191 192\n",
      " 193 194 195 196 197 198 199 200 201 202 203 204 205 206 207 208 209 210\n",
      " 211 212 213 214 215 216 217 218 219 220 221 222 223 224 225 226 227 228\n",
      " 229 230 231 232 233 234 235 236 237 238 239 240 241 242 243 244 245 246\n",
      " 247 248 249 250 251 252 253 254 255]\n",
      "[  0   1   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17  18\n",
      "  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35  36\n",
      "  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53  54\n",
      "  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71  72\n",
      "  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89  90\n",
      "  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107 108\n",
      " 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125 126\n",
      " 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143 144\n",
      " 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161 162\n",
      " 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179 180\n",
      " 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197 198\n",
      " 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215 216\n",
      " 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233 234\n",
      " 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251 252\n",
      " 253 254 255]\n",
      "[  5   6   7   8   9  10  11  12  13  14  15  16  17  18  19  20  21  22\n",
      "  23  24  25  26  27  28  29  30  31  32  33  34  35  36  37  38  39  40\n",
      "  41  42  43  44  45  46  47  48  49  50  51  52  53  54  55  56  57  58\n",
      "  59  60  61  62  63  64  65  66  67  68  69  70  71  72  73  74  75  76\n",
      "  77  78  79  80  81  82  83  84  85  86  87  88  89  90  91  92  93  94\n",
      "  95  96  97  98  99 100 101 102 103 104 105 106 107 108 109 110 111 112\n",
      " 113 114 115 116 117 118 119 120 121 122 123 124 125 126 127 128 129 130\n",
      " 131 132 133 134 135 136 137 138 139 140 141 142 143 144 145 146 147 148\n",
      " 149 150 151 152 153 154 155 156 157 158 159 160 161 162 163 164 165 166\n",
      " 167 168 169 170 171 172 173 174 175 176 177 178 179 180 181 182 183 184\n",
      " 185 186 187 188 189 190 191 192 193 194 195 196 197 198 199 200 201 202\n",
      " 203 204 205 206 207 208 209 210 211 212 213 214 215 216 217 218 219 220\n",
      " 221 222 223 224 225 226 227 228 229 230 231 232 233 234 235 236 237 238\n",
      " 239 240 241 242 243 244 245 246 247 248 249 250 251 252 253 254 255]\n",
      "[  0  15  16  17  18  19  20  21  22  23  24  25  26  27  28  29  30  31\n",
      "  32  33  34  35  36  37  38  39  40  41  42  43  44  45  46  47  48  49\n",
      "  50  51  52  53  54  55  56  57  58  59  60  61  62  63  64  65  66  67\n",
      "  68  69  70  71  72  73  74  75  76  77  78  79  80  81  82  83  84  85\n",
      "  86  87  88  89  90  91  92  93  94  95  96  97  98  99 100 101 102 103\n",
      " 104 105 106 107 108 109 110 111 112 113 114 115 116 117 118 119 120 121\n",
      " 122 123 124 125 126 127 128 129 130 131 132 133 134 135 136 137 138 139\n",
      " 140 141 142 143 144 145 146 147 148 149 150 151 152 153 154 155 156 157\n",
      " 158 159 160 161 162 163 164 165 166 167 168 169 170 171 172 173 174 175\n",
      " 176 177 178 179 180 181 182 183 184 185 186 187 188 189 190 191 192 193\n",
      " 194 195 196 197 198 199 200 201 202 203 204 205 206 207 208 209 210 211\n",
      " 212 213 214 215 216 217 218 219 220 221 222 223 224 225 226 227 228 229\n",
      " 230 231 232 233 234 235 236 237 238 239 240 241 242 243 244 245 246 247\n",
      " 248 249 250 251 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  0   2   3   4   6   7   8   9  10  11  12  13  14  15  16  17  18  19\n",
      "  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35  36  37\n",
      "  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53  54  55\n",
      "  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71  72  73\n",
      "  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89  90  91\n",
      "  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107 108 109\n",
      " 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125 126 127\n",
      " 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143 144 145\n",
      " 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161 162 163\n",
      " 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179 180 181\n",
      " 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197 198 199\n",
      " 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215 216 217\n",
      " 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233 234 235\n",
      " 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251 252 253\n",
      " 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17  18\n",
      "  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35  36\n",
      "  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53  54\n",
      "  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71  72\n",
      "  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89  90\n",
      "  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107 108\n",
      " 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125 126\n",
      " 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143 144\n",
      " 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161 162\n",
      " 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179 180\n",
      " 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197 198\n",
      " 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215 216\n",
      " 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233 234\n",
      " 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251 252\n",
      " 253 254 255]\n",
      "[  0   6   7   8   9  10  11  12  13  14  15  16  17  18  19  20  21  22\n",
      "  23  24  25  26  27  28  29  30  31  32  33  34  35  36  37  38  39  40\n",
      "  41  42  43  44  45  46  47  48  49  50  51  52  53  54  55  56  57  58\n",
      "  59  60  61  62  63  64  65  66  67  68  69  70  71  72  73  74  75  76\n",
      "  77  78  79  80  81  82  83  84  85  86  87  88  89  90  91  92  93  94\n",
      "  95  96  97  98  99 100 101 102 103 104 105 106 107 108 109 110 111 112\n",
      " 113 114 115 116 117 118 119 120 121 122 123 124 125 126 127 128 129 130\n",
      " 131 132 133 134 135 136 137 138 139 140 141 142 143 144 145 146 147 148\n",
      " 149 150 151 152 153 154 155 156 157 158 159 160 161 162 163 164 165 166\n",
      " 167 168 169 170 171 172 173 174 175 176 177 178 179 180 181 182 183 184\n",
      " 185 186 187 188 189 190 191 192 193 194 195 196 197 198 199 200 201 202\n",
      " 203 204 205 206 207 208 209 210 211 212 213 214 215 216 217 218 219 220\n",
      " 221 222 223 224 225 226 227 228 229 230 231 232 233 234 235 236 237 238\n",
      " 239 240 241 242 243 244 245 246 247 248 249 250 251 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17  18  19\n",
      "  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35  36  37\n",
      "  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53  54  55\n",
      "  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71  72  73\n",
      "  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89  90  91\n",
      "  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107 108 109\n",
      " 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125 126 127\n",
      " 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143 144 145\n",
      " 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161 162 163\n",
      " 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179 180 181\n",
      " 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197 198 199\n",
      " 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215 216 217\n",
      " 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233 234 235\n",
      " 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251 252 253\n",
      " 254 255]\n",
      "[  0  10  12  13  14  15  16  17  18  19  20  21  22  23  24  25  26  27\n",
      "  28  29  30  31  32  33  34  35  36  37  38  39  40  41  42  43  44  45\n",
      "  46  47  48  49  50  51  52  53  54  55  56  57  58  59  60  61  62  63\n",
      "  64  65  66  67  68  69  70  71  72  73  74  75  76  77  78  79  80  81\n",
      "  82  83  84  85  86  87  88  89  90  91  92  93  94  95  96  97  98  99\n",
      " 100 101 102 103 104 105 106 107 108 109 110 111 112 113 114 115 116 117\n",
      " 118 119 120 121 122 123 124 125 126 127 128 129 130 131 132 133 134 135\n",
      " 136 137 138 139 140 141 142 143 144 145 146 147 148 149 150 151 152 153\n",
      " 154 155 156 157 158 159 160 161 162 163 164 165 166 167 168 169 170 171\n",
      " 172 173 174 175 176 177 178 179 180 181 182 183 184 185 186 187 188 189\n",
      " 190 191 192 193 194 195 196 197 198 199 200 201 202 203 204 205 206 207\n",
      " 208 209 210 211 212 213 214 215 216 217 218 219 220 221 222 223 224 225\n",
      " 226 227 228 229 230 231 232 233 234 235 236 237 238 239 240 241 242 243\n",
      " 244 245 246 247 248 249 250 251 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  0   5   6   8  10  11  12  13  14  15  16  17  18  19  20  21  22  23\n",
      "  24  25  26  27  28  29  30  31  32  33  34  35  36  37  38  39  40  41\n",
      "  42  43  44  45  46  47  48  49  50  51  52  53  54  55  56  57  58  59\n",
      "  60  61  62  63  64  65  66  67  68  69  70  71  72  73  74  75  76  77\n",
      "  78  79  80  81  82  83  84  85  86  87  88  89  90  91  92  93  94  95\n",
      "  96  97  98  99 100 101 102 103 104 105 106 107 108 109 110 111 112 113\n",
      " 114 115 116 117 118 119 120 121 122 123 124 125 126 127 128 129 130 131\n",
      " 132 133 134 135 136 137 138 139 140 141 142 143 144 145 146 147 148 149\n",
      " 150 151 152 153 154 155 156 157 158 159 160 161 162 163 164 165 166 167\n",
      " 168 169 170 171 172 173 174 175 176 177 178 179 180 181 182 183 184 185\n",
      " 186 187 188 189 190 191 192 193 194 195 196 197 198 199 200 201 202 203\n",
      " 204 205 206 207 208 209 210 211 212 213 214 215 216 217 218 219 220 221\n",
      " 222 223 224 225 226 227 228 229 230 231 232 233 234 235 236 237 238 239\n",
      " 240 241 242 243 244 245 246 247 248 249 250 251 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  4   5   6   7   9  10  11  12  13  14  15  16  17  18  19  20  21  22\n",
      "  23  24  25  26  27  28  29  30  31  32  33  34  35  36  37  38  39  40\n",
      "  41  42  43  44  45  46  47  48  49  50  51  52  53  54  55  56  57  58\n",
      "  59  60  61  62  63  64  65  66  67  68  69  70  71  72  73  74  75  76\n",
      "  77  78  79  80  81  82  83  84  85  86  87  88  89  90  91  92  93  94\n",
      "  95  96  97  98  99 100 101 102 103 104 105 106 107 108 109 110 111 112\n",
      " 113 114 115 116 117 118 119 120 121 122 123 124 125 126 127 128 129 130\n",
      " 131 132 133 134 135 136 137 138 139 140 141 142 143 144 145 146 147 148\n",
      " 149 150 151 152 153 154 155 156 157 158 159 160 161 162 163 164 165 166\n",
      " 167 168 169 170 171 172 173 174 175 176 177 178 179 180 181 182 183 184\n",
      " 185 186 187 188 189 190 191 192 193 194 195 196 197 198 199 200 201 202\n",
      " 203 204 205 206 207 208 209 210 211 212 213 214 215 216 217 218 219 220\n",
      " 221 222 223 224 225 226 227 228 229 230 231 232 233 234 235 236 237 238\n",
      " 239 240 241 242 243 244 245 246 247 248 249 250 251 252 253 254 255]\n",
      "[  0  12  15  16  17  18  19  20  21  22  23  24  25  26  27  28  29  30\n",
      "  31  32  33  34  35  36  37  38  39  40  41  42  43  44  45  46  47  48\n",
      "  49  50  51  52  53  54  55  56  57  58  59  60  61  62  63  64  65  66\n",
      "  67  68  69  70  71  72  73  74  75  76  77  78  79  80  81  82  83  84\n",
      "  85  86  87  88  89  90  91  92  93  94  95  96  97  98  99 100 101 102\n",
      " 103 104 105 106 107 108 109 110 111 112 113 114 115 116 117 118 119 120\n",
      " 121 122 123 124 125 126 127 128 129 130 131 132 133 134 135 136 137 138\n",
      " 139 140 141 142 143 144 145 146 147 148 149 150 151 152 153 154 155 156\n",
      " 157 158 159 160 161 162 163 164 165 166 167 168 169 170 171 172 173 174\n",
      " 175 176 177 178 179 180 181 182 183 184 185 186 187 188 189 190 191 192\n",
      " 193 194 195 196 197 198 199 200 201 202 203 204 205 206 207 208 209 210\n",
      " 211 212 213 214 215 216 217 218 219 220 221 222 223 224 225 226 227 228\n",
      " 229 230 231 232 233 234 235 236 237 238 239 240 241 242 243 244 245 246\n",
      " 247 248 249 250 251 252 253 254 255]\n",
      "[  9  10  11  12  13  14  15  16  17  18  19  20  21  22  23  24  25  26\n",
      "  27  28  29  30  31  32  33  34  35  36  37  38  39  40  41  42  43  44\n",
      "  45  46  47  48  49  50  51  52  53  54  55  56  57  58  59  60  61  62\n",
      "  63  64  65  66  67  68  69  70  71  72  73  74  75  76  77  78  79  80\n",
      "  81  82  83  84  85  86  87  88  89  90  91  92  93  94  95  96  97  98\n",
      "  99 100 101 102 103 104 105 106 107 108 109 110 111 112 113 114 115 116\n",
      " 117 118 119 120 121 122 123 124 125 126 127 128 129 130 131 132 133 134\n",
      " 135 136 137 138 139 140 141 142 143 144 145 146 147 148 149 150 151 152\n",
      " 153 154 155 156 157 158 159 160 161 162 163 164 165 166 167 168 169 170\n",
      " 171 172 173 174 175 176 177 178 179 180 181 182 183 184 185 186 187 188\n",
      " 189 190 191 192 193 194 195 196 197 198 199 200 201 202 203 204 205 206\n",
      " 207 208 209 210 211 212 213 214 215 216 217 218 219 220 221 222 223 224\n",
      " 225 226 227 228 229 230 231 232 233 234 235 236 237 238 239 240 241 242\n",
      " 243 244 245 246 247 248 249 250 251 252 253 254 255]\n",
      "[ 12  14  15  16  17  18  19  20  21  22  23  24  25  26  27  28  29  30\n",
      "  31  32  33  34  35  36  37  38  39  40  41  42  43  44  45  46  47  48\n",
      "  49  50  51  52  53  54  55  56  57  58  59  60  61  62  63  64  65  66\n",
      "  67  68  69  70  71  72  73  74  75  76  77  78  79  80  81  82  83  84\n",
      "  85  86  87  88  89  90  91  92  93  94  95  96  97  98  99 100 101 102\n",
      " 103 104 105 106 107 108 109 110 111 112 113 114 115 116 117 118 119 120\n",
      " 121 122 123 124 125 126 127 128 129 130 131 132 133 134 135 136 137 138\n",
      " 139 140 141 142 143 144 145 146 147 148 149 150 151 152 153 154 155 156\n",
      " 157 158 159 160 161 162 163 164 165 166 167 168 169 170 171 172 173 174\n",
      " 175 176 177 178 179 180 181 182 183 184 185 186 187 188 189 190 191 192\n",
      " 193 194 195 196 197 198 199 200 201 202 203 204 205 206 207 208 209 210\n",
      " 211 212 213 214 215 216 217 218 219 220 221 222 223 224 225 226 227 228\n",
      " 229 230 231 232 233 234 235 236 237 238 239 240 241 242 243 244 245 246\n",
      " 247 248 249 250 251 252 253 254 255]\n",
      "[  0  12  14  15  16  17  18  19  20  21  22  23  24  25  26  27  28  29\n",
      "  30  31  32  33  34  35  36  37  38  39  40  41  42  43  44  45  46  47\n",
      "  48  49  50  51  52  53  54  55  56  57  58  59  60  61  62  63  64  65\n",
      "  66  67  68  69  70  71  72  73  74  75  76  77  78  79  80  81  82  83\n",
      "  84  85  86  87  88  89  90  91  92  93  94  95  96  97  98  99 100 101\n",
      " 102 103 104 105 106 107 108 109 110 111 112 113 114 115 116 117 118 119\n",
      " 120 121 122 123 124 125 126 127 128 129 130 131 132 133 134 135 136 137\n",
      " 138 139 140 141 142 143 144 145 146 147 148 149 150 151 152 153 154 155\n",
      " 156 157 158 159 160 161 162 163 164 165 166 167 168 169 170 171 172 173\n",
      " 174 175 176 177 178 179 180 181 182 183 184 185 186 187 188 189 190 191\n",
      " 192 193 194 195 196 197 198 199 200 201 202 203 204 205 206 207 208 209\n",
      " 210 211 212 213 214 215 216 217 218 219 220 221 222 223 224 225 226 227\n",
      " 228 229 230 231 232 233 234 235 236 237 238 239 240 241 242 243 244 245\n",
      " 246 247 248 249 250 251 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  8   9  10  13  14  15  16  17  18  19  20  21  22  23  24  25  26  27\n",
      "  28  29  30  31  32  33  34  35  36  37  38  39  40  41  42  43  44  45\n",
      "  46  47  48  49  50  51  52  53  54  55  56  57  58  59  60  61  62  63\n",
      "  64  65  66  67  68  69  70  71  72  73  74  75  76  77  78  79  80  81\n",
      "  82  83  84  85  86  87  88  89  90  91  92  93  94  95  96  97  98  99\n",
      " 100 101 102 103 104 105 106 107 108 109 110 111 112 113 114 115 116 117\n",
      " 118 119 120 121 122 123 124 125 126 127 128 129 130 131 132 133 134 135\n",
      " 136 137 138 139 140 141 142 143 144 145 146 147 148 149 150 151 152 153\n",
      " 154 155 156 157 158 159 160 161 162 163 164 165 166 167 168 169 170 171\n",
      " 172 173 174 175 176 177 178 179 180 181 182 183 184 185 186 187 188 189\n",
      " 190 191 192 193 194 195 196 197 198 199 200 201 202 203 204 205 206 207\n",
      " 208 209 210 211 212 213 214 215 216 217 218 219 220 221 222 223 224 225\n",
      " 226 227 228 229 230 231 232 233 234 235 236 237 238 239 240 241 242 243\n",
      " 244 245 246 247 248 249 250 251 252 253 254 255]\n",
      "[  0  16  17  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32\n",
      "  33  34  35  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50\n",
      "  51  52  53  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68\n",
      "  69  70  71  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86\n",
      "  87  88  89  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104\n",
      " 105 106 107 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122\n",
      " 123 124 125 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140\n",
      " 141 142 143 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158\n",
      " 159 160 161 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176\n",
      " 177 178 179 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194\n",
      " 195 196 197 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212\n",
      " 213 214 215 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230\n",
      " 231 232 233 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248\n",
      " 249 250 251 252 253 254 255]\n",
      "[  4   7   8  10  11  12  13  14  15  16  17  18  19  20  21  22  23  24\n",
      "  25  26  27  28  29  30  31  32  33  34  35  36  37  38  39  40  41  42\n",
      "  43  44  45  46  47  48  49  50  51  52  53  54  55  56  57  58  59  60\n",
      "  61  62  63  64  65  66  67  68  69  70  71  72  73  74  75  76  77  78\n",
      "  79  80  81  82  83  84  85  86  87  88  89  90  91  92  93  94  95  96\n",
      "  97  98  99 100 101 102 103 104 105 106 107 108 109 110 111 112 113 114\n",
      " 115 116 117 118 119 120 121 122 123 124 125 126 127 128 129 130 131 132\n",
      " 133 134 135 136 137 138 139 140 141 142 143 144 145 146 147 148 149 150\n",
      " 151 152 153 154 155 156 157 158 159 160 161 162 163 164 165 166 167 168\n",
      " 169 170 171 172 173 174 175 176 177 178 179 180 181 182 183 184 185 186\n",
      " 187 188 189 190 191 192 193 194 195 196 197 198 199 200 201 202 203 204\n",
      " 205 206 207 208 209 210 211 212 213 214 215 216 217 218 219 220 221 222\n",
      " 223 224 225 226 227 228 229 230 231 232 233 234 235 236 237 238 239 240\n",
      " 241 242 243 244 245 246 247 248 249 250 251 252 253 254 255]\n",
      "[ 17  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34\n",
      "  35  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52\n",
      "  53  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70\n",
      "  71  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88\n",
      "  89  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106\n",
      " 107 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124\n",
      " 125 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142\n",
      " 143 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160\n",
      " 161 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178\n",
      " 179 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196\n",
      " 197 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214\n",
      " 215 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232\n",
      " 233 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250\n",
      " 251 252 253 254 255]\n",
      "[  0  17  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33\n",
      "  34  35  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51\n",
      "  52  53  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69\n",
      "  70  71  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87\n",
      "  88  89  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105\n",
      " 106 107 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123\n",
      " 124 125 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141\n",
      " 142 143 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159\n",
      " 160 161 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177\n",
      " 178 179 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195\n",
      " 196 197 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213\n",
      " 214 215 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231\n",
      " 232 233 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249\n",
      " 250 251 252 253 254 255]\n",
      "[  0   1   2   3   4   5   7   8   9  10  11  12  13  14  15  16  17  18\n",
      "  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35  36\n",
      "  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53  54\n",
      "  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71  72\n",
      "  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89  90\n",
      "  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107 108\n",
      " 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125 126\n",
      " 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143 144\n",
      " 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161 162\n",
      " 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179 180\n",
      " 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197 198\n",
      " 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215 216\n",
      " 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233 234\n",
      " 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251 252\n",
      " 253 254 255]\n",
      "[ 10  12  13  14  15  16  17  18  19  20  21  22  23  24  25  26  27  28\n",
      "  29  30  31  32  33  34  35  36  37  38  39  40  41  42  43  44  45  46\n",
      "  47  48  49  50  51  52  53  54  55  56  57  58  59  60  61  62  63  64\n",
      "  65  66  67  68  69  70  71  72  73  74  75  76  77  78  79  80  81  82\n",
      "  83  84  85  86  87  88  89  90  91  92  93  94  95  96  97  98  99 100\n",
      " 101 102 103 104 105 106 107 108 109 110 111 112 113 114 115 116 117 118\n",
      " 119 120 121 122 123 124 125 126 127 128 129 130 131 132 133 134 135 136\n",
      " 137 138 139 140 141 142 143 144 145 146 147 148 149 150 151 152 153 154\n",
      " 155 156 157 158 159 160 161 162 163 164 165 166 167 168 169 170 171 172\n",
      " 173 174 175 176 177 178 179 180 181 182 183 184 185 186 187 188 189 190\n",
      " 191 192 193 194 195 196 197 198 199 200 201 202 203 204 205 206 207 208\n",
      " 209 210 211 212 213 214 215 216 217 218 219 220 221 222 223 224 225 226\n",
      " 227 228 229 230 231 232 233 234 235 236 237 238 239 240 241 242 243 244\n",
      " 245 246 247 248 249 250 251 252 253 254 255]\n",
      "[  0  12  13  14  15  16  17  18  19  20  21  22  23  24  25  26  27  28\n",
      "  29  30  31  32  33  34  35  36  37  38  39  40  41  42  43  44  45  46\n",
      "  47  48  49  50  51  52  53  54  55  56  57  58  59  60  61  62  63  64\n",
      "  65  66  67  68  69  70  71  72  73  74  75  76  77  78  79  80  81  82\n",
      "  83  84  85  86  87  88  89  90  91  92  93  94  95  96  97  98  99 100\n",
      " 101 102 103 104 105 106 107 108 109 110 111 112 113 114 115 116 117 118\n",
      " 119 120 121 122 123 124 125 126 127 128 129 130 131 132 133 134 135 136\n",
      " 137 138 139 140 141 142 143 144 145 146 147 148 149 150 151 152 153 154\n",
      " 155 156 157 158 159 160 161 162 163 164 165 166 167 168 169 170 171 172\n",
      " 173 174 175 176 177 178 179 180 181 182 183 184 185 186 187 188 189 190\n",
      " 191 192 193 194 195 196 197 198 199 200 201 202 203 204 205 206 207 208\n",
      " 209 210 211 212 213 214 215 216 217 218 219 220 221 222 223 224 225 226\n",
      " 227 228 229 230 231 232 233 234 235 236 237 238 239 240 241 242 243 244\n",
      " 245 246 247 248 249 250 251 252 253 254 255]\n",
      "[  1   2   4   6   7   8   9  10  11  12  13  14  15  16  17  18  19  20\n",
      "  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35  36  37  38\n",
      "  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53  54  55  56\n",
      "  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71  72  73  74\n",
      "  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89  90  91  92\n",
      "  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107 108 109 110\n",
      " 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125 126 127 128\n",
      " 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143 144 145 146\n",
      " 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161 162 163 164\n",
      " 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179 180 181 182\n",
      " 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197 198 199 200\n",
      " 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215 216 217 218\n",
      " 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233 234 235 236\n",
      " 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251 252 253 254\n",
      " 255]\n",
      "[  5   8   9  10  11  12  13  14  15  16  17  18  19  20  21  22  23  24\n",
      "  25  26  27  28  29  30  31  32  33  34  35  36  37  38  39  40  41  42\n",
      "  43  44  45  46  47  48  49  50  51  52  53  54  55  56  57  58  59  60\n",
      "  61  62  63  64  65  66  67  68  69  70  71  72  73  74  75  76  77  78\n",
      "  79  80  81  82  83  84  85  86  87  88  89  90  91  92  93  94  95  96\n",
      "  97  98  99 100 101 102 103 104 105 106 107 108 109 110 111 112 113 114\n",
      " 115 116 117 118 119 120 121 122 123 124 125 126 127 128 129 130 131 132\n",
      " 133 134 135 136 137 138 139 140 141 142 143 144 145 146 147 148 149 150\n",
      " 151 152 153 154 155 156 157 158 159 160 161 162 163 164 165 166 167 168\n",
      " 169 170 171 172 173 174 175 176 177 178 179 180 181 182 183 184 185 186\n",
      " 187 188 189 190 191 192 193 194 195 196 197 198 199 200 201 202 203 204\n",
      " 205 206 207 208 209 210 211 212 213 214 215 216 217 218 219 220 221 222\n",
      " 223 224 225 226 227 228 229 230 231 232 233 234 235 236 237 238 239 240\n",
      " 241 242 243 244 245 246 247 248 249 250 251 252 253 254 255]\n",
      "[  0   5   8   9  10  11  12  13  14  15  16  17  18  19  20  21  22  23\n",
      "  24  25  26  27  28  29  30  31  32  33  34  35  36  37  38  39  40  41\n",
      "  42  43  44  45  46  47  48  49  50  51  52  53  54  55  56  57  58  59\n",
      "  60  61  62  63  64  65  66  67  68  69  70  71  72  73  74  75  76  77\n",
      "  78  79  80  81  82  83  84  85  86  87  88  89  90  91  92  93  94  95\n",
      "  96  97  98  99 100 101 102 103 104 105 106 107 108 109 110 111 112 113\n",
      " 114 115 116 117 118 119 120 121 122 123 124 125 126 127 128 129 130 131\n",
      " 132 133 134 135 136 137 138 139 140 141 142 143 144 145 146 147 148 149\n",
      " 150 151 152 153 154 155 156 157 158 159 160 161 162 163 164 165 166 167\n",
      " 168 169 170 171 172 173 174 175 176 177 178 179 180 181 182 183 184 185\n",
      " 186 187 188 189 190 191 192 193 194 195 196 197 198 199 200 201 202 203\n",
      " 204 205 206 207 208 209 210 211 212 213 214 215 216 217 218 219 220 221\n",
      " 222 223 224 225 226 227 228 229 230 231 232 233 234 235 236 237 238 239\n",
      " 240 241 242 243 244 245 246 247 248 249 250 251 252 253 254 255]\n",
      "[  7   8   9  10  11  12  13  14  15  16  17  18  19  20  21  22  23  24\n",
      "  25  26  27  28  29  30  31  32  33  34  35  36  37  38  39  40  41  42\n",
      "  43  44  45  46  47  48  49  50  51  52  53  54  55  56  57  58  59  60\n",
      "  61  62  63  64  65  66  67  68  69  70  71  72  73  74  75  76  77  78\n",
      "  79  80  81  82  83  84  85  86  87  88  89  90  91  92  93  94  95  96\n",
      "  97  98  99 100 101 102 103 104 105 106 107 108 109 110 111 112 113 114\n",
      " 115 116 117 118 119 120 121 122 123 124 125 126 127 128 129 130 131 132\n",
      " 133 134 135 136 137 138 139 140 141 142 143 144 145 146 147 148 149 150\n",
      " 151 152 153 154 155 156 157 158 159 160 161 162 163 164 165 166 167 168\n",
      " 169 170 171 172 173 174 175 176 177 178 179 180 181 182 183 184 185 186\n",
      " 187 188 189 190 191 192 193 194 195 196 197 198 199 200 201 202 203 204\n",
      " 205 206 207 208 209 210 211 212 213 214 215 216 217 218 219 220 221 222\n",
      " 223 224 225 226 227 228 229 230 231 232 233 234 235 236 237 238 239 240\n",
      " 241 242 243 244 245 246 247 248 249 250 251 252 253 254 255]\n",
      "[ 10  11  12  13  14  15  16  17  18  19  20  21  22  23  24  25  26  27\n",
      "  28  29  30  31  32  33  34  35  36  37  38  39  40  41  42  43  44  45\n",
      "  46  47  48  49  50  51  52  53  54  55  56  57  58  59  60  61  62  63\n",
      "  64  65  66  67  68  69  70  71  72  73  74  75  76  77  78  79  80  81\n",
      "  82  83  84  85  86  87  88  89  90  91  92  93  94  95  96  97  98  99\n",
      " 100 101 102 103 104 105 106 107 108 109 110 111 112 113 114 115 116 117\n",
      " 118 119 120 121 122 123 124 125 126 127 128 129 130 131 132 133 134 135\n",
      " 136 137 138 139 140 141 142 143 144 145 146 147 148 149 150 151 152 153\n",
      " 154 155 156 157 158 159 160 161 162 163 164 165 166 167 168 169 170 171\n",
      " 172 173 174 175 176 177 178 179 180 181 182 183 184 185 186 187 188 189\n",
      " 190 191 192 193 194 195 196 197 198 199 200 201 202 203 204 205 206 207\n",
      " 208 209 210 211 212 213 214 215 216 217 218 219 220 221 222 223 224 225\n",
      " 226 227 228 229 230 231 232 233 234 235 236 237 238 239 240 241 242 243\n",
      " 244 245 246 247 248 249 250 251 252 253 254 255]\n",
      "[  0  10  11  12  13  14  15  16  17  18  19  20  21  22  23  24  25  26\n",
      "  27  28  29  30  31  32  33  34  35  36  37  38  39  40  41  42  43  44\n",
      "  45  46  47  48  49  50  51  52  53  54  55  56  57  58  59  60  61  62\n",
      "  63  64  65  66  67  68  69  70  71  72  73  74  75  76  77  78  79  80\n",
      "  81  82  83  84  85  86  87  88  89  90  91  92  93  94  95  96  97  98\n",
      "  99 100 101 102 103 104 105 106 107 108 109 110 111 112 113 114 115 116\n",
      " 117 118 119 120 121 122 123 124 125 126 127 128 129 130 131 132 133 134\n",
      " 135 136 137 138 139 140 141 142 143 144 145 146 147 148 149 150 151 152\n",
      " 153 154 155 156 157 158 159 160 161 162 163 164 165 166 167 168 169 170\n",
      " 171 172 173 174 175 176 177 178 179 180 181 182 183 184 185 186 187 188\n",
      " 189 190 191 192 193 194 195 196 197 198 199 200 201 202 203 204 205 206\n",
      " 207 208 209 210 211 212 213 214 215 216 217 218 219 220 221 222 223 224\n",
      " 225 226 227 228 229 230 231 232 233 234 235 236 237 238 239 240 241 242\n",
      " 243 244 245 246 247 248 249 250 251 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  0   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17  18\n",
      "  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35  36\n",
      "  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53  54\n",
      "  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71  72\n",
      "  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89  90\n",
      "  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107 108\n",
      " 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125 126\n",
      " 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143 144\n",
      " 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161 162\n",
      " 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179 180\n",
      " 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197 198\n",
      " 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215 216\n",
      " 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233 234\n",
      " 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251 252\n",
      " 253 254 255]\n",
      "[  0  17  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35  36\n",
      "  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53  54\n",
      "  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71  72\n",
      "  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89  90\n",
      "  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107 108\n",
      " 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125 126\n",
      " 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143 144\n",
      " 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161 162\n",
      " 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179 180\n",
      " 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197 198\n",
      " 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215 216\n",
      " 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233 234\n",
      " 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251 252\n",
      " 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  0  13  15  16  17  18  19  20  21  22  23  24  25  26  27  28  29  30\n",
      "  31  32  33  34  35  36  37  38  39  40  41  42  43  44  45  46  47  48\n",
      "  49  50  51  52  53  54  55  56  57  58  59  60  61  62  63  64  65  66\n",
      "  67  68  69  70  71  72  73  74  75  76  77  78  79  80  81  82  83  84\n",
      "  85  86  87  88  89  90  91  92  93  94  95  96  97  98  99 100 101 102\n",
      " 103 104 105 106 107 108 109 110 111 112 113 114 115 116 117 118 119 120\n",
      " 121 122 123 124 125 126 127 128 129 130 131 132 133 134 135 136 137 138\n",
      " 139 140 141 142 143 144 145 146 147 148 149 150 151 152 153 154 155 156\n",
      " 157 158 159 160 161 162 163 164 165 166 167 168 169 170 171 172 173 174\n",
      " 175 176 177 178 179 180 181 182 183 184 185 186 187 188 189 190 191 192\n",
      " 193 194 195 196 197 198 199 200 201 202 203 204 205 206 207 208 209 210\n",
      " 211 212 213 214 215 216 217 218 219 220 221 222 223 224 225 226 227 228\n",
      " 229 230 231 232 233 234 235 236 237 238 239 240 241 242 243 244 245 246\n",
      " 247 248 249 250 251 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17  18\n",
      "  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35  36\n",
      "  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53  54\n",
      "  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71  72\n",
      "  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89  90\n",
      "  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107 108\n",
      " 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125 126\n",
      " 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143 144\n",
      " 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161 162\n",
      " 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179 180\n",
      " 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197 198\n",
      " 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215 216\n",
      " 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233 234\n",
      " 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251 252\n",
      " 253 254 255]\n",
      "[  0  11  14  15  16  17  18  19  20  21  22  23  24  25  26  27  28  29\n",
      "  30  31  32  33  34  35  36  37  38  39  40  41  42  43  44  45  46  47\n",
      "  48  49  50  51  52  53  54  55  56  57  58  59  60  61  62  63  64  65\n",
      "  66  67  68  69  70  71  72  73  74  75  76  77  78  79  80  81  82  83\n",
      "  84  85  86  87  88  89  90  91  92  93  94  95  96  97  98  99 100 101\n",
      " 102 103 104 105 106 107 108 109 110 111 112 113 114 115 116 117 118 119\n",
      " 120 121 122 123 124 125 126 127 128 129 130 131 132 133 134 135 136 137\n",
      " 138 139 140 141 142 143 144 145 146 147 148 149 150 151 152 153 154 155\n",
      " 156 157 158 159 160 161 162 163 164 165 166 167 168 169 170 171 172 173\n",
      " 174 175 176 177 178 179 180 181 182 183 184 185 186 187 188 189 190 191\n",
      " 192 193 194 195 196 197 198 199 200 201 202 203 204 205 206 207 208 209\n",
      " 210 211 212 213 214 215 216 217 218 219 220 221 222 223 224 225 226 227\n",
      " 228 229 230 231 232 233 234 235 236 237 238 239 240 241 242 243 244 245\n",
      " 246 247 248 249 250 251 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  0   7  12  13  14  15  16  17  18  19  20  21  22  23  24  25  26  27\n",
      "  28  29  30  31  32  33  34  35  36  37  38  39  40  41  42  43  44  45\n",
      "  46  47  48  49  50  51  52  53  54  55  56  57  58  59  60  61  62  63\n",
      "  64  65  66  67  68  69  70  71  72  73  74  75  76  77  78  79  80  81\n",
      "  82  83  84  85  86  87  88  89  90  91  92  93  94  95  96  97  98  99\n",
      " 100 101 102 103 104 105 106 107 108 109 110 111 112 113 114 115 116 117\n",
      " 118 119 120 121 122 123 124 125 126 127 128 129 130 131 132 133 134 135\n",
      " 136 137 138 139 140 141 142 143 144 145 146 147 148 149 150 151 152 153\n",
      " 154 155 156 157 158 159 160 161 162 163 164 165 166 167 168 169 170 171\n",
      " 172 173 174 175 176 177 178 179 180 181 182 183 184 185 186 187 188 189\n",
      " 190 191 192 193 194 195 196 197 198 199 200 201 202 203 204 205 206 207\n",
      " 208 209 210 211 212 213 214 215 216 217 218 219 220 221 222 223 224 225\n",
      " 226 227 228 229 230 231 232 233 234 235 236 237 238 239 240 241 242 243\n",
      " 244 245 246 247 248 249 250 251 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17  18\n",
      "  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35  36\n",
      "  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53  54\n",
      "  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71  72\n",
      "  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89  90\n",
      "  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107 108\n",
      " 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125 126\n",
      " 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143 144\n",
      " 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161 162\n",
      " 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179 180\n",
      " 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197 198\n",
      " 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215 216\n",
      " 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233 234\n",
      " 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251 252\n",
      " 253 254 255]\n",
      "[  0   3   5   6   7   8   9  10  11  12  13  14  15  16  17  18  19  20\n",
      "  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35  36  37  38\n",
      "  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53  54  55  56\n",
      "  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71  72  73  74\n",
      "  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89  90  91  92\n",
      "  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107 108 109 110\n",
      " 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125 126 127 128\n",
      " 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143 144 145 146\n",
      " 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161 162 163 164\n",
      " 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179 180 181 182\n",
      " 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197 198 199 200\n",
      " 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215 216 217 218\n",
      " 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233 234 235 236\n",
      " 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251 252 253 254\n",
      " 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  0   2   7  10  11  12  13  14  15  16  17  18  19  20  21  22  23  24\n",
      "  25  26  27  28  29  30  31  32  33  34  35  36  37  38  39  40  41  42\n",
      "  43  44  45  46  47  48  49  50  51  52  53  54  55  56  57  58  59  60\n",
      "  61  62  63  64  65  66  67  68  69  70  71  72  73  74  75  76  77  78\n",
      "  79  80  81  82  83  84  85  86  87  88  89  90  91  92  93  94  95  96\n",
      "  97  98  99 100 101 102 103 104 105 106 107 108 109 110 111 112 113 114\n",
      " 115 116 117 118 119 120 121 122 123 124 125 126 127 128 129 130 131 132\n",
      " 133 134 135 136 137 138 139 140 141 142 143 144 145 146 147 148 149 150\n",
      " 151 152 153 154 155 156 157 158 159 160 161 162 163 164 165 166 167 168\n",
      " 169 170 171 172 173 174 175 176 177 178 179 180 181 182 183 184 185 186\n",
      " 187 188 189 190 191 192 193 194 195 196 197 198 199 200 201 202 203 204\n",
      " 205 206 207 208 209 210 211 212 213 214 215 216 217 218 219 220 221 222\n",
      " 223 224 225 226 227 228 229 230 231 232 233 234 235 236 237 238 239 240\n",
      " 241 242 243 244 245 246 247 248 249 250 251 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  0   6  12  13  15  17  19  20  21  22  23  24  25  26  27  28  29  30\n",
      "  31  32  33  34  35  36  37  38  39  40  41  42  43  44  45  46  47  48\n",
      "  49  50  51  52  53  54  55  56  57  58  59  60  61  62  63  64  65  66\n",
      "  67  68  69  70  71  72  73  74  75  76  77  78  79  80  81  82  83  84\n",
      "  85  86  87  88  89  90  91  92  93  94  95  96  97  98  99 100 101 102\n",
      " 103 104 105 106 107 108 109 110 111 112 113 114 115 116 117 118 119 120\n",
      " 121 122 123 124 125 126 127 128 129 130 131 132 133 134 135 136 137 138\n",
      " 139 140 141 142 143 144 145 146 147 148 149 150 151 152 153 154 155 156\n",
      " 157 158 159 160 161 162 163 164 165 166 167 168 169 170 171 172 173 174\n",
      " 175 176 177 178 179 180 181 182 183 184 185 186 187 188 189 190 191 192\n",
      " 193 194 195 196 197 198 199 200 201 202 203 204 205 206 207 208 209 210\n",
      " 211 212 213 214 215 216 217 218 219 220 221 222 223 224 225 226 227 228\n",
      " 229 230 231 232 233 234 235 236 237 238 239 240 241 242 243 244 245 246\n",
      " 247 248 249 250 251 252 253 254 255]\n",
      "[  1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17  18\n",
      "  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35  36\n",
      "  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53  54\n",
      "  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71  72\n",
      "  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89  90\n",
      "  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107 108\n",
      " 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125 126\n",
      " 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143 144\n",
      " 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161 162\n",
      " 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179 180\n",
      " 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197 198\n",
      " 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215 216\n",
      " 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233 234\n",
      " 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251 252\n",
      " 253 254 255]\n",
      "[  3   4   5   6   7   8   9  10  11  12  13  14  15  16  17  18  19  20\n",
      "  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35  36  37  38\n",
      "  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53  54  55  56\n",
      "  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71  72  73  74\n",
      "  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89  90  91  92\n",
      "  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107 108 109 110\n",
      " 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125 126 127 128\n",
      " 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143 144 145 146\n",
      " 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161 162 163 164\n",
      " 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179 180 181 182\n",
      " 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197 198 199 200\n",
      " 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215 216 217 218\n",
      " 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233 234 235 236\n",
      " 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251 252 253 254\n",
      " 255]\n",
      "[  0   9  10  11  12  13  14  15  16  17  18  19  20  21  22  23  24  25\n",
      "  26  27  28  29  30  31  32  33  34  35  36  37  38  39  40  41  42  43\n",
      "  44  45  46  47  48  49  50  51  52  53  54  55  56  57  58  59  60  61\n",
      "  62  63  64  65  66  67  68  69  70  71  72  73  74  75  76  77  78  79\n",
      "  80  81  82  83  84  85  86  87  88  89  90  91  92  93  94  95  96  97\n",
      "  98  99 100 101 102 103 104 105 106 107 108 109 110 111 112 113 114 115\n",
      " 116 117 118 119 120 121 122 123 124 125 126 127 128 129 130 131 132 133\n",
      " 134 135 136 137 138 139 140 141 142 143 144 145 146 147 148 149 150 151\n",
      " 152 153 154 155 156 157 158 159 160 161 162 163 164 165 166 167 168 169\n",
      " 170 171 172 173 174 175 176 177 178 179 180 181 182 183 184 185 186 187\n",
      " 188 189 190 191 192 193 194 195 196 197 198 199 200 201 202 203 204 205\n",
      " 206 207 208 209 210 211 212 213 214 215 216 217 218 219 220 221 222 223\n",
      " 224 225 226 227 228 229 230 231 232 233 234 235 236 237 238 239 240 241\n",
      " 242 243 244 245 246 247 248 249 250 251 252 253 254 255]\n",
      "[  0   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17  18\n",
      "  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35  36\n",
      "  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53  54\n",
      "  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71  72\n",
      "  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89  90\n",
      "  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107 108\n",
      " 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125 126\n",
      " 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143 144\n",
      " 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161 162\n",
      " 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179 180\n",
      " 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197 198\n",
      " 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215 216\n",
      " 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233 234\n",
      " 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251 252\n",
      " 253 254 255]\n",
      "[  6   7   8   9  10  11  12  13  14  15  16  17  18  19  20  21  22  23\n",
      "  24  25  26  27  28  29  30  31  32  33  34  35  36  37  38  39  40  41\n",
      "  42  43  44  45  46  47  48  49  50  51  52  53  54  55  56  57  58  59\n",
      "  60  61  62  63  64  65  66  67  68  69  70  71  72  73  74  75  76  77\n",
      "  78  79  80  81  82  83  84  85  86  87  88  89  90  91  92  93  94  95\n",
      "  96  97  98  99 100 101 102 103 104 105 106 107 108 109 110 111 112 113\n",
      " 114 115 116 117 118 119 120 121 122 123 124 125 126 127 128 129 130 131\n",
      " 132 133 134 135 136 137 138 139 140 141 142 143 144 145 146 147 148 149\n",
      " 150 151 152 153 154 155 156 157 158 159 160 161 162 163 164 165 166 167\n",
      " 168 169 170 171 172 173 174 175 176 177 178 179 180 181 182 183 184 185\n",
      " 186 187 188 189 190 191 192 193 194 195 196 197 198 199 200 201 202 203\n",
      " 204 205 206 207 208 209 210 211 212 213 214 215 216 217 218 219 220 221\n",
      " 222 223 224 225 226 227 228 229 230 231 232 233 234 235 236 237 238 239\n",
      " 240 241 242 243 244 245 246 247 248 249 250 251 252 253 254 255]\n",
      "[  0   6   7   8   9  10  11  12  13  14  15  16  17  18  19  20  21  22\n",
      "  23  24  25  26  27  28  29  30  31  32  33  34  35  36  37  38  39  40\n",
      "  41  42  43  44  45  46  47  48  49  50  51  52  53  54  55  56  57  58\n",
      "  59  60  61  62  63  64  65  66  67  68  69  70  71  72  73  74  75  76\n",
      "  77  78  79  80  81  82  83  84  85  86  87  88  89  90  91  92  93  94\n",
      "  95  96  97  98  99 100 101 102 103 104 105 106 107 108 109 110 111 112\n",
      " 113 114 115 116 117 118 119 120 121 122 123 124 125 126 127 128 129 130\n",
      " 131 132 133 134 135 136 137 138 139 140 141 142 143 144 145 146 147 148\n",
      " 149 150 151 152 153 154 155 156 157 158 159 160 161 162 163 164 165 166\n",
      " 167 168 169 170 171 172 173 174 175 176 177 178 179 180 181 182 183 184\n",
      " 185 186 187 188 189 190 191 192 193 194 195 196 197 198 199 200 201 202\n",
      " 203 204 205 206 207 208 209 210 211 212 213 214 215 216 217 218 219 220\n",
      " 221 222 223 224 225 226 227 228 229 230 231 232 233 234 235 236 237 238\n",
      " 239 240 241 242 243 244 245 246 247 248 249 250 251 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  3   4   5   7   8   9  10  11  12  13  14  15  16  17  18  19  20  21\n",
      "  22  23  24  25  26  27  28  29  30  31  32  33  34  35  36  37  38  39\n",
      "  40  41  42  43  44  45  46  47  48  49  50  51  52  53  54  55  56  57\n",
      "  58  59  60  61  62  63  64  65  66  67  68  69  70  71  72  73  74  75\n",
      "  76  77  78  79  80  81  82  83  84  85  86  87  88  89  90  91  92  93\n",
      "  94  95  96  97  98  99 100 101 102 103 104 105 106 107 108 109 110 111\n",
      " 112 113 114 115 116 117 118 119 120 121 122 123 124 125 126 127 128 129\n",
      " 130 131 132 133 134 135 136 137 138 139 140 141 142 143 144 145 146 147\n",
      " 148 149 150 151 152 153 154 155 156 157 158 159 160 161 162 163 164 165\n",
      " 166 167 168 169 170 171 172 173 174 175 176 177 178 179 180 181 182 183\n",
      " 184 185 186 187 188 189 190 191 192 193 194 195 196 197 198 199 200 201\n",
      " 202 203 204 205 206 207 208 209 210 211 212 213 214 215 216 217 218 219\n",
      " 220 221 222 223 224 225 226 227 228 229 230 231 232 233 234 235 236 237\n",
      " 238 239 240 241 242 243 244 245 246 247 248 249 250 251 252 253 254 255]\n",
      "[  0   7   8   9  10  11  12  13  14  15  16  17  18  19  20  21  22  23\n",
      "  24  25  26  27  28  29  30  31  32  33  34  35  36  37  38  39  40  41\n",
      "  42  43  44  45  46  47  48  49  50  51  52  53  54  55  56  57  58  59\n",
      "  60  61  62  63  64  65  66  67  68  69  70  71  72  73  74  75  76  77\n",
      "  78  79  80  81  82  83  84  85  86  87  88  89  90  91  92  93  94  95\n",
      "  96  97  98  99 100 101 102 103 104 105 106 107 108 109 110 111 112 113\n",
      " 114 115 116 117 118 119 120 121 122 123 124 125 126 127 128 129 130 131\n",
      " 132 133 134 135 136 137 138 139 140 141 142 143 144 145 146 147 148 149\n",
      " 150 151 152 153 154 155 156 157 158 159 160 161 162 163 164 165 166 167\n",
      " 168 169 170 171 172 173 174 175 176 177 178 179 180 181 182 183 184 185\n",
      " 186 187 188 189 190 191 192 193 194 195 196 197 198 199 200 201 202 203\n",
      " 204 205 206 207 208 209 210 211 212 213 214 215 216 217 218 219 220 221\n",
      " 222 223 224 225 226 227 228 229 230 231 232 233 234 235 236 237 238 239\n",
      " 240 241 242 243 244 245 246 247 248 249 250 251 252 253 254 255]\n",
      "[ 10  11  12  13  14  15  16  17  18  19  20  21  22  23  24  25  26  27\n",
      "  28  29  30  31  32  33  34  35  36  37  38  39  40  41  42  43  44  45\n",
      "  46  47  48  49  50  51  52  53  54  55  56  57  58  59  60  61  62  63\n",
      "  64  65  66  67  68  69  70  71  72  73  74  75  76  77  78  79  80  81\n",
      "  82  83  84  85  86  87  88  89  90  91  92  93  94  95  96  97  98  99\n",
      " 100 101 102 103 104 105 106 107 108 109 110 111 112 113 114 115 116 117\n",
      " 118 119 120 121 122 123 124 125 126 127 128 129 130 131 132 133 134 135\n",
      " 136 137 138 139 140 141 142 143 144 145 146 147 148 149 150 151 152 153\n",
      " 154 155 156 157 158 159 160 161 162 163 164 165 166 167 168 169 170 171\n",
      " 172 173 174 175 176 177 178 179 180 181 182 183 184 185 186 187 188 189\n",
      " 190 191 192 193 194 195 196 197 198 199 200 201 202 203 204 205 206 207\n",
      " 208 209 210 211 212 213 214 215 216 217 218 219 220 221 222 223 224 225\n",
      " 226 227 228 229 230 231 232 233 234 235 236 237 238 239 240 241 242 243\n",
      " 244 245 246 247 248 249 250 251 252 253 254 255]\n",
      "[ 13  14  15  16  17  18  19  20  21  22  23  24  25  26  27  28  29  30\n",
      "  31  32  33  34  35  36  37  38  39  40  41  42  43  44  45  46  47  48\n",
      "  49  50  51  52  53  54  55  56  57  58  59  60  61  62  63  64  65  66\n",
      "  67  68  69  70  71  72  73  74  75  76  77  78  79  80  81  82  83  84\n",
      "  85  86  87  88  89  90  91  92  93  94  95  96  97  98  99 100 101 102\n",
      " 103 104 105 106 107 108 109 110 111 112 113 114 115 116 117 118 119 120\n",
      " 121 122 123 124 125 126 127 128 129 130 131 132 133 134 135 136 137 138\n",
      " 139 140 141 142 143 144 145 146 147 148 149 150 151 152 153 154 155 156\n",
      " 157 158 159 160 161 162 163 164 165 166 167 168 169 170 171 172 173 174\n",
      " 175 176 177 178 179 180 181 182 183 184 185 186 187 188 189 190 191 192\n",
      " 193 194 195 196 197 198 199 200 201 202 203 204 205 206 207 208 209 210\n",
      " 211 212 213 214 215 216 217 218 219 220 221 222 223 224 225 226 227 228\n",
      " 229 230 231 232 233 234 235 236 237 238 239 240 241 242 243 244 245 246\n",
      " 247 248 249 250 251 252 253 254 255]\n",
      "[  0  14  15  16  17  18  19  20  21  22  23  24  25  26  27  28  29  30\n",
      "  31  32  33  34  35  36  37  38  39  40  41  42  43  44  45  46  47  48\n",
      "  49  50  51  52  53  54  55  56  57  58  59  60  61  62  63  64  65  66\n",
      "  67  68  69  70  71  72  73  74  75  76  77  78  79  80  81  82  83  84\n",
      "  85  86  87  88  89  90  91  92  93  94  95  96  97  98  99 100 101 102\n",
      " 103 104 105 106 107 108 109 110 111 112 113 114 115 116 117 118 119 120\n",
      " 121 122 123 124 125 126 127 128 129 130 131 132 133 134 135 136 137 138\n",
      " 139 140 141 142 143 144 145 146 147 148 149 150 151 152 153 154 155 156\n",
      " 157 158 159 160 161 162 163 164 165 166 167 168 169 170 171 172 173 174\n",
      " 175 176 177 178 179 180 181 182 183 184 185 186 187 188 189 190 191 192\n",
      " 193 194 195 196 197 198 199 200 201 202 203 204 205 206 207 208 209 210\n",
      " 211 212 213 214 215 216 217 218 219 220 221 222 223 224 225 226 227 228\n",
      " 229 230 231 232 233 234 235 236 237 238 239 240 241 242 243 244 245 246\n",
      " 247 248 249 250 251 252 253 254 255]\n",
      "[  2   5   6   7   8   9  10  11  12  13  14  15  16  17  18  19  20  21\n",
      "  22  23  24  25  26  27  28  29  30  31  32  33  34  35  36  37  38  39\n",
      "  40  41  42  43  44  45  46  47  48  49  50  51  52  53  54  55  56  57\n",
      "  58  59  60  61  62  63  64  65  66  67  68  69  70  71  72  73  74  75\n",
      "  76  77  78  79  80  81  82  83  84  85  86  87  88  89  90  91  92  93\n",
      "  94  95  96  97  98  99 100 101 102 103 104 105 106 107 108 109 110 111\n",
      " 112 113 114 115 116 117 118 119 120 121 122 123 124 125 126 127 128 129\n",
      " 130 131 132 133 134 135 136 137 138 139 140 141 142 143 144 145 146 147\n",
      " 148 149 150 151 152 153 154 155 156 157 158 159 160 161 162 163 164 165\n",
      " 166 167 168 169 170 171 172 173 174 175 176 177 178 179 180 181 182 183\n",
      " 184 185 186 187 188 189 190 191 192 193 194 195 196 197 198 199 200 201\n",
      " 202 203 204 205 206 207 208 209 210 211 212 213 214 215 216 217 218 219\n",
      " 220 221 222 223 224 225 226 227 228 229 230 231 232 233 234 235 236 237\n",
      " 238 239 240 241 242 243 244 245 246 247 248 249 250 251 252 253 254 255]\n",
      "[  6   7   8  10  11  12  13  14  15  16  17  18  19  20  21  22  23  24\n",
      "  25  26  27  28  29  30  31  32  33  34  35  36  37  38  39  40  41  42\n",
      "  43  44  45  46  47  48  49  50  51  52  53  54  55  56  57  58  59  60\n",
      "  61  62  63  64  65  66  67  68  69  70  71  72  73  74  75  76  77  78\n",
      "  79  80  81  82  83  84  85  86  87  88  89  90  91  92  93  94  95  96\n",
      "  97  98  99 100 101 102 103 104 105 106 107 108 109 110 111 112 113 114\n",
      " 115 116 117 118 119 120 121 122 123 124 125 126 127 128 129 130 131 132\n",
      " 133 134 135 136 137 138 139 140 141 142 143 144 145 146 147 148 149 150\n",
      " 151 152 153 154 155 156 157 158 159 160 161 162 163 164 165 166 167 168\n",
      " 169 170 171 172 173 174 175 176 177 178 179 180 181 182 183 184 185 186\n",
      " 187 188 189 190 191 192 193 194 195 196 197 198 199 200 201 202 203 204\n",
      " 205 206 207 208 209 210 211 212 213 214 215 216 217 218 219 220 221 222\n",
      " 223 224 225 226 227 228 229 230 231 232 233 234 235 236 237 238 239 240\n",
      " 241 242 243 244 245 246 247 248 249 250 251 252 253 254 255]\n",
      "[  0   7  10  11  12  13  14  15  16  17  18  19  20  21  22  23  24  25\n",
      "  26  27  28  29  30  31  32  33  34  35  36  37  38  39  40  41  42  43\n",
      "  44  45  46  47  48  49  50  51  52  53  54  55  56  57  58  59  60  61\n",
      "  62  63  64  65  66  67  68  69  70  71  72  73  74  75  76  77  78  79\n",
      "  80  81  82  83  84  85  86  87  88  89  90  91  92  93  94  95  96  97\n",
      "  98  99 100 101 102 103 104 105 106 107 108 109 110 111 112 113 114 115\n",
      " 116 117 118 119 120 121 122 123 124 125 126 127 128 129 130 131 132 133\n",
      " 134 135 136 137 138 139 140 141 142 143 144 145 146 147 148 149 150 151\n",
      " 152 153 154 155 156 157 158 159 160 161 162 163 164 165 166 167 168 169\n",
      " 170 171 172 173 174 175 176 177 178 179 180 181 182 183 184 185 186 187\n",
      " 188 189 190 191 192 193 194 195 196 197 198 199 200 201 202 203 204 205\n",
      " 206 207 208 209 210 211 212 213 214 215 216 217 218 219 220 221 222 223\n",
      " 224 225 226 227 228 229 230 231 232 233 234 235 236 237 238 239 240 241\n",
      " 242 243 244 245 246 247 248 249 250 251 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  8  11  12  13  14  15  16  17  18  19  20  21  22  23  24  25  26  27\n",
      "  28  29  30  31  32  33  34  35  36  37  38  39  40  41  42  43  44  45\n",
      "  46  47  48  49  50  51  52  53  54  55  56  57  58  59  60  61  62  63\n",
      "  64  65  66  67  68  69  70  71  72  73  74  75  76  77  78  79  80  81\n",
      "  82  83  84  85  86  87  88  89  90  91  92  93  94  95  96  97  98  99\n",
      " 100 101 102 103 104 105 106 107 108 109 110 111 112 113 114 115 116 117\n",
      " 118 119 120 121 122 123 124 125 126 127 128 129 130 131 132 133 134 135\n",
      " 136 137 138 139 140 141 142 143 144 145 146 147 148 149 150 151 152 153\n",
      " 154 155 156 157 158 159 160 161 162 163 164 165 166 167 168 169 170 171\n",
      " 172 173 174 175 176 177 178 179 180 181 182 183 184 185 186 187 188 189\n",
      " 190 191 192 193 194 195 196 197 198 199 200 201 202 203 204 205 206 207\n",
      " 208 209 210 211 212 213 214 215 216 217 218 219 220 221 222 223 224 225\n",
      " 226 227 228 229 230 231 232 233 234 235 236 237 238 239 240 241 242 243\n",
      " 244 245 246 247 248 249 250 251 252 253 254 255]\n",
      "[  0   8  11  12  13  14  15  16  17  18  19  20  21  22  23  24  25  26\n",
      "  27  28  29  30  31  32  33  34  35  36  37  38  39  40  41  42  43  44\n",
      "  45  46  47  48  49  50  51  52  53  54  55  56  57  58  59  60  61  62\n",
      "  63  64  65  66  67  68  69  70  71  72  73  74  75  76  77  78  79  80\n",
      "  81  82  83  84  85  86  87  88  89  90  91  92  93  94  95  96  97  98\n",
      "  99 100 101 102 103 104 105 106 107 108 109 110 111 112 113 114 115 116\n",
      " 117 118 119 120 121 122 123 124 125 126 127 128 129 130 131 132 133 134\n",
      " 135 136 137 138 139 140 141 142 143 144 145 146 147 148 149 150 151 152\n",
      " 153 154 155 156 157 158 159 160 161 162 163 164 165 166 167 168 169 170\n",
      " 171 172 173 174 175 176 177 178 179 180 181 182 183 184 185 186 187 188\n",
      " 189 190 191 192 193 194 195 196 197 198 199 200 201 202 203 204 205 206\n",
      " 207 208 209 210 211 212 213 214 215 216 217 218 219 220 221 222 223 224\n",
      " 225 226 227 228 229 230 231 232 233 234 235 236 237 238 239 240 241 242\n",
      " 243 244 245 246 247 248 249 250 251 252 253 254 255]\n",
      "[  0   8  12  13  16  17  18  19  20  21  22  23  24  25  26  27  28  29\n",
      "  30  31  32  33  34  35  36  37  38  39  40  41  42  43  44  45  46  47\n",
      "  48  49  50  51  52  53  54  55  56  57  58  59  60  61  62  63  64  65\n",
      "  66  67  68  69  70  71  72  73  74  75  76  77  78  79  80  81  82  83\n",
      "  84  85  86  87  88  89  90  91  92  93  94  95  96  97  98  99 100 101\n",
      " 102 103 104 105 106 107 108 109 110 111 112 113 114 115 116 117 118 119\n",
      " 120 121 122 123 124 125 126 127 128 129 130 131 132 133 134 135 136 137\n",
      " 138 139 140 141 142 143 144 145 146 147 148 149 150 151 152 153 154 155\n",
      " 156 157 158 159 160 161 162 163 164 165 166 167 168 169 170 171 172 173\n",
      " 174 175 176 177 178 179 180 181 182 183 184 185 186 187 188 189 190 191\n",
      " 192 193 194 195 196 197 198 199 200 201 202 203 204 205 206 207 208 209\n",
      " 210 211 212 213 214 215 216 217 218 219 220 221 222 223 224 225 226 227\n",
      " 228 229 230 231 232 233 234 235 236 237 238 239 240 241 242 243 244 245\n",
      " 246 247 248 249 250 251 252 253 254 255]\n",
      "[ 18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  0  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34\n",
      "  35  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52\n",
      "  53  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70\n",
      "  71  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88\n",
      "  89  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106\n",
      " 107 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124\n",
      " 125 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142\n",
      " 143 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160\n",
      " 161 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178\n",
      " 179 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196\n",
      " 197 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214\n",
      " 215 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232\n",
      " 233 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250\n",
      " 251 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  3   4   5   6   7   8   9  10  11  12  13  14  15  16  17  18  19  20\n",
      "  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35  36  37  38\n",
      "  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53  54  55  56\n",
      "  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71  72  73  74\n",
      "  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89  90  91  92\n",
      "  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107 108 109 110\n",
      " 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125 126 127 128\n",
      " 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143 144 145 146\n",
      " 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161 162 163 164\n",
      " 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179 180 181 182\n",
      " 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197 198 199 200\n",
      " 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215 216 217 218\n",
      " 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233 234 235 236\n",
      " 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251 252 253 254\n",
      " 255]\n",
      "[  0   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17  18  19\n",
      "  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35  36  37\n",
      "  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53  54  55\n",
      "  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71  72  73\n",
      "  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89  90  91\n",
      "  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107 108 109\n",
      " 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125 126 127\n",
      " 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143 144 145\n",
      " 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161 162 163\n",
      " 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179 180 181\n",
      " 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197 198 199\n",
      " 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215 216 217\n",
      " 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233 234 235\n",
      " 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251 252 253\n",
      " 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  2   4   6   7   8   9  10  11  12  13  14  15  16  17  18  19  20  21\n",
      "  22  23  24  25  26  27  28  29  30  31  32  33  34  35  36  37  38  39\n",
      "  40  41  42  43  44  45  46  47  48  49  50  51  52  53  54  55  56  57\n",
      "  58  59  60  61  62  63  64  65  66  67  68  69  70  71  72  73  74  75\n",
      "  76  77  78  79  80  81  82  83  84  85  86  87  88  89  90  91  92  93\n",
      "  94  95  96  97  98  99 100 101 102 103 104 105 106 107 108 109 110 111\n",
      " 112 113 114 115 116 117 118 119 120 121 122 123 124 125 126 127 128 129\n",
      " 130 131 132 133 134 135 136 137 138 139 140 141 142 143 144 145 146 147\n",
      " 148 149 150 151 152 153 154 155 156 157 158 159 160 161 162 163 164 165\n",
      " 166 167 168 169 170 171 172 173 174 175 176 177 178 179 180 181 182 183\n",
      " 184 185 186 187 188 189 190 191 192 193 194 195 196 197 198 199 200 201\n",
      " 202 203 204 205 206 207 208 209 210 211 212 213 214 215 216 217 218 219\n",
      " 220 221 222 223 224 225 226 227 228 229 230 231 232 233 234 235 236 237\n",
      " 238 239 240 241 242 243 244 245 246 247 248 249 250 251 252 253 254 255]\n",
      "[  0   2   4   6   7   8   9  10  11  12  13  14  15  16  17  18  19  20\n",
      "  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35  36  37  38\n",
      "  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53  54  55  56\n",
      "  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71  72  73  74\n",
      "  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89  90  91  92\n",
      "  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107 108 109 110\n",
      " 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125 126 127 128\n",
      " 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143 144 145 146\n",
      " 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161 162 163 164\n",
      " 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179 180 181 182\n",
      " 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197 198 199 200\n",
      " 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215 216 217 218\n",
      " 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233 234 235 236\n",
      " 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251 252 253 254\n",
      " 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  2   4   5   6   7   8   9  10  11  12  13  14  15  16  17  18  19  20\n",
      "  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35  36  37  38\n",
      "  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53  54  55  56\n",
      "  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71  72  73  74\n",
      "  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89  90  91  92\n",
      "  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107 108 109 110\n",
      " 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125 126 127 128\n",
      " 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143 144 145 146\n",
      " 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161 162 163 164\n",
      " 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179 180 181 182\n",
      " 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197 198 199 200\n",
      " 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215 216 217 218\n",
      " 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233 234 235 236\n",
      " 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251 252 253 254\n",
      " 255]\n",
      "[  0   2   4   5   6   7   8   9  10  11  12  13  14  15  16  17  18  19\n",
      "  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35  36  37\n",
      "  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53  54  55\n",
      "  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71  72  73\n",
      "  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89  90  91\n",
      "  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107 108 109\n",
      " 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125 126 127\n",
      " 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143 144 145\n",
      " 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161 162 163\n",
      " 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179 180 181\n",
      " 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197 198 199\n",
      " 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215 216 217\n",
      " 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233 234 235\n",
      " 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251 252 253\n",
      " 254 255]\n",
      "[  0   2   5   6   7   8   9  10  11  12  13  14  15  16  17  18  19  20\n",
      "  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35  36  37  38\n",
      "  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53  54  55  56\n",
      "  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71  72  73  74\n",
      "  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89  90  91  92\n",
      "  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107 108 109 110\n",
      " 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125 126 127 128\n",
      " 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143 144 145 146\n",
      " 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161 162 163 164\n",
      " 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179 180 181 182\n",
      " 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197 198 199 200\n",
      " 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215 216 217 218\n",
      " 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233 234 235 236\n",
      " 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251 252 253 254\n",
      " 255]\n",
      "[  8  10  11  12  13  14  15  16  17  18  19  20  21  22  23  24  25  26\n",
      "  27  28  29  30  31  32  33  34  35  36  37  38  39  40  41  42  43  44\n",
      "  45  46  47  48  49  50  51  52  53  54  55  56  57  58  59  60  61  62\n",
      "  63  64  65  66  67  68  69  70  71  72  73  74  75  76  77  78  79  80\n",
      "  81  82  83  84  85  86  87  88  89  90  91  92  93  94  95  96  97  98\n",
      "  99 100 101 102 103 104 105 106 107 108 109 110 111 112 113 114 115 116\n",
      " 117 118 119 120 121 122 123 124 125 126 127 128 129 130 131 132 133 134\n",
      " 135 136 137 138 139 140 141 142 143 144 145 146 147 148 149 150 151 152\n",
      " 153 154 155 156 157 158 159 160 161 162 163 164 165 166 167 168 169 170\n",
      " 171 172 173 174 175 176 177 178 179 180 181 182 183 184 185 186 187 188\n",
      " 189 190 191 192 193 194 195 196 197 198 199 200 201 202 203 204 205 206\n",
      " 207 208 209 210 211 212 213 214 215 216 217 218 219 220 221 222 223 224\n",
      " 225 226 227 228 229 230 231 232 233 234 235 236 237 238 239 240 241 242\n",
      " 243 244 245 246 247 248 249 250 251 252 253 254 255]\n",
      "[  0   8  10  11  12  13  14  15  16  17  18  19  20  21  22  23  24  25\n",
      "  26  27  28  29  30  31  32  33  34  35  36  37  38  39  40  41  42  43\n",
      "  44  45  46  47  48  49  50  51  52  53  54  55  56  57  58  59  60  61\n",
      "  62  63  64  65  66  67  68  69  70  71  72  73  74  75  76  77  78  79\n",
      "  80  81  82  83  84  85  86  87  88  89  90  91  92  93  94  95  96  97\n",
      "  98  99 100 101 102 103 104 105 106 107 108 109 110 111 112 113 114 115\n",
      " 116 117 118 119 120 121 122 123 124 125 126 127 128 129 130 131 132 133\n",
      " 134 135 136 137 138 139 140 141 142 143 144 145 146 147 148 149 150 151\n",
      " 152 153 154 155 156 157 158 159 160 161 162 163 164 165 166 167 168 169\n",
      " 170 171 172 173 174 175 176 177 178 179 180 181 182 183 184 185 186 187\n",
      " 188 189 190 191 192 193 194 195 196 197 198 199 200 201 202 203 204 205\n",
      " 206 207 208 209 210 211 212 213 214 215 216 217 218 219 220 221 222 223\n",
      " 224 225 226 227 228 229 230 231 232 233 234 235 236 237 238 239 240 241\n",
      " 242 243 244 245 246 247 248 249 250 251 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17  18\n",
      "  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35  36\n",
      "  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53  54\n",
      "  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71  72\n",
      "  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89  90\n",
      "  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107 108\n",
      " 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125 126\n",
      " 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143 144\n",
      " 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161 162\n",
      " 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179 180\n",
      " 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197 198\n",
      " 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215 216\n",
      " 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233 234\n",
      " 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251 252\n",
      " 253 254 255]\n",
      "[  0   6   7   8   9  10  11  12  13  14  15  16  17  18  19  20  21  22\n",
      "  23  24  25  26  27  28  29  30  31  32  33  34  35  36  37  38  39  40\n",
      "  41  42  43  44  45  46  47  48  49  50  51  52  53  54  55  56  57  58\n",
      "  59  60  61  62  63  64  65  66  67  68  69  70  71  72  73  74  75  76\n",
      "  77  78  79  80  81  82  83  84  85  86  87  88  89  90  91  92  93  94\n",
      "  95  96  97  98  99 100 101 102 103 104 105 106 107 108 109 110 111 112\n",
      " 113 114 115 116 117 118 119 120 121 122 123 124 125 126 127 128 129 130\n",
      " 131 132 133 134 135 136 137 138 139 140 141 142 143 144 145 146 147 148\n",
      " 149 150 151 152 153 154 155 156 157 158 159 160 161 162 163 164 165 166\n",
      " 167 168 169 170 171 172 173 174 175 176 177 178 179 180 181 182 183 184\n",
      " 185 186 187 188 189 190 191 192 193 194 195 196 197 198 199 200 201 202\n",
      " 203 204 205 206 207 208 209 210 211 212 213 214 215 216 217 218 219 220\n",
      " 221 222 223 224 225 226 227 228 229 230 231 232 233 234 235 236 237 238\n",
      " 239 240 241 242 243 244 245 246 247 248 249 250 251 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  0  12  13  14  15  16  17  18  19  20  21  22  23  24  25  26  27  28\n",
      "  29  30  31  32  33  34  35  36  37  38  39  40  41  42  43  44  45  46\n",
      "  47  48  49  50  51  52  53  54  55  56  57  58  59  60  61  62  63  64\n",
      "  65  66  67  68  69  70  71  72  73  74  75  76  77  78  79  80  81  82\n",
      "  83  84  85  86  87  88  89  90  91  92  93  94  95  96  97  98  99 100\n",
      " 101 102 103 104 105 106 107 108 109 110 111 112 113 114 115 116 117 118\n",
      " 119 120 121 122 123 124 125 126 127 128 129 130 131 132 133 134 135 136\n",
      " 137 138 139 140 141 142 143 144 145 146 147 148 149 150 151 152 153 154\n",
      " 155 156 157 158 159 160 161 162 163 164 165 166 167 168 169 170 171 172\n",
      " 173 174 175 176 177 178 179 180 181 182 183 184 185 186 187 188 189 190\n",
      " 191 192 193 194 195 196 197 198 199 200 201 202 203 204 205 206 207 208\n",
      " 209 210 211 212 213 214 215 216 217 218 219 220 221 222 223 224 225 226\n",
      " 227 228 229 230 231 232 233 234 235 236 237 238 239 240 241 242 243 244\n",
      " 245 246 247 248 249 250 251 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  0  14  16  17  18  19  20  21  22  23  24  25  26  27  28  29  30  31\n",
      "  32  33  34  35  36  37  38  39  40  41  42  43  44  45  46  47  48  49\n",
      "  50  51  52  53  54  55  56  57  58  59  60  61  62  63  64  65  66  67\n",
      "  68  69  70  71  72  73  74  75  76  77  78  79  80  81  82  83  84  85\n",
      "  86  87  88  89  90  91  92  93  94  95  96  97  98  99 100 101 102 103\n",
      " 104 105 106 107 108 109 110 111 112 113 114 115 116 117 118 119 120 121\n",
      " 122 123 124 125 126 127 128 129 130 131 132 133 134 135 136 137 138 139\n",
      " 140 141 142 143 144 145 146 147 148 149 150 151 152 153 154 155 156 157\n",
      " 158 159 160 161 162 163 164 165 166 167 168 169 170 171 172 173 174 175\n",
      " 176 177 178 179 180 181 182 183 184 185 186 187 188 189 190 191 192 193\n",
      " 194 195 196 197 198 199 200 201 202 203 204 205 206 207 208 209 210 211\n",
      " 212 213 214 215 216 217 218 219 220 221 222 223 224 225 226 227 228 229\n",
      " 230 231 232 233 234 235 236 237 238 239 240 241 242 243 244 245 246 247\n",
      " 248 249 250 251 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  0  10  12  13  14  15  16  17  18  19  20  21  22  23  24  25  26  27\n",
      "  28  29  30  31  32  33  34  35  36  37  38  39  40  41  42  43  44  45\n",
      "  46  47  48  49  50  51  52  53  54  55  56  57  58  59  60  61  62  63\n",
      "  64  65  66  67  68  69  70  71  72  73  74  75  76  77  78  79  80  81\n",
      "  82  83  84  85  86  87  88  89  90  91  92  93  94  95  96  97  98  99\n",
      " 100 101 102 103 104 105 106 107 108 109 110 111 112 113 114 115 116 117\n",
      " 118 119 120 121 122 123 124 125 126 127 128 129 130 131 132 133 134 135\n",
      " 136 137 138 139 140 141 142 143 144 145 146 147 148 149 150 151 152 153\n",
      " 154 155 156 157 158 159 160 161 162 163 164 165 166 167 168 169 170 171\n",
      " 172 173 174 175 176 177 178 179 180 181 182 183 184 185 186 187 188 189\n",
      " 190 191 192 193 194 195 196 197 198 199 200 201 202 203 204 205 206 207\n",
      " 208 209 210 211 212 213 214 215 216 217 218 219 220 221 222 223 224 225\n",
      " 226 227 228 229 230 231 232 233 234 235 236 237 238 239 240 241 242 243\n",
      " 244 245 246 247 248 249 250 251 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  0  11  12  13  14  15  16  17  18  19  20  21  22  23  24  25  26  27\n",
      "  28  29  30  31  32  33  34  35  36  37  38  39  40  41  42  43  44  45\n",
      "  46  47  48  49  50  51  52  53  54  55  56  57  58  59  60  61  62  63\n",
      "  64  65  66  67  68  69  70  71  72  73  74  75  76  77  78  79  80  81\n",
      "  82  83  84  85  86  87  88  89  90  91  92  93  94  95  96  97  98  99\n",
      " 100 101 102 103 104 105 106 107 108 109 110 111 112 113 114 115 116 117\n",
      " 118 119 120 121 122 123 124 125 126 127 128 129 130 131 132 133 134 135\n",
      " 136 137 138 139 140 141 142 143 144 145 146 147 148 149 150 151 152 153\n",
      " 154 155 156 157 158 159 160 161 162 163 164 165 166 167 168 169 170 171\n",
      " 172 173 174 175 176 177 178 179 180 181 182 183 184 185 186 187 188 189\n",
      " 190 191 192 193 194 195 196 197 198 199 200 201 202 203 204 205 206 207\n",
      " 208 209 210 211 212 213 214 215 216 217 218 219 220 221 222 223 224 225\n",
      " 226 227 228 229 230 231 232 233 234 235 236 237 238 239 240 241 242 243\n",
      " 244 245 246 247 248 249 250 251 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  0   8  13  15  16  17  18  19  20  21  22  23  24  25  26  27  28  29\n",
      "  30  31  32  33  34  35  36  37  38  39  40  41  42  43  44  45  46  47\n",
      "  48  49  50  51  52  53  54  55  56  57  58  59  60  61  62  63  64  65\n",
      "  66  67  68  69  70  71  72  73  74  75  76  77  78  79  80  81  82  83\n",
      "  84  85  86  87  88  89  90  91  92  93  94  95  96  97  98  99 100 101\n",
      " 102 103 104 105 106 107 108 109 110 111 112 113 114 115 116 117 118 119\n",
      " 120 121 122 123 124 125 126 127 128 129 130 131 132 133 134 135 136 137\n",
      " 138 139 140 141 142 143 144 145 146 147 148 149 150 151 152 153 154 155\n",
      " 156 157 158 159 160 161 162 163 164 165 166 167 168 169 170 171 172 173\n",
      " 174 175 176 177 178 179 180 181 182 183 184 185 186 187 188 189 190 191\n",
      " 192 193 194 195 196 197 198 199 200 201 202 203 204 205 206 207 208 209\n",
      " 210 211 212 213 214 215 216 217 218 219 220 221 222 223 224 225 226 227\n",
      " 228 229 230 231 232 233 234 235 236 237 238 239 240 241 242 243 244 245\n",
      " 246 247 248 249 250 251 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17  18\n",
      "  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35  36\n",
      "  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53  54\n",
      "  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71  72\n",
      "  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89  90\n",
      "  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107 108\n",
      " 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125 126\n",
      " 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143 144\n",
      " 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161 162\n",
      " 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179 180\n",
      " 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197 198\n",
      " 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215 216\n",
      " 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233 234\n",
      " 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251 252\n",
      " 253 254 255]\n",
      "[  0  21  25  26  27  28  29  30  31  32  33  34  35  36  37  38  39  40\n",
      "  41  42  43  44  45  46  47  48  49  50  51  52  53  54  55  56  57  58\n",
      "  59  60  61  62  63  64  65  66  67  68  69  70  71  72  73  74  75  76\n",
      "  77  78  79  80  81  82  83  84  85  86  87  88  89  90  91  92  93  94\n",
      "  95  96  97  98  99 100 101 102 103 104 105 106 107 108 109 110 111 112\n",
      " 113 114 115 116 117 118 119 120 121 122 123 124 125 126 127 128 129 130\n",
      " 131 132 133 134 135 136 137 138 139 140 141 142 143 144 145 146 147 148\n",
      " 149 150 151 152 153 154 155 156 157 158 159 160 161 162 163 164 165 166\n",
      " 167 168 169 170 171 172 173 174 175 176 177 178 179 180 181 182 183 184\n",
      " 185 186 187 188 189 190 191 192 193 194 195 196 197 198 199 200 201 202\n",
      " 203 204 205 206 207 208 209 210 211 212 213 214 215 216 217 218 219 220\n",
      " 221 222 223 224 225 226 227 228 229 230 231 232 233 234 235 236 237 238\n",
      " 239 240 241 242 243 244 245 246 247 248 249 250 251 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  5   6   7   8   9  10  11  12  13  14  15  16  17  18  19  20  21  22\n",
      "  23  24  25  26  27  28  29  30  31  32  33  34  35  36  37  38  39  40\n",
      "  41  42  43  44  45  46  47  48  49  50  51  52  53  54  55  56  57  58\n",
      "  59  60  61  62  63  64  65  66  67  68  69  70  71  72  73  74  75  76\n",
      "  77  78  79  80  81  82  83  84  85  86  87  88  89  90  91  92  93  94\n",
      "  95  96  97  98  99 100 101 102 103 104 105 106 107 108 109 110 111 112\n",
      " 113 114 115 116 117 118 119 120 121 122 123 124 125 126 127 128 129 130\n",
      " 131 132 133 134 135 136 137 138 139 140 141 142 143 144 145 146 147 148\n",
      " 149 150 151 152 153 154 155 156 157 158 159 160 161 162 163 164 165 166\n",
      " 167 168 169 170 171 172 173 174 175 176 177 178 179 180 181 182 183 184\n",
      " 185 186 187 188 189 190 191 192 193 194 195 196 197 198 199 200 201 202\n",
      " 203 204 205 206 207 208 209 210 211 212 213 214 215 216 217 218 219 220\n",
      " 221 222 223 224 225 226 227 228 229 230 231 232 233 234 235 236 237 238\n",
      " 239 240 241 242 243 244 245 246 247 248 249 250 251 252 253 254 255]\n",
      "[  0  10  11  12  13  14  15  16  17  18  19  20  21  22  23  24  25  26\n",
      "  27  28  29  30  31  32  33  34  35  36  37  38  39  40  41  42  43  44\n",
      "  45  46  47  48  49  50  51  52  53  54  55  56  57  58  59  60  61  62\n",
      "  63  64  65  66  67  68  69  70  71  72  73  74  75  76  77  78  79  80\n",
      "  81  82  83  84  85  86  87  88  89  90  91  92  93  94  95  96  97  98\n",
      "  99 100 101 102 103 104 105 106 107 108 109 110 111 112 113 114 115 116\n",
      " 117 118 119 120 121 122 123 124 125 126 127 128 129 130 131 132 133 134\n",
      " 135 136 137 138 139 140 141 142 143 144 145 146 147 148 149 150 151 152\n",
      " 153 154 155 156 157 158 159 160 161 162 163 164 165 166 167 168 169 170\n",
      " 171 172 173 174 175 176 177 178 179 180 181 182 183 184 185 186 187 188\n",
      " 189 190 191 192 193 194 195 196 197 198 199 200 201 202 203 204 205 206\n",
      " 207 208 209 210 211 212 213 214 215 216 217 218 219 220 221 222 223 224\n",
      " 225 226 227 228 229 230 231 232 233 234 235 236 237 238 239 240 241 242\n",
      " 243 244 245 246 247 248 249 250 251 252 253 254 255]\n",
      "[ 16  17  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33\n",
      "  34  35  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51\n",
      "  52  53  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69\n",
      "  70  71  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87\n",
      "  88  89  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105\n",
      " 106 107 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123\n",
      " 124 125 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141\n",
      " 142 143 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159\n",
      " 160 161 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177\n",
      " 178 179 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195\n",
      " 196 197 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213\n",
      " 214 215 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231\n",
      " 232 233 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249\n",
      " 250 251 252 253 254 255]\n",
      "[ 17  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34\n",
      "  35  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52\n",
      "  53  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70\n",
      "  71  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88\n",
      "  89  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106\n",
      " 107 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124\n",
      " 125 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142\n",
      " 143 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160\n",
      " 161 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178\n",
      " 179 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196\n",
      " 197 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214\n",
      " 215 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232\n",
      " 233 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250\n",
      " 251 252 253 254 255]\n",
      "[  0  17  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33\n",
      "  34  35  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51\n",
      "  52  53  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69\n",
      "  70  71  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87\n",
      "  88  89  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105\n",
      " 106 107 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123\n",
      " 124 125 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141\n",
      " 142 143 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159\n",
      " 160 161 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177\n",
      " 178 179 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195\n",
      " 196 197 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213\n",
      " 214 215 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231\n",
      " 232 233 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249\n",
      " 250 251 252 253 254 255]\n",
      "[  0   1   2   4   5   6   7   8   9  10  11  12  13  14  15  16  17  18\n",
      "  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35  36\n",
      "  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53  54\n",
      "  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71  72\n",
      "  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89  90\n",
      "  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107 108\n",
      " 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125 126\n",
      " 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143 144\n",
      " 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161 162\n",
      " 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179 180\n",
      " 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197 198\n",
      " 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215 216\n",
      " 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233 234\n",
      " 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251 252\n",
      " 253 254 255]\n",
      "[ 10  14  15  16  17  18  19  20  21  22  23  24  25  26  27  28  29  30\n",
      "  31  32  33  34  35  36  37  38  39  40  41  42  43  44  45  46  47  48\n",
      "  49  50  51  52  53  54  55  56  57  58  59  60  61  62  63  64  65  66\n",
      "  67  68  69  70  71  72  73  74  75  76  77  78  79  80  81  82  83  84\n",
      "  85  86  87  88  89  90  91  92  93  94  95  96  97  98  99 100 101 102\n",
      " 103 104 105 106 107 108 109 110 111 112 113 114 115 116 117 118 119 120\n",
      " 121 122 123 124 125 126 127 128 129 130 131 132 133 134 135 136 137 138\n",
      " 139 140 141 142 143 144 145 146 147 148 149 150 151 152 153 154 155 156\n",
      " 157 158 159 160 161 162 163 164 165 166 167 168 169 170 171 172 173 174\n",
      " 175 176 177 178 179 180 181 182 183 184 185 186 187 188 189 190 191 192\n",
      " 193 194 195 196 197 198 199 200 201 202 203 204 205 206 207 208 209 210\n",
      " 211 212 213 214 215 216 217 218 219 220 221 222 223 224 225 226 227 228\n",
      " 229 230 231 232 233 234 235 236 237 238 239 240 241 242 243 244 245 246\n",
      " 247 248 249 250 251 252 253 254 255]\n",
      "[  0  10  14  15  16  17  18  19  20  21  22  23  24  25  26  27  28  29\n",
      "  30  31  32  33  34  35  36  37  38  39  40  41  42  43  44  45  46  47\n",
      "  48  49  50  51  52  53  54  55  56  57  58  59  60  61  62  63  64  65\n",
      "  66  67  68  69  70  71  72  73  74  75  76  77  78  79  80  81  82  83\n",
      "  84  85  86  87  88  89  90  91  92  93  94  95  96  97  98  99 100 101\n",
      " 102 103 104 105 106 107 108 109 110 111 112 113 114 115 116 117 118 119\n",
      " 120 121 122 123 124 125 126 127 128 129 130 131 132 133 134 135 136 137\n",
      " 138 139 140 141 142 143 144 145 146 147 148 149 150 151 152 153 154 155\n",
      " 156 157 158 159 160 161 162 163 164 165 166 167 168 169 170 171 172 173\n",
      " 174 175 176 177 178 179 180 181 182 183 184 185 186 187 188 189 190 191\n",
      " 192 193 194 195 196 197 198 199 200 201 202 203 204 205 206 207 208 209\n",
      " 210 211 212 213 214 215 216 217 218 219 220 221 222 223 224 225 226 227\n",
      " 228 229 230 231 232 233 234 235 236 237 238 239 240 241 242 243 244 245\n",
      " 246 247 248 249 250 251 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  4   7   9  11  13  14  15  16  17  18  19  20  21  22  23  24  25  26\n",
      "  27  28  29  30  31  32  33  34  35  36  37  38  39  40  41  42  43  44\n",
      "  45  46  47  48  49  50  51  52  53  54  55  56  57  58  59  60  61  62\n",
      "  63  64  65  66  67  68  69  70  71  72  73  74  75  76  77  78  79  80\n",
      "  81  82  83  84  85  86  87  88  89  90  91  92  93  94  95  96  97  98\n",
      "  99 100 101 102 103 104 105 106 107 108 109 110 111 112 113 114 115 116\n",
      " 117 118 119 120 121 122 123 124 125 126 127 128 129 130 131 132 133 134\n",
      " 135 136 137 138 139 140 141 142 143 144 145 146 147 148 149 150 151 152\n",
      " 153 154 155 156 157 158 159 160 161 162 163 164 165 166 167 168 169 170\n",
      " 171 172 173 174 175 176 177 178 179 180 181 182 183 184 185 186 187 188\n",
      " 189 190 191 192 193 194 195 196 197 198 199 200 201 202 203 204 205 206\n",
      " 207 208 209 210 211 212 213 214 215 216 217 218 219 220 221 222 223 224\n",
      " 225 226 227 228 229 230 231 232 233 234 235 236 237 238 239 240 241 242\n",
      " 243 244 245 246 247 248 249 250 251 252 253 254 255]\n",
      "[  0   4  11  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32\n",
      "  33  34  35  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50\n",
      "  51  52  53  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68\n",
      "  69  70  71  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86\n",
      "  87  88  89  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104\n",
      " 105 106 107 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122\n",
      " 123 124 125 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140\n",
      " 141 142 143 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158\n",
      " 159 160 161 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176\n",
      " 177 178 179 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194\n",
      " 195 196 197 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212\n",
      " 213 214 215 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230\n",
      " 231 232 233 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248\n",
      " 249 250 251 252 253 254 255]\n",
      "[  4   6   8   9  11  12  13  14  15  16  17  18  19  20  21  22  23  24\n",
      "  25  26  27  28  29  30  31  32  33  34  35  36  37  38  39  40  41  42\n",
      "  43  44  45  46  47  48  49  50  51  52  53  54  55  56  57  58  59  60\n",
      "  61  62  63  64  65  66  67  68  69  70  71  72  73  74  75  76  77  78\n",
      "  79  80  81  82  83  84  85  86  87  88  89  90  91  92  93  94  95  96\n",
      "  97  98  99 100 101 102 103 104 105 106 107 108 109 110 111 112 113 114\n",
      " 115 116 117 118 119 120 121 122 123 124 125 126 127 128 129 130 131 132\n",
      " 133 134 135 136 137 138 139 140 141 142 143 144 145 146 147 148 149 150\n",
      " 151 152 153 154 155 156 157 158 159 160 161 162 163 164 165 166 167 168\n",
      " 169 170 171 172 173 174 175 176 177 178 179 180 181 182 183 184 185 186\n",
      " 187 188 189 190 191 192 193 194 195 196 197 198 199 200 201 202 203 204\n",
      " 205 206 207 208 209 210 211 212 213 214 215 216 217 218 219 220 221 222\n",
      " 223 224 225 226 227 228 229 230 231 232 233 234 235 236 237 238 239 240\n",
      " 241 242 243 244 245 246 247 248 249 250 251 252 253 254 255]\n",
      "[ 18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  0  18  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  3   5   6   7   8   9  10  11  12  13  14  15  16  17  18  19  20  21\n",
      "  22  23  24  25  26  27  28  29  30  31  32  33  34  35  36  37  38  39\n",
      "  40  41  42  43  44  45  46  47  48  49  50  51  52  53  54  55  56  57\n",
      "  58  59  60  61  62  63  64  65  66  67  68  69  70  71  72  73  74  75\n",
      "  76  77  78  79  80  81  82  83  84  85  86  87  88  89  90  91  92  93\n",
      "  94  95  96  97  98  99 100 101 102 103 104 105 106 107 108 109 110 111\n",
      " 112 113 114 115 116 117 118 119 120 121 122 123 124 125 126 127 128 129\n",
      " 130 131 132 133 134 135 136 137 138 139 140 141 142 143 144 145 146 147\n",
      " 148 149 150 151 152 153 154 155 156 157 158 159 160 161 162 163 164 165\n",
      " 166 167 168 169 170 171 172 173 174 175 176 177 178 179 180 181 182 183\n",
      " 184 185 186 187 188 189 190 191 192 193 194 195 196 197 198 199 200 201\n",
      " 202 203 204 205 206 207 208 209 210 211 212 213 214 215 216 217 218 219\n",
      " 220 221 222 223 224 225 226 227 228 229 230 231 232 233 234 235 236 237\n",
      " 238 239 240 241 242 243 244 245 246 247 248 249 250 251 252 253 254 255]\n",
      "[ 10  12  14  15  16  17  18  19  20  21  22  23  24  25  26  27  28  29\n",
      "  30  31  32  33  34  35  36  37  38  39  40  41  42  43  44  45  46  47\n",
      "  48  49  50  51  52  53  54  55  56  57  58  59  60  61  62  63  64  65\n",
      "  66  67  68  69  70  71  72  73  74  75  76  77  78  79  80  81  82  83\n",
      "  84  85  86  87  88  89  90  91  92  93  94  95  96  97  98  99 100 101\n",
      " 102 103 104 105 106 107 108 109 110 111 112 113 114 115 116 117 118 119\n",
      " 120 121 122 123 124 125 126 127 128 129 130 131 132 133 134 135 136 137\n",
      " 138 139 140 141 142 143 144 145 146 147 148 149 150 151 152 153 154 155\n",
      " 156 157 158 159 160 161 162 163 164 165 166 167 168 169 170 171 172 173\n",
      " 174 175 176 177 178 179 180 181 182 183 184 185 186 187 188 189 190 191\n",
      " 192 193 194 195 196 197 198 199 200 201 202 203 204 205 206 207 208 209\n",
      " 210 211 212 213 214 215 216 217 218 219 220 221 222 223 224 225 226 227\n",
      " 228 229 230 231 232 233 234 235 236 237 238 239 240 241 242 243 244 245\n",
      " 246 247 248 249 250 251 252 253 254 255]\n",
      "[  0  10  12  14  15  16  17  18  19  20  21  22  23  24  25  26  27  28\n",
      "  29  30  31  32  33  34  35  36  37  38  39  40  41  42  43  44  45  46\n",
      "  47  48  49  50  51  52  53  54  55  56  57  58  59  60  61  62  63  64\n",
      "  65  66  67  68  69  70  71  72  73  74  75  76  77  78  79  80  81  82\n",
      "  83  84  85  86  87  88  89  90  91  92  93  94  95  96  97  98  99 100\n",
      " 101 102 103 104 105 106 107 108 109 110 111 112 113 114 115 116 117 118\n",
      " 119 120 121 122 123 124 125 126 127 128 129 130 131 132 133 134 135 136\n",
      " 137 138 139 140 141 142 143 144 145 146 147 148 149 150 151 152 153 154\n",
      " 155 156 157 158 159 160 161 162 163 164 165 166 167 168 169 170 171 172\n",
      " 173 174 175 176 177 178 179 180 181 182 183 184 185 186 187 188 189 190\n",
      " 191 192 193 194 195 196 197 198 199 200 201 202 203 204 205 206 207 208\n",
      " 209 210 211 212 213 214 215 216 217 218 219 220 221 222 223 224 225 226\n",
      " 227 228 229 230 231 232 233 234 235 236 237 238 239 240 241 242 243 244\n",
      " 245 246 247 248 249 250 251 252 253 254 255]\n",
      "[  4   9  11  13  15  16  17  18  19  20  21  22  23  24  25  26  27  28\n",
      "  29  30  31  32  33  34  35  36  37  38  39  40  41  42  43  44  45  46\n",
      "  47  48  49  50  51  52  53  54  55  56  57  58  59  60  61  62  63  64\n",
      "  65  66  67  68  69  70  71  72  73  74  75  76  77  78  79  80  81  82\n",
      "  83  84  85  86  87  88  89  90  91  92  93  94  95  96  97  98  99 100\n",
      " 101 102 103 104 105 106 107 108 109 110 111 112 113 114 115 116 117 118\n",
      " 119 120 121 122 123 124 125 126 127 128 129 130 131 132 133 134 135 136\n",
      " 137 138 139 140 141 142 143 144 145 146 147 148 149 150 151 152 153 154\n",
      " 155 156 157 158 159 160 161 162 163 164 165 166 167 168 169 170 171 172\n",
      " 173 174 175 176 177 178 179 180 181 182 183 184 185 186 187 188 189 190\n",
      " 191 192 193 194 195 196 197 198 199 200 201 202 203 204 205 206 207 208\n",
      " 209 210 211 212 213 214 215 216 217 218 219 220 221 222 223 224 225 226\n",
      " 227 228 229 230 231 232 233 234 235 236 237 238 239 240 241 242 243 244\n",
      " 245 246 247 248 249 250 251 252 253 254 255]\n",
      "[ 12  19  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35  36\n",
      "  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53  54\n",
      "  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71  72\n",
      "  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89  90\n",
      "  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107 108\n",
      " 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125 126\n",
      " 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143 144\n",
      " 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161 162\n",
      " 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179 180\n",
      " 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197 198\n",
      " 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215 216\n",
      " 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233 234\n",
      " 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251 252\n",
      " 253 254 255]\n",
      "[  0  25  26  27  28  29  30  31  32  33  34  35  36  37  38  39  40  41\n",
      "  42  43  44  45  46  47  48  49  50  51  52  53  54  55  56  57  58  59\n",
      "  60  61  62  63  64  65  66  67  68  69  70  71  72  73  74  75  76  77\n",
      "  78  79  80  81  82  83  84  85  86  87  88  89  90  91  92  93  94  95\n",
      "  96  97  98  99 100 101 102 103 104 105 106 107 108 109 110 111 112 113\n",
      " 114 115 116 117 118 119 120 121 122 123 124 125 126 127 128 129 130 131\n",
      " 132 133 134 135 136 137 138 139 140 141 142 143 144 145 146 147 148 149\n",
      " 150 151 152 153 154 155 156 157 158 159 160 161 162 163 164 165 166 167\n",
      " 168 169 170 171 172 173 174 175 176 177 178 179 180 181 182 183 184 185\n",
      " 186 187 188 189 190 191 192 193 194 195 196 197 198 199 200 201 202 203\n",
      " 204 205 206 207 208 209 210 211 212 213 214 215 216 217 218 219 220 221\n",
      " 222 223 224 225 226 227 228 229 230 231 232 233 234 235 236 237 238 239\n",
      " 240 241 242 243 244 245 246 247 248 249 250 251 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17  18\n",
      "  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35  36\n",
      "  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53  54\n",
      "  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71  72\n",
      "  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89  90\n",
      "  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107 108\n",
      " 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125 126\n",
      " 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143 144\n",
      " 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161 162\n",
      " 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179 180\n",
      " 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197 198\n",
      " 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215 216\n",
      " 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233 234\n",
      " 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251 252\n",
      " 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17  18\n",
      "  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35  36\n",
      "  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53  54\n",
      "  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71  72\n",
      "  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89  90\n",
      "  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107 108\n",
      " 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125 126\n",
      " 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143 144\n",
      " 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161 162\n",
      " 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179 180\n",
      " 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197 198\n",
      " 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215 216\n",
      " 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233 234\n",
      " 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251 252\n",
      " 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  4   5   6   7   8   9  10  11  12  13  14  15  16  17  18  19  20  21\n",
      "  22  23  24  25  26  27  28  29  30  31  32  33  34  35  36  37  38  39\n",
      "  40  41  42  43  44  45  46  47  48  49  50  51  52  53  54  55  56  57\n",
      "  58  59  60  61  62  63  64  65  66  67  68  69  70  71  72  73  74  75\n",
      "  76  77  78  79  80  81  82  83  84  85  86  87  88  89  90  91  92  93\n",
      "  94  95  96  97  98  99 100 101 102 103 104 105 106 107 108 109 110 111\n",
      " 112 113 114 115 116 117 118 119 120 121 122 123 124 125 126 127 128 129\n",
      " 130 131 132 133 134 135 136 137 138 139 140 141 142 143 144 145 146 147\n",
      " 148 149 150 151 152 153 154 155 156 157 158 159 160 161 162 163 164 165\n",
      " 166 167 168 169 170 171 172 173 174 175 176 177 178 179 180 181 182 183\n",
      " 184 185 186 187 188 189 190 191 192 193 194 195 196 197 198 199 200 201\n",
      " 202 203 204 205 206 207 208 209 210 211 212 213 214 215 216 217 218 219\n",
      " 220 221 222 223 224 225 226 227 228 229 230 231 232 233 234 235 236 237\n",
      " 238 239 240 241 242 243 244 245 246 247 248 249 250 251 252 253 254 255]\n",
      "[  0   4   5   6   7   8   9  10  11  12  13  14  15  16  17  18  19  20\n",
      "  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35  36  37  38\n",
      "  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53  54  55  56\n",
      "  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71  72  73  74\n",
      "  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89  90  91  92\n",
      "  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107 108 109 110\n",
      " 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125 126 127 128\n",
      " 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143 144 145 146\n",
      " 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161 162 163 164\n",
      " 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179 180 181 182\n",
      " 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197 198 199 200\n",
      " 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215 216 217 218\n",
      " 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233 234 235 236\n",
      " 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251 252 253 254\n",
      " 255]\n",
      "[  0   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17  18\n",
      "  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35  36\n",
      "  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53  54\n",
      "  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71  72\n",
      "  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89  90\n",
      "  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107 108\n",
      " 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125 126\n",
      " 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143 144\n",
      " 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161 162\n",
      " 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179 180\n",
      " 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197 198\n",
      " 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215 216\n",
      " 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233 234\n",
      " 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251 252\n",
      " 253 254 255]\n",
      "[  5   6   7   8   9  10  11  12  13  14  15  16  17  18  19  20  21  22\n",
      "  23  24  25  26  27  28  29  30  31  32  33  34  35  36  37  38  39  40\n",
      "  41  42  43  44  45  46  47  48  49  50  51  52  53  54  55  56  57  58\n",
      "  59  60  61  62  63  64  65  66  67  68  69  70  71  72  73  74  75  76\n",
      "  77  78  79  80  81  82  83  84  85  86  87  88  89  90  91  92  93  94\n",
      "  95  96  97  98  99 100 101 102 103 104 105 106 107 108 109 110 111 112\n",
      " 113 114 115 116 117 118 119 120 121 122 123 124 125 126 127 128 129 130\n",
      " 131 132 133 134 135 136 137 138 139 140 141 142 143 144 145 146 147 148\n",
      " 149 150 151 152 153 154 155 156 157 158 159 160 161 162 163 164 165 166\n",
      " 167 168 169 170 171 172 173 174 175 176 177 178 179 180 181 182 183 184\n",
      " 185 186 187 188 189 190 191 192 193 194 195 196 197 198 199 200 201 202\n",
      " 203 204 205 206 207 208 209 210 211 212 213 214 215 216 217 218 219 220\n",
      " 221 222 223 224 225 226 227 228 229 230 231 232 233 234 235 236 237 238\n",
      " 239 240 241 242 243 244 245 246 247 248 249 250 251 252 253 254 255]\n",
      "[  0   5   6   7   8   9  10  11  12  13  14  15  16  17  18  19  20  21\n",
      "  22  23  24  25  26  27  28  29  30  31  32  33  34  35  36  37  38  39\n",
      "  40  41  42  43  44  45  46  47  48  49  50  51  52  53  54  55  56  57\n",
      "  58  59  60  61  62  63  64  65  66  67  68  69  70  71  72  73  74  75\n",
      "  76  77  78  79  80  81  82  83  84  85  86  87  88  89  90  91  92  93\n",
      "  94  95  96  97  98  99 100 101 102 103 104 105 106 107 108 109 110 111\n",
      " 112 113 114 115 116 117 118 119 120 121 122 123 124 125 126 127 128 129\n",
      " 130 131 132 133 134 135 136 137 138 139 140 141 142 143 144 145 146 147\n",
      " 148 149 150 151 152 153 154 155 156 157 158 159 160 161 162 163 164 165\n",
      " 166 167 168 169 170 171 172 173 174 175 176 177 178 179 180 181 182 183\n",
      " 184 185 186 187 188 189 190 191 192 193 194 195 196 197 198 199 200 201\n",
      " 202 203 204 205 206 207 208 209 210 211 212 213 214 215 216 217 218 219\n",
      " 220 221 222 223 224 225 226 227 228 229 230 231 232 233 234 235 236 237\n",
      " 238 239 240 241 242 243 244 245 246 247 248 249 250 251 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  0   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17  18  19\n",
      "  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35  36  37\n",
      "  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53  54  55\n",
      "  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71  72  73\n",
      "  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89  90  91\n",
      "  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107 108 109\n",
      " 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125 126 127\n",
      " 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143 144 145\n",
      " 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161 162 163\n",
      " 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179 180 181\n",
      " 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197 198 199\n",
      " 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215 216 217\n",
      " 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233 234 235\n",
      " 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251 252 253\n",
      " 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  0  16  17  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32\n",
      "  33  34  35  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50\n",
      "  51  52  53  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68\n",
      "  69  70  71  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86\n",
      "  87  88  89  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104\n",
      " 105 106 107 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122\n",
      " 123 124 125 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140\n",
      " 141 142 143 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158\n",
      " 159 160 161 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176\n",
      " 177 178 179 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194\n",
      " 195 196 197 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212\n",
      " 213 214 215 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230\n",
      " 231 232 233 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248\n",
      " 249 250 251 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  0  14  15  16  17  18  19  20  21  22  23  24  25  26  27  28  29  30\n",
      "  31  32  33  34  35  36  37  38  39  40  41  42  43  44  45  46  47  48\n",
      "  49  50  51  52  53  54  55  56  57  58  59  60  61  62  63  64  65  66\n",
      "  67  68  69  70  71  72  73  74  75  76  77  78  79  80  81  82  83  84\n",
      "  85  86  87  88  89  90  91  92  93  94  95  96  97  98  99 100 101 102\n",
      " 103 104 105 106 107 108 109 110 111 112 113 114 115 116 117 118 119 120\n",
      " 121 122 123 124 125 126 127 128 129 130 131 132 133 134 135 136 137 138\n",
      " 139 140 141 142 143 144 145 146 147 148 149 150 151 152 153 154 155 156\n",
      " 157 158 159 160 161 162 163 164 165 166 167 168 169 170 171 172 173 174\n",
      " 175 176 177 178 179 180 181 182 183 184 185 186 187 188 189 190 191 192\n",
      " 193 194 195 196 197 198 199 200 201 202 203 204 205 206 207 208 209 210\n",
      " 211 212 213 214 215 216 217 218 219 220 221 222 223 224 225 226 227 228\n",
      " 229 230 231 232 233 234 235 236 237 238 239 240 241 242 243 244 245 246\n",
      " 247 248 249 250 251 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  0   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17  18\n",
      "  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35  36\n",
      "  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53  54\n",
      "  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71  72\n",
      "  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89  90\n",
      "  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107 108\n",
      " 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125 126\n",
      " 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143 144\n",
      " 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161 162\n",
      " 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179 180\n",
      " 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197 198\n",
      " 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215 216\n",
      " 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233 234\n",
      " 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251 252\n",
      " 253 254 255]\n",
      "[  0  14  15  17  18  19  20  21  22  23  24  25  26  27  28  29  30  31\n",
      "  32  33  34  35  36  37  38  39  40  41  42  43  44  45  46  47  48  49\n",
      "  50  51  52  53  54  55  56  57  58  59  60  61  62  63  64  65  66  67\n",
      "  68  69  70  71  72  73  74  75  76  77  78  79  80  81  82  83  84  85\n",
      "  86  87  88  89  90  91  92  93  94  95  96  97  98  99 100 101 102 103\n",
      " 104 105 106 107 108 109 110 111 112 113 114 115 116 117 118 119 120 121\n",
      " 122 123 124 125 126 127 128 129 130 131 132 133 134 135 136 137 138 139\n",
      " 140 141 142 143 144 145 146 147 148 149 150 151 152 153 154 155 156 157\n",
      " 158 159 160 161 162 163 164 165 166 167 168 169 170 171 172 173 174 175\n",
      " 176 177 178 179 180 181 182 183 184 185 186 187 188 189 190 191 192 193\n",
      " 194 195 196 197 198 199 200 201 202 203 204 205 206 207 208 209 210 211\n",
      " 212 213 214 215 216 217 218 219 220 221 222 223 224 225 226 227 228 229\n",
      " 230 231 232 233 234 235 236 237 238 239 240 241 242 243 244 245 246 247\n",
      " 248 249 250 251 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17  18\n",
      "  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35  36\n",
      "  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53  54\n",
      "  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71  72\n",
      "  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89  90\n",
      "  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107 108\n",
      " 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125 126\n",
      " 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143 144\n",
      " 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161 162\n",
      " 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179 180\n",
      " 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197 198\n",
      " 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215 216\n",
      " 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233 234\n",
      " 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251 252\n",
      " 253 254 255]\n",
      "[  0   5   8  11  12  13  14  15  16  17  18  19  20  21  22  23  24  25\n",
      "  26  27  28  29  30  31  32  33  34  35  36  37  38  39  40  41  42  43\n",
      "  44  45  46  47  48  49  50  51  52  53  54  55  56  57  58  59  60  61\n",
      "  62  63  64  65  66  67  68  69  70  71  72  73  74  75  76  77  78  79\n",
      "  80  81  82  83  84  85  86  87  88  89  90  91  92  93  94  95  96  97\n",
      "  98  99 100 101 102 103 104 105 106 107 108 109 110 111 112 113 114 115\n",
      " 116 117 118 119 120 121 122 123 124 125 126 127 128 129 130 131 132 133\n",
      " 134 135 136 137 138 139 140 141 142 143 144 145 146 147 148 149 150 151\n",
      " 152 153 154 155 156 157 158 159 160 161 162 163 164 165 166 167 168 169\n",
      " 170 171 172 173 174 175 176 177 178 179 180 181 182 183 184 185 186 187\n",
      " 188 189 190 191 192 193 194 195 196 197 198 199 200 201 202 203 204 205\n",
      " 206 207 208 209 210 211 212 213 214 215 216 217 218 219 220 221 222 223\n",
      " 224 225 226 227 228 229 230 231 232 233 234 235 236 237 238 239 240 241\n",
      " 242 243 244 245 246 247 248 249 250 251 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17  18\n",
      "  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35  36\n",
      "  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53  54\n",
      "  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71  72\n",
      "  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89  90\n",
      "  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107 108\n",
      " 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125 126\n",
      " 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143 144\n",
      " 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161 162\n",
      " 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179 180\n",
      " 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197 198\n",
      " 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215 216\n",
      " 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233 234\n",
      " 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251 252\n",
      " 253 254 255]\n",
      "[  0   1   5   9  10  11  12  13  14  15  16  17  18  19  20  21  22  23\n",
      "  24  25  26  27  28  29  30  31  32  33  34  35  36  37  38  39  40  41\n",
      "  42  43  44  45  46  47  48  49  50  51  52  53  54  55  56  57  58  59\n",
      "  60  61  62  63  64  65  66  67  68  69  70  71  72  73  74  75  76  77\n",
      "  78  79  80  81  82  83  84  85  86  87  88  89  90  91  92  93  94  95\n",
      "  96  97  98  99 100 101 102 103 104 105 106 107 108 109 110 111 112 113\n",
      " 114 115 116 117 118 119 120 121 122 123 124 125 126 127 128 129 130 131\n",
      " 132 133 134 135 136 137 138 139 140 141 142 143 144 145 146 147 148 149\n",
      " 150 151 152 153 154 155 156 157 158 159 160 161 162 163 164 165 166 167\n",
      " 168 169 170 171 172 173 174 175 176 177 178 179 180 181 182 183 184 185\n",
      " 186 187 188 189 190 191 192 193 194 195 196 197 198 199 200 201 202 203\n",
      " 204 205 206 207 208 209 210 211 212 213 214 215 216 217 218 219 220 221\n",
      " 222 223 224 225 226 227 228 229 230 231 232 233 234 235 236 237 238 239\n",
      " 240 241 242 243 244 245 246 247 248 249 250 251 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  0   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17  18\n",
      "  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35  36\n",
      "  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53  54\n",
      "  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71  72\n",
      "  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89  90\n",
      "  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107 108\n",
      " 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125 126\n",
      " 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143 144\n",
      " 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161 162\n",
      " 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179 180\n",
      " 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197 198\n",
      " 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215 216\n",
      " 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233 234\n",
      " 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251 252\n",
      " 253 254 255]\n",
      "[  0  10  11  12  13  14  15  16  17  18  19  20  21  22  23  24  25  26\n",
      "  27  28  29  30  31  32  33  34  35  36  37  38  39  40  41  42  43  44\n",
      "  45  46  47  48  49  50  51  52  53  54  55  56  57  58  59  60  61  62\n",
      "  63  64  65  66  67  68  69  70  71  72  73  74  75  76  77  78  79  80\n",
      "  81  82  83  84  85  86  87  88  89  90  91  92  93  94  95  96  97  98\n",
      "  99 100 101 102 103 104 105 106 107 108 109 110 111 112 113 114 115 116\n",
      " 117 118 119 120 121 122 123 124 125 126 127 128 129 130 131 132 133 134\n",
      " 135 136 137 138 139 140 141 142 143 144 145 146 147 148 149 150 151 152\n",
      " 153 154 155 156 157 158 159 160 161 162 163 164 165 166 167 168 169 170\n",
      " 171 172 173 174 175 176 177 178 179 180 181 182 183 184 185 186 187 188\n",
      " 189 190 191 192 193 194 195 196 197 198 199 200 201 202 203 204 205 206\n",
      " 207 208 209 210 211 212 213 214 215 216 217 218 219 220 221 222 223 224\n",
      " 225 226 227 228 229 230 231 232 233 234 235 236 237 238 239 240 241 242\n",
      " 243 244 245 246 247 248 249 250 251 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  0  17  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33\n",
      "  34  35  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51\n",
      "  52  53  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69\n",
      "  70  71  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87\n",
      "  88  89  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105\n",
      " 106 107 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123\n",
      " 124 125 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141\n",
      " 142 143 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159\n",
      " 160 161 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177\n",
      " 178 179 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195\n",
      " 196 197 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213\n",
      " 214 215 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231\n",
      " 232 233 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249\n",
      " 250 251 252 253 254 255]\n",
      "[  7   8   9  10  11  12  13  14  15  16  17  18  19  20  21  22  23  24\n",
      "  25  26  27  28  29  30  31  32  33  34  35  36  37  38  39  40  41  42\n",
      "  43  44  45  46  47  48  49  50  51  52  53  54  55  56  57  58  59  60\n",
      "  61  62  63  64  65  66  67  68  69  70  71  72  73  74  75  76  77  78\n",
      "  79  80  81  82  83  84  85  86  87  88  89  90  91  92  93  94  95  96\n",
      "  97  98  99 100 101 102 103 104 105 106 107 108 109 110 111 112 113 114\n",
      " 115 116 117 118 119 120 121 122 123 124 125 126 127 128 129 130 131 132\n",
      " 133 134 135 136 137 138 139 140 141 142 143 144 145 146 147 148 149 150\n",
      " 151 152 153 154 155 156 157 158 159 160 161 162 163 164 165 166 167 168\n",
      " 169 170 171 172 173 174 175 176 177 178 179 180 181 182 183 184 185 186\n",
      " 187 188 189 190 191 192 193 194 195 196 197 198 199 200 201 202 203 204\n",
      " 205 206 207 208 209 210 211 212 213 214 215 216 217 218 219 220 221 222\n",
      " 223 224 225 226 227 228 229 230 231 232 233 234 235 236 237 238 239 240\n",
      " 241 242 243 244 245 246 247 248 249 250 251 252 253 254 255]\n",
      "[  7   8   9  10  11  12  13  14  15  16  17  18  19  20  21  22  23  24\n",
      "  25  26  27  28  29  30  31  32  33  34  35  36  37  38  39  40  41  42\n",
      "  43  44  45  46  47  48  49  50  51  52  53  54  55  56  57  58  59  60\n",
      "  61  62  63  64  65  66  67  68  69  70  71  72  73  74  75  76  77  78\n",
      "  79  80  81  82  83  84  85  86  87  88  89  90  91  92  93  94  95  96\n",
      "  97  98  99 100 101 102 103 104 105 106 107 108 109 110 111 112 113 114\n",
      " 115 116 117 118 119 120 121 122 123 124 125 126 127 128 129 130 131 132\n",
      " 133 134 135 136 137 138 139 140 141 142 143 144 145 146 147 148 149 150\n",
      " 151 152 153 154 155 156 157 158 159 160 161 162 163 164 165 166 167 168\n",
      " 169 170 171 172 173 174 175 176 177 178 179 180 181 182 183 184 185 186\n",
      " 187 188 189 190 191 192 193 194 195 196 197 198 199 200 201 202 203 204\n",
      " 205 206 207 208 209 210 211 212 213 214 215 216 217 218 219 220 221 222\n",
      " 223 224 225 226 227 228 229 230 231 232 233 234 235 236 237 238 239 240\n",
      " 241 242 243 244 245 246 247 248 249 250 251 252 253 254 255]\n",
      "[  0  10  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33\n",
      "  34  35  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51\n",
      "  52  53  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69\n",
      "  70  71  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87\n",
      "  88  89  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105\n",
      " 106 107 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123\n",
      " 124 125 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141\n",
      " 142 143 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159\n",
      " 160 161 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177\n",
      " 178 179 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195\n",
      " 196 197 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213\n",
      " 214 215 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231\n",
      " 232 233 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249\n",
      " 250 251 252 253 254 255]\n",
      "[  9  10  12  13  14  15  16  17  18  19  20  21  22  23  24  25  26  27\n",
      "  28  29  30  31  32  33  34  35  36  37  38  39  40  41  42  43  44  45\n",
      "  46  47  48  49  50  51  52  53  54  55  56  57  58  59  60  61  62  63\n",
      "  64  65  66  67  68  69  70  71  72  73  74  75  76  77  78  79  80  81\n",
      "  82  83  84  85  86  87  88  89  90  91  92  93  94  95  96  97  98  99\n",
      " 100 101 102 103 104 105 106 107 108 109 110 111 112 113 114 115 116 117\n",
      " 118 119 120 121 122 123 124 125 126 127 128 129 130 131 132 133 134 135\n",
      " 136 137 138 139 140 141 142 143 144 145 146 147 148 149 150 151 152 153\n",
      " 154 155 156 157 158 159 160 161 162 163 164 165 166 167 168 169 170 171\n",
      " 172 173 174 175 176 177 178 179 180 181 182 183 184 185 186 187 188 189\n",
      " 190 191 192 193 194 195 196 197 198 199 200 201 202 203 204 205 206 207\n",
      " 208 209 210 211 212 213 214 215 216 217 218 219 220 221 222 223 224 225\n",
      " 226 227 228 229 230 231 232 233 234 235 236 237 238 239 240 241 242 243\n",
      " 244 245 246 247 248 249 250 251 252 253 254 255]\n",
      "[ 17  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34\n",
      "  35  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52\n",
      "  53  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70\n",
      "  71  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88\n",
      "  89  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106\n",
      " 107 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124\n",
      " 125 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142\n",
      " 143 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160\n",
      " 161 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178\n",
      " 179 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196\n",
      " 197 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214\n",
      " 215 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232\n",
      " 233 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250\n",
      " 251 252 253 254 255]\n",
      "[  0  17  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33\n",
      "  34  35  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51\n",
      "  52  53  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69\n",
      "  70  71  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87\n",
      "  88  89  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105\n",
      " 106 107 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123\n",
      " 124 125 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141\n",
      " 142 143 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159\n",
      " 160 161 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177\n",
      " 178 179 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195\n",
      " 196 197 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213\n",
      " 214 215 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231\n",
      " 232 233 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249\n",
      " 250 251 252 253 254 255]\n",
      "[  1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17  18\n",
      "  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35  36\n",
      "  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53  54\n",
      "  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71  72\n",
      "  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89  90\n",
      "  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107 108\n",
      " 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125 126\n",
      " 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143 144\n",
      " 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161 162\n",
      " 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179 180\n",
      " 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197 198\n",
      " 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215 216\n",
      " 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233 234\n",
      " 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251 252\n",
      " 253 254 255]\n",
      "[  3   4   5   6   7   8   9  10  11  12  13  14  15  16  17  18  19  20\n",
      "  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35  36  37  38\n",
      "  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53  54  55  56\n",
      "  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71  72  73  74\n",
      "  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89  90  91  92\n",
      "  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107 108 109 110\n",
      " 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125 126 127 128\n",
      " 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143 144 145 146\n",
      " 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161 162 163 164\n",
      " 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179 180 181 182\n",
      " 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197 198 199 200\n",
      " 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215 216 217 218\n",
      " 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233 234 235 236\n",
      " 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251 252 253 254\n",
      " 255]\n",
      "[  0   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17  18  19\n",
      "  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35  36  37\n",
      "  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53  54  55\n",
      "  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71  72  73\n",
      "  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89  90  91\n",
      "  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107 108 109\n",
      " 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125 126 127\n",
      " 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143 144 145\n",
      " 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161 162 163\n",
      " 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179 180 181\n",
      " 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197 198 199\n",
      " 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215 216 217\n",
      " 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233 234 235\n",
      " 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251 252 253\n",
      " 254 255]\n",
      "[  0   1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17\n",
      "  18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  5   6   7   8   9  11  12  13  14  15  16  17  18  19  20  21  22  23\n",
      "  24  25  26  27  28  29  30  31  32  33  34  35  36  37  38  39  40  41\n",
      "  42  43  44  45  46  47  48  49  50  51  52  53  54  55  56  57  58  59\n",
      "  60  61  62  63  64  65  66  67  68  69  70  71  72  73  74  75  76  77\n",
      "  78  79  80  81  82  83  84  85  86  87  88  89  90  91  92  93  94  95\n",
      "  96  97  98  99 100 101 102 103 104 105 106 107 108 109 110 111 112 113\n",
      " 114 115 116 117 118 119 120 121 122 123 124 125 126 127 128 129 130 131\n",
      " 132 133 134 135 136 137 138 139 140 141 142 143 144 145 146 147 148 149\n",
      " 150 151 152 153 154 155 156 157 158 159 160 161 162 163 164 165 166 167\n",
      " 168 169 170 171 172 173 174 175 176 177 178 179 180 181 182 183 184 185\n",
      " 186 187 188 189 190 191 192 193 194 195 196 197 198 199 200 201 202 203\n",
      " 204 205 206 207 208 209 210 211 212 213 214 215 216 217 218 219 220 221\n",
      " 222 223 224 225 226 227 228 229 230 231 232 233 234 235 236 237 238 239\n",
      " 240 241 242 243 244 245 246 247 248 249 250 251 252 253 254 255]\n",
      "[  0   5   6   7   8   9  11  12  13  14  15  16  17  18  19  20  21  22\n",
      "  23  24  25  26  27  28  29  30  31  32  33  34  35  36  37  38  39  40\n",
      "  41  42  43  44  45  46  47  48  49  50  51  52  53  54  55  56  57  58\n",
      "  59  60  61  62  63  64  65  66  67  68  69  70  71  72  73  74  75  76\n",
      "  77  78  79  80  81  82  83  84  85  86  87  88  89  90  91  92  93  94\n",
      "  95  96  97  98  99 100 101 102 103 104 105 106 107 108 109 110 111 112\n",
      " 113 114 115 116 117 118 119 120 121 122 123 124 125 126 127 128 129 130\n",
      " 131 132 133 134 135 136 137 138 139 140 141 142 143 144 145 146 147 148\n",
      " 149 150 151 152 153 154 155 156 157 158 159 160 161 162 163 164 165 166\n",
      " 167 168 169 170 171 172 173 174 175 176 177 178 179 180 181 182 183 184\n",
      " 185 186 187 188 189 190 191 192 193 194 195 196 197 198 199 200 201 202\n",
      " 203 204 205 206 207 208 209 210 211 212 213 214 215 216 217 218 219 220\n",
      " 221 222 223 224 225 226 227 228 229 230 231 232 233 234 235 236 237 238\n",
      " 239 240 241 242 243 244 245 246 247 248 249 250 251 252 253 254 255]\n",
      "[ 14  15  16  17  18  19  20  21  22  23  24  25  26  27  28  29  30  31\n",
      "  32  33  34  35  36  37  38  39  40  41  42  43  44  45  46  47  48  49\n",
      "  50  51  52  53  54  55  56  57  58  59  60  61  62  63  64  65  66  67\n",
      "  68  69  70  71  72  73  74  75  76  77  78  79  80  81  82  83  84  85\n",
      "  86  87  88  89  90  91  92  93  94  95  96  97  98  99 100 101 102 103\n",
      " 104 105 106 107 108 109 110 111 112 113 114 115 116 117 118 119 120 121\n",
      " 122 123 124 125 126 127 128 129 130 131 132 133 134 135 136 137 138 139\n",
      " 140 141 142 143 144 145 146 147 148 149 150 151 152 153 154 155 156 157\n",
      " 158 159 160 161 162 163 164 165 166 167 168 169 170 171 172 173 174 175\n",
      " 176 177 178 179 180 181 182 183 184 185 186 187 188 189 190 191 192 193\n",
      " 194 195 196 197 198 199 200 201 202 203 204 205 206 207 208 209 210 211\n",
      " 212 213 214 215 216 217 218 219 220 221 222 223 224 225 226 227 228 229\n",
      " 230 231 232 233 234 235 236 237 238 239 240 241 242 243 244 245 246 247\n",
      " 248 249 250 251 252 253 254 255]\n",
      "[ 18  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35\n",
      "  36  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53\n",
      "  54  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71\n",
      "  72  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89\n",
      "  90  91  92  93  94  95  96  97  98  99 100 101 102 103 104 105 106 107\n",
      " 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125\n",
      " 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143\n",
      " 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161\n",
      " 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179\n",
      " 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197\n",
      " 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215\n",
      " 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233\n",
      " 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251\n",
      " 252 253 254 255]\n",
      "[  0  29  32  33  34  35  36  37  38  39  40  41  42  43  44  45  46  47\n",
      "  48  49  50  51  52  53  54  55  56  57  58  59  60  61  62  63  64  65\n",
      "  66  67  68  69  70  71  72  73  74  75  76  77  78  79  80  81  82  83\n",
      "  84  85  86  87  88  89  90  91  92  93  94  95  96  97  98  99 100 101\n",
      " 102 103 104 105 106 107 108 109 110 111 112 113 114 115 116 117 118 119\n",
      " 120 121 122 123 124 125 126 127 128 129 130 131 132 133 134 135 136 137\n",
      " 138 139 140 141 142 143 144 145 146 147 148 149 150 151 152 153 154 155\n",
      " 156 157 158 159 160 161 162 163 164 165 166 167 168 169 170 171 172 173\n",
      " 174 175 176 177 178 179 180 181 182 183 184 185 186 187 188 189 190 191\n",
      " 192 193 194 195 196 197 198 199 200 201 202 203 204 205 206 207 208 209\n",
      " 210 211 212 213 214 215 216 217 218 219 220 221 222 223 224 225 226 227\n",
      " 228 229 230 231 232 233 234 235 236 237 238 239 240 241 242 243 244 245\n",
      " 246 247 248 249 250 251 252 253 254 255]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1 2]\n",
      "[0 1]\n",
      "[0 1]\n",
      "[0 1]\n",
      "(alignment) "
     ]
    }
   ],
   "source": [
    "conda activate alignment\n",
    "cd /scratch2/NSF_GWAS/ImageAlignment/\n",
    "file_list_input=\"${data}/file_list.csv\"\n",
    "python main.py \\\n",
    "--hyper-img Grids_to_align-selected/20itemgrid_F1.9_I3.0_L50_cyan_114229_0_0_0_index130.csv \\\n",
    "--rgb-img Grids_to_align-selected/20itemgrid_F1.9_I3.0_L50_cyan_114229_0_0_0_rgb.jpg \\\n",
    "--img-csv $file_list_input \\\n",
    "--mode 2\n",
    "conda deactivate\n",
    "conda deactivate"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 4. Cross-analyze deep segmentation and regression results"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Scripts in the <a href=\"https://github.com/naglemi/GMOlabeler\" target=\"_blank\">GMOlabeler repository</a> are used to cross-reference results from deep segmentation of RGB images and regression of hyperspectral imaging, apply thresholding parameters to classify tissues as transgenic or escapes, and produce plots."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<img src=\"Figures/GMOlabeler.png\">"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Figure: The various steps of data processing in GMOlabeler are illustrated for an example explant from an experiment on CIM optimization for cottonwood. Images of plates are cropped to a sub-image for each explant. RGB segmentation results and hyperspectral regression results are cross-referenced to calculate fluorecent proteins in specific tissues and infer whether these tissues are transgenic."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### 4.1. Prepare sample datasheet input"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Prepare input file we will use for making plots. This file contains paths to CLS results, RGB images, and hyperspectral images."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"”2020-10-27”\"\n",
      "[1] \"Looking for CLS data in: /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//\"\n",
      "[1] \"2020-10-27\"\n",
      "[1] \"Looking for CLS files in directory /scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//\"\n",
      "[1] \"How many CLS files? 90\"\n",
      "[1] \"Writing 90 rows to /scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///samples_pre_labeling.csv\"\n"
     ]
    }
   ],
   "source": [
    "cd /scratch2/NSF_GWAS/GMOdetectoR/\n",
    "Rscript wrappers/pre_label_a3.R \\\n",
    "-r \"${data}/\" \\\n",
    "-R \"/scratch2/NSF_GWAS/gmodetector_out/\" \\\n",
    "-i 1 \\\n",
    "-d $datestamp"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### 4.2. Cross-reference RGB and hyperspectral data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [],
   "source": [
    "#cat \"${data}/samples_pre_labeling.csv\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "12\n"
     ]
    }
   ],
   "source": [
    "echo $grid"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/scratch2/NSF_GWAS/macroPhor_Array/grids/grids_left_facing_125208_1_0_1_rgb_processed.jpg JPEG 1419x1566 1419x1566+0+0 8-bit sRGB 346442B 0.000u 0:00.000\n"
     ]
    }
   ],
   "source": [
    "identify $grid_file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(base) (base) Grid type 12\n",
      "Loading plate 0 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA4_I5.0_F1.9_L100_175854_5_2_6_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA4_I5.0_F1.9_L100_175854_5_2_6_weights.hdf\n",
      "Loading plate 1 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA4_I5.0_F1.9_L100_175759_4_2_5_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA4_I5.0_F1.9_L100_175759_4_2_5_weights.hdf\n",
      "Loading plate 2 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA4_I5.0_F1.9_L100_175704_3_2_4_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA4_I5.0_F1.9_L100_175704_3_2_4_weights.hdf\n",
      "Loading plate 3 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA4_I5.0_F1.9_L100_175610_2_2_3_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA4_I5.0_F1.9_L100_175610_2_2_3_weights.hdf\n",
      "Loading plate 4 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA4_I5.0_F1.9_L100_175515_1_2_2_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA4_I5.0_F1.9_L100_175515_1_2_2_weights.hdf\n",
      "Loading plate 5 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA4_I5.0_F1.9_L100_175417_0_2_1_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA4_I5.0_F1.9_L100_175417_0_2_1_weights.hdf\n",
      "Loading plate 6 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA5_I5.0_F1.9_L100_174910_11_1_0_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA5_I5.0_F1.9_L100_174910_11_1_0_weights.hdf\n",
      "Loading plate 7 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA5_I5.0_F1.9_L100_174815_10_1_1_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA5_I5.0_F1.9_L100_174815_10_1_1_weights.hdf\n",
      "Loading plate 8 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA5_I5.0_F1.9_L100_174720_9_1_2_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA5_I5.0_F1.9_L100_174720_9_1_2_weights.hdf\n",
      "Loading plate 9 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA5_I5.0_F1.9_L100_174625_8_1_3_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA5_I5.0_F1.9_L100_174625_8_1_3_weights.hdf\n",
      "Loading plate 10 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA5_I5.0_F1.9_L100_174528_7_1_4_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA5_I5.0_F1.9_L100_174528_7_1_4_weights.hdf\n",
      "Loading plate 11 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA5_I5.0_F1.9_L100_174432_6_0_6_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA5_I5.0_F1.9_L100_174432_6_0_6_weights.hdf\n",
      "Loading plate 12 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA5_I5.0_F1.9_L100_174335_5_0_5_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA5_I5.0_F1.9_L100_174335_5_0_5_weights.hdf\n",
      "Loading plate 13 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA5_I5.0_F1.9_L100_174241_4_0_4_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA5_I5.0_F1.9_L100_174241_4_0_4_weights.hdf\n",
      "Loading plate 14 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA5_I5.0_F1.9_L100_174146_3_0_3_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA5_I5.0_F1.9_L100_174146_3_0_3_weights.hdf\n",
      "Loading plate 15 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA5_I5.0_F1.9_L100_174051_2_0_2_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA5_I5.0_F1.9_L100_174051_2_0_2_weights.hdf\n",
      "Loading plate 16 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA5_I5.0_F1.9_L100_173952_1_0_1_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA5_I5.0_F1.9_L100_173952_1_0_1_weights.hdf\n",
      "Loading plate 17 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA5_I5.0_F1.9_L100_173911_0_0_0_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA5_I5.0_F1.9_L100_173911_0_0_0_weights.hdf\n",
      "Loading plate 18 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA4_I5.0_F1.9_L100_171951_13_2_0_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA4_I5.0_F1.9_L100_171951_13_2_0_weights.hdf\n",
      "Loading plate 19 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA4_I5.0_F1.9_L100_171855_12_1_0_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA4_I5.0_F1.9_L100_171855_12_1_0_weights.hdf\n",
      "Loading plate 20 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA4_I5.0_F1.9_L100_171759_11_1_1_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA4_I5.0_F1.9_L100_171759_11_1_1_weights.hdf\n",
      "Loading plate 21 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA4_I5.0_F1.9_L100_171705_10_1_2_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA4_I5.0_F1.9_L100_171705_10_1_2_weights.hdf\n",
      "Loading plate 22 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA4_I5.0_F1.9_L100_171610_9_1_3_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA4_I5.0_F1.9_L100_171610_9_1_3_weights.hdf\n",
      "Loading plate 23 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA4_I5.0_F1.9_L100_171515_8_1_4_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA4_I5.0_F1.9_L100_171515_8_1_4_weights.hdf\n",
      "Loading plate 24 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA4_I5.0_F1.9_L100_171419_7_1_5_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA4_I5.0_F1.9_L100_171419_7_1_5_weights.hdf\n",
      "Loading plate 25 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA4_I5.0_F1.9_L100_171330_6_1_6_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA4_I5.0_F1.9_L100_171330_6_1_6_weights.hdf\n",
      "Loading plate 26 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA4_I5.0_F1.9_L100_171234_5_0_5_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA4_I5.0_F1.9_L100_171234_5_0_5_weights.hdf\n",
      "Loading plate 27 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA4_I5.0_F1.9_L100_171140_4_0_4_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA4_I5.0_F1.9_L100_171140_4_0_4_weights.hdf\n",
      "Loading plate 28 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA4_I5.0_F1.9_L100_171045_3_0_3_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA4_I5.0_F1.9_L100_171045_3_0_3_weights.hdf\n",
      "Loading plate 29 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA4_I5.0_F1.9_L100_170950_2_0_2_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA4_I5.0_F1.9_L100_170950_2_0_2_weights.hdf\n",
      "Loading plate 30 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA4_I5.0_F1.9_L100_170854_1_0_1_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA4_I5.0_F1.9_L100_170854_1_0_1_weights.hdf\n",
      "Loading plate 31 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA4_I5.0_F1.9_L100_170813_0_0_0_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA4_I5.0_F1.9_L100_170813_0_0_0_weights.hdf\n",
      "Loading plate 32 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA2_I5.0_F1.9_L100_165800_17_2_6_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA2_I5.0_F1.9_L100_165800_17_2_6_weights.hdf\n",
      "Loading plate 33 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA2_I5.0_F1.9_L100_165705_16_2_5_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA2_I5.0_F1.9_L100_165705_16_2_5_weights.hdf\n",
      "Loading plate 34 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA2_I5.0_F1.9_L100_165610_15_2_4_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA2_I5.0_F1.9_L100_165610_15_2_4_weights.hdf\n",
      "Loading plate 35 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA2_I5.0_F1.9_L100_165513_14_2_3_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA2_I5.0_F1.9_L100_165513_14_2_3_weights.hdf\n",
      "Loading plate 36 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA2_I5.0_F1.9_L100_165418_13_2_2_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA2_I5.0_F1.9_L100_165418_13_2_2_weights.hdf\n",
      "Loading plate 37 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA2_I5.0_F1.9_L100_165325_12_2_1_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA2_I5.0_F1.9_L100_165325_12_2_1_weights.hdf\n",
      "Loading plate 38 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA2_I5.0_F1.9_L100_165239_11_2_0_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA2_I5.0_F1.9_L100_165239_11_2_0_weights.hdf\n",
      "Loading plate 39 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA2_I5.0_F1.9_L100_165144_10_1_0_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA2_I5.0_F1.9_L100_165144_10_1_0_weights.hdf\n",
      "Loading plate 40 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA2_I5.0_F1.9_L100_165049_9_1_1_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA2_I5.0_F1.9_L100_165049_9_1_1_weights.hdf\n",
      "Loading plate 41 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA2_I5.0_F1.9_L100_164955_8_1_2_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA2_I5.0_F1.9_L100_164955_8_1_2_weights.hdf\n",
      "Loading plate 42 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA2_I5.0_F1.9_L100_164901_7_1_3_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA2_I5.0_F1.9_L100_164901_7_1_3_weights.hdf\n",
      "Loading plate 43 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA2_I5.0_F1.9_L100_164807_6_1_4_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA2_I5.0_F1.9_L100_164807_6_1_4_weights.hdf\n",
      "Loading plate 44 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA2_I5.0_F1.9_L100_164708_5_1_5_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA2_I5.0_F1.9_L100_164708_5_1_5_weights.hdf\n",
      "Loading plate 45 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA2_I5.0_F1.9_L100_164623_4_1_6_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA2_I5.0_F1.9_L100_164623_4_1_6_weights.hdf\n",
      "Loading plate 46 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA2_I5.0_F1.9_L100_164528_3_0_6_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA2_I5.0_F1.9_L100_164528_3_0_6_weights.hdf\n",
      "Loading plate 47 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA2_I5.0_F1.9_L100_164433_2_0_5_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA2_I5.0_F1.9_L100_164433_2_0_5_weights.hdf\n",
      "Loading plate 48 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA2_I5.0_F1.9_L100_164338_1_0_4_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA2_I5.0_F1.9_L100_164338_1_0_4_weights.hdf\n",
      "Loading plate 49 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA2_I5.0_F1.9_L100_164229_0_0_3_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA2_I5.0_F1.9_L100_164229_0_0_3_weights.hdf\n",
      "Loading plate 50 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA3_I5.0_F1.9_L100_163834_19_2_6_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA3_I5.0_F1.9_L100_163834_19_2_6_weights.hdf\n",
      "Loading plate 51 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA3_I5.0_F1.9_L100_163739_18_2_5_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA3_I5.0_F1.9_L100_163739_18_2_5_weights.hdf\n",
      "Loading plate 52 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA3_I5.0_F1.9_L100_163639_17_2_4_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA3_I5.0_F1.9_L100_163639_17_2_4_weights.hdf\n",
      "Loading plate 53 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA3_I5.0_F1.9_L100_163544_16_2_2_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA3_I5.0_F1.9_L100_163544_16_2_2_weights.hdf\n",
      "Loading plate 54 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA3_I5.0_F1.9_L100_163450_15_2_1_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA3_I5.0_F1.9_L100_163450_15_2_1_weights.hdf\n",
      "Loading plate 55 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA1_I5.0_F1.9_L100_163432_19_2_6_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA1_I5.0_F1.9_L100_163432_19_2_6_weights.hdf\n",
      "Loading plate 56 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA3_I5.0_F1.9_L100_163401_14_2_0_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA3_I5.0_F1.9_L100_163401_14_2_0_weights.hdf\n",
      "Loading plate 57 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA1_I5.0_F1.9_L100_163322_18_2_5_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA1_I5.0_F1.9_L100_163322_18_2_5_weights.hdf\n",
      "Loading plate 58 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA3_I5.0_F1.9_L100_163306_13_1_0_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA3_I5.0_F1.9_L100_163306_13_1_0_weights.hdf\n",
      "Loading plate 59 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA1_I5.0_F1.9_L100_163228_17_2_4_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA1_I5.0_F1.9_L100_163228_17_2_4_weights.hdf\n",
      "Loading plate 60 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA3_I5.0_F1.9_L100_163211_12_1_1_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA3_I5.0_F1.9_L100_163211_12_1_1_weights.hdf\n",
      "Loading plate 61 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA1_I5.0_F1.9_L100_163133_16_2_3_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA1_I5.0_F1.9_L100_163133_16_2_3_weights.hdf\n",
      "Loading plate 62 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA3_I5.0_F1.9_L100_163114_11_1_2_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA3_I5.0_F1.9_L100_163114_11_1_2_weights.hdf\n",
      "Loading plate 63 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA1_I5.0_F1.9_L100_163039_15_2_2_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA1_I5.0_F1.9_L100_163039_15_2_2_weights.hdf\n",
      "Loading plate 64 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA3_I5.0_F1.9_L100_163019_10_1_3_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA3_I5.0_F1.9_L100_163019_10_1_3_weights.hdf\n",
      "Loading plate 65 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA1_I5.0_F1.9_L100_162939_14_2_1_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA1_I5.0_F1.9_L100_162939_14_2_1_weights.hdf\n",
      "Loading plate 66 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA3_I5.0_F1.9_L100_162924_9_1_4_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA3_I5.0_F1.9_L100_162924_9_1_4_weights.hdf\n",
      "Loading plate 67 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA1_I5.0_F1.9_L100_162853_13_2_0_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA1_I5.0_F1.9_L100_162853_13_2_0_weights.hdf\n",
      "Loading plate 68 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA3_I5.0_F1.9_L100_162828_8_1_5_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA3_I5.0_F1.9_L100_162828_8_1_5_weights.hdf\n",
      "Loading plate 69 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA1_I5.0_F1.9_L100_162757_12_1_0_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA1_I5.0_F1.9_L100_162757_12_1_0_weights.hdf\n",
      "Loading plate 70 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA3_I5.0_F1.9_L100_162743_7_1_6_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA3_I5.0_F1.9_L100_162743_7_1_6_weights.hdf\n",
      "Loading plate 71 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA1_I5.0_F1.9_L100_162649_11_1_2_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA1_I5.0_F1.9_L100_162649_11_1_2_weights.hdf\n",
      "Loading plate 72 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA3_I5.0_F1.9_L100_162648_6_0_6_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA3_I5.0_F1.9_L100_162648_6_0_6_weights.hdf\n",
      "Loading plate 73 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA1_I5.0_F1.9_L100_162555_10_1_3_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA1_I5.0_F1.9_L100_162555_10_1_3_weights.hdf\n",
      "Loading plate 74 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA3_I5.0_F1.9_L100_162553_5_0_5_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA3_I5.0_F1.9_L100_162553_5_0_5_weights.hdf\n",
      "Loading plate 75 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA3_I5.0_F1.9_L100_162457_4_0_4_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA3_I5.0_F1.9_L100_162457_4_0_4_weights.hdf\n",
      "Loading plate 76 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA1_I5.0_F1.9_L100_162450_9_1_4_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA1_I5.0_F1.9_L100_162450_9_1_4_weights.hdf\n",
      "Loading plate 77 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA3_I5.0_F1.9_L100_162402_3_0_3_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA3_I5.0_F1.9_L100_162402_3_0_3_weights.hdf\n",
      "Loading plate 78 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA1_I5.0_F1.9_L100_162354_8_1_5_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA1_I5.0_F1.9_L100_162354_8_1_5_weights.hdf\n",
      "Loading plate 79 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA1_I5.0_F1.9_L100_162308_7_1_6_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA1_I5.0_F1.9_L100_162308_7_1_6_weights.hdf\n",
      "Loading plate 80 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA3_I5.0_F1.9_L100_162308_2_0_2_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA3_I5.0_F1.9_L100_162308_2_0_2_weights.hdf\n",
      "Loading plate 81 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA1_I5.0_F1.9_L100_162212_6_0_6_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA1_I5.0_F1.9_L100_162212_6_0_6_weights.hdf\n",
      "Loading plate 82 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA3_I5.0_F1.9_L100_162212_1_0_1_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA3_I5.0_F1.9_L100_162212_1_0_1_weights.hdf\n",
      "Loading plate 83 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA3_I5.0_F1.9_L100_162131_0_0_0_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA3_I5.0_F1.9_L100_162131_0_0_0_weights.hdf\n",
      "Loading plate 84 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA1_I5.0_F1.9_L100_162117_5_0_5_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA1_I5.0_F1.9_L100_162117_5_0_5_weights.hdf\n",
      "Loading plate 85 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA1_I5.0_F1.9_L100_162020_4_0_4_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA1_I5.0_F1.9_L100_162020_4_0_4_weights.hdf\n",
      "Loading plate 86 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA1_I5.0_F1.9_L100_161923_3_0_3_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA1_I5.0_F1.9_L100_161923_3_0_3_weights.hdf\n",
      "Loading plate 87 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA1_I5.0_F1.9_L100_161827_2_0_2_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA1_I5.0_F1.9_L100_161827_2_0_2_weights.hdf\n",
      "Loading plate 88 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA1_I5.0_F1.9_L100_161732_1_0_1_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA1_I5.0_F1.9_L100_161732_1_0_1_weights.hdf\n",
      "Loading plate 89 of 90\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA1_I5.0_F1.9_L100_161651_0_0_0_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/scratch2/NSF_GWAS/gmodetector_out/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA1_I5.0_F1.9_L100_161651_0_0_0_weights.hdf\n",
      "(base) "
     ]
    }
   ],
   "source": [
    "conda activate\n",
    "cd /scratch2/NSF_GWAS/GMOlabeler/\n",
    "python main.py \\\n",
    "\"${data}/samples_pre_labeling.csv\" \\\n",
    "$grid_file \\\n",
    "$reporter_threshold \\\n",
    "$reporter \\\n",
    "$grid \\\n",
    "\"hdf\"\n",
    "conda deactivate"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### 4.3. Calculate sums of statistics over combined segments"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We are interested in all regenerated tissue (callus + shoot) as well as all tissue (including stem as well). We will calculate aggregate statistics over these groups."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[1] \"Writing output with sums statistics calculated over combined tissue segments to: /scratch2/NSF_GWAS/GMOlabeler/output/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//stats_with_sums_over_tissues.csv\"\n"
     ]
    }
   ],
   "source": [
    "Rscript calculate_sum_stats_over_combined_segments.R \\\n",
    "--datapath \"${data_folder}/\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### 4.4. Make plots of results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/GFPA_randomized list_MNrevised.xlsx\n",
      "3\n",
      "\n",
      "1\n",
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//output.csv\n",
      "12\n",
      "1\n",
      "5\n",
      "9\n"
     ]
    }
   ],
   "source": [
    "echo \"${data_folder}/\"\n",
    "echo $randomization_datasheet\n",
    "echo $pixel_threshold\n",
    "echo $variable_type\n",
    "echo 1\n",
    "echo $missing_explants\n",
    "echo $grid\n",
    "echo 1\n",
    "echo $height\n",
    "echo $width"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//output.csv\n"
     ]
    }
   ],
   "source": [
    "echo $missing_explants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/GFPA_randomized list_MNrevised.xlsx\n"
     ]
    }
   ],
   "source": [
    "echo $randomization_datasheet"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Warning messages:\n",
      "1: replacing previous import ‘data.table::melt’ by ‘reshape2::melt’ when loading ‘GMOdetectoR’ \n",
      "2: replacing previous import ‘data.table::dcast’ by ‘reshape2::dcast’ when loading ‘GMOdetectoR’ \n",
      "Warning message:\n",
      "In storage.mode(default) <- type : NAs introduced by coercion\n",
      "[1] \"Saving list of input arguments to : /scratch2/NSF_GWAS/GMOlabeler/plots/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//gridplot_args.rds\"\n",
      "[1] \"Reading in output from GMOlabeler at path: /scratch2/NSF_GWAS/GMOlabeler/output/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//stats_with_sums_over_tissues.csv\"\n",
      "\n",
      "[1] \"Rows in output from GMOlabeler: 4359\"\n",
      "\n",
      "[1] \"Max n_pixels_passing_threshold in output from GMOlabeler: 2626\"\n",
      "\n",
      "[1] \"Max total_signal in output from GMOlabeler: 471922.685884808\"\n",
      "\n",
      "[1] \"Look at the top of output from GMOlabeler\"\n",
      "[1] \"\"                                                                                                                       \n",
      "[2] \"/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA4_I5.0_F1.9_L100_175854_5_2_6.png\"\n",
      "[3] \"/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA4_I5.0_F1.9_L100_175854_5_2_6.png\"\n",
      "[4] \"/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA4_I5.0_F1.9_L100_175854_5_2_6.png\"\n",
      "[5] \"/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA4_I5.0_F1.9_L100_175854_5_2_6.png\"\n",
      "[6] \"/scratch2/NSF_GWAS/macroPhor_Array/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent///GFPA4_I5.0_F1.9_L100_175854_5_2_6.png\"\n",
      "\n",
      "[1] \"Look at the top of missing explant data\"\n",
      "\u001b[90m# A tibble: 6 x 3\u001b[39m\n",
      "  filename                                  grid_item present\n",
      "  \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m                                     \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m     \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m  \n",
      "\u001b[90m1\u001b[39m GFPA1_I5.0_F1.9_L100_161651_0_0_0_rgb.jpg 1         NC     \n",
      "\u001b[90m2\u001b[39m GFPA1_I5.0_F1.9_L100_161651_0_0_0_rgb.jpg 2         NC     \n",
      "\u001b[90m3\u001b[39m GFPA1_I5.0_F1.9_L100_161651_0_0_0_rgb.jpg 3         C      \n",
      "\u001b[90m4\u001b[39m GFPA1_I5.0_F1.9_L100_161651_0_0_0_rgb.jpg 4         NC     \n",
      "\u001b[90m5\u001b[39m GFPA1_I5.0_F1.9_L100_161651_0_0_0_rgb.jpg 5         NC     \n",
      "\u001b[90m6\u001b[39m GFPA1_I5.0_F1.9_L100_161651_0_0_0_rgb.jpg 6         NC     \n",
      "\n",
      "[1] \"Look at the head of filenames for missing explant data\"\n",
      "[1] \"GFPA1_I5.0_F1.9_L100_161651_0_0_0_rgb.jpg\"\n",
      "[2] \"GFPA1_I5.0_F1.9_L100_161651_0_0_0_rgb.jpg\"\n",
      "[3] \"GFPA1_I5.0_F1.9_L100_161651_0_0_0_rgb.jpg\"\n",
      "[4] \"GFPA1_I5.0_F1.9_L100_161651_0_0_0_rgb.jpg\"\n",
      "[5] \"GFPA1_I5.0_F1.9_L100_161651_0_0_0_rgb.jpg\"\n",
      "[6] \"GFPA1_I5.0_F1.9_L100_161651_0_0_0_rgb.jpg\"\n",
      "\n",
      "[1] \"Look at the colnames of output from GMOlabeler\"\n",
      " [1] \"V1\"                         \"filename\"                  \n",
      " [3] \"grid_item\"                  \"segment_hex\"               \n",
      " [5] \"intensity_threshold\"        \"n_pixels_passing_threshold\"\n",
      " [7] \"mean_signal\"                \"max_signal\"                \n",
      " [9] \"total_signal\"               \"total_pixels\"              \n",
      "\n",
      "[1] \"Look at the colnames of missing explant data\"\n",
      "[1] \"filename\"  \"grid_item\" \"present\"  \n",
      "\n",
      "[1] \"Before merging the GMOlabeler and missing output data, they have rows of 4359 and 1080 respectively.\"\n",
      "\n",
      "[1] \"After merging but before removing missing/contminated explants, merged output has 4359 rows.\"\n",
      "\n",
      "[1] \"After removing the missing/contaminated explants, merged output has 4359\"\n",
      "\n",
      "[1] \"Afer processing GMOlabeler output to merge with missing explant data, Head of data (w/ first 5 col): \"\n",
      "                                filename grid_item          V1 segment_hex\n",
      "1:                                            <NA> observation            \n",
      "2: GFPA1_I5.0_F1.9_L100_161651_0_0_0.png         1 observation      00CC11\n",
      "3: GFPA1_I5.0_F1.9_L100_161651_0_0_0.png         1 observation      0006CC\n",
      "4: GFPA1_I5.0_F1.9_L100_161651_0_0_0.png         1 observation      CC0000\n",
      "5: GFPA1_I5.0_F1.9_L100_161651_0_0_0.png         1 observation  All_tissue\n",
      "   intensity_threshold\n",
      "1:                  NA\n",
      "2:                  38\n",
      "3:                  38\n",
      "4:                  38\n",
      "5:                  NA\n",
      "New names:\n",
      "* `` -> ...12\n",
      "[1] \"Upon loading randomization_datasheet, it has how many rows? 96\"\n",
      "\n",
      "[1] \"Calculating total numbers of missing explants per plate\"\n",
      "[1] \"Let's figure out why contaminated/missing plates are being said to have 0 instead of NA\"\n",
      "\n",
      "[1] \"Dimensions of missing explant data with totals\"\n",
      "[1] 90 14\n",
      "[1] \"Minimum number of explants on any plate after removing missing and contaminated explants is...\"\n",
      "[1] 2\n",
      "                                  image_name 1 2 3 4 5 6 7 8 9 10 11 12\n",
      "1: GFPA1_I5.0_F1.9_L100_161651_0_0_0_rgb.jpg 1 1 0 1 1 1 1 1 1  1  1  1\n",
      "2: GFPA1_I5.0_F1.9_L100_161732_1_0_1_rgb.jpg 1 1 1 1 1 1 1 1 1  1  1  1\n",
      "3: GFPA1_I5.0_F1.9_L100_161827_2_0_2_rgb.jpg 1 1 1 1 1 1 1 1 1  1  1  1\n",
      "4: GFPA1_I5.0_F1.9_L100_161923_3_0_3_rgb.jpg 1 1 1 1 1 1 1 1 1  1  1  1\n",
      "5: GFPA1_I5.0_F1.9_L100_162020_4_0_4_rgb.jpg 1 1 1 1 1 1 1 1 1  0  0  0\n",
      "6: GFPA1_I5.0_F1.9_L100_162117_5_0_5_rgb.jpg 1 1 1 1 1 1 1 1 1  1  1  1\n",
      "   total_explants\n",
      "1:             11\n",
      "2:             12\n",
      "3:             12\n",
      "4:             12\n",
      "5:              9\n",
      "6:             12\n",
      "\n",
      "[1] \"About to add IDs to missing explant data based on filenames. Here is head of filenames.\"\n",
      "[1] \"GFPA1_I5.0_F1.9_L100_161651_0_0_0\" \"GFPA1_I5.0_F1.9_L100_161732_1_0_1\"\n",
      "[3] \"GFPA1_I5.0_F1.9_L100_161827_2_0_2\" \"GFPA1_I5.0_F1.9_L100_161923_3_0_3\"\n",
      "[5] \"GFPA1_I5.0_F1.9_L100_162020_4_0_4\" \"GFPA1_I5.0_F1.9_L100_162117_5_0_5\"\n",
      "\n",
      "[1] \"We have added IDs to missing explant data based on filenames. Here is head of IDs.\"\n",
      "[1] \"GFPA1_1\" \"GFPA1_2\" \"GFPA1_3\" \"GFPA1_4\" \"GFPA1_5\" \"GFPA1_6\"\n",
      "\n",
      "[1] \"N rows of randomization datasheet before merging with missing explant data: \"\n",
      "[1] 96\n",
      "[1] \"Head of IDs in missing explant data\"\n",
      "[1] \"GFPA1_1\" \"GFPA1_2\" \"GFPA1_3\" \"GFPA1_4\" \"GFPA1_5\" \"GFPA1_6\"\n",
      "[1] \"Head of IDs in randomization datasheet\"\n",
      "[1] \"GFPA1_1\" \"GFPA1_2\" \"GFPA1_3\" \"GFPA1_4\" \"GFPA1_5\" \"GFPA1_6\"\n",
      "\n",
      "[1] \"Now about to merge randomization_datasheet and missing explant data.\"\n",
      "[1] \"Rows in randomization_datasheet: 96\"\n",
      "[1] \"Rows in missing explant data: 90\"\n",
      "\n",
      "[1] \"Now about to merge randomization datasheet and missing explant data, by ID. Look at head of IDs for both.\"\n",
      "[1] \"GFPA1_1\" \"GFPA1_2\" \"GFPA1_3\" \"GFPA1_4\" \"GFPA1_5\" \"GFPA1_6\"\n",
      "[1] \"GFPA1_1\" \"GFPA1_2\" \"GFPA1_3\" \"GFPA1_4\" \"GFPA1_5\" \"GFPA1_6\"\n",
      "        ID Image # Tray ID Rand # Date on Media *Plate # Index# in GH\n",
      "1  GFPA1_1       1   GFPA1      1    2020-09-17  3-2-1-2           R7\n",
      "2 GFPA1_10      10   GFPA1     10    2020-09-17  3-2-1-5           R7\n",
      "3 GFPA1_11      11   GFPA1     11    2020-09-17  1-2-1-1          140\n",
      "4 GFPA1_12      12   GFPA1     12    2020-09-17  4-2-1-3          R21\n",
      "5 GFPA1_13      13   GFPA1     13    2020-09-17  4-2-1-6          R21\n",
      "6 GFPA1_14      14   GFPA1     14    2020-09-17  3-1-2-6           R7\n",
      "  Genotype_ID Costruct Media Treatment name Replicate ...12     LableTop\n",
      "1   Nisqually      Mar   CTR        CTR Mar         2    NA   GFPA1_1_R1\n",
      "2   Nisqually      Mar   CTR        CTR Mar         5    NA GFPA1_10_R10\n",
      "3    BESC-134      Mar   CTR        CTR Mar         1    NA GFPA1_11_R11\n",
      "4   SKWB-24-2      Mar   CTR        CTR Mar         3    NA GFPA1_12_R12\n",
      "5   SKWB-24-2      Mar   CTR        CTR Mar         6    NA GFPA1_13_R13\n",
      "6   Nisqually     HSPT    AS        AS HSPT         6    NA GFPA1_14_R14\n",
      "             LableBot Plate#      Rand                           filename 1 2 3\n",
      "1 Nisqually_Mar_CTR_2     62 0.3583730  GFPA1_I5.0_F1.9_L100_161651_0_0_0 1 1 0\n",
      "2 Nisqually_Mar_CTR_5     65 0.8666917 GFPA1_I5.0_F1.9_L100_162649_11_1_2 1 1 1\n",
      "3  BESC-134_Mar_CTR_1     13 0.5296783 GFPA1_I5.0_F1.9_L100_162555_10_1_3 1 1 1\n",
      "4 SKWB-24-2_Mar_CTR_3     87 0.8703313  GFPA1_I5.0_F1.9_L100_162450_9_1_4 1 1 1\n",
      "5 SKWB-24-2_Mar_CTR_6     90 0.4970979  GFPA1_I5.0_F1.9_L100_162354_8_1_5 1 1 1\n",
      "6 Nisqually_HSPT_AS_6     60 0.6426024  GFPA1_I5.0_F1.9_L100_162308_7_1_6 1 1 1\n",
      "  4 5 6 7 8 9 10 11 12 total_explants\n",
      "1 1 1 1 1 1 1  1  1  1             11\n",
      "2 1 1 1 1 1 1  1  1  1             12\n",
      "3 1 0 1 1 1 1  1  1  1             11\n",
      "4 1 1 1 1 1 1  1  1  1             12\n",
      "5 1 1 0 1 1 1  1  1  1             11\n",
      "6 1 1 1 0 1 1  1  1  1             11\n",
      " [1] 11 12 11 12 11 11 10 11 12 11 10 12 11 11 12 12  9 12 12  9 NA NA 12 12  7\n",
      "[26] 12 11 11 11  8  9 11 NA  7 12 NA  7 10  9 12 11 11  8 10 11 11 11 12  5 12\n",
      "[51] 12 NA 10  7 11 10  9 11 12 12 11 12 12  9  8  6 11 12 12 10 12 11 11  9 12\n",
      "[76] 12 12 12 11 12  9 NA  8 10  8 12 10 10  2  7 12 12 10  9  8  8\n",
      "[1] \"Still figuring, now in merged rd, out why contaminated/missing plates are being said to have 0 instead of NA\"\n",
      "\n",
      "[1] \"Dimensions of missing explant data with totals\"\n",
      "[1] 90 15\n",
      "[1] \"Levels of total explants on any plate after removing missing and contaminated explants is...\"\n",
      "[1] \"2\"  \"5\"  \"6\"  \"7\"  \"8\"  \"9\"  \"10\" \"11\" \"12\"\n",
      "[1] \"Minimum number of explants on any plate after removing missing and contaminated explants is...\"\n",
      "[1] 2\n",
      "\n",
      "[1] \"Rows in merged output: 96\"\n",
      " [1] 11 12 11 12 11 11 10 11 12 11 10 12 11 11 12 12  9 12 12  9 NA NA 12 12  7\n",
      "[26] 12 11 11 11  8  9 11 NA  7 12 NA  7 10  9 12 11 11  8 10 11 11 11 12  5 12\n",
      "[51] 12 NA 10  7 11 10  9 11 12 12 11 12 12  9  8  6 11 12 12 10 12 11 11  9 12\n",
      "[76] 12 12 12 11 12  9 NA  8 10  8 12 10 10  2  7 12 12 10  9  8  8\n",
      "\n",
      "[1] \"Assuming for all explants for which we have no missing explant data that there are:\"\n",
      "[1] 12\n",
      "\n",
      "[1] \"N rows of randomization datasheet after merging with missing explant data: \"\n",
      "[1] 96\n",
      "[1] \"Column names of randomization datasheet afer merging with missing explant data: \"\n",
      " [1] \"ID\"             \"Image #\"        \"Tray ID\"        \"Rand #\"        \n",
      " [5] \"Date on Media\"  \"*Plate #\"       \"Index# in GH\"   \"Genotype_ID\"   \n",
      " [9] \"Costruct\"       \"Media\"          \"Treatment name\" \"Replicate\"     \n",
      "[13] \"...12\"          \"LableTop\"       \"LableBot\"       \"Plate#\"        \n",
      "[17] \"Rand\"           \"filename\"       \"1\"              \"2\"             \n",
      "[21] \"3\"              \"4\"              \"5\"              \"6\"             \n",
      "[25] \"7\"              \"8\"              \"9\"              \"10\"            \n",
      "[29] \"11\"             \"12\"             \"total_explants\"\n",
      "\n",
      "[1] \"In output, Maximum observations for a segment in a grid item in a filename (should be 1): 1\"\n",
      "[1] \"In randomization datasheet, Maximum observations for an ID (should be 1): 1\"\n",
      "\n",
      "[1] \"Before calculating totals, let's look at the column total_explants in randomization datasheet\"\n",
      "[1] 11 12 11 12 11 11\n",
      "[1] \"2\"  \"5\"  \"6\"  \"7\"  \"8\"  \"9\"  \"10\" \"11\" \"12\"\n",
      "\n",
      "\n",
      "[1] \"Nrow of output before dropping missing explants is: 4359\"\n",
      "[1] \"Head of ID_exp columns from both output data table and IDs_to_drop data tbales, respectively:\"\n",
      "[1] \"__NA\"       \"GFPA1_1_1\"  \"GFPA1_1_1\"  \"GFPA1_1_1\"  \"GFPA1_1_1\" \n",
      "[6] \"GFPA1_1_10\"\n",
      "[1] \"GFPA1_1_3\"  \"GFPA1_5_10\" \"GFPA1_5_11\" \"GFPA1_5_12\" \"GFPA1_14_7\"\n",
      "[6] \"GFPA1_13_6\"\n",
      "[1] \"Nrow of output AFTER dropping missing explants is: 3759\"\n",
      "[1] \"Head of output\"\n",
      "   filename grid_item          V1 segment_hex intensity_threshold\n",
      "1:               <NA> observation                              NA\n",
      "   n_pixels_passing_threshold mean_signal max_signal total_signal total_pixels\n",
      "1:                         NA          NA         NA           NA           NA\n",
      "   present ID transgenic segment_present ID_exp\n",
      "1:       Y  _          0               0   __NA\n",
      "\n",
      "\u001b[90m# A tibble: 6 x 5\u001b[39m\n",
      "  filename                              grid_item present ID       ID_exp    \n",
      "  \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m                                 \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m     \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m   \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m    \u001b[3m\u001b[90m<chr>\u001b[39m\u001b[23m     \n",
      "\u001b[90m1\u001b[39m GFPA1_I5.0_F1.9_L100_161651_0_0_0.jpg 3         C       GFPA1_1  GFPA1_1_3 \n",
      "\u001b[90m2\u001b[39m GFPA1_I5.0_F1.9_L100_162020_4_0_4.jpg 10        M       GFPA1_5  GFPA1_5_10\n",
      "\u001b[90m3\u001b[39m GFPA1_I5.0_F1.9_L100_162020_4_0_4.jpg 11        M       GFPA1_5  GFPA1_5_11\n",
      "\u001b[90m4\u001b[39m GFPA1_I5.0_F1.9_L100_162020_4_0_4.jpg 12        M       GFPA1_5  GFPA1_5_12\n",
      "\u001b[90m5\u001b[39m GFPA1_I5.0_F1.9_L100_162308_7_1_6.jpg 7         C       GFPA1_14 GFPA1_14_7\n",
      "\u001b[90m6\u001b[39m GFPA1_I5.0_F1.9_L100_162354_8_1_5.jpg 6         C       GFPA1_13 GFPA1_13_6\n",
      " [1] \"filename\"                   \"grid_item\"                 \n",
      " [3] \"V1\"                         \"segment_hex\"               \n",
      " [5] \"intensity_threshold\"        \"n_pixels_passing_threshold\"\n",
      " [7] \"mean_signal\"                \"max_signal\"                \n",
      " [9] \"total_signal\"               \"total_pixels\"              \n",
      "[11] \"present\"                    \"ID\"                        \n",
      "[13] \"transgenic\"                 \"segment_present\"           \n",
      "[15] \"ID_exp\"                    \n",
      "[1] \"Maximum # grid positions with transgenic stem in any plate: 8\"\n",
      "[1] \"Maximum # grid positions with transgenic callus in any plate: 1\"\n",
      "[1] \"Maximum # grid positions with transgenic shoot in any plate: 0\"\n",
      "[1] \"Minimum # grid positions with explants in any plate: 2\"\n",
      "[1] \"Maximum # grid positions with explants in any plate: 12\"\n",
      "[1] \"Maximum # grid positions with callus in any plate: 3\"\n",
      "[1] \"Maximum # grid positions with shoot in any plate: 0\"\n",
      "[1] \"Column of interest is portion_transgenic_Shoot\"\n",
      "[1] \"For transgenic tissue, What is the range of values for portion for Shoot\"\n",
      "[1] 0\n",
      "[1] 0\n",
      "[1] \"What are factor levels?\"\n",
      "[1] \"0\"\n",
      "\n",
      "[1] \"Column of interest is portion_Shoot\"\n",
      "[1] \"For total tissue, What is the range of values for portion for Shoot\"\n",
      "[1] 0\n",
      "[1] 0\n",
      "[1] \"What are factor levels?\"\n",
      "[1] \"0\"\n",
      "\n",
      "[1] \"Column of interest is portion_transgenic_Callus\"\n",
      "[1] \"For transgenic tissue, What is the range of values for portion for Callus\"\n",
      "[1] 0\n",
      "[1] 0.1428571\n",
      "[1] \"What are factor levels?\"\n",
      "[1] \"0\"                  \"0.0833333333333333\" \"0.1\"               \n",
      "[4] \"0.125\"              \"0.142857142857143\" \n",
      "\n",
      "[1] \"Column of interest is portion_Callus\"\n",
      "[1] \"For total tissue, What is the range of values for portion for Callus\"\n",
      "[1] 0\n",
      "[1] 0.2727273\n",
      "[1] \"What are factor levels?\"\n",
      " [1] \"0\"                  \"0.0833333333333333\" \"0.0909090909090909\"\n",
      " [4] \"0.1\"                \"0.111111111111111\"  \"0.125\"             \n",
      " [7] \"0.142857142857143\"  \"0.166666666666667\"  \"0.2\"               \n",
      "[10] \"0.272727272727273\" \n",
      "\n",
      "[1] \"Column of interest is portion_transgenic_Stem\"\n",
      "[1] \"For transgenic tissue, What is the range of values for portion for Stem\"\n",
      "[1] 0\n",
      "[1] 0.875\n",
      "[1] \"What are factor levels?\"\n",
      " [1] \"0\"                  \"0.0833333333333333\" \"0.0909090909090909\"\n",
      " [4] \"0.1\"                \"0.111111111111111\"  \"0.142857142857143\" \n",
      " [7] \"0.166666666666667\"  \"0.181818181818182\"  \"0.25\"              \n",
      "[10] \"0.333333333333333\"  \"0.363636363636364\"  \"0.4\"               \n",
      "[13] \"0.416666666666667\"  \"0.454545454545455\"  \"0.5\"               \n",
      "[16] \"0.545454545454545\"  \"0.555555555555556\"  \"0.571428571428571\" \n",
      "[19] \"0.583333333333333\"  \"0.6\"                \"0.625\"             \n",
      "[22] \"0.666666666666667\"  \"0.7\"                \"0.75\"              \n",
      "[25] \"0.8\"                \"0.875\"             \n",
      "\n",
      "[1] \"Column of interest is portion_Stem\"\n",
      "[1] \"For total tissue, What is the range of values for portion for Stem\"\n",
      "[1] 0\n",
      "[1] 1\n",
      "[1] \"What are factor levels?\"\n",
      "[1] \"0\"                 \"0.916666666666667\" \"1\"                \n",
      "\n",
      "[1] \"Column of interest is portion_transgenic_All_tissue\"\n",
      "[1] \"For transgenic tissue, What is the range of values for portion for All_tissue\"\n",
      "[1] 0\n",
      "[1] 0.875\n",
      "[1] \"What are factor levels?\"\n",
      " [1] \"0\"                  \"0.0833333333333333\" \"0.0909090909090909\"\n",
      " [4] \"0.1\"                \"0.111111111111111\"  \"0.142857142857143\" \n",
      " [7] \"0.166666666666667\"  \"0.181818181818182\"  \"0.25\"              \n",
      "[10] \"0.333333333333333\"  \"0.363636363636364\"  \"0.4\"               \n",
      "[13] \"0.416666666666667\"  \"0.454545454545455\"  \"0.5\"               \n",
      "[16] \"0.545454545454545\"  \"0.555555555555556\"  \"0.571428571428571\" \n",
      "[19] \"0.583333333333333\"  \"0.6\"                \"0.625\"             \n",
      "[22] \"0.666666666666667\"  \"0.7\"                \"0.714285714285714\" \n",
      "[25] \"0.75\"               \"0.8\"                \"0.875\"             \n",
      "\n",
      "[1] \"Column of interest is portion_All_tissue\"\n",
      "[1] \"For total tissue, What is the range of values for portion for All_tissue\"\n",
      "[1] 0\n",
      "[1] 1\n",
      "[1] \"What are factor levels?\"\n",
      "[1] \"0\" \"1\"\n",
      "\n",
      "[1] \"Column of interest is portion_transgenic_All_regenerated_tissue\"\n",
      "[1] \"For transgenic tissue, What is the range of values for portion for All_regenerated_tissue\"\n",
      "[1] 0\n",
      "[1] 0.1428571\n",
      "[1] \"What are factor levels?\"\n",
      "[1] \"0\"                  \"0.0833333333333333\" \"0.1\"               \n",
      "[4] \"0.125\"              \"0.142857142857143\" \n",
      "\n",
      "[1] \"Column of interest is portion_All_regenerated_tissue\"\n",
      "[1] \"For total tissue, What is the range of values for portion for All_regenerated_tissue\"\n",
      "[1] 0\n",
      "[1] 0.2727273\n",
      "[1] \"What are factor levels?\"\n",
      " [1] \"0\"                  \"0.0833333333333333\" \"0.0909090909090909\"\n",
      " [4] \"0.1\"                \"0.111111111111111\"  \"0.125\"             \n",
      " [7] \"0.142857142857143\"  \"0.166666666666667\"  \"0.2\"               \n",
      "[10] \"0.272727272727273\" \n",
      "\n",
      "[1] \"Column of interest is portion_transgenic_Background\"\n",
      "[1] \"For transgenic tissue, What is the range of values for portion for Background\"\n",
      "[1] 0\n",
      "[1] 0\n",
      "[1] \"What are factor levels?\"\n",
      "[1] \"0\"\n",
      "\n",
      "[1] \"Column of interest is portion_Background\"\n",
      "[1] \"For total tissue, What is the range of values for portion for Background\"\n",
      "[1] 0\n",
      "[1] 0\n",
      "[1] \"What are factor levels?\"\n",
      "[1] \"0\"\n",
      "\n",
      "[1] \"N rows of randomization datasheet: \"\n",
      "[1] 96\n",
      "[1] \"N rows of randomization datasheet: \"\n",
      "[1] 96\n",
      "\n",
      "[1] \"Plots will be saved to: \"\n",
      "[1] \"/scratch2/NSF_GWAS/GMOlabeler/plots/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent//\"\n",
      "\n",
      "\n",
      "[1] \"How many groups per treatment?\"\n",
      "\n",
      " AS HSPT   AS Mar CTR HSPT  CTR Mar \n",
      "      24       24       24       24 \n",
      "[1] \"How many rows in randomization datasheet?\"\n",
      "[1] 96\n",
      "[1] \"How many rows in randomization datasheet with portion_Callus data?\"\n",
      "[1] 96\n",
      "Warning message:\n",
      "`fun.y` is deprecated. Use `fun` instead. \n",
      "Warning message:\n",
      "`fun.y` is deprecated. Use `fun` instead. \n",
      "Warning message:\n",
      "`fun.y` is deprecated. Use `fun` instead. \n",
      "Warning message:\n",
      "`fun.y` is deprecated. Use `fun` instead. \n",
      "Warning message:\n",
      "`fun.y` is deprecated. Use `fun` instead. \n",
      "Warning message:\n",
      "`fun.y` is deprecated. Use `fun` instead. \n",
      "Warning message:\n",
      "`fun.y` is deprecated. Use `fun` instead. \n",
      "Warning message:\n",
      "`fun.y` is deprecated. Use `fun` instead. \n",
      "\n",
      "[1] \"How many groups per treatment?\"\n",
      "\n",
      " AS HSPT   AS Mar CTR HSPT  CTR Mar \n",
      "      24       24       24       24 \n",
      "[1] \"How many rows in randomization datasheet?\"\n",
      "[1] 96\n",
      "[1] \"How many rows in randomization datasheet with portion_Callus data?\"\n",
      "[1] 96\n",
      "Warning messages:\n",
      "1: Ignoring unknown parameters: outlier.shape \n",
      "2: `fun.y` is deprecated. Use `fun` instead. \n",
      "Warning messages:\n",
      "1: Ignoring unknown parameters: outlier.shape \n",
      "2: `fun.y` is deprecated. Use `fun` instead. \n",
      "Warning messages:\n",
      "1: Ignoring unknown parameters: outlier.shape \n",
      "2: `fun.y` is deprecated. Use `fun` instead. \n",
      "Warning messages:\n",
      "1: Ignoring unknown parameters: outlier.shape \n",
      "2: `fun.y` is deprecated. Use `fun` instead. \n",
      "Warning messages:\n",
      "1: Ignoring unknown parameters: outlier.shape \n",
      "2: `fun.y` is deprecated. Use `fun` instead. \n",
      "Warning messages:\n",
      "1: Ignoring unknown parameters: outlier.shape \n",
      "2: `fun.y` is deprecated. Use `fun` instead. \n",
      "Warning messages:\n",
      "1: Ignoring unknown parameters: outlier.shape \n",
      "2: `fun.y` is deprecated. Use `fun` instead. \n",
      "Warning messages:\n",
      "1: Ignoring unknown parameters: outlier.shape \n",
      "2: `fun.y` is deprecated. Use `fun` instead. \n",
      "\n",
      "[1] \"How many groups per treatment?\"\n",
      "\n",
      " AS HSPT   AS Mar CTR HSPT  CTR Mar \n",
      "      24       24       24       24 \n",
      "[1] \"How many rows in randomization datasheet?\"\n",
      "[1] 96\n",
      "[1] \"How many rows in randomization datasheet with n_Callus data?\"\n",
      "[1] 96\n",
      "[1] \"Head:\"\n",
      "        ID Image # Tray ID Rand # Date on Media *Plate # Index# in GH\n",
      "1  GFPA1_1       1   GFPA1      1    2020-09-17  3-2-1-2           R7\n",
      "2 GFPA1_10      10   GFPA1     10    2020-09-17  3-2-1-5           R7\n",
      "3 GFPA1_11      11   GFPA1     11    2020-09-17  1-2-1-1          140\n",
      "4 GFPA1_12      12   GFPA1     12    2020-09-17  4-2-1-3          R21\n",
      "5 GFPA1_13      13   GFPA1     13    2020-09-17  4-2-1-6          R21\n",
      "6 GFPA1_14      14   GFPA1     14    2020-09-17  3-1-2-6           R7\n",
      "  Genotype_ID Costruct Media Treatment name Replicate ...12     LableTop\n",
      "1   Nisqually      Mar   CTR        CTR Mar         2    NA   GFPA1_1_R1\n",
      "2   Nisqually      Mar   CTR        CTR Mar         5    NA GFPA1_10_R10\n",
      "3    BESC-134      Mar   CTR        CTR Mar         1    NA GFPA1_11_R11\n",
      "4   SKWB-24-2      Mar   CTR        CTR Mar         3    NA GFPA1_12_R12\n",
      "5   SKWB-24-2      Mar   CTR        CTR Mar         6    NA GFPA1_13_R13\n",
      "6   Nisqually     HSPT    AS        AS HSPT         6    NA GFPA1_14_R14\n",
      "             LableBot Plate#      Rand                           filename 1 2 3\n",
      "1 Nisqually_Mar_CTR_2     62 0.3583730  GFPA1_I5.0_F1.9_L100_161651_0_0_0 1 1 0\n",
      "2 Nisqually_Mar_CTR_5     65 0.8666917 GFPA1_I5.0_F1.9_L100_162649_11_1_2 1 1 1\n",
      "3  BESC-134_Mar_CTR_1     13 0.5296783 GFPA1_I5.0_F1.9_L100_162555_10_1_3 1 1 1\n",
      "4 SKWB-24-2_Mar_CTR_3     87 0.8703313  GFPA1_I5.0_F1.9_L100_162450_9_1_4 1 1 1\n",
      "5 SKWB-24-2_Mar_CTR_6     90 0.4970979  GFPA1_I5.0_F1.9_L100_162354_8_1_5 1 1 1\n",
      "6 Nisqually_HSPT_AS_6     60 0.6426024  GFPA1_I5.0_F1.9_L100_162308_7_1_6 1 1 1\n",
      "  4 5 6 7 8 9 10 11 12 total_explants n_transgenic_Shoot\n",
      "1 1 1 1 1 1 1  1  1  1             11                  0\n",
      "2 1 1 1 1 1 1  1  1  1             12                  0\n",
      "3 1 0 1 1 1 1  1  1  1             11                  0\n",
      "4 1 1 1 1 1 1  1  1  1             12                  0\n",
      "5 1 1 0 1 1 1  1  1  1             11                  0\n",
      "6 1 1 1 0 1 1  1  1  1             11                  0\n",
      "  portion_transgenic_Shoot n_transgenic_Callus portion_transgenic_Callus\n",
      "1                        0                   0                         0\n",
      "2                        0                   0                         0\n",
      "3                        0                   0                         0\n",
      "4                        0                   0                         0\n",
      "5                        0                   0                         0\n",
      "6                        0                   0                         0\n",
      "  n_transgenic_Stem portion_transgenic_Stem n_transgenic_All_tissue\n",
      "1                 6              0.54545455                       6\n",
      "2                 4              0.33333333                       4\n",
      "3                 0              0.00000000                       0\n",
      "4                 0              0.00000000                       0\n",
      "5                 0              0.00000000                       0\n",
      "6                 1              0.09090909                       1\n",
      "  portion_transgenic_All_tissue n_transgenic_All_regenerated_tissue\n",
      "1                    0.54545455                                   0\n",
      "2                    0.33333333                                   0\n",
      "3                    0.00000000                                   0\n",
      "4                    0.00000000                                   0\n",
      "5                    0.00000000                                   0\n",
      "6                    0.09090909                                   0\n",
      "  portion_transgenic_All_regenerated_tissue n_transgenic_Background\n",
      "1                                         0                       0\n",
      "2                                         0                       0\n",
      "3                                         0                       0\n",
      "4                                         0                       0\n",
      "5                                         0                       0\n",
      "6                                         0                       0\n",
      "  portion_transgenic_Background n_Shoot portion_Shoot n_Callus portion_Callus\n",
      "1                             0       0             0        0     0.00000000\n",
      "2                             0       0             0        0     0.00000000\n",
      "3                             0       0             0        0     0.00000000\n",
      "4                             0       0             0        0     0.00000000\n",
      "5                             0       0             0        1     0.09090909\n",
      "6                             0       0             0        0     0.00000000\n",
      "  n_Stem portion_Stem n_All_tissue portion_All_tissue n_All_regenerated_tissue\n",
      "1     11            1           11                  1                        0\n",
      "2     12            1           12                  1                        0\n",
      "3     11            1           11                  1                        0\n",
      "4     12            1           12                  1                        0\n",
      "5     11            1           11                  1                        1\n",
      "6     11            1           11                  1                        0\n",
      "  portion_All_regenerated_tissue n_Background portion_Background\n",
      "1                     0.00000000            0                  0\n",
      "2                     0.00000000            0                  0\n",
      "3                     0.00000000            0                  0\n",
      "4                     0.00000000            0                  0\n",
      "5                     0.09090909            0                  0\n",
      "6                     0.00000000            0                  0\n",
      "Warning message:\n",
      "`fun.y` is deprecated. Use `fun` instead. \n",
      "Warning message:\n",
      "`fun.y` is deprecated. Use `fun` instead. \n",
      "Warning message:\n",
      "`fun.y` is deprecated. Use `fun` instead. \n",
      "Warning message:\n",
      "`fun.y` is deprecated. Use `fun` instead. \n",
      "Warning message:\n",
      "`fun.y` is deprecated. Use `fun` instead. \n",
      "Warning message:\n",
      "`fun.y` is deprecated. Use `fun` instead. \n",
      "Warning message:\n",
      "`fun.y` is deprecated. Use `fun` instead. \n",
      "Warning message:\n",
      "`fun.y` is deprecated. Use `fun` instead. \n",
      "\n",
      "[1] \"List of treatments in this randomization_datasheet: \"\n",
      "[1] \"AS HSPT\"  \"AS Mar\"   \"CTR HSPT\" \"CTR Mar\" \n",
      "[1] \"Now about to merge randomization_datasheet and GMOlabeler output.\"\n",
      "[1] \"Rows in randomization_datasheet: 96\"\n",
      "[1] \"Rows in GMOlabeler output: 3759\"\n",
      "[1] \"Before merging, look at table of ID in both. \"\n",
      "\n",
      " GFPA1_1 GFPA1_10 GFPA1_11 GFPA1_12 GFPA1_13 GFPA1_14 GFPA1_15 GFPA1_16 \n",
      "       1        1        1        1        1        1        1        1 \n",
      "GFPA1_17 GFPA1_18 GFPA1_19  GFPA1_2 GFPA1_20 GFPA1_21  GFPA1_3  GFPA1_4 \n",
      "       1        1        1        1        1        1        1        1 \n",
      " GFPA1_5  GFPA1_6  GFPA1_7  GFPA1_8  GFPA1_9  GFPA2_1 GFPA2_10 GFPA2_11 \n",
      "       1        1        1        1        1        1        1        1 \n",
      "GFPA2_12 GFPA2_13 GFPA2_14 GFPA2_15 GFPA2_16 GFPA2_17 GFPA2_18 GFPA2_19 \n",
      "       1        1        1        1        1        1        1        1 \n",
      " GFPA2_2 GFPA2_20 GFPA2_21  GFPA2_3  GFPA2_4  GFPA2_5  GFPA2_6  GFPA2_7 \n",
      "       1        1        1        1        1        1        1        1 \n",
      " GFPA2_8  GFPA2_9  GFPA3_1 GFPA3_10 GFPA3_11 GFPA3_12 GFPA3_13 GFPA3_14 \n",
      "       1        1        1        1        1        1        1        1 \n",
      "GFPA3_15 GFPA3_16 GFPA3_17 GFPA3_18 GFPA3_19  GFPA3_2 GFPA3_20 GFPA3_21 \n",
      "       1        1        1        1        1        1        1        1 \n",
      " GFPA3_3  GFPA3_4  GFPA3_5  GFPA3_6  GFPA3_7  GFPA3_8  GFPA3_9  GFPA4_1 \n",
      "       1        1        1        1        1        1        1        1 \n",
      "GFPA4_10 GFPA4_11 GFPA4_12 GFPA4_13 GFPA4_14 GFPA4_15 GFPA4_16 GFPA4_17 \n",
      "       1        1        1        1        1        1        1        1 \n",
      "GFPA4_18 GFPA4_19  GFPA4_2 GFPA4_20 GFPA4_21  GFPA4_3  GFPA4_4  GFPA4_5 \n",
      "       1        1        1        1        1        1        1        1 \n",
      " GFPA4_6  GFPA4_7  GFPA4_8  GFPA4_9  GFPA5_1 GFPA5_10 GFPA5_11 GFPA5_12 \n",
      "       1        1        1        1        1        1        1        1 \n",
      " GFPA5_2  GFPA5_3  GFPA5_4  GFPA5_5  GFPA5_6  GFPA5_7  GFPA5_8  GFPA5_9 \n",
      "       1        1        1        1        1        1        1        1 \n",
      "\n",
      "       _  GFPA1_1 GFPA1_10 GFPA1_11 GFPA1_12 GFPA1_13 GFPA1_14 GFPA1_15 \n",
      "       1       44       48       44       48       45       44       40 \n",
      "GFPA1_16 GFPA1_17 GFPA1_18 GFPA1_19  GFPA1_2 GFPA1_20 GFPA1_21  GFPA1_3 \n",
      "      44       48       44       40       48       44       44       49 \n",
      " GFPA1_4  GFPA1_5  GFPA1_6  GFPA1_7  GFPA1_8 GFPA2_10 GFPA2_11 GFPA2_12 \n",
      "      48       37       48       49       36       49       48       28 \n",
      "GFPA2_13 GFPA2_14 GFPA2_15 GFPA2_16 GFPA2_17 GFPA2_18 GFPA2_19 GFPA2_20 \n",
      "      48       44       44       44       32       36       45       29 \n",
      "GFPA2_21  GFPA2_4  GFPA2_5  GFPA2_6  GFPA2_7  GFPA2_8  GFPA2_9  GFPA3_1 \n",
      "      49       28       40       36       48       44       44       32 \n",
      "GFPA3_10 GFPA3_11 GFPA3_12 GFPA3_13 GFPA3_14 GFPA3_15 GFPA3_16 GFPA3_17 \n",
      "      41       44       44       44       48       21       50       48 \n",
      "GFPA3_19  GFPA3_2 GFPA3_20 GFPA3_21  GFPA3_3  GFPA3_4  GFPA3_5  GFPA3_6 \n",
      "      41       29       45       41       36       44       49       49 \n",
      " GFPA3_7  GFPA3_8  GFPA3_9  GFPA4_1 GFPA4_10 GFPA4_11 GFPA4_12 GFPA4_13 \n",
      "      45       48       48       36       32       24       47       48 \n",
      "GFPA4_14 GFPA4_15 GFPA4_16 GFPA4_17 GFPA4_18 GFPA4_19  GFPA4_2 GFPA4_20 \n",
      "      49       41       49       44       44       36       48       50 \n",
      "GFPA4_21  GFPA4_3  GFPA4_4  GFPA4_5  GFPA4_6  GFPA4_8  GFPA4_9  GFPA5_1 \n",
      "      49       50       44       49       36       32       40       32 \n",
      "GFPA5_10 GFPA5_11 GFPA5_12  GFPA5_2  GFPA5_3  GFPA5_4  GFPA5_5  GFPA5_6 \n",
      "      48       40       41        8       28       48       49       40 \n",
      " GFPA5_7  GFPA5_8  GFPA5_9 \n",
      "      36       32       33 \n",
      "[1] \"Rows in merged output: 3759\"\n",
      "[1] \"List of treatments in this combined data: \"\n",
      "[1] \"AS HSPT\"  \"AS Mar\"   \"CTR HSPT\" \"CTR Mar\" \n",
      "\n",
      "[1] 931  69\n",
      "[1] 931  69\n",
      "[1] 931  69\n",
      "[1] 931  69\n",
      "[1] 34 69\n",
      "There were 50 or more warnings (use warnings() to see the first 50)\n"
     ]
    }
   ],
   "source": [
    "cd /scratch2/NSF_GWAS/GMOlabeler/\n",
    "Rscript grid_item_plots_a13.R \\\n",
    "-d \"${data_folder}/\" \\\n",
    "-r \"$randomization_datasheet\" \\\n",
    "-p $pixel_threshold \\\n",
    "-v categorical \\\n",
    "-m 1 \\\n",
    "-M $missing_explants \\\n",
    "-g $grid \\\n",
    "--sort 1 \\\n",
    "--height $height \\\n",
    "--width $width"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 5. Email plots to user"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### 5.1. ZIP results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/scratch2/NSF_GWAS/GMOlabeler/plotsGREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent/\n"
     ]
    }
   ],
   "source": [
    "echo \"/scratch2/NSF_GWAS/GMOlabeler/plots${data_folder}\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [],
   "source": [
    "cd \"/scratch2/NSF_GWAS/GMOlabeler/plots/${data_folder}\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [],
   "source": [
    "rm -f ./plants_over_plates.csv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [],
   "source": [
    "cp \"/scratch2/NSF_GWAS/GMOlabeler/output/${data_folder}/plants_over_plates.csv\" ./"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [],
   "source": [
    "rm -f Rplots.pdf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [],
   "source": [
    "cd ../"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This messy substitution is explained here: https://superuser.com/questions/1068031/replace-backslash-with-forward-slash-in-a-variable-in-bash"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [],
   "source": [
    "data_folder_Compress=\"${data_folder////-}.zip\"\n",
    "data_folder_Compress=${data_folder_Compress#?};"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001b[0m\u001b[01;34mFluorescent\u001b[0m\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "updating: 2A1c_portion_Callus_total.png (deflated 11%)\n",
      "updating: 2A1_portion_Callus_total.png (deflated 13%)\n",
      "updating: 2A2c_portion_Callus_transgenic.png (deflated 12%)\n",
      "updating: 2A2_portion_Callus_transgenic.png (deflated 29%)\n",
      "updating: 2B1c_portion_Shoot_total.png (deflated 32%)\n",
      "updating: 2B1_portion_Shoot_total.png (deflated 19%)\n",
      "updating: 2B2c_portion_Shoot_transgenic.png (deflated 32%)\n",
      "updating: 2B2_portion_Shoot_transgenic.png (deflated 20%)\n",
      "updating: 2C1c_portion_All_regenerated_tissue_total.png (deflated 11%)\n",
      "updating: 2C1_portion_All_regenerated_tissue_total.png (deflated 14%)\n",
      "updating: 2C2c_portion_All_regenerated_tissue_transgenic.png (deflated 12%)\n",
      "updating: 2C2_portion_All_regenerated_tissue_transgenic.png (deflated 27%)\n",
      "updating: 2D1c_portion_All_tissue_total.png (deflated 11%)\n",
      "updating: 2D1_portion_All_tissue_total.png (deflated 15%)\n",
      "updating: 2D2c_portion_All_tissue_transgenic.png (deflated 8%)\n",
      "updating: 2D2_portion_All_tissue_transgenic.png (deflated 12%)\n",
      "updating: 3A1_Callus_total.png (deflated 15%)\n",
      "updating: 3A2_Callus_transgenic.png (deflated 30%)\n",
      "updating: 3B1_Shoot_total.png (deflated 20%)\n",
      "updating: 3B2_Shoot_transgenic.png (deflated 20%)\n",
      "updating: 3D1_All_regenerated_tissue_total.png (deflated 14%)\n",
      "updating: 3D1_All_tissue_total.png (deflated 13%)\n",
      "updating: 3D2_All_regenerated_tissue_transgenic.png (deflated 28%)\n",
      "updating: 3D2_All_tissue_transgenic.png (deflated 12%)\n",
      "updating: A1_Shoot_Total_signal.png (deflated 43%)\n",
      "updating: A2_Callus_Total_signal.png (deflated 16%)\n",
      "updating: A3_Stem_Total_signal.png (deflated 13%)\n",
      "updating: A4_All_tissue_Total_signal.png (deflated 13%)\n",
      "updating: A5_All_regenerated_tissue_Total_signal.png (deflated 16%)\n",
      "updating: B1_Shoot_Max_signal.png (deflated 41%)\n",
      "updating: B2_Callus_Max_signal.png (deflated 16%)\n",
      "updating: B3_Stem_Max_signal.png (deflated 13%)\n",
      "updating: B4_All_tissue_Max_signal.png (deflated 13%)\n",
      "updating: B5_All_regenerated_tissue_Max_signal.png (deflated 16%)\n",
      "updating: C1_Shoot_Mean_signal.png (deflated 41%)\n",
      "updating: C2_Callus_Mean_signal.png (deflated 17%)\n",
      "updating: C3_Stem_Mean_signal.png (deflated 14%)\n",
      "updating: gridplot_args.rds (stored 0%)\n",
      "updating: plants_over_plates.csv (deflated 80%)\n"
     ]
    }
   ],
   "source": [
    "cd $timepoint\n",
    "zip -r $data_folder_Compress ./*"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### 5.2. Write email"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [],
   "source": [
    "duration=$(( SECONDS - start ))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "https://unix.stackexchange.com/questions/53841/how-to-use-a-timer-in-bash"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "metadata": {},
   "outputs": [],
   "source": [
    "rm -f /scratch2/NSF_GWAS/GMOlabeler/email_to_send.txt\n",
    "cp /scratch2/NSF_GWAS/GMOlabeler/email_to_send_template.txt /scratch2/NSF_GWAS/GMOlabeler/email_to_send.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [],
   "source": [
    "echo \"\" >> /scratch2/NSF_GWAS/GMOlabeler/email_to_send.txt\n",
    "echo \"Number of samples run: \" >> /scratch2/NSF_GWAS/GMOlabeler/email_to_send.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [],
   "source": [
    "cat \"${data}/test.csv\" | wc -l >> /scratch2/NSF_GWAS/GMOlabeler/email_to_send.txt\n",
    "echo \"\" >> /scratch2/NSF_GWAS/GMOlabeler/email_to_send.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [],
   "source": [
    "if (( $SECONDS > 3600 )) ; then\n",
    "    let \"hours=SECONDS/3600\"\n",
    "    let \"minutes=(SECONDS%3600)/60\"\n",
    "    let \"seconds=(SECONDS%3600)%60\"\n",
    "    echo \"Completed in $hours hour(s), $minutes minute(s) and $seconds second(s)\" >> /scratch2/NSF_GWAS/GMOlabeler/email_to_send.txt\n",
    "elif (( $SECONDS > 60 )) ; then\n",
    "    let \"minutes=(SECONDS%3600)/60\"\n",
    "    let \"seconds=(SECONDS%3600)%60\"\n",
    "    echo \"Completed in $minutes minute(s) and $seconds second(s)\" >> /scratch2/NSF_GWAS/GMOlabeler/email_to_send.txt\n",
    "else\n",
    "    echo \"Completed in $SECONDS seconds\" >> /scratch2/NSF_GWAS/GMOlabeler/email_to_send.txt\n",
    "fi"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 69,
   "metadata": {},
   "outputs": [],
   "source": [
    "echo \"\" >> /scratch2/NSF_GWAS/GMOlabeler/email_to_send.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### 5.3. Send email with results to user"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/scratch2/NSF_GWAS/GMOlabeler/plots/GREGSCORNER/GFPA_cottonwoods/wk3/Fluorescent\n"
     ]
    }
   ],
   "source": [
    "pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [],
   "source": [
    "mail -a $data_folder_Compress -s $data_folder $email < /scratch2/NSF_GWAS/GMOlabeler/email_to_send.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3.5M\tREGSCORNER-GFPA_cottonwoods-wk3-Fluorescent-.zip\n"
     ]
    }
   ],
   "source": [
    "du -sh $data_folder_Compress"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Your job is complete!\n",
      "\n",
      "Number of samples run: \n",
      "90\n",
      "\n",
      "Completed in 57 minute(s) and 37 second(s)\n",
      "\n"
     ]
    }
   ],
   "source": [
    "cat /scratch2/NSF_GWAS/GMOlabeler/email_to_send.txt"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### Time analysis ends"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Tue Oct 27 21:06:27 PDT 2020\n"
     ]
    }
   ],
   "source": [
    "echo $(date)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
