{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# *GMOnotebook*\n",
    "#### **Notebook template for applying routine hyperspectral/segmentation cross-analysis phenomics workflow over new datasets**"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "###### Notebook template v0.2.03 (Oct 14, 2020)\n",
    "###### v0.2.0 onward applies the Python `GMOdetector` package instead of the deprecated R `GMOdetectoR` package, for much faster runtime\n",
    "###### v0.2.01 has calls to `conda` cleaned up with `conda activate` and `conda deactivate` in the same code blocks to avoid any problems with warnings/errors preventing automatic execution of subsequent blocks\n",
    "###### v0.2.02 is cleaned up using path variables (declared in notebook) instead of hardcoded paths, for portability\n",
    "###### v0.2.03 is more clean to avoid running alignment-processed images through `inference.py` if an analysis if being repeated... also has better instructions"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "v.0.2.05 is updated to work with FAA wk7. Specifically, there are updates to deal with increased image resolution (y~=2000), off-center images, and rotated images, including metadata normalization and user inputs for parameterized grid cropping."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<img src=\"Figures/WorkflowFlowchart.png\">"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "In this workflow, images taken with the macroPhor Array dual RGB/hyperspectral imaging platform are analyzed by a workflow in which regression quantifies fluorescent signals in hyperspectral images, deep learning segments RGB images into different tissues, and these datasets are cross-referenced to produce statistics on growth of transgenic callus and shoot."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Instructions for use\n",
    "1.  Enter information for the experiment below\n",
    "2. Set <font color=blue>variables</font> for data paths and parameters, as instructed by colored boxes.\n",
    "3. \"Save as\" with filename describing experiment and anything special about this analysis (e.g. T18_OD_TAO_wk7_automation_test_attempt2.ipynb)\n",
    "4. Run notebook from console, by either one of two methods:\n",
    "- In the top bar of JupyterLab, select \"Run\" and \"Run all cells\"\n",
    "- In a terminal, enter the below command with the notebook filename inserted<br>\n",
    "```jupyter nbconvert --to HTML --ExecutePreprocessor.timeout=-1 --allow-errors --execute insert_filename_here```\n",
    "5. Wait for email"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Experiment ID and quick description:"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<div class=\"alert alert-block alert-success\">\n",
    "Provide a short description of the experiment in the below box. This should include unique identifier codes for the experiment, along with a short description of genotypes and treatments studied. The timepoint should also be included. </div>"
   ]
  },
  {
   "cell_type": "raw",
   "metadata": {},
   "source": [
    "FAE: Testing new DEV gene plasmids\n",
    "Week 7 timepoint – After 3w callus induction and 4w shoot induction"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Parameters for analysis:"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<div class=\"alert alert-block alert-success\">\n",
    "The below variables must be modified appropriately every time this workflow is run over new images.\n",
    "</div>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### Data location\n",
    "The `data` variable below provides the **complete** path to the folder containing data to be analyzed. This should include all folders and subfolders in which the data of interest is organized by. For the organizational system used for our lab's data, this should follow the format \"/Experiment/Subexperiment/Timepoint/\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "data_prefix=\"/run/media/labgroup/Elements_9/\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "data=\"${data_prefix}/Transformation/FAE_Dev_Genes/wk7/Fluorescence/\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### Sample information\n",
    "Every experiment has a randomization datasheet, which was used to organize treatment and genotype information for each plate, prepare labels, and randomize plates. The path to this file is provided through the `randomization_datasheets` variable below. This workflow requires this datasheet in order to know which plates have which genotype/treatment. At a later date, we will integrate an ability to read this data directly from labels."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 116,
   "metadata": {},
   "outputs": [],
   "source": [
    "randomization_datasheet=\"/run/media/labgroup/Elements_9/Transformation/FAE_Dev_Genes/FAE_randomization_list_labels.xlsx\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### Exclusion of missing/contaminated explants from analysis"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Set the `missing_explants` variable to `\"Automatic\"` if using model to automatically detect missing and contaminated explants. Note that this model is only supported for plates with 12 explants. Otherwise, provide an appropriately formatted `.csv` spreadsheet (see example)  of manually scored contamination / missing explant data."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "missing_explants=\"Automatic\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Enter your email where results will be sent"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "email=michael.nagle@oregonstate.edu"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "<div class=\"alert alert-block alert-warning\">\n",
    "All variables below should be modified only as needed to indicate the fluorescent proteins in samples and the grid layout of explants. </div>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### Provide information pertinent to the fluorophores in the sample"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below should be all known fluorescent components contained in the sample. This includes each fluorescent protein, as well as a \"noise\" or \"diffraction\" term if applicable. All of these components must exist in the user's spectral library. `GMOdetector` currently comes with a spectral library that includes, by default:\n",
    "- DsRed\n",
    "- ZsYellow\n",
    "- GFP\n",
    "- Chl\n",
    "- ChlA\n",
    "- ChlB\n",
    "- Diffraction"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "fluorophores=(DsRed Chl Diffraction) # An explanation of array variables in bash is here: https://tldp.org/LDP/Bash-Beginners-Guide/html/sect_10_02.html"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The user has the option of limiting loading of hyperspectral data and subsequent regression to a specific range of wavelengths, using the `desired_wavelength_range` array variable. This range should cover all fluorophores provided in the `fluorophores` array variable able. Aside from runtime, there is no disadvantage to including a wider range than is needed."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "desired_wavelength_range=(500 900)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### To assist user inspection of regression results, false color plots will be produced by `GMOdetector` to show results of regression over whole samples. \n",
    "###### Note: In v0.2.0 of the workflow, these parameters are independent of those used later by `GMOlabeler` to produce by-explant plots including false color plots. These will be made the same in a later update.\n",
    "The `FalseColor_channels` array variable indicates the components to be plotted as red, green and blue, in that order."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "FalseColor_channels=(Chl DsRed Diffraction)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The `FalseColor_caps` array variable indicates an upper limit of signal for each of these component. Any signal at or above these values will appear with maximum brightness; thus, these variables are comparable to exposure on a camera. If caps are too high, not much signal at lower ranges will be seen. If cap for a given component is too low, the false color images will appear overexposed with respect to the component."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "FalseColor_caps=(200 400 50)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### `GMOlabeler` will use the below parameters to classify individual explants on plates as transgenic or not."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#reporter=GFP\n",
    "reporter=DsRed"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Parameters for reporter signal threshold and pixel threshold must be provided by the user. Our grid search yielded several noted below. These were most recently calculated from statistics produced with Python `GMOdetector`."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<img src=\"Figures/GMOlabeler_parameters.png\">"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "reporter_threshold=38\n",
    "pixel_threshold=3"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### Grid information"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Currently, `GMOlabeler` distinguishes explants by their expected position on a plate. This is supported for plates with 12 or 20 explants, arranged in specific positions on a grid. False positives may result in cases in which one explant grows to such an extent that it intrudes into the grid square meant for an adjacent explant. <br>In the future, an additional neural network will be used to segment individual explants on a per-pixel basis, avoiding the need for crude cropping, and the outputs from the resulting model will be integrated with `GMOlabeler`."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Cropping to grid squares is supported for standard grids with 12 or 20 explants, as indicated by the `grid` variable. An image of each grid square (with a file path provided by the `grid_file` variable) will be superimposed over each image during `GMOlabeler` analysis to allow the user to easily inspect and verify grid positions."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "grid=12"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<div class=\"alert alert-block alert-danger\">\n",
    "The below variables do not need to be modified during any routine use of the workflow.\n",
    "</div>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "For the following variables, use 0 (False) or 1 (True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "composite=0"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Set dimensions for plot outputs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#width=15\n",
    "width=9\n",
    "height=5"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "parallel=0"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Paths to workflow components on local machine"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "cwd=\"/home/labgroup/code/GMOnotebook\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#gmodetector_wd=\"/scratch2/NSF_GWAS/notebooks/gmodetector_py/\"\n",
    "gmodetector_wd=\"/home/labgroup/code/gmodetector_py/\"\n",
    "spectral_library_path=\"${gmodetector_wd}spectral_library/\"\n",
    "#output_directory_prefix=\"/scratch2/NSF_GWAS/gmodetector_out/\"\n",
    "#deeplab_path=\"/scratch2/NSF_GWAS/deeplab/\"\n",
    "deeplab_path=\"/home/labgroup/code/deeplab/\"\n",
    "#alignment_path=\"/scratch2/NSF_GWAS/ImageAlignment/\"\n",
    "alignment_path=\"/home/labgroup/code/ImageAlignment/\"\n",
    "#gmolabeler_path=\"/scratch2/NSF_GWAS/GMOlabeler/\"\n",
    "gmolabeler_path=\"/home/labgroup/code/GMOlabeler/\"\n",
    "#contamination_path=\"/scratch2/NSF_GWAS/Contamination\"\n",
    "contamination_path=\"/home/labgroup/code/DenseNet\"\n",
    "data_prefix=\"/home/labgroup/data/\"\n",
    "output_directory_prefix=\"${data_prefix}gmodetector_out/\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#grid_file=\"/scratch2/NSF_GWAS/macroPhor_Array/grids/grid20_post_processed.png\"\n",
    "#grid_file=\"/scratch2/NSF_GWAS/macroPhor_Array/grids/grids_left_facing_125208_1_0_1_rgb_processed.jpg\"\n",
    "\n",
    "# For data from before when imager went off-center\n",
    "# grid_file=\"${gmolabeler_path}/grids/grid_12_explants_left-facing_processed.jpg\"\n",
    "\n",
    "# For offcenter images\n",
    "grid_file=\"${alignment_path}/output/chromagrid_I5.0_F1.9_L100_122503_0_0_0_rgb_aligned.jpg\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<div class=\"alert alert-block alert-info\">\n",
    "With all above variables set, please \"Save as...\" with a filename referencing this specific dataset. <br>Finally, deploy the workflow (Step 4 in above instructions).\n",
    "</div>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Automated workflow to be deployed"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "See the below code for a walkthrough of how GMOnotebook works, or view the outputs after running the workflow for help troubleshooting errors in specific steps of analysis."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<div class=\"alert alert-block alert-danger\"> <b>Danger:</b> Do not modify any below code without creating a new version of the template notebook. During routine usage, this workflow should be customized only by modifying variables above, while leaving the below code unmodified. </div>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "These internal variables are set automatically."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "datestamp=$(date +”%Y-%m-%d”)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "data_folder=$(echo $data | cut -d/ -f5-)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "timepoint=\"$(basename -- $data_folder)\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "output_directory_full=\"$output_directory_prefix$data_folder\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "dataset_name=$(echo $data_folder | sed -e 's/\\//-/g')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### Time analysis begins:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Wed Jan 20 21:54:03 PST 2021\n"
     ]
    }
   ],
   "source": [
    "echo $(date)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 1. Quantification of fluorescent proteins by regression"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The Python package `GMOdetector` is used to quantify fluorescent proteins in each pixel of hyperspectral images via linear regression. Hyperspectral images are regressed over spectra of known components, and pixelwise maps of test-statistics are constructed for each component in the sample. This approach to quantifying components of hyperspectral images is described in-depth in the Methods section from <a href=\"https://link.springer.com/article/10.1007/s40789-019-0252-7\" target=\"_blank\">Böhme, et al. 2019</a>. Code and documentation for `GMOdetector` is on <a href=\"https://github.com/naglemi/GMOdetector_py\" target=\"_blank\">Github</a>."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "cd $gmodetector_wd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "job_list_name=\"$dataset_name.jobs\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "rm -rf $job_list_name"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "for file in $data/*.hdr\n",
    "do\n",
    " if [[ \"$file\" != *'hroma'* ]]; then\n",
    "  echo \"python wrappers/analyze_sample.py \\\n",
    "--file_path $file \\\n",
    "--fluorophores ${fluorophores[*]} \\\n",
    "--min_desired_wavelength ${desired_wavelength_range[0]} \\\n",
    "--max_desired_wavelength ${desired_wavelength_range[1]} \\\n",
    "--red_channel ${FalseColor_channels[0]} \\\n",
    "--green_channel ${FalseColor_channels[1]} \\\n",
    "--blue_channel ${FalseColor_channels[2]} \\\n",
    "--red_cap ${FalseColor_caps[0]} \\\n",
    "--green_cap ${FalseColor_caps[1]} \\\n",
    "--blue_cap ${FalseColor_caps[2]} \\\n",
    "--spectral_library_path \"$spectral_library_path\" \\\n",
    "--output_dir $output_directory_full\" >> $job_list_name\n",
    " fi\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Elements_9--Transformation-FAE_Dev_Genes-wk7-Fluorescence-.jobs\n"
     ]
    }
   ],
   "source": [
    "echo $job_list_name"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(test-environment) (test-environment) (test-environment) (test-environment) During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "(test-environment) (test-environment) (base) "
     ]
    }
   ],
   "source": [
    "conda activate test-environment\n",
    "\n",
    "if [ $parallel -eq 1 ]\n",
    "then\n",
    "    parallel -a $job_list_name >> $data/log_gmodetector.txt\n",
    "fi\n",
    "\n",
    "if [ $parallel -eq 0 ]\n",
    "then\n",
    "    bash $job_list_name >> $data/log_gmodetector.txt\n",
    "fi\n",
    "\n",
    "conda deactivate\n",
    "conda deactivate"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### Time regression completes:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Thu Jan 21 01:38:46 PST 2021\n"
     ]
    }
   ],
   "source": [
    "echo $(date)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 2. Neural networks to segment tissues, classify missing/contaminated explants"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 2.1. Semantic segmentation of tissues"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Images are segmented into specific plant tissues by a deep neural network of the state-of-the-art Deeplab v3 architecture <a href=\"https://arxiv.org/abs/1706.05587\" target=\"_blank\">Liang-Chieh et al., 2017</a>. The model has been trained using training sets generated with our annotation GUI Intelligent DEep Annotator for Segmentation (IDEAS, available on <a href=\"https://bitbucket.org/JialinYuan/image-annotator/src/master/\" target=\"_blank\">Bitbucket</a>, publication pending). Our branch of the Deeplab v3 repo, including a Jupyter walkthrough for training, can be found on Github."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Training is completed upstream of this notebook, which only entails analysis of test data using the latest model."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<img src=\"Figures/downsized/segmentation_composite2.png\">"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Figure: This example image was taken from an experiment on the effects of different CIMs on cottonwood regeneration. This composite image illustrates that for every sample, tissues are segmented into stem (red), callus (blue) and shoot (green). These composite images, useful for manual inspection of results, are produced when the 'composite' option is on."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### 2.1.1. Pre-processing\n",
    "##### 2.1.1.1. Normalize orientation, crop to remove labels, and resize images to 900x600"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We desire for images to all be in the same orientation. At one point, the camera on the *macroPhor Array* was set to automatically detect orientation, which led to images randomly being in portrait or landscape. Here we will standardize the orientation."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(alignment)     1 image files updated\n",
      "    1 image files updated\n",
      "    1 image files updated\n",
      "    1 image files updated\n",
      "    1 image files updated\n",
      "    1 image files updated\n",
      "    1 image files updated\n",
      "    1 image files updated\n",
      "    1 image files updated\n",
      "    1 image files updated\n",
      "    1 image files updated\n",
      "    1 image files updated\n",
      "    1 image files updated\n",
      "    1 image files updated\n",
      "    1 image files updated\n",
      "    1 image files updated\n",
      "    1 image files updated\n",
      "    1 image files updated\n",
      "    1 image files updated\n",
      "    1 image files updated\n",
      "    1 image files updated\n",
      "    1 image files updated\n",
      "    1 image files updated\n",
      "    1 image files updated\n",
      "    1 image files updated\n",
      "    1 image files updated\n",
      "    1 image files updated\n",
      "    1 image files updated\n",
      "    1 image files updated\n",
      "    1 image files updated\n",
      "    1 image files updated\n",
      "    1 image files updated\n",
      "    1 image files updated\n",
      "    1 image files updated\n",
      "    1 image files updated\n",
      "    1 image files updated\n",
      "    1 image files updated\n",
      "    1 image files updated\n",
      "    1 image files updated\n",
      "    1 image files updated\n",
      "    1 image files updated\n",
      "    1 image files updated\n",
      "    1 image files updated\n",
      "    1 image files updated\n",
      "    1 image files updated\n",
      "    1 image files updated\n",
      "(alignment) "
     ]
    }
   ],
   "source": [
    "conda activate alignment\n",
    "for filename in $data/*.jpg; do\n",
    "    #echo $filename\n",
    "    exiftool -Orientation=8 -n $filename\n",
    "    done\n",
    "conda deactivate"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "rm -f $data/*original*"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This script resizes images to 900x900 and then crops away top and bottom 150 pixels for a final image size of 900x600."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [],
   "source": [
    "cd ${cwd}/intermediates/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Note, the `standardize_orientation.py` script was used for a situation where image rotation was attempted manually in Mac OS X but metadata was left behind, causing downstream confusion. This approach is no longer needed since we are using `exiftool` (above) to make `Orientation` in metadata consistent."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(base) (base) (base) "
     ]
    }
   ],
   "source": [
    "conda activate base\n",
    "#python standardize_orientation.py $data\n",
    "python crop.py $data\n",
    "conda deactivate"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### 2.1.1.2. prepare test.csv file for inference"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Make a list of all our image files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [],
   "source": [
    "cd $data\n",
    "ls -d $PWD/* $data | grep -i \"rgb_cropped.jpg\" > test.csv"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Remove the chroma standard from list of RGB image data to be segmented"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [],
   "source": [
    "sed -i '/hroma/d' \"${data}/test.csv\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### 2.1.2. Inference"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The trained model is deployed to perform semantic segmentation of experimental images. A list of RGB images to be segmented by the trained model is passed through the --image-list option. For each of these images, we will obtain an output mask (.png) of labeled tissues"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Dependencies include `opencv`, `scipy`, `yaml` and `tensorflow` (version 1.14)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(deeplab) WARNING:tensorflow:From inference.py:34: The name tf.gfile.GFile is deprecated. Please use tf.io.gfile.GFile instead.\n",
      "\n",
      "WARNING:tensorflow:From inference.py:35: The name tf.GraphDef is deprecated. Please use tf.compat.v1.GraphDef instead.\n",
      "\n",
      "WARNING:tensorflow:From inference.py:41: The name tf.Session is deprecated. Please use tf.compat.v1.Session instead.\n",
      "\n",
      "2021-01-21 02:07:55.769087: I tensorflow/core/platform/cpu_feature_guard.cc:142] Your CPU supports instructions that this TensorFlow binary was not compiled to use: SSE4.1 SSE4.2 AVX\n",
      "2021-01-21 02:07:56.033456: I tensorflow/core/platform/profile_utils/cpu_utils.cc:94] CPU Frequency: 2494325000 Hz\n",
      "2021-01-21 02:07:56.034855: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x562851455550 executing computations on platform Host. Devices:\n",
      "2021-01-21 02:07:56.035025: I tensorflow/compiler/xla/service/service.cc:175]   StreamExecutor device (0): <undefined>, <undefined>\n",
      "2021-01-21 02:08:01.653759: W tensorflow/compiler/jit/mark_for_compilation_pass.cc:1412] (One-time warning): Not using XLA:CPU for cluster because envvar TF_XLA_FLAGS=--tf_xla_cpu_global_jit was not set.  If you want XLA:CPU, either set that envvar, or use experimental_jit_scope to enable XLA:CPU.  To confirm that XLA is active, pass --vmodule=xla_compilation_cache=1 (as a proper command-line flag, not via TF_XLA_FLAGS) or set the envvar XLA_FLAGS=--xla_hlo_profile.\n",
      "(deeplab) "
     ]
    }
   ],
   "source": [
    "cd $deeplab_path\n",
    "conda activate deeplab\n",
    "python -W ignore inference.py --image-list \"${data}/test.csv\" >> $data/log_inference.txt\n",
    "conda deactivate\n",
    "conda deactivate"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### 2.3. Post-processing"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Name outputs to reflect that they are segmentation results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence/\n"
     ]
    }
   ],
   "source": [
    "echo $data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [],
   "source": [
    "cd $data\n",
    "for file in *_rgb_cropped.png; do mv -f \"$file\" \"${file%_rgb_cropped.png}_segment_cropped.png\"; done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Re-expand segment outputs to same size as original RGB files"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Dependencies include `scikit-image`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(alignment) (alignment) (alignment) "
     ]
    }
   ],
   "source": [
    "conda activate alignment\n",
    "cd $alignment_path\n",
    "python expand.py $data >> $data/log_expand.txt\n",
    "conda deactivate"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Make composite images with side-by-side RGB, segmentation outputs and blended images"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence/\n"
     ]
    }
   ],
   "source": [
    "echo $data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(test-environment) (test-environment) "
     ]
    }
   ],
   "source": [
    "conda activate test-environment\n",
    "if [ $composite -eq 1 ]\n",
    "then\n",
    "    cd $gmolabeler_path\n",
    "    python image_blender.py $data 0.75 'both' 1 180 >> $data/log_blend.txt\n",
    "fi\n",
    "conda deactivate"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### 2.4. Classification of contaminated/missing explants"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Plates are cropped into sub-images for each explant and each is analyzed to determine if the explant position should be excluded from analysis due to being missing or contamination. Missing and contaminated explants are recognized using a trained Densenet model (<a href=\"https://github.com/Contamination-Classification/DenseNet\" target=\"_blank\">Huang, et al. 2018</a>). Our fork of the Densenet repository is available on <a href=\"https://arxiv.org/abs/1608.06993\" target=\"_blank\">GitHub</a>."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<img src=\"Figures/Densenet.png\">\n",
    "Figure: These are four examples of contaminated explants used in the training set for this pre-trained model"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "If the mode for missing explant data is automatic, prepare input file for script to detect missing explants and run this script."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [],
   "source": [
    "img_list_path=\"${data}/rgb_list.txt\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Dependencies include `keras-preprocessing`, `termcolor`,  `protobuf` and `absl-py`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//rgb_list.txt\n"
     ]
    }
   ],
   "source": [
    "echo $img_list_path"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "When running `inference.py` to detect missing or contaminated explants, the user should provide dimensions for cropping down to the grid borders. Note, these dimensions apply after the image is rescaled to 2000x2000. Default dimensions, used before the imager began to take \"off-center\" images, are (310, 515, 1750, 1610). Dimensions for cropping for the off-center images are (260, 600, 1700, 1710)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Missing explants will be inferred.\n",
      "Using TensorFlow backend.\n",
      "/home/labgroup/anaconda3/envs/densenet/lib/python3.6/site-packages/tensorflow/python/framework/dtypes.py:516: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.\n",
      "  _np_qint8 = np.dtype([(\"qint8\", np.int8, 1)])\n",
      "/home/labgroup/anaconda3/envs/densenet/lib/python3.6/site-packages/tensorflow/python/framework/dtypes.py:517: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.\n",
      "  _np_quint8 = np.dtype([(\"quint8\", np.uint8, 1)])\n",
      "/home/labgroup/anaconda3/envs/densenet/lib/python3.6/site-packages/tensorflow/python/framework/dtypes.py:518: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.\n",
      "  _np_qint16 = np.dtype([(\"qint16\", np.int16, 1)])\n",
      "/home/labgroup/anaconda3/envs/densenet/lib/python3.6/site-packages/tensorflow/python/framework/dtypes.py:519: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.\n",
      "  _np_quint16 = np.dtype([(\"quint16\", np.uint16, 1)])\n",
      "/home/labgroup/anaconda3/envs/densenet/lib/python3.6/site-packages/tensorflow/python/framework/dtypes.py:520: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.\n",
      "  _np_qint32 = np.dtype([(\"qint32\", np.int32, 1)])\n",
      "/home/labgroup/anaconda3/envs/densenet/lib/python3.6/site-packages/tensorflow/python/framework/dtypes.py:525: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.\n",
      "  np_resource = np.dtype([(\"resource\", np.ubyte, 1)])\n",
      "/home/labgroup/anaconda3/envs/densenet/lib/python3.6/site-packages/tensorboard/compat/tensorflow_stub/dtypes.py:541: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.\n",
      "  _np_qint8 = np.dtype([(\"qint8\", np.int8, 1)])\n",
      "/home/labgroup/anaconda3/envs/densenet/lib/python3.6/site-packages/tensorboard/compat/tensorflow_stub/dtypes.py:542: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.\n",
      "  _np_quint8 = np.dtype([(\"quint8\", np.uint8, 1)])\n",
      "/home/labgroup/anaconda3/envs/densenet/lib/python3.6/site-packages/tensorboard/compat/tensorflow_stub/dtypes.py:543: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.\n",
      "  _np_qint16 = np.dtype([(\"qint16\", np.int16, 1)])\n",
      "/home/labgroup/anaconda3/envs/densenet/lib/python3.6/site-packages/tensorboard/compat/tensorflow_stub/dtypes.py:544: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.\n",
      "  _np_quint16 = np.dtype([(\"quint16\", np.uint16, 1)])\n",
      "/home/labgroup/anaconda3/envs/densenet/lib/python3.6/site-packages/tensorboard/compat/tensorflow_stub/dtypes.py:545: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.\n",
      "  _np_qint32 = np.dtype([(\"qint32\", np.int32, 1)])\n",
      "/home/labgroup/anaconda3/envs/densenet/lib/python3.6/site-packages/tensorboard/compat/tensorflow_stub/dtypes.py:550: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.\n",
      "  np_resource = np.dtype([(\"resource\", np.ubyte, 1)])\n",
      "2021-01-21 21:23:17.702326: I tensorflow/core/platform/cpu_feature_guard.cc:142] Your CPU supports instructions that this TensorFlow binary was not compiled to use: SSE4.1 SSE4.2 AVX\n",
      "2021-01-21 21:23:17.918503: I tensorflow/core/platform/profile_utils/cpu_utils.cc:94] CPU Frequency: 2494325000 Hz\n",
      "2021-01-21 21:23:17.919154: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x55c24d5bbd10 executing computations on platform Host. Devices:\n",
      "2021-01-21 21:23:17.919227: I tensorflow/compiler/xla/service/service.cc:175]   StreamExecutor device (0): <undefined>, <undefined>\n",
      "2021-01-21 21:23:18.201904: W tensorflow/compiler/jit/mark_for_compilation_pass.cc:1412] (One-time warning): Not using XLA:CPU for cluster because envvar TF_XLA_FLAGS=--tf_xla_cpu_global_jit was not set.  If you want XLA:CPU, either set that envvar, or use experimental_jit_scope to enable XLA:CPU.  To confirm that XLA is active, pass --vmodule=xla_compilation_cache=1 (as a proper command-line flag, not via TF_XLA_FLAGS) or set the envvar XLA_FLAGS=--xla_hlo_profile.\n",
      "WARNING:tensorflow:From /home/labgroup/anaconda3/envs/densenet/lib/python3.6/site-packages/keras/backend/tensorflow_backend.py:4070: The name tf.nn.max_pool is deprecated. Please use tf.nn.max_pool2d instead.\n",
      "\n",
      "WARNING:tensorflow:From /home/labgroup/anaconda3/envs/densenet/lib/python3.6/site-packages/keras/backend/tensorflow_backend.py:4074: The name tf.nn.avg_pool is deprecated. Please use tf.nn.avg_pool2d instead.\n",
      "\n",
      "WARNING:tensorflow:From /home/labgroup/anaconda3/envs/densenet/lib/python3.6/site-packages/keras/backend/tensorflow_backend.py:422: The name tf.global_variables is deprecated. Please use tf.compat.v1.global_variables instead.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "if [ $missing_explants = \"Automatic\" ]; then\n",
    "    conda activate densenet\n",
    "    echo \"Missing explants will be inferred.\"\n",
    "    cd $data\n",
    "    ls -d $PWD/* $data | grep -i \"rgb.jpg\" > rgb_list.txt\n",
    "    sed -i '/hroma/d' rgb_list.txt\n",
    "    #sed -i 's/.jpg//g' rgb_list.txt\n",
    "    #cat rgb_list.txt\n",
    "    cd $contamination_path\n",
    "    img_list_path=\"${data}/rgb_list.txt\"\n",
    "    #echo $img_list_path\n",
    "    python inference.py \\\n",
    "    --img-list=$img_list_path \\\n",
    "    --crop_dims \"(260,600,1700,1710)\" \\\n",
    "    --output_file=output.csv >> $data/log_contam.txt\n",
    "    mv -f output.csv \"${data}/output.csv\"\n",
    "    missing_explants=\"${data}/output.csv\"\n",
    "    conda deactivate\n",
    "else\n",
    "    echo \"Missing explants input manually by user, in file: \"\n",
    "    echo $missing_explants\n",
    "fi"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 3. Alignment of RGB and hyperspectral layers"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "To match the frame and angle of RGB and hyperspectral image layers, we apply a scale-invariant feature transformation (<a href=\"https://github.com/NSF-Image-alignment/ImageAlignment\" target=\"_blank\">GitHub</a>). Using a pair of standard images, a homography matrix is calculated for the necessary transformation of RGB images to align with hyperspectral images. The transformation can then be applied to large batches of images rapidly, as long as the RGB and hyperspectral cameras remain in the same positions relative to one another (as they do in the macroPhor Array platform)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<img src=\"Figures/Alignment.png\">\n",
    "Figure: To enable precise calculation of a homography matrix for transformation of RGB images to match hyperspectral images, we used images of a piece of paper with grid marks. These images are provided by the user inputs to --hyper-img and --rgb-img in the below call to the alignment script. If using a phenotyping platform other than the macroPhor Array, or using updated camera settings, these variables will need to be replaced."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### 3.1. Prepare inputs for alignment"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We need to produce a csv with two columns with headers `hyper_img` and `rgb_images`. For each RGB image being transformed in batch alignment (mode 2), we can test the alignment by producing superimposed images of the transformed RGB images together with a hyperspectral layer. The hyperspectral layer can either be for a grid (fast) or can be for a layer extracted from the hyperspectral image of each channel (slow, but useful for making sure a certain homography matrix works reliably to transform a batch of images)."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Let's first generate the list that will go into the column `rgb_images`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [],
   "source": [
    "cd $data\n",
    "ls | grep -i 'rgb\\.jpg' > file_list_part1.csv\n",
    "ls | grep -i 'segment_uncropped\\.png' > file_list_part2.csv\n",
    "cat file_list_part* > file_list.csv\n",
    "# sed -i '/hroma/d' file_list.csv\n",
    "cwd2=$(pwd)/\n",
    "awk -v prefix=\"$cwd2\" '{print prefix $0}' file_list.csv > temp\n",
    "mv -f temp file_list.csv\n",
    "echo 'rgb_images' | cat - file_list.csv > temp && mv -f temp file_list.csv"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now to modify the list going into the `rgb_images` column to produce the list of filenames for `hyper_img`:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [],
   "source": [
    "cp file_list.csv file_list_hyper_channel.csv\n",
    "sed -i 's/_rgb.jpg/_hyperchannel.csv/g' file_list_hyper_channel.csv\n",
    "sed -i 's/rgb_images/hyper_img/g' file_list_hyper_channel.csv\n",
    "sed -i 's/_segment_uncropped.png/_hyperchannel.csv/g' file_list_hyper_channel.csv\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Bind these columns together..."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [],
   "source": [
    "paste --delimiters=',' file_list_hyper_channel.csv file_list.csv > rgb_and_hyper_channel_lists.csv"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### 3.2 Run alignment"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "grid_hdr=$(ls $data | grep -i \"grid\" | grep -i \"hdr\")\n",
    "grid_jpg=$(ls $data | grep -i \"grid\" | grep -i \"rgb.jpg\")\n",
    "\n",
    "echo $grid_hdr\n",
    "echo $grid_jpg"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "FAE1_I5.0_F1.9_L100_143813_0_0_0_Fluorescence.hdr\n",
      "FAE1_I5.0_F1.9_L100_143813_0_0_0_Fluorescence.raw\n",
      "FAE1_I5.0_F1.9_L100_143813_0_0_0_hyperchannel.csv\n",
      "FAE1_I5.0_F1.9_L100_143813_0_0_0_rgb_cropped.jpg\n",
      "FAE1_I5.0_F1.9_L100_143813_0_0_0_rgb.jpg\n",
      "FAE1_I5.0_F1.9_L100_143813_0_0_0_segment_cropped.png\n",
      "FAE1_I5.0_F1.9_L100_143813_0_0_0_segment_uncropped.png\n",
      "FAE1_I5.0_F1.9_L100_143859_1_0_1_Fluorescence.hdr\n",
      "FAE1_I5.0_F1.9_L100_143859_1_0_1_Fluorescence.raw\n",
      "FAE1_I5.0_F1.9_L100_143859_1_0_1_hyperchannel.csv\n",
      "FAE1_I5.0_F1.9_L100_143859_1_0_1_rgb_cropped.jpg\n",
      "FAE1_I5.0_F1.9_L100_143859_1_0_1_rgb.jpg\n",
      "FAE1_I5.0_F1.9_L100_143859_1_0_1_segment_cropped.png\n",
      "FAE1_I5.0_F1.9_L100_143859_1_0_1_segment_uncropped.png\n",
      "FAE1_I5.0_F1.9_L100_143959_2_0_2_Fluorescence.hdr\n",
      "FAE1_I5.0_F1.9_L100_143959_2_0_2_Fluorescence.raw\n",
      "FAE1_I5.0_F1.9_L100_143959_2_0_2_hyperchannel.csv\n",
      "FAE1_I5.0_F1.9_L100_143959_2_0_2_rgb_cropped.jpg\n",
      "FAE1_I5.0_F1.9_L100_143959_2_0_2_rgb.jpg\n",
      "FAE1_I5.0_F1.9_L100_143959_2_0_2_segment_cropped.png\n",
      "FAE1_I5.0_F1.9_L100_143959_2_0_2_segment_uncropped.png\n",
      "FAE1_I5.0_F1.9_L100_144101_3_0_3_Fluorescence.hdr\n",
      "FAE1_I5.0_F1.9_L100_144101_3_0_3_Fluorescence.raw\n",
      "FAE1_I5.0_F1.9_L100_144101_3_0_3_hyperchannel.csv\n",
      "FAE1_I5.0_F1.9_L100_144101_3_0_3_rgb_cropped.jpg\n",
      "FAE1_I5.0_F1.9_L100_144101_3_0_3_rgb.jpg\n",
      "FAE1_I5.0_F1.9_L100_144101_3_0_3_segment_cropped.png\n",
      "FAE1_I5.0_F1.9_L100_144101_3_0_3_segment_uncropped.png\n",
      "FAE1_I5.0_F1.9_L100_144200_4_0_4_Fluorescence.hdr\n",
      "FAE1_I5.0_F1.9_L100_144200_4_0_4_Fluorescence.raw\n",
      "FAE1_I5.0_F1.9_L100_144200_4_0_4_hyperchannel.csv\n",
      "FAE1_I5.0_F1.9_L100_144200_4_0_4_rgb_cropped.jpg\n",
      "FAE1_I5.0_F1.9_L100_144200_4_0_4_rgb.jpg\n",
      "FAE1_I5.0_F1.9_L100_144200_4_0_4_segment_cropped.png\n",
      "FAE1_I5.0_F1.9_L100_144200_4_0_4_segment_uncropped.png\n",
      "FAE1_I5.0_F1.9_L100_144259_5_0_5_Fluorescence.hdr\n",
      "FAE1_I5.0_F1.9_L100_144259_5_0_5_Fluorescence.raw\n",
      "FAE1_I5.0_F1.9_L100_144259_5_0_5_hyperchannel.csv\n",
      "FAE1_I5.0_F1.9_L100_144259_5_0_5_rgb_cropped.jpg\n",
      "FAE1_I5.0_F1.9_L100_144259_5_0_5_rgb.jpg\n",
      "FAE1_I5.0_F1.9_L100_144259_5_0_5_segment_cropped.png\n",
      "FAE1_I5.0_F1.9_L100_144259_5_0_5_segment_uncropped.png\n",
      "FAE1_I5.0_F1.9_L100_144359_6_0_6_Fluorescence.hdr\n",
      "FAE1_I5.0_F1.9_L100_144359_6_0_6_Fluorescence.raw\n",
      "FAE1_I5.0_F1.9_L100_144359_6_0_6_hyperchannel.csv\n",
      "FAE1_I5.0_F1.9_L100_144359_6_0_6_rgb_cropped.jpg\n",
      "FAE1_I5.0_F1.9_L100_144359_6_0_6_rgb.jpg\n",
      "FAE1_I5.0_F1.9_L100_144359_6_0_6_segment_cropped.png\n",
      "FAE1_I5.0_F1.9_L100_144359_6_0_6_segment_uncropped.png\n",
      "FAE1_I5.0_F1.9_L100_144459_7_1_6_Fluorescence.hdr\n",
      "FAE1_I5.0_F1.9_L100_144459_7_1_6_Fluorescence.raw\n",
      "FAE1_I5.0_F1.9_L100_144459_7_1_6_hyperchannel.csv\n",
      "FAE1_I5.0_F1.9_L100_144459_7_1_6_rgb_cropped.jpg\n",
      "FAE1_I5.0_F1.9_L100_144459_7_1_6_rgb.jpg\n",
      "FAE1_I5.0_F1.9_L100_144459_7_1_6_segment_cropped.png\n",
      "FAE1_I5.0_F1.9_L100_144459_7_1_6_segment_uncropped.png\n",
      "FAE1_I5.0_F1.9_L100_144550_8_1_5_Fluorescence.hdr\n",
      "FAE1_I5.0_F1.9_L100_144550_8_1_5_Fluorescence.raw\n",
      "FAE1_I5.0_F1.9_L100_144550_8_1_5_hyperchannel.csv\n",
      "FAE1_I5.0_F1.9_L100_144550_8_1_5_rgb_cropped.jpg\n",
      "FAE1_I5.0_F1.9_L100_144550_8_1_5_rgb.jpg\n",
      "FAE1_I5.0_F1.9_L100_144550_8_1_5_segment_cropped.png\n",
      "FAE1_I5.0_F1.9_L100_144550_8_1_5_segment_uncropped.png\n",
      "FAE1_I5.0_F1.9_L100_144650_9_1_4_Fluorescence.hdr\n",
      "FAE1_I5.0_F1.9_L100_144650_9_1_4_Fluorescence.raw\n",
      "FAE1_I5.0_F1.9_L100_144650_9_1_4_hyperchannel.csv\n",
      "FAE1_I5.0_F1.9_L100_144650_9_1_4_rgb_cropped.jpg\n",
      "FAE1_I5.0_F1.9_L100_144650_9_1_4_rgb.jpg\n",
      "FAE1_I5.0_F1.9_L100_144650_9_1_4_segment_cropped.png\n",
      "FAE1_I5.0_F1.9_L100_144650_9_1_4_segment_uncropped.png\n",
      "FAE1_I5.0_F1.9_L100_144749_10_1_3_Fluorescence.hdr\n",
      "FAE1_I5.0_F1.9_L100_144749_10_1_3_Fluorescence.raw\n",
      "FAE1_I5.0_F1.9_L100_144749_10_1_3_hyperchannel.csv\n",
      "FAE1_I5.0_F1.9_L100_144749_10_1_3_rgb_cropped.jpg\n",
      "FAE1_I5.0_F1.9_L100_144749_10_1_3_rgb.jpg\n",
      "FAE1_I5.0_F1.9_L100_144749_10_1_3_segment_cropped.png\n",
      "FAE1_I5.0_F1.9_L100_144749_10_1_3_segment_uncropped.png\n",
      "FAE1_I5.0_F1.9_L100_144848_11_1_2_Fluorescence.hdr\n",
      "FAE1_I5.0_F1.9_L100_144848_11_1_2_Fluorescence.raw\n",
      "FAE1_I5.0_F1.9_L100_144848_11_1_2_hyperchannel.csv\n",
      "FAE1_I5.0_F1.9_L100_144848_11_1_2_rgb_cropped.jpg\n",
      "FAE1_I5.0_F1.9_L100_144848_11_1_2_rgb.jpg\n",
      "FAE1_I5.0_F1.9_L100_144848_11_1_2_segment_cropped.png\n",
      "FAE1_I5.0_F1.9_L100_144848_11_1_2_segment_uncropped.png\n",
      "FAE1_I5.0_F1.9_L100_144948_12_1_1_Fluorescence.hdr\n",
      "FAE1_I5.0_F1.9_L100_144948_12_1_1_Fluorescence.raw\n",
      "FAE1_I5.0_F1.9_L100_144948_12_1_1_hyperchannel.csv\n",
      "FAE1_I5.0_F1.9_L100_144948_12_1_1_rgb_cropped.jpg\n",
      "FAE1_I5.0_F1.9_L100_144948_12_1_1_rgb.jpg\n",
      "FAE1_I5.0_F1.9_L100_144948_12_1_1_segment_cropped.png\n",
      "FAE1_I5.0_F1.9_L100_144948_12_1_1_segment_uncropped.png\n",
      "FAE1_I5.0_F1.9_L100_145051_13_1_0_Fluorescence.hdr\n",
      "FAE1_I5.0_F1.9_L100_145051_13_1_0_Fluorescence.raw\n",
      "FAE1_I5.0_F1.9_L100_145051_13_1_0_hyperchannel.csv\n",
      "FAE1_I5.0_F1.9_L100_145051_13_1_0_rgb_cropped.jpg\n",
      "FAE1_I5.0_F1.9_L100_145051_13_1_0_rgb.jpg\n",
      "FAE1_I5.0_F1.9_L100_145051_13_1_0_segment_cropped.png\n",
      "FAE1_I5.0_F1.9_L100_145051_13_1_0_segment_uncropped.png\n",
      "FAE1_I5.0_F1.9_L100_145151_14_2_0_Fluorescence.hdr\n",
      "FAE1_I5.0_F1.9_L100_145151_14_2_0_Fluorescence.raw\n",
      "FAE1_I5.0_F1.9_L100_145151_14_2_0_hyperchannel.csv\n",
      "FAE1_I5.0_F1.9_L100_145151_14_2_0_rgb_cropped.jpg\n",
      "FAE1_I5.0_F1.9_L100_145151_14_2_0_rgb.jpg\n",
      "FAE1_I5.0_F1.9_L100_145151_14_2_0_segment_cropped.png\n",
      "FAE1_I5.0_F1.9_L100_145151_14_2_0_segment_uncropped.png\n",
      "FAE1_I5.0_F1.9_L100_145243_15_2_1_Fluorescence.hdr\n",
      "FAE1_I5.0_F1.9_L100_145243_15_2_1_Fluorescence.raw\n",
      "FAE1_I5.0_F1.9_L100_145243_15_2_1_hyperchannel.csv\n",
      "FAE1_I5.0_F1.9_L100_145243_15_2_1_rgb_cropped.jpg\n",
      "FAE1_I5.0_F1.9_L100_145243_15_2_1_rgb.jpg\n",
      "FAE1_I5.0_F1.9_L100_145243_15_2_1_segment_cropped.png\n",
      "FAE1_I5.0_F1.9_L100_145243_15_2_1_segment_uncropped.png\n",
      "FAE1_I5.0_F1.9_L100_145343_16_2_2_Fluorescence.hdr\n",
      "FAE1_I5.0_F1.9_L100_145343_16_2_2_Fluorescence.raw\n",
      "FAE1_I5.0_F1.9_L100_145343_16_2_2_hyperchannel.csv\n",
      "FAE1_I5.0_F1.9_L100_145343_16_2_2_rgb_cropped.jpg\n",
      "FAE1_I5.0_F1.9_L100_145343_16_2_2_rgb.jpg\n",
      "FAE1_I5.0_F1.9_L100_145343_16_2_2_segment_cropped.png\n",
      "FAE1_I5.0_F1.9_L100_145343_16_2_2_segment_uncropped.png\n",
      "FAE1_I5.0_F1.9_L100_145442_17_2_3_Fluorescence.hdr\n",
      "FAE1_I5.0_F1.9_L100_145442_17_2_3_Fluorescence.raw\n",
      "FAE1_I5.0_F1.9_L100_145442_17_2_3_hyperchannel.csv\n",
      "FAE1_I5.0_F1.9_L100_145442_17_2_3_rgb_cropped.jpg\n",
      "FAE1_I5.0_F1.9_L100_145442_17_2_3_rgb.jpg\n",
      "FAE1_I5.0_F1.9_L100_145442_17_2_3_segment_cropped.png\n",
      "FAE1_I5.0_F1.9_L100_145442_17_2_3_segment_uncropped.png\n",
      "FAE1_I5.0_F1.9_L100_145544_18_2_4_Fluorescence.hdr\n",
      "FAE1_I5.0_F1.9_L100_145544_18_2_4_Fluorescence.raw\n",
      "FAE1_I5.0_F1.9_L100_145544_18_2_4_hyperchannel.csv\n",
      "FAE1_I5.0_F1.9_L100_145544_18_2_4_rgb_cropped.jpg\n",
      "FAE1_I5.0_F1.9_L100_145544_18_2_4_rgb.jpg\n",
      "FAE1_I5.0_F1.9_L100_145544_18_2_4_segment_cropped.png\n",
      "FAE1_I5.0_F1.9_L100_145544_18_2_4_segment_uncropped.png\n",
      "FAE1_I5.0_F1.9_L100_145643_19_2_5_Fluorescence.hdr\n",
      "FAE1_I5.0_F1.9_L100_145643_19_2_5_Fluorescence.raw\n",
      "FAE1_I5.0_F1.9_L100_145643_19_2_5_hyperchannel.csv\n",
      "FAE1_I5.0_F1.9_L100_145643_19_2_5_rgb_cropped.jpg\n",
      "FAE1_I5.0_F1.9_L100_145643_19_2_5_rgb.jpg\n",
      "FAE1_I5.0_F1.9_L100_145643_19_2_5_segment_cropped.png\n",
      "FAE1_I5.0_F1.9_L100_145643_19_2_5_segment_uncropped.png\n",
      "FAE1_I5.0_F1.9_L100_145744_20_2_6_Fluorescence.hdr\n",
      "FAE1_I5.0_F1.9_L100_145744_20_2_6_Fluorescence.raw\n",
      "FAE1_I5.0_F1.9_L100_145744_20_2_6_hyperchannel.csv\n",
      "FAE1_I5.0_F1.9_L100_145744_20_2_6_rgb_cropped.jpg\n",
      "FAE1_I5.0_F1.9_L100_145744_20_2_6_rgb.jpg\n",
      "FAE1_I5.0_F1.9_L100_145744_20_2_6_segment_cropped.png\n",
      "FAE1_I5.0_F1.9_L100_145744_20_2_6_segment_uncropped.png\n",
      "FAE1_I5.0_F1.9_L100_152151_0_1_6_Fluorescence.hdr\n",
      "FAE1_I5.0_F1.9_L100_152151_0_1_6_Fluorescence.raw\n",
      "FAE1_I5.0_F1.9_L100_152151_0_1_6_hyperchannel.csv\n",
      "FAE1_I5.0_F1.9_L100_152151_0_1_6_rgb_cropped.jpg\n",
      "FAE1_I5.0_F1.9_L100_152151_0_1_6_rgb.jpg\n",
      "FAE1_I5.0_F1.9_L100_152151_0_1_6_segment_cropped.png\n",
      "FAE1_I5.0_F1.9_L100_152151_0_1_6_segment_uncropped.png\n",
      "FAE2_I5.0_F1.9_L100_150028_0_0_0_Fluorescence.hdr\n",
      "FAE2_I5.0_F1.9_L100_150028_0_0_0_Fluorescence.raw\n",
      "FAE2_I5.0_F1.9_L100_150028_0_0_0_hyperchannel.csv\n",
      "FAE2_I5.0_F1.9_L100_150028_0_0_0_rgb_cropped.jpg\n",
      "FAE2_I5.0_F1.9_L100_150028_0_0_0_rgb.jpg\n",
      "FAE2_I5.0_F1.9_L100_150028_0_0_0_segment_cropped.png\n",
      "FAE2_I5.0_F1.9_L100_150028_0_0_0_segment_uncropped.png\n",
      "FAE2_I5.0_F1.9_L100_150113_1_0_1_Fluorescence.hdr\n",
      "FAE2_I5.0_F1.9_L100_150113_1_0_1_Fluorescence.raw\n",
      "FAE2_I5.0_F1.9_L100_150113_1_0_1_hyperchannel.csv\n",
      "FAE2_I5.0_F1.9_L100_150113_1_0_1_rgb_cropped.jpg\n",
      "FAE2_I5.0_F1.9_L100_150113_1_0_1_rgb.jpg\n",
      "FAE2_I5.0_F1.9_L100_150113_1_0_1_segment_cropped.png\n",
      "FAE2_I5.0_F1.9_L100_150113_1_0_1_segment_uncropped.png\n",
      "FAE2_I5.0_F1.9_L100_150214_2_0_2_Fluorescence.hdr\n",
      "FAE2_I5.0_F1.9_L100_150214_2_0_2_Fluorescence.raw\n",
      "FAE2_I5.0_F1.9_L100_150214_2_0_2_hyperchannel.csv\n",
      "FAE2_I5.0_F1.9_L100_150214_2_0_2_rgb_cropped.jpg\n",
      "FAE2_I5.0_F1.9_L100_150214_2_0_2_rgb.jpg\n",
      "FAE2_I5.0_F1.9_L100_150214_2_0_2_segment_cropped.png\n",
      "FAE2_I5.0_F1.9_L100_150214_2_0_2_segment_uncropped.png\n",
      "FAE2_I5.0_F1.9_L100_150313_3_0_3_Fluorescence.hdr\n",
      "FAE2_I5.0_F1.9_L100_150313_3_0_3_Fluorescence.raw\n",
      "FAE2_I5.0_F1.9_L100_150313_3_0_3_hyperchannel.csv\n",
      "FAE2_I5.0_F1.9_L100_150313_3_0_3_rgb_cropped.jpg\n",
      "FAE2_I5.0_F1.9_L100_150313_3_0_3_rgb.jpg\n",
      "FAE2_I5.0_F1.9_L100_150313_3_0_3_segment_cropped.png\n",
      "FAE2_I5.0_F1.9_L100_150313_3_0_3_segment_uncropped.png\n",
      "FAE2_I5.0_F1.9_L100_150413_4_0_4_Fluorescence.hdr\n",
      "FAE2_I5.0_F1.9_L100_150413_4_0_4_Fluorescence.raw\n",
      "FAE2_I5.0_F1.9_L100_150413_4_0_4_hyperchannel.csv\n",
      "FAE2_I5.0_F1.9_L100_150413_4_0_4_rgb_cropped.jpg\n",
      "FAE2_I5.0_F1.9_L100_150413_4_0_4_rgb.jpg\n",
      "FAE2_I5.0_F1.9_L100_150413_4_0_4_segment_cropped.png\n",
      "FAE2_I5.0_F1.9_L100_150413_4_0_4_segment_uncropped.png\n",
      "FAE2_I5.0_F1.9_L100_150513_5_0_5_Fluorescence.hdr\n",
      "FAE2_I5.0_F1.9_L100_150513_5_0_5_Fluorescence.raw\n",
      "FAE2_I5.0_F1.9_L100_150513_5_0_5_hyperchannel.csv\n",
      "FAE2_I5.0_F1.9_L100_150513_5_0_5_rgb_cropped.jpg\n",
      "FAE2_I5.0_F1.9_L100_150513_5_0_5_rgb.jpg\n",
      "FAE2_I5.0_F1.9_L100_150513_5_0_5_segment_cropped.png\n",
      "FAE2_I5.0_F1.9_L100_150513_5_0_5_segment_uncropped.png\n",
      "FAE2_I5.0_F1.9_L100_150613_6_0_6_Fluorescence.hdr\n",
      "FAE2_I5.0_F1.9_L100_150613_6_0_6_Fluorescence.raw\n",
      "FAE2_I5.0_F1.9_L100_150613_6_0_6_hyperchannel.csv\n",
      "FAE2_I5.0_F1.9_L100_150613_6_0_6_rgb_cropped.jpg\n",
      "FAE2_I5.0_F1.9_L100_150613_6_0_6_rgb.jpg\n",
      "FAE2_I5.0_F1.9_L100_150613_6_0_6_segment_cropped.png\n",
      "FAE2_I5.0_F1.9_L100_150613_6_0_6_segment_uncropped.png\n",
      "FAE2_I5.0_F1.9_L100_150713_7_1_6_Fluorescence.hdr\n",
      "FAE2_I5.0_F1.9_L100_150713_7_1_6_Fluorescence.raw\n",
      "FAE2_I5.0_F1.9_L100_150713_7_1_6_hyperchannel.csv\n",
      "FAE2_I5.0_F1.9_L100_150713_7_1_6_rgb_cropped.jpg\n",
      "FAE2_I5.0_F1.9_L100_150713_7_1_6_rgb.jpg\n",
      "FAE2_I5.0_F1.9_L100_150713_7_1_6_segment_cropped.png\n",
      "FAE2_I5.0_F1.9_L100_150713_7_1_6_segment_uncropped.png\n",
      "FAE2_I5.0_F1.9_L100_150803_8_1_5_Fluorescence.hdr\n",
      "FAE2_I5.0_F1.9_L100_150803_8_1_5_Fluorescence.raw\n",
      "FAE2_I5.0_F1.9_L100_150803_8_1_5_hyperchannel.csv\n",
      "FAE2_I5.0_F1.9_L100_150803_8_1_5_rgb_cropped.jpg\n",
      "FAE2_I5.0_F1.9_L100_150803_8_1_5_rgb.jpg\n",
      "FAE2_I5.0_F1.9_L100_150803_8_1_5_segment_cropped.png\n",
      "FAE2_I5.0_F1.9_L100_150803_8_1_5_segment_uncropped.png\n",
      "FAE2_I5.0_F1.9_L100_150902_9_1_4_Fluorescence.hdr\n",
      "FAE2_I5.0_F1.9_L100_150902_9_1_4_Fluorescence.raw\n",
      "FAE2_I5.0_F1.9_L100_150902_9_1_4_hyperchannel.csv\n",
      "FAE2_I5.0_F1.9_L100_150902_9_1_4_rgb_cropped.jpg\n",
      "FAE2_I5.0_F1.9_L100_150902_9_1_4_rgb.jpg\n",
      "FAE2_I5.0_F1.9_L100_150902_9_1_4_segment_cropped.png\n",
      "FAE2_I5.0_F1.9_L100_150902_9_1_4_segment_uncropped.png\n",
      "FAE2_I5.0_F1.9_L100_151002_10_1_3_Fluorescence.hdr\n",
      "FAE2_I5.0_F1.9_L100_151002_10_1_3_Fluorescence.raw\n",
      "FAE2_I5.0_F1.9_L100_151002_10_1_3_hyperchannel.csv\n",
      "FAE2_I5.0_F1.9_L100_151002_10_1_3_rgb_cropped.jpg\n",
      "FAE2_I5.0_F1.9_L100_151002_10_1_3_rgb.jpg\n",
      "FAE2_I5.0_F1.9_L100_151002_10_1_3_segment_cropped.png\n",
      "FAE2_I5.0_F1.9_L100_151002_10_1_3_segment_uncropped.png\n",
      "FAE2_I5.0_F1.9_L100_151102_11_1_1_Fluorescence.hdr\n",
      "FAE2_I5.0_F1.9_L100_151102_11_1_1_Fluorescence.raw\n",
      "FAE2_I5.0_F1.9_L100_151102_11_1_1_hyperchannel.csv\n",
      "FAE2_I5.0_F1.9_L100_151102_11_1_1_rgb_cropped.jpg\n",
      "FAE2_I5.0_F1.9_L100_151102_11_1_1_rgb.jpg\n",
      "FAE2_I5.0_F1.9_L100_151102_11_1_1_segment_cropped.png\n",
      "FAE2_I5.0_F1.9_L100_151102_11_1_1_segment_uncropped.png\n",
      "FAE2_I5.0_F1.9_L100_151202_12_1_0_Fluorescence.hdr\n",
      "FAE2_I5.0_F1.9_L100_151202_12_1_0_Fluorescence.raw\n",
      "FAE2_I5.0_F1.9_L100_151202_12_1_0_hyperchannel.csv\n",
      "FAE2_I5.0_F1.9_L100_151202_12_1_0_rgb_cropped.jpg\n",
      "FAE2_I5.0_F1.9_L100_151202_12_1_0_rgb.jpg\n",
      "FAE2_I5.0_F1.9_L100_151202_12_1_0_segment_cropped.png\n",
      "FAE2_I5.0_F1.9_L100_151202_12_1_0_segment_uncropped.png\n",
      "FAE2_I5.0_F1.9_L100_151302_13_2_0_Fluorescence.hdr\n",
      "FAE2_I5.0_F1.9_L100_151302_13_2_0_Fluorescence.raw\n",
      "FAE2_I5.0_F1.9_L100_151302_13_2_0_hyperchannel.csv\n",
      "FAE2_I5.0_F1.9_L100_151302_13_2_0_rgb_cropped.jpg\n",
      "FAE2_I5.0_F1.9_L100_151302_13_2_0_rgb.jpg\n",
      "FAE2_I5.0_F1.9_L100_151302_13_2_0_segment_cropped.png\n",
      "FAE2_I5.0_F1.9_L100_151302_13_2_0_segment_uncropped.png\n",
      "FAE2_I5.0_F1.9_L100_151354_14_2_1_Fluorescence.hdr\n",
      "FAE2_I5.0_F1.9_L100_151354_14_2_1_Fluorescence.raw\n",
      "FAE2_I5.0_F1.9_L100_151354_14_2_1_hyperchannel.csv\n",
      "FAE2_I5.0_F1.9_L100_151354_14_2_1_rgb_cropped.jpg\n",
      "FAE2_I5.0_F1.9_L100_151354_14_2_1_rgb.jpg\n",
      "FAE2_I5.0_F1.9_L100_151354_14_2_1_segment_cropped.png\n",
      "FAE2_I5.0_F1.9_L100_151354_14_2_1_segment_uncropped.png\n",
      "FAE2_I5.0_F1.9_L100_151456_15_2_2_Fluorescence.hdr\n",
      "FAE2_I5.0_F1.9_L100_151456_15_2_2_Fluorescence.raw\n",
      "FAE2_I5.0_F1.9_L100_151456_15_2_2_hyperchannel.csv\n",
      "FAE2_I5.0_F1.9_L100_151456_15_2_2_rgb_cropped.jpg\n",
      "FAE2_I5.0_F1.9_L100_151456_15_2_2_rgb.jpg\n",
      "FAE2_I5.0_F1.9_L100_151456_15_2_2_segment_cropped.png\n",
      "FAE2_I5.0_F1.9_L100_151456_15_2_2_segment_uncropped.png\n",
      "FAE2_I5.0_F1.9_L100_151556_16_2_3_Fluorescence.hdr\n",
      "FAE2_I5.0_F1.9_L100_151556_16_2_3_Fluorescence.raw\n",
      "FAE2_I5.0_F1.9_L100_151556_16_2_3_hyperchannel.csv\n",
      "FAE2_I5.0_F1.9_L100_151556_16_2_3_rgb_cropped.jpg\n",
      "FAE2_I5.0_F1.9_L100_151556_16_2_3_rgb.jpg\n",
      "FAE2_I5.0_F1.9_L100_151556_16_2_3_segment_cropped.png\n",
      "FAE2_I5.0_F1.9_L100_151556_16_2_3_segment_uncropped.png\n",
      "FAE2_I5.0_F1.9_L100_151657_17_2_4_Fluorescence.hdr\n",
      "FAE2_I5.0_F1.9_L100_151657_17_2_4_Fluorescence.raw\n",
      "FAE2_I5.0_F1.9_L100_151657_17_2_4_hyperchannel.csv\n",
      "FAE2_I5.0_F1.9_L100_151657_17_2_4_rgb_cropped.jpg\n",
      "FAE2_I5.0_F1.9_L100_151657_17_2_4_rgb.jpg\n",
      "FAE2_I5.0_F1.9_L100_151657_17_2_4_segment_cropped.png\n",
      "FAE2_I5.0_F1.9_L100_151657_17_2_4_segment_uncropped.png\n",
      "FAE2_I5.0_F1.9_L100_151757_18_2_6_Fluorescence.hdr\n",
      "FAE2_I5.0_F1.9_L100_151757_18_2_6_Fluorescence.raw\n",
      "FAE2_I5.0_F1.9_L100_151757_18_2_6_hyperchannel.csv\n",
      "FAE2_I5.0_F1.9_L100_151757_18_2_6_rgb_cropped.jpg\n",
      "FAE2_I5.0_F1.9_L100_151757_18_2_6_rgb.jpg\n",
      "FAE2_I5.0_F1.9_L100_151757_18_2_6_segment_cropped.png\n",
      "FAE2_I5.0_F1.9_L100_151757_18_2_6_segment_uncropped.png\n",
      "FAE3_I5.0_F1.9_L100_143049_0_0_0_Fluorescence.hdr\n",
      "FAE3_I5.0_F1.9_L100_143049_0_0_0_Fluorescence.raw\n",
      "FAE3_I5.0_F1.9_L100_143049_0_0_0_hyperchannel.csv\n",
      "FAE3_I5.0_F1.9_L100_143049_0_0_0_rgb_cropped.jpg\n",
      "FAE3_I5.0_F1.9_L100_143049_0_0_0_rgb.jpg\n",
      "FAE3_I5.0_F1.9_L100_143049_0_0_0_segment_cropped.png\n",
      "FAE3_I5.0_F1.9_L100_143049_0_0_0_segment_uncropped.png\n",
      "FAE3_I5.0_F1.9_L100_143135_1_0_1_Fluorescence.hdr\n",
      "FAE3_I5.0_F1.9_L100_143135_1_0_1_Fluorescence.raw\n",
      "FAE3_I5.0_F1.9_L100_143135_1_0_1_hyperchannel.csv\n",
      "FAE3_I5.0_F1.9_L100_143135_1_0_1_rgb_cropped.jpg\n",
      "FAE3_I5.0_F1.9_L100_143135_1_0_1_rgb.jpg\n",
      "FAE3_I5.0_F1.9_L100_143135_1_0_1_segment_cropped.png\n",
      "FAE3_I5.0_F1.9_L100_143135_1_0_1_segment_uncropped.png\n",
      "FAE3_I5.0_F1.9_L100_143235_2_0_2_Fluorescence.hdr\n",
      "FAE3_I5.0_F1.9_L100_143235_2_0_2_Fluorescence.raw\n",
      "FAE3_I5.0_F1.9_L100_143235_2_0_2_hyperchannel.csv\n",
      "FAE3_I5.0_F1.9_L100_143235_2_0_2_rgb_cropped.jpg\n",
      "FAE3_I5.0_F1.9_L100_143235_2_0_2_rgb.jpg\n",
      "FAE3_I5.0_F1.9_L100_143235_2_0_2_segment_cropped.png\n",
      "FAE3_I5.0_F1.9_L100_143235_2_0_2_segment_uncropped.png\n",
      "FAE3_I5.0_F1.9_L100_143334_3_0_3_Fluorescence.hdr\n",
      "FAE3_I5.0_F1.9_L100_143334_3_0_3_Fluorescence.raw\n",
      "FAE3_I5.0_F1.9_L100_143334_3_0_3_hyperchannel.csv\n",
      "FAE3_I5.0_F1.9_L100_143334_3_0_3_rgb_cropped.jpg\n",
      "FAE3_I5.0_F1.9_L100_143334_3_0_3_rgb.jpg\n",
      "FAE3_I5.0_F1.9_L100_143334_3_0_3_segment_cropped.png\n",
      "FAE3_I5.0_F1.9_L100_143334_3_0_3_segment_uncropped.png\n",
      "FAE3_I5.0_F1.9_L100_143434_4_0_4_Fluorescence.hdr\n",
      "FAE3_I5.0_F1.9_L100_143434_4_0_4_Fluorescence.raw\n",
      "FAE3_I5.0_F1.9_L100_143434_4_0_4_hyperchannel.csv\n",
      "FAE3_I5.0_F1.9_L100_143434_4_0_4_rgb_cropped.jpg\n",
      "FAE3_I5.0_F1.9_L100_143434_4_0_4_rgb.jpg\n",
      "FAE3_I5.0_F1.9_L100_143434_4_0_4_segment_cropped.png\n",
      "FAE3_I5.0_F1.9_L100_143434_4_0_4_segment_uncropped.png\n",
      "file_list.csv\n",
      "file_list_hyper_channel.csv\n",
      "file_list_part1.csv\n",
      "file_list_part2.csv\n",
      "hdr_backup\n",
      "log_contam.txt\n",
      "log_expand.txt\n",
      "log_gmodetector.txt\n",
      "log_inference.txt\n",
      "output.csv\n",
      "rgb_and_hyper_channel_lists.csv\n",
      "rgb_list.txt\n",
      "test.csv\n"
     ]
    }
   ],
   "source": [
    "ls $data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### Generate `csv` of a hyperspectral channel for the grid standard"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(test-environment) (test-environment) Reading hyperspectral image /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence/\n",
      "Traceback (most recent call last):\n",
      "  File \"hypercube_to_csv.py\", line 40, in <module>\n",
      "    main(hdr_path = sys.argv[1],\n",
      "  File \"hypercube_to_csv.py\", line 7, in main\n",
      "    image1 = envi.open(hdr_path)\n",
      "  File \"/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py\", line 289, in open\n",
      "    header_path = find_file_path(file)\n",
      "  File \"/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/spyfile.py\", line 120, in find_file_path\n",
      "    raise FileNotFoundError(msg)\n",
      "spectral.io.spyfile.FileNotFoundError: Unable to locate file \"/run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence/\". If the file exists, use its full path or place its directory in the SPECTRAL_DATA environment variable.\n",
      "(test-environment) "
     ]
    }
   ],
   "source": [
    "conda activate test-environment\n",
    "cd $alignment_path\n",
    "python hypercube_to_csv.py \\\n",
    "$data$grid_hdr \\\n",
    "700 \\\n",
    "130 \\\n",
    "$data\"hypercube_to_csv.csv\" \\\n",
    "$data\"hypercube_to_csv.jpg\"\n",
    "conda deactivate"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### Run alignment to generate homography matrix"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [],
   "source": [
    "aligned=$(echo $grid_jpg | sed 's/rgb/rgb_aligned/g')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence/hypercube_to_csv.csv\n",
      "/run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence/\n"
     ]
    }
   ],
   "source": [
    "echo $data\"hypercube_to_csv.csv\"\n",
    "echo $data$grid_jpg"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(alignment) (alignment) Traceback (most recent call last):\n",
      "  File \"main.py\", line 215, in <module>\n",
      "    main(args)\n",
      "  File \"main.py\", line 57, in main\n",
      "    hyp_img = genfromtxt(hyper_img_path, delimiter=',')\n",
      "  File \"/home/labgroup/anaconda3/envs/alignment/lib/python3.7/site-packages/numpy/lib/npyio.py\", line 1749, in genfromtxt\n",
      "    fid = np.lib._datasource.open(fname, 'rt', encoding=encoding)\n",
      "  File \"/home/labgroup/anaconda3/envs/alignment/lib/python3.7/site-packages/numpy/lib/_datasource.py\", line 195, in open\n",
      "    return ds.open(path, mode, encoding=encoding, newline=newline)\n",
      "  File \"/home/labgroup/anaconda3/envs/alignment/lib/python3.7/site-packages/numpy/lib/_datasource.py\", line 535, in open\n",
      "    raise IOError(\"%s not found.\" % path)\n",
      "OSError: /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence/hypercube_to_csv.csv not found.\n",
      "(alignment) "
     ]
    }
   ],
   "source": [
    "conda activate alignment\n",
    "cd $alignment_path\n",
    "python main.py \\\n",
    "--hyper-img $data\"hypercube_to_csv.csv\" \\\n",
    "--rgb-img $data$grid_jpg \\\n",
    "--mode 1 \\\n",
    "--image_thresh_high=120 \\\n",
    "--image_thresh_low=50 \\\n",
    "--distance=0.6 \\\n",
    "--gaussian_sigma 0.55\n",
    "conda deactivate"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "cat: -: Is a directory\n",
      "Not a valid image: b''"
     ]
    }
   ],
   "source": [
    "display < output/$aligned"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [],
   "source": [
    "# conda activate alignment\n",
    "# python main.py \\\n",
    "# --hyper-img '/home/labgroup/Downloads/input/chromagrid_I5.0_F1.9_L100_131132_0_0_0_rgb.csv' \\\n",
    "# --rgb-img '/home/labgroup/Downloads/input/chromagrid_I5.0_F1.9_L100_131132_0_0_0_rgb.jpg' \\\n",
    "# --mode 1 \\\n",
    "# --image_thresh_high=120 \\\n",
    "# --image_thresh_low=50 \\\n",
    "# --distance=0.6 \\\n",
    "# --gaussian_sigma 0.57\n",
    "# conda deactivate"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### Generate `csv` of hyperspectral channel for every image in dataset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [],
   "source": [
    "hdr_files=$data/*.hdr"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(test-environment) Reading hyperspectral image /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE1_I5.0_F1.9_L100_143813_0_0_0_Fluorescence.hdr\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "Extracting channel at index 200\n",
      "Writing channel matrix to /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE1_I5.0_F1.9_L100_143813_0_0_0_hyperchannel.csv\n",
      "Making image for inspection and writing to jpg_file\n",
      "Reading hyperspectral image /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE1_I5.0_F1.9_L100_143859_1_0_1_Fluorescence.hdr\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "Extracting channel at index 200\n",
      "Writing channel matrix to /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE1_I5.0_F1.9_L100_143859_1_0_1_hyperchannel.csv\n",
      "Making image for inspection and writing to jpg_file\n",
      "Reading hyperspectral image /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE1_I5.0_F1.9_L100_143959_2_0_2_Fluorescence.hdr\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "Extracting channel at index 200\n",
      "Writing channel matrix to /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE1_I5.0_F1.9_L100_143959_2_0_2_hyperchannel.csv\n",
      "Making image for inspection and writing to jpg_file\n",
      "Reading hyperspectral image /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE1_I5.0_F1.9_L100_144101_3_0_3_Fluorescence.hdr\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "Extracting channel at index 200\n",
      "Writing channel matrix to /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE1_I5.0_F1.9_L100_144101_3_0_3_hyperchannel.csv\n",
      "Making image for inspection and writing to jpg_file\n",
      "Reading hyperspectral image /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE1_I5.0_F1.9_L100_144200_4_0_4_Fluorescence.hdr\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "Extracting channel at index 200\n",
      "Writing channel matrix to /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE1_I5.0_F1.9_L100_144200_4_0_4_hyperchannel.csv\n",
      "Making image for inspection and writing to jpg_file\n",
      "Reading hyperspectral image /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE1_I5.0_F1.9_L100_144259_5_0_5_Fluorescence.hdr\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "Extracting channel at index 200\n",
      "Writing channel matrix to /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE1_I5.0_F1.9_L100_144259_5_0_5_hyperchannel.csv\n",
      "Making image for inspection and writing to jpg_file\n",
      "Reading hyperspectral image /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE1_I5.0_F1.9_L100_144359_6_0_6_Fluorescence.hdr\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "Extracting channel at index 200\n",
      "Writing channel matrix to /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE1_I5.0_F1.9_L100_144359_6_0_6_hyperchannel.csv\n",
      "Making image for inspection and writing to jpg_file\n",
      "Reading hyperspectral image /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE1_I5.0_F1.9_L100_144459_7_1_6_Fluorescence.hdr\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "Extracting channel at index 200\n",
      "Writing channel matrix to /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE1_I5.0_F1.9_L100_144459_7_1_6_hyperchannel.csv\n",
      "Making image for inspection and writing to jpg_file\n",
      "Reading hyperspectral image /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE1_I5.0_F1.9_L100_144550_8_1_5_Fluorescence.hdr\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "Extracting channel at index 200\n",
      "Writing channel matrix to /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE1_I5.0_F1.9_L100_144550_8_1_5_hyperchannel.csv\n",
      "Making image for inspection and writing to jpg_file\n",
      "Reading hyperspectral image /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE1_I5.0_F1.9_L100_144650_9_1_4_Fluorescence.hdr\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "Extracting channel at index 200\n",
      "Writing channel matrix to /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE1_I5.0_F1.9_L100_144650_9_1_4_hyperchannel.csv\n",
      "Making image for inspection and writing to jpg_file\n",
      "Reading hyperspectral image /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE1_I5.0_F1.9_L100_144749_10_1_3_Fluorescence.hdr\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "Extracting channel at index 200\n",
      "Writing channel matrix to /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE1_I5.0_F1.9_L100_144749_10_1_3_hyperchannel.csv\n",
      "Making image for inspection and writing to jpg_file\n",
      "Reading hyperspectral image /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE1_I5.0_F1.9_L100_144848_11_1_2_Fluorescence.hdr\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "Extracting channel at index 200\n",
      "Writing channel matrix to /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE1_I5.0_F1.9_L100_144848_11_1_2_hyperchannel.csv\n",
      "Making image for inspection and writing to jpg_file\n",
      "Reading hyperspectral image /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE1_I5.0_F1.9_L100_144948_12_1_1_Fluorescence.hdr\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "Extracting channel at index 200\n",
      "Writing channel matrix to /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE1_I5.0_F1.9_L100_144948_12_1_1_hyperchannel.csv\n",
      "Making image for inspection and writing to jpg_file\n",
      "Reading hyperspectral image /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE1_I5.0_F1.9_L100_145051_13_1_0_Fluorescence.hdr\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "Extracting channel at index 200\n",
      "Writing channel matrix to /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE1_I5.0_F1.9_L100_145051_13_1_0_hyperchannel.csv\n",
      "Making image for inspection and writing to jpg_file\n",
      "Reading hyperspectral image /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE1_I5.0_F1.9_L100_145151_14_2_0_Fluorescence.hdr\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "Extracting channel at index 200\n",
      "Writing channel matrix to /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE1_I5.0_F1.9_L100_145151_14_2_0_hyperchannel.csv\n",
      "Making image for inspection and writing to jpg_file\n",
      "Reading hyperspectral image /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE1_I5.0_F1.9_L100_145243_15_2_1_Fluorescence.hdr\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "Extracting channel at index 200\n",
      "Writing channel matrix to /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE1_I5.0_F1.9_L100_145243_15_2_1_hyperchannel.csv\n",
      "Making image for inspection and writing to jpg_file\n",
      "Reading hyperspectral image /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE1_I5.0_F1.9_L100_145343_16_2_2_Fluorescence.hdr\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "Extracting channel at index 200\n",
      "Writing channel matrix to /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE1_I5.0_F1.9_L100_145343_16_2_2_hyperchannel.csv\n",
      "Making image for inspection and writing to jpg_file\n",
      "Reading hyperspectral image /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE1_I5.0_F1.9_L100_145442_17_2_3_Fluorescence.hdr\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "Extracting channel at index 200\n",
      "Writing channel matrix to /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE1_I5.0_F1.9_L100_145442_17_2_3_hyperchannel.csv\n",
      "Making image for inspection and writing to jpg_file\n",
      "Reading hyperspectral image /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE1_I5.0_F1.9_L100_145544_18_2_4_Fluorescence.hdr\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "Extracting channel at index 200\n",
      "Writing channel matrix to /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE1_I5.0_F1.9_L100_145544_18_2_4_hyperchannel.csv\n",
      "Making image for inspection and writing to jpg_file\n",
      "Reading hyperspectral image /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE1_I5.0_F1.9_L100_145643_19_2_5_Fluorescence.hdr\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "Extracting channel at index 200\n",
      "Writing channel matrix to /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE1_I5.0_F1.9_L100_145643_19_2_5_hyperchannel.csv\n",
      "Making image for inspection and writing to jpg_file\n",
      "Reading hyperspectral image /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE1_I5.0_F1.9_L100_145744_20_2_6_Fluorescence.hdr\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "Extracting channel at index 200\n",
      "Writing channel matrix to /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE1_I5.0_F1.9_L100_145744_20_2_6_hyperchannel.csv\n",
      "Making image for inspection and writing to jpg_file\n",
      "Reading hyperspectral image /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE1_I5.0_F1.9_L100_152151_0_1_6_Fluorescence.hdr\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "Extracting channel at index 200\n",
      "Writing channel matrix to /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE1_I5.0_F1.9_L100_152151_0_1_6_hyperchannel.csv\n",
      "Making image for inspection and writing to jpg_file\n",
      "Reading hyperspectral image /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE2_I5.0_F1.9_L100_150028_0_0_0_Fluorescence.hdr\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "Extracting channel at index 200\n",
      "Writing channel matrix to /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE2_I5.0_F1.9_L100_150028_0_0_0_hyperchannel.csv\n",
      "Making image for inspection and writing to jpg_file\n",
      "Reading hyperspectral image /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE2_I5.0_F1.9_L100_150113_1_0_1_Fluorescence.hdr\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "Extracting channel at index 200\n",
      "Writing channel matrix to /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE2_I5.0_F1.9_L100_150113_1_0_1_hyperchannel.csv\n",
      "Making image for inspection and writing to jpg_file\n",
      "Reading hyperspectral image /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE2_I5.0_F1.9_L100_150214_2_0_2_Fluorescence.hdr\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "Extracting channel at index 200\n",
      "Writing channel matrix to /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE2_I5.0_F1.9_L100_150214_2_0_2_hyperchannel.csv\n",
      "Making image for inspection and writing to jpg_file\n",
      "Reading hyperspectral image /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE2_I5.0_F1.9_L100_150313_3_0_3_Fluorescence.hdr\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "Extracting channel at index 200\n",
      "Writing channel matrix to /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE2_I5.0_F1.9_L100_150313_3_0_3_hyperchannel.csv\n",
      "Making image for inspection and writing to jpg_file\n",
      "Reading hyperspectral image /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE2_I5.0_F1.9_L100_150413_4_0_4_Fluorescence.hdr\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "Extracting channel at index 200\n",
      "Writing channel matrix to /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE2_I5.0_F1.9_L100_150413_4_0_4_hyperchannel.csv\n",
      "Making image for inspection and writing to jpg_file\n",
      "Reading hyperspectral image /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE2_I5.0_F1.9_L100_150513_5_0_5_Fluorescence.hdr\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "Extracting channel at index 200\n",
      "Writing channel matrix to /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE2_I5.0_F1.9_L100_150513_5_0_5_hyperchannel.csv\n",
      "Making image for inspection and writing to jpg_file\n",
      "Reading hyperspectral image /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE2_I5.0_F1.9_L100_150613_6_0_6_Fluorescence.hdr\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "Extracting channel at index 200\n",
      "Writing channel matrix to /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE2_I5.0_F1.9_L100_150613_6_0_6_hyperchannel.csv\n",
      "Making image for inspection and writing to jpg_file\n",
      "Reading hyperspectral image /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE2_I5.0_F1.9_L100_150713_7_1_6_Fluorescence.hdr\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "Extracting channel at index 200\n",
      "Writing channel matrix to /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE2_I5.0_F1.9_L100_150713_7_1_6_hyperchannel.csv\n",
      "Making image for inspection and writing to jpg_file\n",
      "Reading hyperspectral image /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE2_I5.0_F1.9_L100_150803_8_1_5_Fluorescence.hdr\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "Extracting channel at index 200\n",
      "Writing channel matrix to /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE2_I5.0_F1.9_L100_150803_8_1_5_hyperchannel.csv\n",
      "Making image for inspection and writing to jpg_file\n",
      "Reading hyperspectral image /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE2_I5.0_F1.9_L100_150902_9_1_4_Fluorescence.hdr\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "Extracting channel at index 200\n",
      "Writing channel matrix to /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE2_I5.0_F1.9_L100_150902_9_1_4_hyperchannel.csv\n",
      "Making image for inspection and writing to jpg_file\n",
      "Reading hyperspectral image /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE2_I5.0_F1.9_L100_151002_10_1_3_Fluorescence.hdr\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "Extracting channel at index 200\n",
      "Writing channel matrix to /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE2_I5.0_F1.9_L100_151002_10_1_3_hyperchannel.csv\n",
      "Making image for inspection and writing to jpg_file\n",
      "Reading hyperspectral image /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE2_I5.0_F1.9_L100_151102_11_1_1_Fluorescence.hdr\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "Extracting channel at index 200\n",
      "Writing channel matrix to /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE2_I5.0_F1.9_L100_151102_11_1_1_hyperchannel.csv\n",
      "Making image for inspection and writing to jpg_file\n",
      "Reading hyperspectral image /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE2_I5.0_F1.9_L100_151202_12_1_0_Fluorescence.hdr\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "Extracting channel at index 200\n",
      "Writing channel matrix to /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE2_I5.0_F1.9_L100_151202_12_1_0_hyperchannel.csv\n",
      "Making image for inspection and writing to jpg_file\n",
      "Reading hyperspectral image /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE2_I5.0_F1.9_L100_151302_13_2_0_Fluorescence.hdr\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "Extracting channel at index 200\n",
      "Writing channel matrix to /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE2_I5.0_F1.9_L100_151302_13_2_0_hyperchannel.csv\n",
      "Making image for inspection and writing to jpg_file\n",
      "Reading hyperspectral image /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE2_I5.0_F1.9_L100_151354_14_2_1_Fluorescence.hdr\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "Extracting channel at index 200\n",
      "Writing channel matrix to /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE2_I5.0_F1.9_L100_151354_14_2_1_hyperchannel.csv\n",
      "Making image for inspection and writing to jpg_file\n",
      "Reading hyperspectral image /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE2_I5.0_F1.9_L100_151456_15_2_2_Fluorescence.hdr\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "Extracting channel at index 200\n",
      "Writing channel matrix to /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE2_I5.0_F1.9_L100_151456_15_2_2_hyperchannel.csv\n",
      "Making image for inspection and writing to jpg_file\n",
      "Reading hyperspectral image /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE2_I5.0_F1.9_L100_151556_16_2_3_Fluorescence.hdr\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "Extracting channel at index 200\n",
      "Writing channel matrix to /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE2_I5.0_F1.9_L100_151556_16_2_3_hyperchannel.csv\n",
      "Making image for inspection and writing to jpg_file\n",
      "Reading hyperspectral image /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE2_I5.0_F1.9_L100_151657_17_2_4_Fluorescence.hdr\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "Extracting channel at index 200\n",
      "Writing channel matrix to /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE2_I5.0_F1.9_L100_151657_17_2_4_hyperchannel.csv\n",
      "Making image for inspection and writing to jpg_file\n",
      "Reading hyperspectral image /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE2_I5.0_F1.9_L100_151757_18_2_6_Fluorescence.hdr\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "Extracting channel at index 200\n",
      "Writing channel matrix to /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE2_I5.0_F1.9_L100_151757_18_2_6_hyperchannel.csv\n",
      "Making image for inspection and writing to jpg_file\n",
      "Reading hyperspectral image /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE3_I5.0_F1.9_L100_143049_0_0_0_Fluorescence.hdr\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "Extracting channel at index 200\n",
      "Writing channel matrix to /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE3_I5.0_F1.9_L100_143049_0_0_0_hyperchannel.csv\n",
      "Making image for inspection and writing to jpg_file\n",
      "Reading hyperspectral image /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE3_I5.0_F1.9_L100_143135_1_0_1_Fluorescence.hdr\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "Extracting channel at index 200\n",
      "Writing channel matrix to /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE3_I5.0_F1.9_L100_143135_1_0_1_hyperchannel.csv\n",
      "Making image for inspection and writing to jpg_file\n",
      "Reading hyperspectral image /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE3_I5.0_F1.9_L100_143235_2_0_2_Fluorescence.hdr\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "Extracting channel at index 200\n",
      "Writing channel matrix to /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE3_I5.0_F1.9_L100_143235_2_0_2_hyperchannel.csv\n",
      "Making image for inspection and writing to jpg_file\n",
      "Reading hyperspectral image /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE3_I5.0_F1.9_L100_143334_3_0_3_Fluorescence.hdr\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "Extracting channel at index 200\n",
      "Writing channel matrix to /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE3_I5.0_F1.9_L100_143334_3_0_3_hyperchannel.csv\n",
      "Making image for inspection and writing to jpg_file\n",
      "Reading hyperspectral image /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE3_I5.0_F1.9_L100_143434_4_0_4_Fluorescence.hdr\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "Extracting channel at index 200\n",
      "Writing channel matrix to /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//FAE3_I5.0_F1.9_L100_143434_4_0_4_hyperchannel.csv\n",
      "Making image for inspection and writing to jpg_file\n",
      "(test-environment) "
     ]
    }
   ],
   "source": [
    "cd $alignment_path\n",
    "conda activate test-environment\n",
    "for hdr_file in $hdr_files\n",
    "do\n",
    " csv_file=$(sed 's/Fluorescence.hdr/hyperchannel.csv/g' <<< $hdr_file)\n",
    " csv_file=$(sed 's/.hdr/hyperchannel.csv/g' <<< $csv_file)\n",
    " csv_jpg=$(sed 's/csv/jpg/g' <<< csv_file)\n",
    " python hypercube_to_csv.py \\\n",
    " $hdr_file \\\n",
    " 500 \\\n",
    " 200 \\\n",
    " $csv_file \\\n",
    " $csv_jpg\n",
    "done\n",
    "conda deactivate"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### Run alignment to apply homography matrix to all images"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(alignment) (alignment) (alignment) (alignment) (base) "
     ]
    }
   ],
   "source": [
    "conda activate alignment\n",
    "cd $alignment_path\n",
    "file_list_input=\"${data}/rgb_and_hyper_channel_lists.csv\"\n",
    "python main.py \\\n",
    "--hyper-img \"/run/media/labgroup/Elements_9//Transformation/FAD_Dev_Genes/wk7/Fluorescent/hypercube_to_csv.csv\" \\\n",
    "--img-csv $file_list_input \\\n",
    "--mode 2 \\\n",
    "--h_matrix_path \"output/chromagrid_I5.0_F1.9_L100LED30_125827_0_0_0_rgb_homography.npy\" >> $data/log_alignment_mode2.txt\n",
    "conda deactivate\n",
    "conda deactivate"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "If using this method, we should have RGB and hyperspectral images of a grid. Here, we will assume they have \"grid\" in their name and can be found in the same folder as the sample images."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We will now extract a channel from the grid hypercube."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Note, for recent \"chromagrid\" images, alignment requires the parameter `--gaussian_sigma 2.5` be added."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### 3.1.2. Generate homography matrix for alignment"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below block generates homography matrix from a pair of an archived RGB / hyperspectral images channel layers we know to align well with these parameters."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### Apply homography matrix to transform a batch of images"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "# conda activate alignment\n",
    "# cd $alignment_path\n",
    "# file_list_input=\"${data}/file_list.csv\"\n",
    "# python main.py \\\n",
    "# --img-csv $file_list_input \\\n",
    "# --mode 2 \\\n",
    "# --h_matrix_path \"output/chromagrid_I5.0_F1.9_L100_131132_0_0_0_rgb\" >> $data/log_alignment.txt\n",
    "# conda deactivate\n",
    "# conda deactivate"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below commented-out part is for if we are re-doing alignment every time instead of taking a standard pair of images that align well."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "aligned=$(echo $grid_jpg | sed 's/rgb/rgb_aligned/g')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "# conda activate alignment\n",
    "# python main.py \\\n",
    "# --hyper-img $data\"hypercube_to_csv.csv\" \\\n",
    "# --rgb-img $data$grid_jpg \\\n",
    "# --mode 1 \\\n",
    "# --image_thresh_high=120 \\\n",
    "# --image_thresh_low=50 \\\n",
    "# --distance=0.6 \\\n",
    "# --gaussian_sigma 0.649\n",
    "# conda deactivate"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "# display < $alignment_path/output/$aligned"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "# conda activate alignment\n",
    "# cd $alignment_path\n",
    "# file_list_input=\"${data}/file_list.csv\"\n",
    "# python main.py \\\n",
    "# --hyper-img $data\"hypercube_to_csv.csv\" \\\n",
    "# --rgb-img $data$grid_jpg \\\n",
    "# --img-csv $file_list_input \\\n",
    "# --mode 2 \\\n",
    "# --image_thresh_high=120 \\\n",
    "# --image_thresh_low=50 \\\n",
    "# --distance=0.6 \\\n",
    "# --gaussian_sigma 0.57 >> $data/log_align.txt\n",
    "# conda deactivate\n",
    "# conda deactivate"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 4. Cross-analyze deep segmentation and regression results"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Scripts in the <a href=\"https://github.com/naglemi/GMOlabeler\" target=\"_blank\">GMOlabeler repository</a> are used to cross-reference results from deep segmentation of RGB images and regression of hyperspectral imaging, apply thresholding parameters to classify tissues as transgenic or escapes, and produce plots."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<img src=\"Figures/GMOlabeler.png\">"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Figure: The various steps of data processing in GMOlabeler are illustrated for an example explant from an experiment on CIM optimization for cottonwood. Images of plates are cropped to a sub-image for each explant. RGB segmentation results and hyperspectral regression results are cross-referenced to calculate fluorecent proteins in specific tissues and infer whether these tissues are transgenic."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### 4.1. Prepare sample datasheet input"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Prepare input file we will use for making plots. This file contains paths to CLS results, RGB images, and hyperspectral images."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence/\n"
     ]
    }
   ],
   "source": [
    "echo $data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence/\n"
     ]
    }
   ],
   "source": [
    "echo $data$aligned"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(gmolabeler) (gmolabeler) [1] \"”2021-01-21”\"\n",
      "[1] \"Looking for CLS data in: /home/labgroup/data/gmodetector_out/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//\"\n",
      "[1] \"2021-01-21\"\n",
      "[1] \"Looking for CLS files in directory /home/labgroup/data/gmodetector_out/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//\"\n",
      "[1] \"How many CLS files? 46\"\n",
      "[1] \"We will only include in analysis the FINAL of multiple images for FAE1_1_6\"\n",
      "[1] \"Writing 45 rows to /run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///samples_pre_labeling.csv\"\n",
      "(gmolabeler) "
     ]
    }
   ],
   "source": [
    "conda activate gmolabeler\n",
    "cd \"${cwd}/intermediates/\"\n",
    "Rscript pre_label.R \\\n",
    "-r \"${data}/\" \\\n",
    "-R \"${output_directory_prefix}\" \\\n",
    "-i 1 \\\n",
    "-d $datestamp\n",
    "conda deactivate"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### 4.2. Cross-reference RGB and hyperspectral data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "On 1/11/20, we altered `gmolabeler` to accept user-defined borders of the grid box, rather than hardcoding these borders in for specific grids with 12 or 20 explants. Thus, if providing `grid_borders`, `grid` can be left as `None`, or vice versa. To determine the borders, open the *aligned* grid file in a GUI like GIMP or Microsoft Paint and use the crosshairs to find the pixel positions of each border. These are then provided in order \"left right bottom top\"."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# NOTE THE CHANGE HERE for FAE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [],
   "source": [
    "data_old=/run/media/labgroup/Elements_9//Transformation/FAD_Dev_Genes/wk7/Fluorescent/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [],
   "source": [
    "aligned_grid=$(ls $data_old | grep -i \"grid\" | grep -i \"rgb_processed.jpg\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [],
   "source": [
    "#aligned_grid=$(ls $data | grep -i \"grid\" | grep -i \"rgb_processed.jpg\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chromagrid_I5.0_F1.9_L100LED30_125827_0_0_0_rgb_processed.jpg\n"
     ]
    }
   ],
   "source": [
    "echo $aligned_grid"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Some notes on grid copping parameters:\n",
    "- Submit a space-delimited string with parameters for borders of grid as \"left right bottom top\"\n",
    "- For FAA wk3, use parameters \"139 1580 1250 312\"\n",
    "- For FAA wk7 and FAD wk7, use \"156 1600 1280 335\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(gmolabeler) (gmolabeler) Grid type 12\n",
      "Loading plate 0 of 45\n",
      "/run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE1_I5.0_F1.9_L100_152151_0_1_6_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE1_I5.0_F1.9_L100_152151_0_1_6_Fluorescence_weights.hdf\n",
      "Loading plate 1 of 45\n",
      "/run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE2_I5.0_F1.9_L100_151757_18_2_6_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE2_I5.0_F1.9_L100_151757_18_2_6_Fluorescence_weights.hdf\n",
      "Loading plate 2 of 45\n",
      "/run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE2_I5.0_F1.9_L100_151657_17_2_4_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE2_I5.0_F1.9_L100_151657_17_2_4_Fluorescence_weights.hdf\n",
      "Loading plate 3 of 45\n",
      "/run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE2_I5.0_F1.9_L100_151556_16_2_3_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE2_I5.0_F1.9_L100_151556_16_2_3_Fluorescence_weights.hdf\n",
      "Loading plate 4 of 45\n",
      "/run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE2_I5.0_F1.9_L100_151456_15_2_2_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE2_I5.0_F1.9_L100_151456_15_2_2_Fluorescence_weights.hdf\n",
      "Loading plate 5 of 45\n",
      "/run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE2_I5.0_F1.9_L100_151354_14_2_1_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE2_I5.0_F1.9_L100_151354_14_2_1_Fluorescence_weights.hdf\n",
      "Loading plate 6 of 45\n",
      "/run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE2_I5.0_F1.9_L100_151302_13_2_0_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE2_I5.0_F1.9_L100_151302_13_2_0_Fluorescence_weights.hdf\n",
      "Loading plate 7 of 45\n",
      "/run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE2_I5.0_F1.9_L100_151202_12_1_0_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE2_I5.0_F1.9_L100_151202_12_1_0_Fluorescence_weights.hdf\n",
      "Loading plate 8 of 45\n",
      "/run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE2_I5.0_F1.9_L100_151102_11_1_1_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE2_I5.0_F1.9_L100_151102_11_1_1_Fluorescence_weights.hdf\n",
      "Loading plate 9 of 45\n",
      "/run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE2_I5.0_F1.9_L100_151002_10_1_3_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE2_I5.0_F1.9_L100_151002_10_1_3_Fluorescence_weights.hdf\n",
      "Loading plate 10 of 45\n",
      "/run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE2_I5.0_F1.9_L100_150902_9_1_4_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE2_I5.0_F1.9_L100_150902_9_1_4_Fluorescence_weights.hdf\n",
      "Loading plate 11 of 45\n",
      "/run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE2_I5.0_F1.9_L100_150803_8_1_5_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE2_I5.0_F1.9_L100_150803_8_1_5_Fluorescence_weights.hdf\n",
      "Loading plate 12 of 45\n",
      "/run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE2_I5.0_F1.9_L100_150713_7_1_6_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE2_I5.0_F1.9_L100_150713_7_1_6_Fluorescence_weights.hdf\n",
      "Loading plate 13 of 45\n",
      "/run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE2_I5.0_F1.9_L100_150613_6_0_6_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE2_I5.0_F1.9_L100_150613_6_0_6_Fluorescence_weights.hdf\n",
      "Loading plate 14 of 45\n",
      "/run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE2_I5.0_F1.9_L100_150513_5_0_5_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE2_I5.0_F1.9_L100_150513_5_0_5_Fluorescence_weights.hdf\n",
      "Loading plate 15 of 45\n",
      "/run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE2_I5.0_F1.9_L100_150413_4_0_4_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE2_I5.0_F1.9_L100_150413_4_0_4_Fluorescence_weights.hdf\n",
      "Loading plate 16 of 45\n",
      "/run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE2_I5.0_F1.9_L100_150313_3_0_3_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE2_I5.0_F1.9_L100_150313_3_0_3_Fluorescence_weights.hdf\n",
      "Loading plate 17 of 45\n",
      "/run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE2_I5.0_F1.9_L100_150214_2_0_2_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE2_I5.0_F1.9_L100_150214_2_0_2_Fluorescence_weights.hdf\n",
      "Loading plate 18 of 45\n",
      "/run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE2_I5.0_F1.9_L100_150113_1_0_1_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE2_I5.0_F1.9_L100_150113_1_0_1_Fluorescence_weights.hdf\n",
      "Loading plate 19 of 45\n",
      "/run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE2_I5.0_F1.9_L100_150028_0_0_0_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE2_I5.0_F1.9_L100_150028_0_0_0_Fluorescence_weights.hdf\n",
      "Loading plate 20 of 45\n",
      "/run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE1_I5.0_F1.9_L100_145744_20_2_6_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE1_I5.0_F1.9_L100_145744_20_2_6_Fluorescence_weights.hdf\n",
      "Loading plate 21 of 45\n",
      "/run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE1_I5.0_F1.9_L100_145643_19_2_5_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE1_I5.0_F1.9_L100_145643_19_2_5_Fluorescence_weights.hdf\n",
      "Loading plate 22 of 45\n",
      "/run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE1_I5.0_F1.9_L100_145544_18_2_4_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE1_I5.0_F1.9_L100_145544_18_2_4_Fluorescence_weights.hdf\n",
      "Loading plate 23 of 45\n",
      "/run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE1_I5.0_F1.9_L100_145442_17_2_3_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE1_I5.0_F1.9_L100_145442_17_2_3_Fluorescence_weights.hdf\n",
      "Loading plate 24 of 45\n",
      "/run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE1_I5.0_F1.9_L100_145343_16_2_2_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE1_I5.0_F1.9_L100_145343_16_2_2_Fluorescence_weights.hdf\n",
      "Loading plate 25 of 45\n",
      "/run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE1_I5.0_F1.9_L100_145243_15_2_1_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE1_I5.0_F1.9_L100_145243_15_2_1_Fluorescence_weights.hdf\n",
      "Loading plate 26 of 45\n",
      "/run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE1_I5.0_F1.9_L100_145151_14_2_0_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE1_I5.0_F1.9_L100_145151_14_2_0_Fluorescence_weights.hdf\n",
      "Loading plate 27 of 45\n",
      "/run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE1_I5.0_F1.9_L100_145051_13_1_0_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE1_I5.0_F1.9_L100_145051_13_1_0_Fluorescence_weights.hdf\n",
      "Loading plate 28 of 45\n",
      "/run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE1_I5.0_F1.9_L100_144948_12_1_1_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE1_I5.0_F1.9_L100_144948_12_1_1_Fluorescence_weights.hdf\n",
      "Loading plate 29 of 45\n",
      "/run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE1_I5.0_F1.9_L100_144848_11_1_2_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE1_I5.0_F1.9_L100_144848_11_1_2_Fluorescence_weights.hdf\n",
      "Loading plate 30 of 45\n",
      "/run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE1_I5.0_F1.9_L100_144749_10_1_3_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE1_I5.0_F1.9_L100_144749_10_1_3_Fluorescence_weights.hdf\n",
      "Loading plate 31 of 45\n",
      "/run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE1_I5.0_F1.9_L100_144650_9_1_4_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE1_I5.0_F1.9_L100_144650_9_1_4_Fluorescence_weights.hdf\n",
      "Loading plate 32 of 45\n",
      "/run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE1_I5.0_F1.9_L100_144550_8_1_5_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE1_I5.0_F1.9_L100_144550_8_1_5_Fluorescence_weights.hdf\n",
      "Loading plate 33 of 45\n",
      "/run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE1_I5.0_F1.9_L100_144359_6_0_6_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE1_I5.0_F1.9_L100_144359_6_0_6_Fluorescence_weights.hdf\n",
      "Loading plate 34 of 45\n",
      "/run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE1_I5.0_F1.9_L100_144259_5_0_5_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE1_I5.0_F1.9_L100_144259_5_0_5_Fluorescence_weights.hdf\n",
      "Loading plate 35 of 45\n",
      "/run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE1_I5.0_F1.9_L100_144200_4_0_4_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE1_I5.0_F1.9_L100_144200_4_0_4_Fluorescence_weights.hdf\n",
      "Loading plate 36 of 45\n",
      "/run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE1_I5.0_F1.9_L100_144101_3_0_3_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE1_I5.0_F1.9_L100_144101_3_0_3_Fluorescence_weights.hdf\n",
      "Loading plate 37 of 45\n",
      "/run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE1_I5.0_F1.9_L100_143959_2_0_2_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE1_I5.0_F1.9_L100_143959_2_0_2_Fluorescence_weights.hdf\n",
      "Loading plate 38 of 45\n",
      "/run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE1_I5.0_F1.9_L100_143859_1_0_1_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE1_I5.0_F1.9_L100_143859_1_0_1_Fluorescence_weights.hdf\n",
      "Loading plate 39 of 45\n",
      "/run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE1_I5.0_F1.9_L100_143813_0_0_0_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE1_I5.0_F1.9_L100_143813_0_0_0_Fluorescence_weights.hdf\n",
      "Loading plate 40 of 45\n",
      "/run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE3_I5.0_F1.9_L100_143434_4_0_4_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE3_I5.0_F1.9_L100_143434_4_0_4_Fluorescence_weights.hdf\n",
      "Loading plate 41 of 45\n",
      "/run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE3_I5.0_F1.9_L100_143334_3_0_3_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE3_I5.0_F1.9_L100_143334_3_0_3_Fluorescence_weights.hdf\n",
      "Loading plate 42 of 45\n",
      "/run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE3_I5.0_F1.9_L100_143235_2_0_2_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE3_I5.0_F1.9_L100_143235_2_0_2_Fluorescence_weights.hdf\n",
      "Loading plate 43 of 45\n",
      "/run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE3_I5.0_F1.9_L100_143135_1_0_1_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE3_I5.0_F1.9_L100_143135_1_0_1_Fluorescence_weights.hdf\n",
      "Loading plate 44 of 45\n",
      "/run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE3_I5.0_F1.9_L100_143049_0_0_0_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE3_I5.0_F1.9_L100_143049_0_0_0_Fluorescence_weights.hdf\n",
      "(gmolabeler) "
     ]
    }
   ],
   "source": [
    "conda activate gmolabeler\n",
    "cd $gmolabeler_path\n",
    "python main.py \\\n",
    "\"${data}/samples_pre_labeling.csv\" \\\n",
    "$data_old$aligned_grid \\\n",
    "$reporter_threshold \\\n",
    "$reporter \\\n",
    "$grid \\\n",
    "\"hdf\" \\\n",
    "$gmolabeler_path \\\n",
    "\"156 1600 1280 335\"\n",
    "conda deactivate"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### 4.3. Calculate sums of statistics over combined segments"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We are interested in all regenerated tissue (callus + shoot) as well as all tissue (including stem as well). We will calculate aggregate statistics over these groups."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(gmolabeler) [1] \"Writing output with sums statistics calculated over combined tissue segments to: /home/labgroup/code/GMOlabeler//output/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//stats_with_sums_over_tissues.csv\"\n",
      "(gmolabeler) "
     ]
    }
   ],
   "source": [
    "conda activate gmolabeler\n",
    "Rscript calculate_sum_stats_over_combined_segments.R \\\n",
    "--output_dir \"${gmolabeler_path}/output/\" \\\n",
    "--datapath \"${data_folder}/\"\n",
    "conda deactivate"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### 4.4. Make plots of results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(gmolabeler) (gmolabeler) Warning message:\n",
      "In storage.mode(default) <- type : NAs introduced by coercion\n",
      "[1] \"Saving list of input arguments to : /home/labgroup/code/GMOlabeler/plots/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//gridplot_args.rds\"\n",
      "[1] \"Reading in output from GMOlabeler at path: /home/labgroup/code/GMOlabeler/output/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//stats_with_sums_over_tissues.csv\"\n",
      "\n",
      "[1] \"Rows in output from GMOlabeler: 2491\"\n",
      "\n",
      "[1] \"Max n_pixels_passing_threshold in output from GMOlabeler: 59483\"\n",
      "\n",
      "[1] \"Max total_signal in output from GMOlabeler: 22149221.7910449\"\n",
      "\n",
      "[1] \"Look at the top of output from GMOlabeler\"\n",
      "[1] \"\"                                                                                                                    \n",
      "[2] \"/run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE1_I5.0_F1.9_L100_152151_0_1_6.png\"\n",
      "[3] \"/run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE1_I5.0_F1.9_L100_152151_0_1_6.png\"\n",
      "[4] \"/run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE1_I5.0_F1.9_L100_152151_0_1_6.png\"\n",
      "[5] \"/run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE1_I5.0_F1.9_L100_152151_0_1_6.png\"\n",
      "[6] \"/run/media/labgroup/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence///FAE1_I5.0_F1.9_L100_152151_0_1_6.png\"\n",
      "\n",
      "[1] \"Look at the top of missing explant data\"\n",
      "# A tibble: 6 x 3\n",
      "  filename                                 grid_item present\n",
      "  <chr>                                    <chr>     <chr>  \n",
      "1 FAE1_I5.0_F1.9_L100_143813_0_0_0_rgb.jpg 1         NC     \n",
      "2 FAE1_I5.0_F1.9_L100_143813_0_0_0_rgb.jpg 2         NC     \n",
      "3 FAE1_I5.0_F1.9_L100_143813_0_0_0_rgb.jpg 3         NC     \n",
      "4 FAE1_I5.0_F1.9_L100_143813_0_0_0_rgb.jpg 4         NC     \n",
      "5 FAE1_I5.0_F1.9_L100_143813_0_0_0_rgb.jpg 5         NC     \n",
      "6 FAE1_I5.0_F1.9_L100_143813_0_0_0_rgb.jpg 6         NC     \n",
      "\n",
      "[1] \"Look at the head of filenames for missing explant data\"\n",
      "[1] \"FAE1_I5.0_F1.9_L100_143813_0_0_0_rgb.jpg\"\n",
      "[2] \"FAE1_I5.0_F1.9_L100_143813_0_0_0_rgb.jpg\"\n",
      "[3] \"FAE1_I5.0_F1.9_L100_143813_0_0_0_rgb.jpg\"\n",
      "[4] \"FAE1_I5.0_F1.9_L100_143813_0_0_0_rgb.jpg\"\n",
      "[5] \"FAE1_I5.0_F1.9_L100_143813_0_0_0_rgb.jpg\"\n",
      "[6] \"FAE1_I5.0_F1.9_L100_143813_0_0_0_rgb.jpg\"\n",
      "\n",
      "[1] \"Look at the colnames of output from GMOlabeler\"\n",
      " [1] \"V1\"                         \"filename\"                  \n",
      " [3] \"grid_item\"                  \"segment_hex\"               \n",
      " [5] \"intensity_threshold\"        \"n_pixels_passing_threshold\"\n",
      " [7] \"mean_signal\"                \"max_signal\"                \n",
      " [9] \"total_signal\"               \"total_pixels\"              \n",
      "\n",
      "[1] \"Look at the colnames of missing explant data\"\n",
      "[1] \"filename\"  \"grid_item\" \"present\"  \n",
      "\n",
      "[1] \"Before merging the GMOlabeler and missing output data, they have rows of 2491 and 552 respectively.\"\n",
      "\n",
      "[1] \"After merging but before removing missing/contminated explants, merged output has 2491 rows.\"\n",
      "\n",
      "[1] \"After removing the missing/contaminated explants, merged output has 2491\"\n",
      "\n",
      "[1] \"Afer processing GMOlabeler output to merge with missing explant data, Head of data (w/ first 5 col): \"\n",
      "                               filename grid_item          V1 segment_hex\n",
      "1:                                           <NA> observation            \n",
      "2: FAE1_I5.0_F1.9_L100_143813_0_0_0.png         1 observation      00CC11\n",
      "3: FAE1_I5.0_F1.9_L100_143813_0_0_0.png         1 observation      0006CC\n",
      "4: FAE1_I5.0_F1.9_L100_143813_0_0_0.png         1 observation      CC0000\n",
      "5: FAE1_I5.0_F1.9_L100_143813_0_0_0.png         1 observation  All_tissue\n",
      "   intensity_threshold\n",
      "1:                  NA\n",
      "2:                  38\n",
      "3:                  38\n",
      "4:                  38\n",
      "5:                  NA\n",
      "New names:\n",
      "* `` -> ...10\n",
      "* `` -> ...12\n",
      "[1] \"Upon loading randomization_datasheet, it has how many rows? 47\"\n",
      "\n",
      "[1] \"Calculating total numbers of missing explants per plate\"\n",
      "[1] \"Let's figure out why contaminated/missing plates are being said to have 0 instead of NA\"\n",
      "\n",
      "[1] \"Dimensions of missing explant data with totals\"\n",
      "[1] 46 14\n",
      "[1] \"Minimum number of explants on any plate after removing missing and contaminated explants is...\"\n",
      "[1] 0\n",
      "                                 image_name 1 2 3 4 5 6 7 8 9 10 11 12\n",
      "1: FAE1_I5.0_F1.9_L100_143813_0_0_0_rgb.jpg 1 1 1 1 1 1 1 1 1  1  1  1\n",
      "2: FAE1_I5.0_F1.9_L100_143859_1_0_1_rgb.jpg 1 1 1 1 1 0 1 1 1  1  1  1\n",
      "3: FAE1_I5.0_F1.9_L100_143959_2_0_2_rgb.jpg 1 1 1 1 1 1 1 1 0  1  1  1\n",
      "4: FAE1_I5.0_F1.9_L100_144101_3_0_3_rgb.jpg 1 1 1 1 1 1 1 1 1  1  1  1\n",
      "5: FAE1_I5.0_F1.9_L100_144200_4_0_4_rgb.jpg 1 1 1 1 1 1 1 1 1  1  1  1\n",
      "6: FAE1_I5.0_F1.9_L100_144259_5_0_5_rgb.jpg 1 1 1 1 1 1 1 1 1  1  1  1\n",
      "   total_explants\n",
      "1:             12\n",
      "2:             11\n",
      "3:             11\n",
      "4:             12\n",
      "5:             12\n",
      "6:             12\n",
      "\n",
      "[1] \"About to add IDs to missing explant data based on filenames. Here is head of filenames.\"\n",
      "[1] \"FAE1_I5.0_F1.9_L100_143813_0_0_0\" \"FAE1_I5.0_F1.9_L100_143859_1_0_1\"\n",
      "[3] \"FAE1_I5.0_F1.9_L100_143959_2_0_2\" \"FAE1_I5.0_F1.9_L100_144101_3_0_3\"\n",
      "[5] \"FAE1_I5.0_F1.9_L100_144200_4_0_4\" \"FAE1_I5.0_F1.9_L100_144259_5_0_5\"\n",
      "\n",
      "[1] \"We have added IDs to missing explant data based on filenames. Here is head of IDs.\"\n",
      "[1] \"FAE1_1\" \"FAE1_2\" \"FAE1_3\" \"FAE1_4\" \"FAE1_5\" \"FAE1_6\"\n",
      "\n",
      "[1] \"N rows of randomization datasheet before merging with missing explant data: \"\n",
      "[1] 47\n",
      "[1] \"Head of IDs in missing explant data\"\n",
      "[1] \"FAE1_1\" \"FAE1_2\" \"FAE1_3\" \"FAE1_4\" \"FAE1_5\" \"FAE1_6\"\n",
      "[1] \"Head of IDs in randomization datasheet\"\n",
      "[1] \"FAE1_1\" \"FAE1_2\" \"FAE1_3\" \"FAE1_4\" \"FAE1_5\" \"FAE1_6\"\n",
      "\n",
      "[1] \"Now about to merge randomization_datasheet and missing explant data.\"\n",
      "[1] \"Rows in randomization_datasheet: 47\"\n",
      "[1] \"Rows in missing explant data: 46\"\n",
      "\n",
      "[1] \"Now about to merge randomization datasheet and missing explant data, by ID. Look at head of IDs for both.\"\n",
      "[1] \"FAE1_1\" \"FAE1_2\" \"FAE1_3\" \"FAE1_4\" \"FAE1_5\" \"FAE1_6\"\n",
      "[1] \"FAE1_1\" \"FAE1_2\" \"FAE1_3\" \"FAE1_4\" \"FAE1_5\" \"FAE1_6\"\n",
      "       ID Image # Tray ID Rand # Date of transform Date on Media *Plate #\n",
      "1  FAE1_1       1    FAE1      1        2020-11-17    2020-11-19        7\n",
      "2 FAE1_10      10    FAE1     10        2020-11-17    2020-11-19       14\n",
      "3 FAE1_11      11    FAE1     11        2020-11-17    2020-11-19       13\n",
      "4 FAE1_12      12    FAE1     12        2020-11-17    2020-11-19       27\n",
      "5 FAE1_13      13    FAE1     13        2020-11-17    2020-11-19       12\n",
      "6 FAE1_14      14    FAE1     14        2020-11-17    2020-11-19       44\n",
      "  Genotype_ID Treatment name Replicate ...10       Rand ...12    LabelTop\n",
      "1         717            CTR         7    NA 0.72638474    NA   FAE1_1_R1\n",
      "2         717          WOX5a         7    NA 0.91179032    NA FAE1_10_R10\n",
      "3         717          WOX5a         6    NA 0.60061333    NA FAE1_11_R11\n",
      "4         353            CTR         5    NA 0.70703430    NA FAE1_12_R12\n",
      "5         717          WOX5a         5    NA 0.53159478    NA FAE1_13_R13\n",
      "6         353          WOX11         5    NA 0.05430878    NA FAE1_14_R14\n",
      "     LabelBot                          filename 1 2 3 4 5 6 7 8 9 10 11 12\n",
      "1   717_CTR_7  FAE1_I5.0_F1.9_L100_143813_0_0_0 1 1 1 1 1 1 1 1 1  1  1  1\n",
      "2 717_WOX5a_7 FAE1_I5.0_F1.9_L100_144848_11_1_2 1 1 1 1 1 1 1 1 0  1  1  1\n",
      "3 717_WOX5a_6 FAE1_I5.0_F1.9_L100_144749_10_1_3 1 1 1 1 1 1 1 1 1  1  1  1\n",
      "4   353_CTR_5  FAE1_I5.0_F1.9_L100_144650_9_1_4 1 1 1 1 1 1 1 1 0  1  1  1\n",
      "5 717_WOX5a_5  FAE1_I5.0_F1.9_L100_144550_8_1_5 1 1 1 0 0 1 1 1 1  1  1  1\n",
      "6 353_WOX11_5  FAE1_I5.0_F1.9_L100_144459_7_1_6 0 0 0 0 0 0 0 0 0  0  0  0\n",
      "  total_explants\n",
      "1             12\n",
      "2             11\n",
      "3             12\n",
      "4             11\n",
      "5             10\n",
      "6              0\n",
      " [1] 12 11 12 11 10  0 12 11 12 12 12 12 11 12  6 11 12 12 12 12 12  4 10 NA 12\n",
      "[26] 12 12 12 12 11 10 12 12 12 NA 12 12 12 11 12 11 11 12 10 10 12 12 12\n",
      "[1] \"Still figuring, now in merged rd, out why contaminated/missing plates are being said to have 0 instead of NA\"\n",
      "\n",
      "[1] \"Dimensions of missing explant data with totals\"\n",
      "[1] 46 15\n",
      "[1] \"Levels of total explants on any plate after removing missing and contaminated explants is...\"\n",
      "[1] \"0\"  \"4\"  \"6\"  \"10\" \"11\" \"12\"\n",
      "[1] \"Minimum number of explants on any plate after removing missing and contaminated explants is...\"\n",
      "[1] 0\n",
      "\n",
      "[1] \"Rows in merged output: 48\"\n",
      " [1] 12 11 12 11 10  0 12 11 12 12 12 12 11 12  6 11 12 12 12 12 12  4 10 NA 12\n",
      "[26] 12 12 12 12 11 10 12 12 12 NA 12 12 12 11 12 11 11 12 10 10 12 12 12\n",
      "\n",
      "[1] \"Assuming for all explants for which we have no missing explant data that there are:\"\n",
      "[1] 12\n",
      "\n",
      "[1] \"N rows of randomization datasheet after merging with missing explant data: \"\n",
      "[1] 48\n",
      "[1] \"Column names of randomization datasheet afer merging with missing explant data: \"\n",
      " [1] \"ID\"                \"Image #\"           \"Tray ID\"          \n",
      " [4] \"Rand #\"            \"Date of transform\" \"Date on Media\"    \n",
      " [7] \"*Plate #\"          \"Genotype_ID\"       \"Treatment name\"   \n",
      "[10] \"Replicate\"         \"...10\"             \"Rand\"             \n",
      "[13] \"...12\"             \"LabelTop\"          \"LabelBot\"         \n",
      "[16] \"filename\"          \"1\"                 \"2\"                \n",
      "[19] \"3\"                 \"4\"                 \"5\"                \n",
      "[22] \"6\"                 \"7\"                 \"8\"                \n",
      "[25] \"9\"                 \"10\"                \"11\"               \n",
      "[28] \"12\"                \"total_explants\"   \n",
      "\n",
      "[1] \"In output, Maximum observations for a segment in a grid item in a filename (should be 1): 1\"\n",
      "[1] \"In randomization datasheet, Maximum observations for an ID (should be 1): 2\"\n",
      "\n",
      "[1] \"Before calculating totals, let's look at the column total_explants in randomization datasheet\"\n",
      "[1] 12 11 12 11 10  0\n",
      "[1] \"0\"  \"4\"  \"6\"  \"10\" \"11\" \"12\"\n",
      "\n",
      "\n",
      "[1] \"Nrow of output before dropping missing explants is: 2491\"\n",
      "[1] \"Head of ID_exp columns from both output data table and IDs_to_drop data tbales, respectively:\"\n",
      "[1] \"__NA\"     \"FAE1_1_1\" \"FAE1_1_1\" \"FAE1_1_1\" \"FAE1_1_1\" \"FAE1_1_1\"\n",
      "[1] \"FAE1_2_6\"  \"FAE1_3_9\"  \"FAE1_14_1\" \"FAE1_14_2\" \"FAE1_14_3\" \"FAE1_14_4\"\n",
      "[1] \"Nrow of output AFTER dropping missing explants is: 2311\"\n",
      "[1] \"Head of output\"\n",
      "   filename grid_item          V1 segment_hex intensity_threshold\n",
      "1:               <NA> observation                              NA\n",
      "   n_pixels_passing_threshold mean_signal max_signal total_signal total_pixels\n",
      "1:                         NA          NA         NA           NA           NA\n",
      "   present ID transgenic segment_present ID_exp\n",
      "1:       Y  _          0               0   __NA\n",
      "\n",
      "# A tibble: 6 x 5\n",
      "  filename                             grid_item present ID      ID_exp   \n",
      "  <chr>                                <chr>     <chr>   <chr>   <chr>    \n",
      "1 FAE1_I5.0_F1.9_L100_143859_1_0_1.jpg 6         C       FAE1_2  FAE1_2_6 \n",
      "2 FAE1_I5.0_F1.9_L100_143959_2_0_2.jpg 9         C       FAE1_3  FAE1_3_9 \n",
      "3 FAE1_I5.0_F1.9_L100_144459_7_1_6.jpg 1         C       FAE1_14 FAE1_14_1\n",
      "4 FAE1_I5.0_F1.9_L100_144459_7_1_6.jpg 2         C       FAE1_14 FAE1_14_2\n",
      "5 FAE1_I5.0_F1.9_L100_144459_7_1_6.jpg 3         C       FAE1_14 FAE1_14_3\n",
      "6 FAE1_I5.0_F1.9_L100_144459_7_1_6.jpg 4         C       FAE1_14 FAE1_14_4\n",
      " [1] \"filename\"                   \"grid_item\"                 \n",
      " [3] \"V1\"                         \"segment_hex\"               \n",
      " [5] \"intensity_threshold\"        \"n_pixels_passing_threshold\"\n",
      " [7] \"mean_signal\"                \"max_signal\"                \n",
      " [9] \"total_signal\"               \"total_pixels\"              \n",
      "[11] \"present\"                    \"ID\"                        \n",
      "[13] \"transgenic\"                 \"segment_present\"           \n",
      "[15] \"ID_exp\"                    \n",
      "[1] \"Maximum # grid positions with transgenic stem in any plate: 9\"\n",
      "[1] \"Maximum # grid positions with transgenic callus in any plate: 11\"\n",
      "[1] \"Maximum # grid positions with transgenic shoot in any plate: 9\"\n",
      "[1] \"Minimum # grid positions with explants in any plate: 0\"\n",
      "[1] \"Maximum # grid positions with explants in any plate: 12\"\n",
      "[1] \"Maximum # grid positions with callus in any plate: 12\"\n",
      "[1] \"Maximum # grid positions with shoot in any plate: 10\"\n",
      "[1] \"Column of interest is portion_transgenic_Shoot\"\n",
      "[1] \"For transgenic tissue, What is the range of values for portion for Shoot\"\n",
      "[1] NA\n",
      "[1] NA\n",
      "[1] \"What are factor levels?\"\n",
      " [1] \"0\"                  \"0.0833333333333333\" \"0.166666666666667\" \n",
      " [4] \"0.181818181818182\"  \"0.2\"                \"0.25\"              \n",
      " [7] \"0.333333333333333\"  \"0.363636363636364\"  \"0.416666666666667\" \n",
      "[10] \"0.454545454545455\"  \"0.5\"                \"0.636363636363636\" \n",
      "[13] \"0.666666666666667\"  \"0.727272727272727\"  \"0.75\"              \n",
      "\n",
      "[1] \"Column of interest is portion_Shoot\"\n",
      "[1] \"For total tissue, What is the range of values for portion for Shoot\"\n",
      "[1] NaN\n",
      "[1] NaN\n",
      "[1] \"What are factor levels?\"\n",
      " [1] \"0\"                  \"0.0833333333333333\" \"0.1\"               \n",
      " [4] \"0.166666666666667\"  \"0.181818181818182\"  \"0.2\"               \n",
      " [7] \"0.25\"               \"0.333333333333333\"  \"0.363636363636364\" \n",
      "[10] \"0.416666666666667\"  \"0.454545454545455\"  \"0.5\"               \n",
      "[13] \"0.545454545454545\"  \"0.636363636363636\"  \"0.666666666666667\" \n",
      "[16] \"0.727272727272727\"  \"0.75\"               \"0.833333333333333\" \n",
      "[19] \"NaN\"               \n",
      "\n",
      "[1] \"Column of interest is portion_transgenic_Callus\"\n",
      "[1] \"For transgenic tissue, What is the range of values for portion for Callus\"\n",
      "[1] NA\n",
      "[1] NA\n",
      "[1] \"What are factor levels?\"\n",
      " [1] \"0\"                  \"0.0833333333333333\" \"0.0909090909090909\"\n",
      " [4] \"0.1\"                \"0.166666666666667\"  \"0.2\"               \n",
      " [7] \"0.25\"               \"0.272727272727273\"  \"0.3\"               \n",
      "[10] \"0.333333333333333\"  \"0.363636363636364\"  \"0.416666666666667\" \n",
      "[13] \"0.454545454545455\"  \"0.5\"                \"0.545454545454545\" \n",
      "[16] \"0.583333333333333\"  \"0.636363636363636\"  \"0.666666666666667\" \n",
      "[19] \"0.833333333333333\"  \"0.916666666666667\" \n",
      "\n",
      "[1] \"Column of interest is portion_Callus\"\n",
      "[1] \"For total tissue, What is the range of values for portion for Callus\"\n",
      "[1] NaN\n",
      "[1] NaN\n",
      "[1] \"What are factor levels?\"\n",
      " [1] \"0\"                 \"0.25\"              \"0.333333333333333\"\n",
      " [4] \"0.363636363636364\" \"0.4\"               \"0.416666666666667\"\n",
      " [7] \"0.5\"               \"0.583333333333333\" \"0.6\"              \n",
      "[10] \"0.666666666666667\" \"0.727272727272727\" \"0.75\"             \n",
      "[13] \"0.818181818181818\" \"0.833333333333333\" \"0.9\"              \n",
      "[16] \"0.909090909090909\" \"1\"                 \"NaN\"              \n",
      "\n",
      "[1] \"Column of interest is portion_transgenic_Stem\"\n",
      "[1] \"For transgenic tissue, What is the range of values for portion for Stem\"\n",
      "[1] NA\n",
      "[1] NA\n",
      "[1] \"What are factor levels?\"\n",
      " [1] \"0\"                  \"0.0833333333333333\" \"0.0909090909090909\"\n",
      " [4] \"0.1\"                \"0.166666666666667\"  \"0.181818181818182\" \n",
      " [7] \"0.2\"                \"0.25\"               \"0.272727272727273\" \n",
      "[10] \"0.3\"                \"0.333333333333333\"  \"0.363636363636364\" \n",
      "[13] \"0.4\"                \"0.416666666666667\"  \"0.5\"               \n",
      "[16] \"0.75\"              \n",
      "\n",
      "[1] \"Column of interest is portion_Stem\"\n",
      "[1] \"For total tissue, What is the range of values for portion for Stem\"\n",
      "[1] NaN\n",
      "[1] NaN\n",
      "[1] \"What are factor levels?\"\n",
      " [1] \"0\"                 \"0.333333333333333\" \"0.363636363636364\"\n",
      " [4] \"0.416666666666667\" \"0.454545454545455\" \"0.5\"              \n",
      " [7] \"0.545454545454545\" \"0.583333333333333\" \"0.666666666666667\"\n",
      "[10] \"0.7\"               \"0.727272727272727\" \"0.75\"             \n",
      "[13] \"0.8\"               \"0.818181818181818\" \"0.833333333333333\"\n",
      "[16] \"0.9\"               \"0.909090909090909\" \"0.916666666666667\"\n",
      "[19] \"1\"                 \"NaN\"              \n",
      "\n",
      "[1] \"Column of interest is portion_transgenic_All_tissue\"\n",
      "[1] \"For transgenic tissue, What is the range of values for portion for All_tissue\"\n",
      "[1] NA\n",
      "[1] NA\n",
      "[1] \"What are factor levels?\"\n",
      " [1] \"0\"                 \"0.1\"               \"0.166666666666667\"\n",
      " [4] \"0.25\"              \"0.272727272727273\" \"0.333333333333333\"\n",
      " [7] \"0.4\"               \"0.416666666666667\" \"0.5\"              \n",
      "[10] \"0.545454545454545\" \"0.583333333333333\" \"0.6\"              \n",
      "[13] \"0.636363636363636\" \"0.666666666666667\" \"0.727272727272727\"\n",
      "[16] \"0.75\"              \"0.818181818181818\" \"0.833333333333333\"\n",
      "[19] \"0.916666666666667\" \"1\"                \n",
      "\n",
      "[1] \"Column of interest is portion_All_tissue\"\n",
      "[1] \"For total tissue, What is the range of values for portion for All_tissue\"\n",
      "[1] NaN\n",
      "[1] NaN\n",
      "[1] \"What are factor levels?\"\n",
      "[1] \"0\"   \"1\"   \"NaN\"\n",
      "\n",
      "[1] \"Column of interest is portion_transgenic_All_regenerated_tissue\"\n",
      "[1] \"For transgenic tissue, What is the range of values for portion for All_regenerated_tissue\"\n",
      "[1] NA\n",
      "[1] NA\n",
      "[1] \"What are factor levels?\"\n",
      " [1] \"0\"                  \"0.0833333333333333\" \"0.0909090909090909\"\n",
      " [4] \"0.1\"                \"0.166666666666667\"  \"0.181818181818182\" \n",
      " [7] \"0.2\"                \"0.25\"               \"0.272727272727273\" \n",
      "[10] \"0.3\"                \"0.333333333333333\"  \"0.4\"               \n",
      "[13] \"0.416666666666667\"  \"0.454545454545455\"  \"0.5\"               \n",
      "[16] \"0.545454545454545\"  \"0.583333333333333\"  \"0.636363636363636\" \n",
      "[19] \"0.666666666666667\"  \"0.75\"               \"0.818181818181818\" \n",
      "[22] \"0.833333333333333\"  \"1\"                 \n",
      "\n",
      "[1] \"Column of interest is portion_All_regenerated_tissue\"\n",
      "[1] \"For total tissue, What is the range of values for portion for All_regenerated_tissue\"\n",
      "[1] NaN\n",
      "[1] NaN\n",
      "[1] \"What are factor levels?\"\n",
      " [1] \"0\"                 \"0.25\"              \"0.363636363636364\"\n",
      " [4] \"0.4\"               \"0.416666666666667\" \"0.5\"              \n",
      " [7] \"0.583333333333333\" \"0.6\"               \"0.666666666666667\"\n",
      "[10] \"0.75\"              \"0.818181818181818\" \"0.833333333333333\"\n",
      "[13] \"0.9\"               \"0.909090909090909\" \"0.916666666666667\"\n",
      "[16] \"1\"                 \"NaN\"              \n",
      "\n",
      "[1] \"Column of interest is portion_transgenic_Background\"\n",
      "[1] \"For transgenic tissue, What is the range of values for portion for Background\"\n",
      "[1] NA\n",
      "[1] NA\n",
      "[1] \"What are factor levels?\"\n",
      "[1] \"0\"\n",
      "\n",
      "[1] \"Column of interest is portion_Background\"\n",
      "[1] \"For total tissue, What is the range of values for portion for Background\"\n",
      "[1] NaN\n",
      "[1] NaN\n",
      "[1] \"What are factor levels?\"\n",
      "[1] \"0\"   \"NaN\"\n",
      "\n",
      "[1] \"N rows of randomization datasheet: \"\n",
      "[1] 48\n",
      "[1] \"N rows of randomization datasheet: \"\n",
      "[1] 48\n",
      "\n",
      "[1] \"Plots will be saved to: \"\n",
      "[1] \"/home/labgroup/code/GMOlabeler/plots/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//\"\n",
      "\n",
      "\n",
      "[1] \"How many groups per treatment?\"\n",
      "\n",
      "  CTR WOX11 WOX5a \n",
      "   15    17    16 \n",
      "[1] \"How many rows in randomization datasheet?\"\n",
      "[1] 48\n",
      "[1] \"How many rows in randomization datasheet with portion_Callus data?\"\n",
      "[1] 47\n",
      "Warning messages:\n",
      "1: `fun.y` is deprecated. Use `fun` instead. \n",
      "2: Removed 1 rows containing non-finite values (stat_boxplot). \n",
      "3: Removed 1 rows containing non-finite values (stat_summary). \n",
      "4: Removed 1 rows containing missing values (geom_point). \n",
      "5: Removed 1 rows containing missing values (geom_point). \n",
      "Warning messages:\n",
      "1: Removed 1 rows containing non-finite values (stat_boxplot). \n",
      "2: Removed 1 rows containing non-finite values (stat_summary). \n",
      "3: Removed 1 rows containing missing values (geom_point). \n",
      "4: Removed 1 rows containing missing values (geom_point). \n",
      "Warning messages:\n",
      "1: `fun.y` is deprecated. Use `fun` instead. \n",
      "2: Removed 1 rows containing non-finite values (stat_boxplot). \n",
      "3: Removed 1 rows containing non-finite values (stat_summary). \n",
      "4: Removed 1 rows containing missing values (geom_point). \n",
      "Warning messages:\n",
      "1: Removed 1 rows containing non-finite values (stat_boxplot). \n",
      "2: Removed 1 rows containing non-finite values (stat_summary). \n",
      "3: Removed 1 rows containing missing values (geom_point). \n",
      "Warning messages:\n",
      "1: `fun.y` is deprecated. Use `fun` instead. \n",
      "2: Removed 1 rows containing non-finite values (stat_boxplot). \n",
      "3: Removed 1 rows containing non-finite values (stat_summary). \n",
      "4: Removed 1 rows containing missing values (geom_point). \n",
      "5: Removed 1 rows containing missing values (geom_point). \n",
      "Warning messages:\n",
      "1: Removed 1 rows containing non-finite values (stat_boxplot). \n",
      "2: Removed 1 rows containing non-finite values (stat_summary). \n",
      "3: Removed 1 rows containing missing values (geom_point). \n",
      "4: Removed 1 rows containing missing values (geom_point). \n",
      "Warning messages:\n",
      "1: `fun.y` is deprecated. Use `fun` instead. \n",
      "2: Removed 1 rows containing non-finite values (stat_boxplot). \n",
      "3: Removed 1 rows containing non-finite values (stat_summary). \n",
      "4: Removed 1 rows containing missing values (geom_point). \n",
      "Warning messages:\n",
      "1: Removed 1 rows containing non-finite values (stat_boxplot). \n",
      "2: Removed 1 rows containing non-finite values (stat_summary). \n",
      "3: Removed 1 rows containing missing values (geom_point). \n",
      "Warning messages:\n",
      "1: `fun.y` is deprecated. Use `fun` instead. \n",
      "2: Removed 1 rows containing non-finite values (stat_boxplot). \n",
      "3: Removed 1 rows containing non-finite values (stat_summary). \n",
      "4: Removed 1 rows containing missing values (geom_point). \n",
      "Warning messages:\n",
      "1: Removed 1 rows containing non-finite values (stat_boxplot). \n",
      "2: Removed 1 rows containing non-finite values (stat_summary). \n",
      "3: Removed 1 rows containing missing values (geom_point). \n",
      "Warning messages:\n",
      "1: `fun.y` is deprecated. Use `fun` instead. \n",
      "2: Removed 1 rows containing non-finite values (stat_boxplot). \n",
      "3: Removed 1 rows containing non-finite values (stat_summary). \n",
      "4: Removed 1 rows containing missing values (geom_point). \n",
      "5: Removed 1 rows containing missing values (geom_point). \n",
      "Warning messages:\n",
      "1: Removed 1 rows containing non-finite values (stat_boxplot). \n",
      "2: Removed 1 rows containing non-finite values (stat_summary). \n",
      "3: Removed 1 rows containing missing values (geom_point). \n",
      "4: Removed 1 rows containing missing values (geom_point). \n",
      "Warning messages:\n",
      "1: `fun.y` is deprecated. Use `fun` instead. \n",
      "2: Removed 1 rows containing non-finite values (stat_boxplot). \n",
      "3: Removed 1 rows containing non-finite values (stat_summary). \n",
      "4: Removed 1 rows containing missing values (geom_point). \n",
      "Warning messages:\n",
      "1: Removed 1 rows containing non-finite values (stat_boxplot). \n",
      "2: Removed 1 rows containing non-finite values (stat_summary). \n",
      "3: Removed 1 rows containing missing values (geom_point). \n",
      "Warning messages:\n",
      "1: `fun.y` is deprecated. Use `fun` instead. \n",
      "2: Removed 1 rows containing non-finite values (stat_boxplot). \n",
      "3: Removed 1 rows containing non-finite values (stat_summary). \n",
      "4: Removed 1 rows containing missing values (geom_point). \n",
      "Warning messages:\n",
      "1: Removed 1 rows containing non-finite values (stat_boxplot). \n",
      "2: Removed 1 rows containing non-finite values (stat_summary). \n",
      "3: Removed 1 rows containing missing values (geom_point). \n",
      "\n",
      "[1] \"How many groups per treatment?\"\n",
      "\n",
      "  CTR WOX11 WOX5a \n",
      "   15    17    16 \n",
      "[1] \"How many rows in randomization datasheet?\"\n",
      "[1] 48\n",
      "[1] \"How many rows in randomization datasheet with portion_Callus data?\"\n",
      "[1] 47\n",
      "Warning messages:\n",
      "1: Ignoring unknown parameters: outlier.shape \n",
      "2: `fun.y` is deprecated. Use `fun` instead. \n",
      "3: Removed 1 rows containing non-finite values (stat_ydensity). \n",
      "4: Removed 1 rows containing non-finite values (stat_summary). \n",
      "Warning messages:\n",
      "1: Removed 1 rows containing non-finite values (stat_ydensity). \n",
      "2: Removed 1 rows containing non-finite values (stat_summary). \n",
      "Warning messages:\n",
      "1: Ignoring unknown parameters: outlier.shape \n",
      "2: `fun.y` is deprecated. Use `fun` instead. \n",
      "3: Removed 1 rows containing non-finite values (stat_ydensity). \n",
      "4: Removed 1 rows containing non-finite values (stat_summary). \n",
      "Warning messages:\n",
      "1: Removed 1 rows containing non-finite values (stat_ydensity). \n",
      "2: Removed 1 rows containing non-finite values (stat_summary). \n",
      "Warning messages:\n",
      "1: Ignoring unknown parameters: outlier.shape \n",
      "2: `fun.y` is deprecated. Use `fun` instead. \n",
      "3: Removed 1 rows containing non-finite values (stat_ydensity). \n",
      "4: Removed 1 rows containing non-finite values (stat_summary). \n",
      "Warning messages:\n",
      "1: Removed 1 rows containing non-finite values (stat_ydensity). \n",
      "2: Removed 1 rows containing non-finite values (stat_summary). \n",
      "Warning messages:\n",
      "1: Ignoring unknown parameters: outlier.shape \n",
      "2: `fun.y` is deprecated. Use `fun` instead. \n",
      "3: Removed 1 rows containing non-finite values (stat_ydensity). \n",
      "4: Removed 1 rows containing non-finite values (stat_summary). \n",
      "Warning messages:\n",
      "1: Removed 1 rows containing non-finite values (stat_ydensity). \n",
      "2: Removed 1 rows containing non-finite values (stat_summary). \n",
      "Warning messages:\n",
      "1: Ignoring unknown parameters: outlier.shape \n",
      "2: `fun.y` is deprecated. Use `fun` instead. \n",
      "3: Removed 1 rows containing non-finite values (stat_ydensity). \n",
      "4: Removed 1 rows containing non-finite values (stat_summary). \n",
      "Warning messages:\n",
      "1: Removed 1 rows containing non-finite values (stat_ydensity). \n",
      "2: Removed 1 rows containing non-finite values (stat_summary). \n",
      "Warning messages:\n",
      "1: Ignoring unknown parameters: outlier.shape \n",
      "2: `fun.y` is deprecated. Use `fun` instead. \n",
      "3: Removed 1 rows containing non-finite values (stat_ydensity). \n",
      "4: Removed 1 rows containing non-finite values (stat_summary). \n",
      "Warning messages:\n",
      "1: Removed 1 rows containing non-finite values (stat_ydensity). \n",
      "2: Removed 1 rows containing non-finite values (stat_summary). \n",
      "Warning messages:\n",
      "1: Ignoring unknown parameters: outlier.shape \n",
      "2: `fun.y` is deprecated. Use `fun` instead. \n",
      "3: Removed 1 rows containing non-finite values (stat_ydensity). \n",
      "4: Removed 1 rows containing non-finite values (stat_summary). \n",
      "Warning messages:\n",
      "1: Removed 1 rows containing non-finite values (stat_ydensity). \n",
      "2: Removed 1 rows containing non-finite values (stat_summary). \n",
      "Warning messages:\n",
      "1: Ignoring unknown parameters: outlier.shape \n",
      "2: `fun.y` is deprecated. Use `fun` instead. \n",
      "3: Removed 1 rows containing non-finite values (stat_ydensity). \n",
      "4: Removed 1 rows containing non-finite values (stat_summary). \n",
      "Warning messages:\n",
      "1: Removed 1 rows containing non-finite values (stat_ydensity). \n",
      "2: Removed 1 rows containing non-finite values (stat_summary). \n",
      "\n",
      "[1] \"How many groups per treatment?\"\n",
      "\n",
      "  CTR WOX11 WOX5a \n",
      "   15    17    16 \n",
      "[1] \"How many rows in randomization datasheet?\"\n",
      "[1] 48\n",
      "[1] \"How many rows in randomization datasheet with n_Callus data?\"\n",
      "[1] 48\n",
      "[1] \"Head:\"\n",
      "       ID Image # Tray ID Rand # Date of transform Date on Media *Plate #\n",
      "1  FAE1_1       1    FAE1      1        2020-11-17    2020-11-19        7\n",
      "2 FAE1_10      10    FAE1     10        2020-11-17    2020-11-19       14\n",
      "3 FAE1_11      11    FAE1     11        2020-11-17    2020-11-19       13\n",
      "4 FAE1_12      12    FAE1     12        2020-11-17    2020-11-19       27\n",
      "5 FAE1_13      13    FAE1     13        2020-11-17    2020-11-19       12\n",
      "6 FAE1_14      14    FAE1     14        2020-11-17    2020-11-19       44\n",
      "  Genotype_ID Treatment name Replicate ...10       Rand ...12    LabelTop\n",
      "1         717            CTR         7    NA 0.72638474    NA   FAE1_1_R1\n",
      "2         717          WOX5a         7    NA 0.91179032    NA FAE1_10_R10\n",
      "3         717          WOX5a         6    NA 0.60061333    NA FAE1_11_R11\n",
      "4         353            CTR         5    NA 0.70703430    NA FAE1_12_R12\n",
      "5         717          WOX5a         5    NA 0.53159478    NA FAE1_13_R13\n",
      "6         353          WOX11         5    NA 0.05430878    NA FAE1_14_R14\n",
      "     LabelBot                          filename 1 2 3 4 5 6 7 8 9 10 11 12\n",
      "1   717_CTR_7  FAE1_I5.0_F1.9_L100_143813_0_0_0 1 1 1 1 1 1 1 1 1  1  1  1\n",
      "2 717_WOX5a_7 FAE1_I5.0_F1.9_L100_144848_11_1_2 1 1 1 1 1 1 1 1 0  1  1  1\n",
      "3 717_WOX5a_6 FAE1_I5.0_F1.9_L100_144749_10_1_3 1 1 1 1 1 1 1 1 1  1  1  1\n",
      "4   353_CTR_5  FAE1_I5.0_F1.9_L100_144650_9_1_4 1 1 1 1 1 1 1 1 0  1  1  1\n",
      "5 717_WOX5a_5  FAE1_I5.0_F1.9_L100_144550_8_1_5 1 1 1 0 0 1 1 1 1  1  1  1\n",
      "6 353_WOX11_5  FAE1_I5.0_F1.9_L100_144459_7_1_6 0 0 0 0 0 0 0 0 0  0  0  0\n",
      "  total_explants n_transgenic_Shoot portion_transgenic_Shoot\n",
      "1             12                  5                0.4166667\n",
      "2             11                  5                0.4545455\n",
      "3             12                  8                0.6666667\n",
      "4             11                  2                0.1818182\n",
      "5             10                  2                0.2000000\n",
      "6              0                  0                       NA\n",
      "  n_transgenic_Callus portion_transgenic_Callus n_transgenic_Stem\n",
      "1                   7                 0.5833333                 2\n",
      "2                   7                 0.6363636                 1\n",
      "3                   7                 0.5833333                 4\n",
      "4                   3                 0.2727273                 1\n",
      "5                   3                 0.3000000                 3\n",
      "6                   0                        NA                 0\n",
      "  portion_transgenic_Stem n_transgenic_All_tissue portion_transgenic_All_tissue\n",
      "1              0.16666667                       7                     0.5833333\n",
      "2              0.09090909                       7                     0.6363636\n",
      "3              0.33333333                      11                     0.9166667\n",
      "4              0.09090909                       3                     0.2727273\n",
      "5              0.30000000                       5                     0.5000000\n",
      "6                      NA                       0                            NA\n",
      "  n_transgenic_All_regenerated_tissue portion_transgenic_All_regenerated_tissue\n",
      "1                                   7                                 0.5833333\n",
      "2                                   7                                 0.6363636\n",
      "3                                   9                                 0.7500000\n",
      "4                                   3                                 0.2727273\n",
      "5                                   3                                 0.3000000\n",
      "6                                   0                                        NA\n",
      "  n_transgenic_Background portion_transgenic_Background n_Shoot portion_Shoot\n",
      "1                       0                             0       5     0.4166667\n",
      "2                       0                             0       6     0.5454545\n",
      "3                       0                             0       8     0.6666667\n",
      "4                       0                             0       2     0.1818182\n",
      "5                       0                             0       2     0.2000000\n",
      "6                       0                            NA       0           NaN\n",
      "  n_Callus portion_Callus n_Stem portion_Stem n_All_tissue portion_All_tissue\n",
      "1       12      1.0000000      5    0.4166667           12                  1\n",
      "2        9      0.8181818      6    0.5454545           11                  1\n",
      "3       10      0.8333333      7    0.5833333           12                  1\n",
      "4        4      0.3636364     10    0.9090909           11                  1\n",
      "5        4      0.4000000      7    0.7000000           10                  1\n",
      "6        0            NaN      0          NaN            0                NaN\n",
      "  n_All_regenerated_tissue portion_All_regenerated_tissue n_Background\n",
      "1                       12                      1.0000000            0\n",
      "2                        9                      0.8181818            0\n",
      "3                       10                      0.8333333            0\n",
      "4                        4                      0.3636364            0\n",
      "5                        4                      0.4000000            0\n",
      "6                        0                            NaN            0\n",
      "  portion_Background\n",
      "1                  0\n",
      "2                  0\n",
      "3                  0\n",
      "4                  0\n",
      "5                  0\n",
      "6                NaN\n",
      "Warning message:\n",
      "`fun.y` is deprecated. Use `fun` instead. \n",
      "Warning message:\n",
      "`fun.y` is deprecated. Use `fun` instead. \n",
      "Warning message:\n",
      "`fun.y` is deprecated. Use `fun` instead. \n",
      "Warning message:\n",
      "`fun.y` is deprecated. Use `fun` instead. \n",
      "Warning message:\n",
      "`fun.y` is deprecated. Use `fun` instead. \n",
      "Warning message:\n",
      "`fun.y` is deprecated. Use `fun` instead. \n",
      "Warning message:\n",
      "`fun.y` is deprecated. Use `fun` instead. \n",
      "Warning message:\n",
      "`fun.y` is deprecated. Use `fun` instead. \n",
      "\n",
      "[1] \"List of treatments in this randomization_datasheet: \"\n",
      "[1] \"CTR\"   \"WOX11\" \"WOX5a\"\n",
      "[1] \"Now about to merge randomization_datasheet and GMOlabeler output.\"\n",
      "[1] \"Rows in randomization_datasheet: 48\"\n",
      "[1] \"Rows in GMOlabeler output: 2311\"\n",
      "[1] \"Before merging, look at table of ID in both. \"\n",
      "\n",
      " FAE1_1 FAE1_10 FAE1_11 FAE1_12 FAE1_13 FAE1_14 FAE1_15 FAE1_16 FAE1_17 FAE1_18 \n",
      "      1       1       1       1       1       2       1       1       1       1 \n",
      "FAE1_19  FAE1_2 FAE1_20 FAE1_21  FAE1_3  FAE1_4  FAE1_5  FAE1_6  FAE1_7  FAE1_8 \n",
      "      1       1       1       1       1       1       1       1       1       1 \n",
      " FAE1_9  FAE2_1 FAE2_10 FAE2_11 FAE2_12 FAE2_13 FAE2_14 FAE2_15 FAE2_16 FAE2_17 \n",
      "      1       1       1       1       1       1       1       1       1       1 \n",
      "FAE2_18 FAE2_19  FAE2_2 FAE2_20 FAE2_21  FAE2_3  FAE2_4  FAE2_5  FAE2_6  FAE2_7 \n",
      "      1       1       1       1       1       1       1       1       1       1 \n",
      " FAE2_8  FAE2_9  FAE3_1  FAE3_2  FAE3_3  FAE3_4  FAE3_5 \n",
      "      1       1       1       1       1       1       1 \n",
      "\n",
      "      _  FAE1_1 FAE1_10 FAE1_11 FAE1_12 FAE1_13 FAE1_15 FAE1_16 FAE1_17 FAE1_18 \n",
      "      1      60      53      58      48      44      48      55      54      56 \n",
      "FAE1_19  FAE1_2 FAE1_20 FAE1_21  FAE1_3  FAE1_4  FAE1_5  FAE1_6  FAE1_7  FAE1_8 \n",
      "     55      53      57      29      53      55      54      60      53      58 \n",
      " FAE1_9  FAE2_1 FAE2_11 FAE2_12 FAE2_13 FAE2_14 FAE2_15 FAE2_16 FAE2_17 FAE2_18 \n",
      "     20      44      59      56      56      55      53      54      46      55 \n",
      "FAE2_19  FAE2_2 FAE2_21  FAE2_3  FAE2_4  FAE2_5  FAE2_6  FAE2_7  FAE2_8  FAE2_9 \n",
      "     58      55      57      59      55      48      57      54      53      51 \n",
      " FAE3_1  FAE3_2  FAE3_3  FAE3_4  FAE3_5 \n",
      "     49      46      60      54      53 \n",
      "[1] \"Rows in merged output: 2311\"\n",
      "[1] \"List of treatments in this combined data: \"\n",
      "[1] \"CTR\"   \"WOX11\" \"WOX5a\"\n",
      "\n",
      "[1] \"Making final plots for tissue: \"\n",
      "  Tissue hex_code color\n",
      "1  Shoot   00CC11 green\n",
      "[1] \"Dim before and after subsetting combined data to this tissue only\"\n",
      "[1] 2311   67\n",
      "[1] 495  67\n",
      "[1] \"Max mean_signal\"\n",
      "[1] 354.1415\n",
      "[1] \"Mean signal plots\"\n",
      "[1] \"Finished making mean signal plot... Now to save and make the rest.\"\n",
      "[1] \"Max signal plots\"\n",
      "[1] \"Total signal plots\"\n",
      "[1] \"Making final plots for tissue: \"\n",
      "  Tissue hex_code color\n",
      "2 Callus   0006CC  blue\n",
      "[1] \"Dim before and after subsetting combined data to this tissue only\"\n",
      "[1] 2311   67\n",
      "[1] 495  67\n",
      "[1] \"Max mean_signal\"\n",
      "[1] 794.7745\n",
      "[1] \"Mean signal plots\"\n",
      "[1] \"Finished making mean signal plot... Now to save and make the rest.\"\n",
      "[1] \"Max signal plots\"\n",
      "[1] \"Total signal plots\"\n",
      "[1] \"Making final plots for tissue: \"\n",
      "  Tissue hex_code color\n",
      "3   Stem   CC0000   red\n",
      "[1] \"Dim before and after subsetting combined data to this tissue only\"\n",
      "[1] 2311   67\n",
      "[1] 495  67\n",
      "[1] \"Max mean_signal\"\n",
      "[1] 82.91154\n",
      "[1] \"Mean signal plots\"\n",
      "[1] \"Finished making mean signal plot... Now to save and make the rest.\"\n",
      "[1] \"Max signal plots\"\n",
      "[1] \"Total signal plots\"\n",
      "[1] \"Making final plots for tissue: \"\n",
      "      Tissue   hex_code color\n",
      "4 All_tissue All_tissue black\n",
      "[1] \"Dim before and after subsetting combined data to this tissue only\"\n",
      "[1] 2311   67\n",
      "[1] 495  67\n",
      "[1] \"Max mean_signal\"\n",
      "[1] -Inf\n",
      "[1] \"Making final plots for tissue: \"\n",
      "                  Tissue               hex_code color\n",
      "5 All_regenerated_tissue All_regenerated_tissue  <NA>\n",
      "[1] \"Dim before and after subsetting combined data to this tissue only\"\n",
      "[1] 2311   67\n",
      "[1] 330  67\n",
      "[1] \"Max mean_signal\"\n",
      "[1] -Inf\n",
      "There were 50 or more warnings (use warnings() to see the first 50)\n",
      "(gmolabeler) "
     ]
    }
   ],
   "source": [
    "conda activate gmolabeler\n",
    "cd $gmolabeler_path\n",
    "Rscript grid_item_plots.R \\\n",
    "-d \"${data_folder}/\" \\\n",
    "-r \"$randomization_datasheet\" \\\n",
    "-p $pixel_threshold \\\n",
    "-v categorical \\\n",
    "-m 1 \\\n",
    "-M $missing_explants \\\n",
    "-g $grid \\\n",
    "--sort 1 \\\n",
    "--height $height \\\n",
    "--width $width\n",
    "conda deactivate"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 5. Email plots to user"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### 5.1. ZIP results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/labgroup/code/GMOlabeler//plotsElements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence/\n"
     ]
    }
   ],
   "source": [
    "echo \"${gmolabeler_path}/plots${data_folder}\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "metadata": {},
   "outputs": [],
   "source": [
    "cd \"${gmolabeler_path}/plots/${data_folder}\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "metadata": {},
   "outputs": [],
   "source": [
    "rm -f ./plants_over_plates.csv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/labgroup/code/GMOlabeler//output/Elements_9//Transformation/FAE_Dev_Genes/wk7/Fluorescence//\n"
     ]
    }
   ],
   "source": [
    "echo ${gmolabeler_path}/output/${data_folder}/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "metadata": {},
   "outputs": [],
   "source": [
    "cp \"${gmolabeler_path}/output/${data_folder}/plants_over_plates.csv\" ./"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "metadata": {},
   "outputs": [],
   "source": [
    "cp \"${gmolabeler_path}/output/${data_folder}/stats_with_sums_over_tissues.csv\" ./"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 124,
   "metadata": {},
   "outputs": [],
   "source": [
    "cd $data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 125,
   "metadata": {},
   "outputs": [],
   "source": [
    "rm -f Rplots.pdf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 126,
   "metadata": {},
   "outputs": [],
   "source": [
    "cd ../"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This messy substitution is explained here: https://superuser.com/questions/1068031/replace-backslash-with-forward-slash-in-a-variable-in-bash"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 127,
   "metadata": {},
   "outputs": [],
   "source": [
    "data_folder_Compress=\"${data_folder////-}.zip\"\n",
    "data_folder_Compress=${data_folder_Compress#?};"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 128,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Broadband  Fluorescence\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 129,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  adding: FAE1_I5.0_F1.9_L100_143813_0_0_0_Fluorescence.hdr (deflated 59%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_143813_0_0_0_Fluorescence.raw (deflated 58%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_143813_0_0_0_hyperchannel.csv (deflated 96%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_143813_0_0_0_rgb_cropped.jpg (deflated 0%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_143813_0_0_0_rgb.jpg (deflated 1%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_143813_0_0_0_rgb_processed.jpg (deflated 0%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_143813_0_0_0_segment_cropped.png (deflated 4%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_143813_0_0_0_segment_uncropped.png (deflated 39%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_143813_0_0_0_segment_uncropped_processed.png (deflated 11%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_143859_1_0_1_Fluorescence.hdr (deflated 59%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_143859_1_0_1_Fluorescence.raw (deflated 58%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_143859_1_0_1_hyperchannel.csv (deflated 96%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_143859_1_0_1_rgb_cropped.jpg (deflated 0%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_143859_1_0_1_rgb.jpg (deflated 1%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_143859_1_0_1_rgb_processed.jpg (deflated 0%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_143859_1_0_1_segment_cropped.png (deflated 3%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_143859_1_0_1_segment_uncropped.png (deflated 45%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_143859_1_0_1_segment_uncropped_processed.png (deflated 14%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_143959_2_0_2_Fluorescence.hdr (deflated 59%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_143959_2_0_2_Fluorescence.raw (deflated 57%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_143959_2_0_2_hyperchannel.csv (deflated 96%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_143959_2_0_2_rgb_cropped.jpg (deflated 0%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_143959_2_0_2_rgb.jpg (deflated 1%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_143959_2_0_2_rgb_processed.jpg (deflated 0%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_143959_2_0_2_segment_cropped.png (deflated 3%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_143959_2_0_2_segment_uncropped.png (deflated 31%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_143959_2_0_2_segment_uncropped_processed.png (deflated 11%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144101_3_0_3_Fluorescence.hdr (deflated 59%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144101_3_0_3_Fluorescence.raw (deflated 58%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144101_3_0_3_hyperchannel.csv (deflated 96%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144101_3_0_3_rgb_cropped.jpg (deflated 0%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144101_3_0_3_rgb.jpg (deflated 1%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144101_3_0_3_rgb_processed.jpg (deflated 0%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144101_3_0_3_segment_cropped.png (deflated 3%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144101_3_0_3_segment_uncropped.png (deflated 44%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144101_3_0_3_segment_uncropped_processed.png (deflated 14%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144200_4_0_4_Fluorescence.hdr (deflated 59%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144200_4_0_4_Fluorescence.raw (deflated 58%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144200_4_0_4_hyperchannel.csv (deflated 96%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144200_4_0_4_rgb_cropped.jpg (deflated 0%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144200_4_0_4_rgb.jpg (deflated 1%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144200_4_0_4_rgb_processed.jpg (deflated 0%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144200_4_0_4_segment_cropped.png (deflated 4%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144200_4_0_4_segment_uncropped.png (deflated 47%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144200_4_0_4_segment_uncropped_processed.png (deflated 16%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144259_5_0_5_Fluorescence.hdr (deflated 59%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144259_5_0_5_Fluorescence.raw (deflated 58%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144259_5_0_5_hyperchannel.csv (deflated 96%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144259_5_0_5_rgb_cropped.jpg (deflated 0%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144259_5_0_5_rgb.jpg (deflated 1%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144259_5_0_5_rgb_processed.jpg (deflated 0%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144259_5_0_5_segment_cropped.png (deflated 4%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144259_5_0_5_segment_uncropped.png (deflated 42%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144259_5_0_5_segment_uncropped_processed.png (deflated 13%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144359_6_0_6_Fluorescence.hdr (deflated 59%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144359_6_0_6_Fluorescence.raw (deflated 58%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144359_6_0_6_hyperchannel.csv (deflated 96%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144359_6_0_6_rgb_cropped.jpg (deflated 0%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144359_6_0_6_rgb.jpg (deflated 1%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144359_6_0_6_rgb_processed.jpg (deflated 0%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144359_6_0_6_segment_cropped.png (deflated 6%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144359_6_0_6_segment_uncropped.png (deflated 48%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144359_6_0_6_segment_uncropped_processed.png (deflated 22%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144459_7_1_6_Fluorescence.hdr (deflated 59%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144459_7_1_6_Fluorescence.raw (deflated 58%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144459_7_1_6_hyperchannel.csv (deflated 96%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144459_7_1_6_rgb_cropped.jpg (deflated 95%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144459_7_1_6_rgb.jpg (deflated 67%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144459_7_1_6_rgb_processed.jpg (deflated 43%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144459_7_1_6_segment_cropped.png (deflated 58%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144459_7_1_6_segment_uncropped.png (deflated 97%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144459_7_1_6_segment_uncropped_processed.png (deflated 87%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144550_8_1_5_Fluorescence.hdr (deflated 59%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144550_8_1_5_Fluorescence.raw (deflated 58%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144550_8_1_5_hyperchannel.csv (deflated 96%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144550_8_1_5_rgb_cropped.jpg (deflated 0%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144550_8_1_5_rgb.jpg (deflated 1%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144550_8_1_5_rgb_processed.jpg (deflated 0%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144550_8_1_5_segment_cropped.png (deflated 3%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144550_8_1_5_segment_uncropped.png (deflated 44%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144550_8_1_5_segment_uncropped_processed.png (deflated 13%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144650_9_1_4_Fluorescence.hdr (deflated 59%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144650_9_1_4_Fluorescence.raw (deflated 58%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144650_9_1_4_hyperchannel.csv (deflated 96%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144650_9_1_4_rgb_cropped.jpg (deflated 0%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144650_9_1_4_rgb.jpg (deflated 1%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144650_9_1_4_rgb_processed.jpg (deflated 0%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144650_9_1_4_segment_cropped.png (deflated 4%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144650_9_1_4_segment_uncropped.png (deflated 47%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144650_9_1_4_segment_uncropped_processed.png (deflated 17%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144749_10_1_3_Fluorescence.hdr (deflated 59%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144749_10_1_3_Fluorescence.raw (deflated 57%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144749_10_1_3_hyperchannel.csv (deflated 96%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144749_10_1_3_rgb_cropped.jpg (deflated 0%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144749_10_1_3_rgb.jpg (deflated 1%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144749_10_1_3_rgb_processed.jpg (deflated 0%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144749_10_1_3_segment_cropped.png (deflated 2%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144749_10_1_3_segment_uncropped.png (deflated 38%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144749_10_1_3_segment_uncropped_processed.png (deflated 10%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144848_11_1_2_Fluorescence.hdr (deflated 59%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144848_11_1_2_Fluorescence.raw (deflated 57%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144848_11_1_2_hyperchannel.csv (deflated 96%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144848_11_1_2_rgb_cropped.jpg (deflated 0%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144848_11_1_2_rgb.jpg (deflated 1%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144848_11_1_2_rgb_processed.jpg (deflated 0%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144848_11_1_2_segment_cropped.png (deflated 3%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144848_11_1_2_segment_uncropped.png (deflated 37%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144848_11_1_2_segment_uncropped_processed.png (deflated 12%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144948_12_1_1_Fluorescence.hdr (deflated 59%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144948_12_1_1_Fluorescence.raw (deflated 58%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144948_12_1_1_hyperchannel.csv (deflated 96%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144948_12_1_1_rgb_cropped.jpg (deflated 1%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144948_12_1_1_rgb.jpg (deflated 1%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144948_12_1_1_rgb_processed.jpg (deflated 0%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144948_12_1_1_segment_cropped.png (deflated 7%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144948_12_1_1_segment_uncropped.png (deflated 66%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_144948_12_1_1_segment_uncropped_processed.png (deflated 36%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_145051_13_1_0_Fluorescence.hdr (deflated 59%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_145051_13_1_0_Fluorescence.raw (deflated 58%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_145051_13_1_0_hyperchannel.csv (deflated 96%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_145051_13_1_0_rgb_cropped.jpg (deflated 0%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_145051_13_1_0_rgb.jpg (deflated 1%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_145051_13_1_0_rgb_processed.jpg (deflated 0%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_145051_13_1_0_segment_cropped.png (deflated 5%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_145051_13_1_0_segment_uncropped.png (deflated 45%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_145051_13_1_0_segment_uncropped_processed.png (deflated 14%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_145151_14_2_0_Fluorescence.hdr (deflated 59%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_145151_14_2_0_Fluorescence.raw (deflated 58%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_145151_14_2_0_hyperchannel.csv (deflated 96%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_145151_14_2_0_rgb_cropped.jpg (deflated 0%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_145151_14_2_0_rgb.jpg (deflated 1%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_145151_14_2_0_rgb_processed.jpg (deflated 0%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_145151_14_2_0_segment_cropped.png (deflated 3%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_145151_14_2_0_segment_uncropped.png (deflated 47%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_145151_14_2_0_segment_uncropped_processed.png (deflated 15%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_145243_15_2_1_Fluorescence.hdr (deflated 59%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_145243_15_2_1_Fluorescence.raw (deflated 58%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_145243_15_2_1_hyperchannel.csv (deflated 96%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_145243_15_2_1_rgb_cropped.jpg (deflated 0%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_145243_15_2_1_rgb.jpg (deflated 1%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_145243_15_2_1_rgb_processed.jpg (deflated 0%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_145243_15_2_1_segment_cropped.png (deflated 4%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_145243_15_2_1_segment_uncropped.png (deflated 38%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_145243_15_2_1_segment_uncropped_processed.png (deflated 13%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_145343_16_2_2_Fluorescence.hdr (deflated 59%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_145343_16_2_2_Fluorescence.raw (deflated 58%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_145343_16_2_2_hyperchannel.csv (deflated 96%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_145343_16_2_2_rgb_cropped.jpg (deflated 0%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_145343_16_2_2_rgb.jpg (deflated 1%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_145343_16_2_2_rgb_processed.jpg (deflated 0%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_145343_16_2_2_segment_cropped.png (deflated 5%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_145343_16_2_2_segment_uncropped.png (deflated 47%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_145343_16_2_2_segment_uncropped_processed.png (deflated 17%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_145442_17_2_3_Fluorescence.hdr (deflated 59%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_145442_17_2_3_Fluorescence.raw (deflated 58%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_145442_17_2_3_hyperchannel.csv (deflated 96%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_145442_17_2_3_rgb_cropped.jpg (deflated 0%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_145442_17_2_3_rgb.jpg (deflated 1%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_145442_17_2_3_rgb_processed.jpg (deflated 0%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_145442_17_2_3_segment_cropped.png (deflated 5%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_145442_17_2_3_segment_uncropped.png (deflated 40%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_145442_17_2_3_segment_uncropped_processed.png (deflated 11%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_145544_18_2_4_Fluorescence.hdr (deflated 59%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_145544_18_2_4_Fluorescence.raw (deflated 58%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_145544_18_2_4_hyperchannel.csv (deflated 96%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_145544_18_2_4_rgb_cropped.jpg (deflated 0%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_145544_18_2_4_rgb.jpg (deflated 1%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_145544_18_2_4_rgb_processed.jpg (deflated 0%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_145544_18_2_4_segment_cropped.png (deflated 4%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_145544_18_2_4_segment_uncropped.png (deflated 47%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_145544_18_2_4_segment_uncropped_processed.png (deflated 17%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_145643_19_2_5_Fluorescence.hdr (deflated 59%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_145643_19_2_5_Fluorescence.raw (deflated 57%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_145643_19_2_5_hyperchannel.csv (deflated 96%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_145643_19_2_5_rgb_cropped.jpg (deflated 0%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_145643_19_2_5_rgb.jpg (deflated 1%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_145643_19_2_5_rgb_processed.jpg (deflated 0%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_145643_19_2_5_segment_cropped.png (deflated 3%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_145643_19_2_5_segment_uncropped.png (deflated 32%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_145643_19_2_5_segment_uncropped_processed.png (deflated 10%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_145744_20_2_6_Fluorescence.hdr (deflated 59%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_145744_20_2_6_Fluorescence.raw (deflated 58%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_145744_20_2_6_hyperchannel.csv (deflated 96%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_145744_20_2_6_rgb_cropped.jpg (deflated 0%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_145744_20_2_6_rgb.jpg (deflated 1%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_145744_20_2_6_rgb_processed.jpg (deflated 0%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_145744_20_2_6_segment_cropped.png (deflated 8%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_145744_20_2_6_segment_uncropped.png (deflated 63%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_145744_20_2_6_segment_uncropped_processed.png (deflated 25%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_152151_0_1_6_Fluorescence.hdr (deflated 59%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_152151_0_1_6_Fluorescence.raw (deflated 58%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_152151_0_1_6_hyperchannel.csv (deflated 96%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_152151_0_1_6_rgb_cropped.jpg (deflated 0%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_152151_0_1_6_rgb.jpg (deflated 1%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_152151_0_1_6_rgb_processed.jpg (deflated 0%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_152151_0_1_6_segment_cropped.png (deflated 6%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_152151_0_1_6_segment_uncropped.png (deflated 46%)\n",
      "  adding: FAE1_I5.0_F1.9_L100_152151_0_1_6_segment_uncropped_processed.png (deflated 21%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150028_0_0_0_Fluorescence.hdr (deflated 59%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150028_0_0_0_Fluorescence.raw (deflated 58%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150028_0_0_0_hyperchannel.csv (deflated 96%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150028_0_0_0_rgb_cropped.jpg (deflated 0%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150028_0_0_0_rgb.jpg (deflated 1%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150028_0_0_0_rgb_processed.jpg (deflated 0%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150028_0_0_0_segment_cropped.png (deflated 5%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150028_0_0_0_segment_uncropped.png (deflated 49%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150028_0_0_0_segment_uncropped_processed.png (deflated 17%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150113_1_0_1_Fluorescence.hdr (deflated 59%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150113_1_0_1_Fluorescence.raw (deflated 58%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150113_1_0_1_hyperchannel.csv (deflated 96%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150113_1_0_1_rgb_cropped.jpg (deflated 0%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150113_1_0_1_rgb.jpg (deflated 1%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150113_1_0_1_rgb_processed.jpg (deflated 0%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150113_1_0_1_segment_cropped.png (deflated 3%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150113_1_0_1_segment_uncropped.png (deflated 44%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150113_1_0_1_segment_uncropped_processed.png (deflated 13%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150214_2_0_2_Fluorescence.hdr (deflated 59%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150214_2_0_2_Fluorescence.raw (deflated 57%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150214_2_0_2_hyperchannel.csv (deflated 96%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150214_2_0_2_rgb_cropped.jpg (deflated 0%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150214_2_0_2_rgb.jpg (deflated 1%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150214_2_0_2_rgb_processed.jpg (deflated 0%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150214_2_0_2_segment_cropped.png (deflated 2%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150214_2_0_2_segment_uncropped.png (deflated 25%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150214_2_0_2_segment_uncropped_processed.png (deflated 9%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150313_3_0_3_Fluorescence.hdr (deflated 59%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150313_3_0_3_Fluorescence.raw (deflated 58%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150313_3_0_3_hyperchannel.csv (deflated 96%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150313_3_0_3_rgb_cropped.jpg (deflated 0%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150313_3_0_3_rgb.jpg (deflated 1%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150313_3_0_3_rgb_processed.jpg (deflated 0%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150313_3_0_3_segment_cropped.png (deflated 4%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150313_3_0_3_segment_uncropped.png (deflated 50%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150313_3_0_3_segment_uncropped_processed.png (deflated 16%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150413_4_0_4_Fluorescence.hdr (deflated 59%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150413_4_0_4_Fluorescence.raw (deflated 58%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150413_4_0_4_hyperchannel.csv (deflated 96%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150413_4_0_4_rgb_cropped.jpg (deflated 0%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150413_4_0_4_rgb.jpg (deflated 1%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150413_4_0_4_rgb_processed.jpg (deflated 0%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150413_4_0_4_segment_cropped.png (deflated 4%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150413_4_0_4_segment_uncropped.png (deflated 49%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150413_4_0_4_segment_uncropped_processed.png (deflated 19%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150513_5_0_5_Fluorescence.hdr (deflated 59%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150513_5_0_5_Fluorescence.raw (deflated 58%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150513_5_0_5_hyperchannel.csv (deflated 96%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150513_5_0_5_rgb_cropped.jpg (deflated 0%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150513_5_0_5_rgb.jpg (deflated 1%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150513_5_0_5_rgb_processed.jpg (deflated 0%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150513_5_0_5_segment_cropped.png (deflated 3%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150513_5_0_5_segment_uncropped.png (deflated 45%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150513_5_0_5_segment_uncropped_processed.png (deflated 14%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150613_6_0_6_Fluorescence.hdr (deflated 59%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150613_6_0_6_Fluorescence.raw (deflated 57%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150613_6_0_6_hyperchannel.csv (deflated 96%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150613_6_0_6_rgb_cropped.jpg (deflated 0%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150613_6_0_6_rgb.jpg (deflated 1%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150613_6_0_6_rgb_processed.jpg (deflated 0%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150613_6_0_6_segment_cropped.png (deflated 2%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150613_6_0_6_segment_uncropped.png (deflated 23%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150613_6_0_6_segment_uncropped_processed.png (deflated 7%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150713_7_1_6_Fluorescence.hdr (deflated 59%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150713_7_1_6_Fluorescence.raw (deflated 58%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150713_7_1_6_hyperchannel.csv (deflated 96%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150713_7_1_6_rgb_cropped.jpg (deflated 0%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150713_7_1_6_rgb.jpg (deflated 1%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150713_7_1_6_rgb_processed.jpg (deflated 0%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150713_7_1_6_segment_cropped.png (deflated 5%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150713_7_1_6_segment_uncropped.png (deflated 44%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150713_7_1_6_segment_uncropped_processed.png (deflated 14%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150803_8_1_5_Fluorescence.hdr (deflated 59%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150803_8_1_5_Fluorescence.raw (deflated 58%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150803_8_1_5_hyperchannel.csv (deflated 96%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150803_8_1_5_rgb_cropped.jpg (deflated 0%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150803_8_1_5_rgb.jpg (deflated 1%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150803_8_1_5_rgb_processed.jpg (deflated 0%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150803_8_1_5_segment_cropped.png (deflated 4%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150803_8_1_5_segment_uncropped.png (deflated 43%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150803_8_1_5_segment_uncropped_processed.png (deflated 13%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150902_9_1_4_Fluorescence.hdr (deflated 59%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150902_9_1_4_Fluorescence.raw (deflated 58%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150902_9_1_4_hyperchannel.csv (deflated 96%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150902_9_1_4_rgb_cropped.jpg (deflated 0%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150902_9_1_4_rgb.jpg (deflated 1%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150902_9_1_4_rgb_processed.jpg (deflated 0%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150902_9_1_4_segment_cropped.png (deflated 5%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150902_9_1_4_segment_uncropped.png (deflated 42%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_150902_9_1_4_segment_uncropped_processed.png (deflated 12%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_151002_10_1_3_Fluorescence.hdr (deflated 59%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_151002_10_1_3_Fluorescence.raw (deflated 57%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_151002_10_1_3_hyperchannel.csv (deflated 96%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_151002_10_1_3_rgb_cropped.jpg (deflated 0%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_151002_10_1_3_rgb.jpg (deflated 1%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_151002_10_1_3_rgb_processed.jpg (deflated 0%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_151002_10_1_3_segment_cropped.png (deflated 2%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_151002_10_1_3_segment_uncropped.png (deflated 29%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_151002_10_1_3_segment_uncropped_processed.png (deflated 9%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_151102_11_1_1_Fluorescence.hdr (deflated 59%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_151102_11_1_1_Fluorescence.raw (deflated 58%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_151102_11_1_1_hyperchannel.csv (deflated 96%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_151102_11_1_1_rgb_cropped.jpg (deflated 0%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_151102_11_1_1_rgb.jpg (deflated 1%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_151102_11_1_1_rgb_processed.jpg (deflated 0%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_151102_11_1_1_segment_cropped.png (deflated 4%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_151102_11_1_1_segment_uncropped.png (deflated 46%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_151102_11_1_1_segment_uncropped_processed.png (deflated 20%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_151202_12_1_0_Fluorescence.hdr (deflated 59%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_151202_12_1_0_Fluorescence.raw (deflated 58%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_151202_12_1_0_hyperchannel.csv (deflated 96%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_151202_12_1_0_rgb_cropped.jpg (deflated 0%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_151202_12_1_0_rgb.jpg (deflated 1%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_151202_12_1_0_rgb_processed.jpg (deflated 0%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_151202_12_1_0_segment_cropped.png (deflated 4%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_151202_12_1_0_segment_uncropped.png (deflated 37%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_151202_12_1_0_segment_uncropped_processed.png (deflated 13%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_151302_13_2_0_Fluorescence.hdr (deflated 59%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_151302_13_2_0_Fluorescence.raw (deflated 58%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_151302_13_2_0_hyperchannel.csv (deflated 96%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_151302_13_2_0_rgb_cropped.jpg (deflated 0%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_151302_13_2_0_rgb.jpg (deflated 1%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_151302_13_2_0_rgb_processed.jpg (deflated 0%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_151302_13_2_0_segment_cropped.png (deflated 4%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_151302_13_2_0_segment_uncropped.png (deflated 47%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_151302_13_2_0_segment_uncropped_processed.png (deflated 17%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_151354_14_2_1_Fluorescence.hdr (deflated 59%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_151354_14_2_1_Fluorescence.raw (deflated 57%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_151354_14_2_1_hyperchannel.csv (deflated 96%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_151354_14_2_1_rgb_cropped.jpg (deflated 0%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_151354_14_2_1_rgb.jpg (deflated 1%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_151354_14_2_1_rgb_processed.jpg (deflated 0%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_151354_14_2_1_segment_cropped.png (deflated 4%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_151354_14_2_1_segment_uncropped.png (deflated 31%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_151354_14_2_1_segment_uncropped_processed.png (deflated 9%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_151456_15_2_2_Fluorescence.hdr (deflated 59%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_151456_15_2_2_Fluorescence.raw (deflated 58%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_151456_15_2_2_hyperchannel.csv (deflated 96%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_151456_15_2_2_rgb_cropped.jpg (deflated 0%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_151456_15_2_2_rgb.jpg (deflated 1%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_151456_15_2_2_rgb_processed.jpg (deflated 0%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_151456_15_2_2_segment_cropped.png (deflated 4%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_151456_15_2_2_segment_uncropped.png (deflated 47%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_151456_15_2_2_segment_uncropped_processed.png (deflated 18%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_151556_16_2_3_Fluorescence.hdr (deflated 59%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_151556_16_2_3_Fluorescence.raw (deflated 58%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_151556_16_2_3_hyperchannel.csv (deflated 96%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_151556_16_2_3_rgb_cropped.jpg (deflated 0%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_151556_16_2_3_rgb.jpg (deflated 1%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_151556_16_2_3_rgb_processed.jpg (deflated 0%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_151556_16_2_3_segment_cropped.png (deflated 4%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_151556_16_2_3_segment_uncropped.png (deflated 51%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_151556_16_2_3_segment_uncropped_processed.png (deflated 16%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_151657_17_2_4_Fluorescence.hdr (deflated 59%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_151657_17_2_4_Fluorescence.raw (deflated 57%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_151657_17_2_4_hyperchannel.csv (deflated 96%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_151657_17_2_4_rgb_cropped.jpg (deflated 0%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_151657_17_2_4_rgb.jpg (deflated 1%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_151657_17_2_4_rgb_processed.jpg (deflated 0%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_151657_17_2_4_segment_cropped.png (deflated 2%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_151657_17_2_4_segment_uncropped.png (deflated 27%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_151657_17_2_4_segment_uncropped_processed.png (deflated 10%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_151757_18_2_6_Fluorescence.hdr (deflated 59%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_151757_18_2_6_Fluorescence.raw (deflated 58%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_151757_18_2_6_hyperchannel.csv (deflated 96%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_151757_18_2_6_rgb_cropped.jpg (deflated 0%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_151757_18_2_6_rgb.jpg (deflated 1%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_151757_18_2_6_rgb_processed.jpg (deflated 0%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_151757_18_2_6_segment_cropped.png (deflated 3%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_151757_18_2_6_segment_uncropped.png (deflated 45%)\n",
      "  adding: FAE2_I5.0_F1.9_L100_151757_18_2_6_segment_uncropped_processed.png (deflated 14%)\n",
      "  adding: FAE3_I5.0_F1.9_L100_143049_0_0_0_Fluorescence.hdr (deflated 59%)\n",
      "  adding: FAE3_I5.0_F1.9_L100_143049_0_0_0_Fluorescence.raw (deflated 58%)\n",
      "  adding: FAE3_I5.0_F1.9_L100_143049_0_0_0_hyperchannel.csv (deflated 96%)\n",
      "  adding: FAE3_I5.0_F1.9_L100_143049_0_0_0_rgb_cropped.jpg (deflated 0%)\n",
      "  adding: FAE3_I5.0_F1.9_L100_143049_0_0_0_rgb.jpg (deflated 1%)\n",
      "  adding: FAE3_I5.0_F1.9_L100_143049_0_0_0_rgb_processed.jpg (deflated 0%)\n",
      "  adding: FAE3_I5.0_F1.9_L100_143049_0_0_0_segment_cropped.png (deflated 4%)\n",
      "  adding: FAE3_I5.0_F1.9_L100_143049_0_0_0_segment_uncropped.png (deflated 49%)\n",
      "  adding: FAE3_I5.0_F1.9_L100_143049_0_0_0_segment_uncropped_processed.png (deflated 16%)\n",
      "  adding: FAE3_I5.0_F1.9_L100_143135_1_0_1_Fluorescence.hdr (deflated 59%)\n",
      "  adding: FAE3_I5.0_F1.9_L100_143135_1_0_1_Fluorescence.raw (deflated 58%)\n",
      "  adding: FAE3_I5.0_F1.9_L100_143135_1_0_1_hyperchannel.csv (deflated 96%)\n",
      "  adding: FAE3_I5.0_F1.9_L100_143135_1_0_1_rgb_cropped.jpg (deflated 0%)\n",
      "  adding: FAE3_I5.0_F1.9_L100_143135_1_0_1_rgb.jpg (deflated 1%)\n",
      "  adding: FAE3_I5.0_F1.9_L100_143135_1_0_1_rgb_processed.jpg (deflated 0%)\n",
      "  adding: FAE3_I5.0_F1.9_L100_143135_1_0_1_segment_cropped.png (deflated 4%)\n",
      "  adding: FAE3_I5.0_F1.9_L100_143135_1_0_1_segment_uncropped.png (deflated 49%)\n",
      "  adding: FAE3_I5.0_F1.9_L100_143135_1_0_1_segment_uncropped_processed.png (deflated 18%)\n",
      "  adding: FAE3_I5.0_F1.9_L100_143235_2_0_2_Fluorescence.hdr (deflated 59%)\n",
      "  adding: FAE3_I5.0_F1.9_L100_143235_2_0_2_Fluorescence.raw (deflated 57%)\n",
      "  adding: FAE3_I5.0_F1.9_L100_143235_2_0_2_hyperchannel.csv (deflated 96%)\n",
      "  adding: FAE3_I5.0_F1.9_L100_143235_2_0_2_rgb_cropped.jpg (deflated 0%)\n",
      "  adding: FAE3_I5.0_F1.9_L100_143235_2_0_2_rgb.jpg (deflated 1%)\n",
      "  adding: FAE3_I5.0_F1.9_L100_143235_2_0_2_rgb_processed.jpg (deflated 0%)\n",
      "  adding: FAE3_I5.0_F1.9_L100_143235_2_0_2_segment_cropped.png (deflated 3%)\n",
      "  adding: FAE3_I5.0_F1.9_L100_143235_2_0_2_segment_uncropped.png (deflated 32%)\n",
      "  adding: FAE3_I5.0_F1.9_L100_143235_2_0_2_segment_uncropped_processed.png (deflated 9%)\n",
      "  adding: FAE3_I5.0_F1.9_L100_143334_3_0_3_Fluorescence.hdr (deflated 59%)\n",
      "  adding: FAE3_I5.0_F1.9_L100_143334_3_0_3_Fluorescence.raw (deflated 58%)\n",
      "  adding: FAE3_I5.0_F1.9_L100_143334_3_0_3_hyperchannel.csv (deflated 96%)\n",
      "  adding: FAE3_I5.0_F1.9_L100_143334_3_0_3_rgb_cropped.jpg (deflated 0%)\n",
      "  adding: FAE3_I5.0_F1.9_L100_143334_3_0_3_rgb.jpg (deflated 1%)\n",
      "  adding: FAE3_I5.0_F1.9_L100_143334_3_0_3_rgb_processed.jpg (deflated 0%)\n",
      "  adding: FAE3_I5.0_F1.9_L100_143334_3_0_3_segment_cropped.png (deflated 3%)\n",
      "  adding: FAE3_I5.0_F1.9_L100_143334_3_0_3_segment_uncropped.png (deflated 47%)\n",
      "  adding: FAE3_I5.0_F1.9_L100_143334_3_0_3_segment_uncropped_processed.png (deflated 16%)\n",
      "  adding: FAE3_I5.0_F1.9_L100_143434_4_0_4_Fluorescence.hdr (deflated 59%)\n",
      "  adding: FAE3_I5.0_F1.9_L100_143434_4_0_4_Fluorescence.raw (deflated 58%)\n",
      "  adding: FAE3_I5.0_F1.9_L100_143434_4_0_4_hyperchannel.csv (deflated 96%)\n",
      "  adding: FAE3_I5.0_F1.9_L100_143434_4_0_4_rgb_cropped.jpg (deflated 0%)\n",
      "  adding: FAE3_I5.0_F1.9_L100_143434_4_0_4_rgb.jpg (deflated 1%)\n",
      "  adding: FAE3_I5.0_F1.9_L100_143434_4_0_4_rgb_processed.jpg (deflated 0%)\n",
      "  adding: FAE3_I5.0_F1.9_L100_143434_4_0_4_segment_cropped.png (deflated 4%)\n",
      "  adding: FAE3_I5.0_F1.9_L100_143434_4_0_4_segment_uncropped.png (deflated 44%)\n",
      "  adding: FAE3_I5.0_F1.9_L100_143434_4_0_4_segment_uncropped_processed.png (deflated 16%)\n",
      "  adding: file_list.csv (deflated 94%)\n",
      "  adding: file_list_hyper_channel.csv (deflated 96%)\n",
      "  adding: file_list_part1.csv (deflated 83%)\n",
      "  adding: file_list_part2.csv (deflated 86%)\n",
      "  adding: hdr_backup/ (stored 0%)\n",
      "  adding: hdr_backup/FAE1_I5.0_F1.9_L100_143813_0_0_0_Fluorescence.hdr (deflated 59%)\n",
      "  adding: hdr_backup/FAE1_I5.0_F1.9_L100_143859_1_0_1_Fluorescence.hdr (deflated 59%)\n",
      "  adding: hdr_backup/FAE1_I5.0_F1.9_L100_143959_2_0_2_Fluorescence.hdr (deflated 59%)\n",
      "  adding: hdr_backup/FAE1_I5.0_F1.9_L100_144101_3_0_3_Fluorescence.hdr (deflated 59%)\n",
      "  adding: hdr_backup/FAE1_I5.0_F1.9_L100_144200_4_0_4_Fluorescence.hdr (deflated 59%)\n",
      "  adding: hdr_backup/FAE1_I5.0_F1.9_L100_144259_5_0_5_Fluorescence.hdr (deflated 59%)\n",
      "  adding: hdr_backup/FAE1_I5.0_F1.9_L100_144359_6_0_6_Fluorescence.hdr (deflated 59%)\n",
      "  adding: hdr_backup/FAE1_I5.0_F1.9_L100_144459_7_1_6_Fluorescence.hdr (deflated 59%)\n",
      "  adding: hdr_backup/FAE1_I5.0_F1.9_L100_144550_8_1_5_Fluorescence.hdr (deflated 59%)\n",
      "  adding: hdr_backup/FAE1_I5.0_F1.9_L100_144650_9_1_4_Fluorescence.hdr (deflated 59%)\n",
      "  adding: hdr_backup/FAE1_I5.0_F1.9_L100_144848_11_1_2_Fluorescence.hdr (deflated 59%)\n",
      "  adding: hdr_backup/FAE1_I5.0_F1.9_L100_144948_12_1_1_Fluorescence.hdr (deflated 59%)\n",
      "  adding: hdr_backup/FAE1_I5.0_F1.9_L100_145051_13_1_0_Fluorescence.hdr (deflated 59%)\n",
      "  adding: hdr_backup/FAE1_I5.0_F1.9_L100_145151_14_2_0_Fluorescence.hdr (deflated 59%)\n",
      "  adding: hdr_backup/FAE1_I5.0_F1.9_L100_145243_15_2_1_Fluorescence.hdr (deflated 59%)\n",
      "  adding: hdr_backup/FAE1_I5.0_F1.9_L100_145343_16_2_2_Fluorescence.hdr (deflated 59%)\n",
      "  adding: hdr_backup/FAE1_I5.0_F1.9_L100_145442_17_2_3_Fluorescence.hdr (deflated 59%)\n",
      "  adding: hdr_backup/FAE1_I5.0_F1.9_L100_145544_18_2_4_Fluorescence.hdr (deflated 59%)\n",
      "  adding: hdr_backup/FAE1_I5.0_F1.9_L100_145643_19_2_5_Fluorescence.hdr (deflated 59%)\n",
      "  adding: hdr_backup/FAE1_I5.0_F1.9_L100_145744_20_2_6_Fluorescence.hdr (deflated 59%)\n",
      "  adding: hdr_backup/FAE2_I5.0_F1.9_L100_150028_0_0_0_Fluorescence.hdr (deflated 59%)\n",
      "  adding: hdr_backup/FAE2_I5.0_F1.9_L100_150113_1_0_1_Fluorescence.hdr (deflated 59%)\n",
      "  adding: hdr_backup/FAE2_I5.0_F1.9_L100_150214_2_0_2_Fluorescence.hdr (deflated 59%)\n",
      "  adding: hdr_backup/FAE2_I5.0_F1.9_L100_150313_3_0_3_Fluorescence.hdr (deflated 59%)\n",
      "  adding: hdr_backup/FAE2_I5.0_F1.9_L100_150413_4_0_4_Fluorescence.hdr (deflated 59%)\n",
      "  adding: hdr_backup/FAE2_I5.0_F1.9_L100_150513_5_0_5_Fluorescence.hdr (deflated 59%)\n",
      "  adding: hdr_backup/FAE2_I5.0_F1.9_L100_150613_6_0_6_Fluorescence.hdr (deflated 59%)\n",
      "  adding: hdr_backup/FAE2_I5.0_F1.9_L100_150713_7_1_6_Fluorescence.hdr (deflated 59%)\n",
      "  adding: hdr_backup/FAE2_I5.0_F1.9_L100_150803_8_1_5_Fluorescence.hdr (deflated 59%)\n",
      "  adding: hdr_backup/FAE2_I5.0_F1.9_L100_150902_9_1_4_Fluorescence.hdr (deflated 59%)\n",
      "  adding: hdr_backup/FAE1_I5.0_F1.9_L100_144749_10_1_3_Fluorescence.hdr (deflated 59%)\n",
      "  adding: hdr_backup/FAE1_I5.0_F1.9_L100_152151_0_1_6_Fluorescence.hdr (deflated 59%)\n",
      "  adding: hdr_backup/FAE2_I5.0_F1.9_L100_151002_10_1_3_Fluorescence.hdr (deflated 59%)\n",
      "  adding: hdr_backup/FAE2_I5.0_F1.9_L100_151102_11_1_1_Fluorescence.hdr (deflated 59%)\n",
      "  adding: hdr_backup/FAE2_I5.0_F1.9_L100_151202_12_1_0_Fluorescence.hdr (deflated 59%)\n",
      "  adding: hdr_backup/FAE2_I5.0_F1.9_L100_151302_13_2_0_Fluorescence.hdr (deflated 59%)\n",
      "  adding: hdr_backup/FAE2_I5.0_F1.9_L100_151354_14_2_1_Fluorescence.hdr (deflated 59%)\n",
      "  adding: hdr_backup/FAE2_I5.0_F1.9_L100_151456_15_2_2_Fluorescence.hdr (deflated 59%)\n",
      "  adding: hdr_backup/FAE2_I5.0_F1.9_L100_151556_16_2_3_Fluorescence.hdr (deflated 59%)\n",
      "  adding: hdr_backup/FAE2_I5.0_F1.9_L100_151657_17_2_4_Fluorescence.hdr (deflated 59%)\n",
      "  adding: hdr_backup/FAE2_I5.0_F1.9_L100_151757_18_2_6_Fluorescence.hdr (deflated 59%)\n",
      "  adding: hdr_backup/FAE3_I5.0_F1.9_L100_143049_0_0_0_Fluorescence.hdr (deflated 59%)\n",
      "  adding: hdr_backup/FAE3_I5.0_F1.9_L100_143135_1_0_1_Fluorescence.hdr (deflated 59%)\n",
      "  adding: hdr_backup/FAE3_I5.0_F1.9_L100_143235_2_0_2_Fluorescence.hdr (deflated 59%)\n",
      "  adding: hdr_backup/FAE3_I5.0_F1.9_L100_143334_3_0_3_Fluorescence.hdr (deflated 59%)\n",
      "  adding: hdr_backup/FAE3_I5.0_F1.9_L100_143434_4_0_4_Fluorescence.hdr (deflated 59%)\n",
      "  adding: log_alignment_mode2.txt (deflated 97%)\n",
      "  adding: log_contam.txt (deflated 96%)\n",
      "  adding: log_expand.txt (deflated 90%)\n",
      "  adding: log_gmodetector.txt (deflated 93%)\n",
      "  adding: log_inference.txt (deflated 88%)\n",
      "  adding: output.csv (deflated 87%)\n",
      "  adding: rgb_and_hyper_channel_lists.csv (deflated 96%)\n",
      "  adding: rgb_list.txt (deflated 92%)\n",
      "  adding: samples_pre_labeling.csv (deflated 95%)\n",
      "  adding: stats_with_sums_over_tissues.csv (deflated 87%)\n",
      "  adding: test.csv (deflated 92%)\n"
     ]
    }
   ],
   "source": [
    "cd $timepoint\n",
    "zip -r $data_folder_Compress ./*"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### 5.2. Write email"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 130,
   "metadata": {},
   "outputs": [],
   "source": [
    "duration=$(( SECONDS - start ))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "https://unix.stackexchange.com/questions/53841/how-to-use-a-timer-in-bash"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 131,
   "metadata": {},
   "outputs": [],
   "source": [
    "rm -f \"${gmolabeler_path}/email_to_send.txt\"\n",
    "cp \"${gmolabeler_path}/email_to_send_template.txt\" \"${gmolabeler_path}/email_to_send.txt\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 132,
   "metadata": {},
   "outputs": [],
   "source": [
    "echo \"\" >> \"${gmolabeler_path}/email_to_send.txt\"\n",
    "echo \"Number of samples run: \" >> \"${gmolabeler_path}/email_to_send.txt\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 133,
   "metadata": {},
   "outputs": [],
   "source": [
    "cat \"${data}/test.csv\" | wc -l >> \"${gmolabeler_path}/email_to_send.txt\"\n",
    "echo \"\" >> \"${gmolabeler_path}/email_to_send.txt\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 134,
   "metadata": {},
   "outputs": [],
   "source": [
    "if (( $SECONDS > 3600 )) ; then\n",
    "    let \"hours=SECONDS/3600\"\n",
    "    let \"minutes=(SECONDS%3600)/60\"\n",
    "    let \"seconds=(SECONDS%3600)%60\"\n",
    "    echo \"Completed in $hours hour(s), $minutes minute(s) and $seconds second(s)\" >> \"${gmolabeler_path}/email_to_send.txt\"\n",
    "elif (( $SECONDS > 60 )) ; then\n",
    "    let \"minutes=(SECONDS%3600)/60\"\n",
    "    let \"seconds=(SECONDS%3600)%60\"\n",
    "    echo \"Completed in $minutes minute(s) and $seconds second(s)\" >> \"${gmolabeler_path}/email_to_send.txt\"\n",
    "else\n",
    "    echo \"Completed in $SECONDS seconds\" >> \"${gmolabeler_path}/email_to_send.txt\"\n",
    "fi"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 135,
   "metadata": {},
   "outputs": [],
   "source": [
    "echo \"\" >> \"${gmolabeler_path}/email_to_send.txt\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### 5.3. Send email with results to user"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 136,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/run/media/labgroup/Elements_9/Transformation/FAE_Dev_Genes/wk7/Fluorescence\n"
     ]
    }
   ],
   "source": [
    "pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 137,
   "metadata": {},
   "outputs": [],
   "source": [
    "conda deactivate"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "mail -a $data_folder_Compress -s $data_folder $email < \"${gmolabeler_path}/email_to_send.txt\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "du -sh $data_folder_Compress"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "cat \"${gmolabeler_path}/email_to_send.txt\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### Time analysis ends"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "echo $(date)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
