{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# *GMOnotebook*\n",
    "#### **Notebook template for applying routine hyperspectral/segmentation cross-analysis phenomics workflow over new datasets**"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "###### Notebook template v0.2.03 (Oct 14, 2020)\n",
    "###### v0.2.0 onward applies the Python `GMOdetector` package instead of the deprecated R `GMOdetectoR` package, for much faster runtime\n",
    "###### v0.2.01 has calls to `conda` cleaned up with `conda activate` and `conda deactivate` in the same code blocks to avoid any problems with warnings/errors preventing automatic execution of subsequent blocks\n",
    "###### v0.2.02 is cleaned up using path variables (declared in notebook) instead of hardcoded paths, for portability\n",
    "###### v0.2.03 is more clean to avoid running alignment-processed images through `inference.py` if an analysis if being repeated... also has better instructions"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<img src=\"Figures/WorkflowFlowchart.png\">"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "In this workflow, images taken with the macroPhor Array dual RGB/hyperspectral imaging platform are analyzed by a workflow in which regression quantifies fluorescent signals in hyperspectral images, deep learning segments RGB images into different tissues, and these datasets are cross-referenced to produce statistics on growth of transgenic callus and shoot."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Instructions for use\n",
    "1.  Enter information for the experiment below\n",
    "2. Set <font color=blue>variables</font> for data paths and parameters, as instructed by colored boxes.\n",
    "3. \"Save as\" with filename describing experiment and anything special about this analysis (e.g. T18_OD_TAO_wk7_automation_test_attempt2.ipynb)\n",
    "4. Run notebook from console, by either one of two methods:\n",
    "- In the top bar of JupyterLab, select \"Run\" and \"Run all cells\"\n",
    "- In a terminal, enter the below command with the notebook filename inserted<br>\n",
    "```jupyter nbconvert --to HTML --ExecutePreprocessor.timeout=-1 --allow-errors --execute insert_filename_here```\n",
    "5. Wait for email"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Experiment ID and quick description:"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<div class=\"alert alert-block alert-success\">\n",
    "Provide a short description of the experiment in the below box. This should include unique identifier codes for the experiment, along with a short description of genotypes and treatments studied. The timepoint should also be included. </div>"
   ]
  },
  {
   "cell_type": "raw",
   "metadata": {},
   "source": [
    "GFPA: Cottonwoods transformed with experimental plasmids harboring GFP plasmids that may be used for GWAS of transformation<br>\n",
    "Week 7 timepoint – After 3w callus induction, 4w shoot induction treatment"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Parameters for analysis:"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<div class=\"alert alert-block alert-success\">\n",
    "The below variables must be modified appropriately every time this workflow is run over new images.\n",
    "</div>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### Data location\n",
    "The `data` variable below provides the **complete** path to the folder containing data to be analyzed. This should include all folders and subfolders in which the data of interest is organized by. For the organizational system used for our lab's data, this should follow the format \"/Experiment/Subexperiment/Timepoint/\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(base) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "data_prefix=\"/run/media/labgroup/Elements_7/\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(base) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "data=\"${data_prefix}/GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### Sample information\n",
    "Every experiment has a randomization datasheet, which was used to organize treatment and genotype information for each plate, prepare labels, and randomize plates. The path to this file is provided through the `randomization_datasheets` variable below. This workflow requires this datasheet in order to know which plates have which genotype/treatment. At a later date, we will integrate an ability to read this data directly from labels."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(base) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "randomization_datasheet=\"/home/labgroup/data/GREGSCORNER/GFPA_cottonwoods/GFPA_randomized_list_MNrevised.xlsx\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### Exclusion of missing/contaminated explants from analysis"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Set the `missing_explants` variable to `\"Automatic\"` if using model to automatically detect missing and contaminated explants. Note that this model is only supported for plates with 12 explants. Otherwise, provide an appropriately formatted `.csv` spreadsheet (see example)  of manually scored contamination / missing explant data."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(base) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "missing_explants=\"Automatic\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Enter your email where results will be sent"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(base) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "email=michael.nagle@oregonstate.edu"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<div class=\"alert alert-block alert-warning\">\n",
    "All variables below should be modified only as needed to indicate the fluorescent proteins in samples and the grid layout of explants. </div>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### Provide information pertinent to the fluorophores in the sample"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Below should be all known fluorescent components contained in the sample. This includes each fluorescent protein, as well as a \"noise\" or \"diffraction\" term if applicable. All of these components must exist in the user's spectral library. `GMOdetector` currently comes with a spectral library that includes, by default:\n",
    "- DsRed\n",
    "- ZsYellow\n",
    "- GFP\n",
    "- Chl\n",
    "- ChlA\n",
    "- ChlB\n",
    "- Diffraction"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(base) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "fluorophores=(GFP Chl Diffraction) # An explanation of array variables in bash is here: https://tldp.org/LDP/Bash-Beginners-Guide/html/sect_10_02.html"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The user has the option of limiting loading of hyperspectral data and subsequent regression to a specific range of wavelengths, using the `desired_wavelength_range` array variable. This range should cover all fluorophores provided in the `fluorophores` array variable able. Aside from runtime, there is no disadvantage to including a wider range than is needed."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(base) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "desired_wavelength_range=(500 900)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### To assist user inspection of regression results, false color plots will be produced by `GMOdetector` to show results of regression over whole samples. \n",
    "###### Note: In v0.2.0 of the workflow, these parameters are independent of those used later by `GMOlabeler` to produce by-explant plots including false color plots. These will be made the same in a later update.\n",
    "The `FalseColor_channels` array variable indicates the components to be plotted as red, green and blue, in that order."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(base) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "FalseColor_channels=(Chl GFP Diffraction)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The `FalseColor_caps` array variable indicates an upper limit of signal for each of these component. Any signal at or above these values will appear with maximum brightness; thus, these variables are comparable to exposure on a camera. If caps are too high, not much signal at lower ranges will be seen. If cap for a given component is too low, the false color images will appear overexposed with respect to the component."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(base) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "FalseColor_caps=(200 400 50)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### `GMOlabeler` will use the below parameters to classify individual explants on plates as transgenic or not."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(base) (base) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "reporter=GFP\n",
    "#reporter=DsRed"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Parameters for reporter signal threshold and pixel threshold must be provided by the user. Our grid search yielded several noted below. These were most recently calculated from statistics produced with Python `GMOdetector`."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<img src=\"Figures/GMOlabeler_parameters.png\">"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(base) (base) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "reporter_threshold=38\n",
    "pixel_threshold=3"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### Grid information"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Currently, `GMOlabeler` distinguishes explants by their expected position on a plate. This is supported for plates with 12 or 20 explants, arranged in specific positions on a grid. False positives may result in cases in which one explant grows to such an extent that it intrudes into the grid square meant for an adjacent explant. <br>In the future, an additional neural network will be used to segment individual explants on a per-pixel basis, avoiding the need for crude cropping, and the outputs from the resulting model will be integrated with `GMOlabeler`."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Cropping to grid squares is supported for standard grids with 12 or 20 explants, as indicated by the `grid` variable. An image of each grid square (with a file path provided by the `grid_file` variable) will be superimposed over each image during `GMOlabeler` analysis to allow the user to easily inspect and verify grid positions."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(base) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "grid=12"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<div class=\"alert alert-block alert-danger\">\n",
    "The below variables do not need to be modified during any routine use of the workflow.\n",
    "</div>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "For the following variables, use 0 (False) or 1 (True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(base) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "composite=1"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Set dimensions for plot outputs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(base) (base) (base) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "#width=15\n",
    "width=9\n",
    "height=5"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(base) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "parallel=0"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Paths to workflow components on local machine"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(base) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "cwd=\"/home/labgroup/code/GMOnotebook\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(base) (base) (base) (base) (base) (base) (base) (base) (base) (base) (base) (base) (base) (base) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "#gmodetector_wd=\"/scratch2/NSF_GWAS/notebooks/gmodetector_py/\"\n",
    "gmodetector_wd=\"/home/labgroup/code/gmodetector_py/\"\n",
    "spectral_library_path=\"${gmodetector_wd}spectral_library/\"\n",
    "#output_directory_prefix=\"/scratch2/NSF_GWAS/gmodetector_out/\"\n",
    "#deeplab_path=\"/scratch2/NSF_GWAS/deeplab/\"\n",
    "deeplab_path=\"/home/labgroup/code/deeplab/\"\n",
    "#alignment_path=\"/scratch2/NSF_GWAS/ImageAlignment/\"\n",
    "alignment_path=\"/home/labgroup/code/ImageAlignment/\"\n",
    "#gmolabeler_path=\"/scratch2/NSF_GWAS/GMOlabeler/\"\n",
    "gmolabeler_path=\"/home/labgroup/code/GMOlabeler/\"\n",
    "#contamination_path=\"/scratch2/NSF_GWAS/Contamination\"\n",
    "contamination_path=\"/home/labgroup/code/DenseNet\"\n",
    "data_prefix=\"/home/labgroup/data/\"\n",
    "output_directory_prefix=\"${data_prefix}gmodetector_out/\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(base) (base) (base) (base) (base) (base) (base) (base) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "#grid_file=\"/scratch2/NSF_GWAS/macroPhor_Array/grids/grid20_post_processed.png\"\n",
    "#grid_file=\"/scratch2/NSF_GWAS/macroPhor_Array/grids/grids_left_facing_125208_1_0_1_rgb_processed.jpg\"\n",
    "\n",
    "# For data from before when imager went off-center\n",
    "# grid_file=\"${gmolabeler_path}/grids/grid_12_explants_left-facing_processed.jpg\"\n",
    "\n",
    "# For offcenter images\n",
    "grid_file=\"${alignment_path}/output/12itemoffcenterleftfacinga3_F1.9_I5.0_L80_cyan_172334_0_0_0_rgb_transformed_rgb.jpg\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<div class=\"alert alert-block alert-info\">\n",
    "With all above variables set, please \"Save as...\" with a filename referencing this specific dataset. <br>Finally, deploy the workflow (Step 4 in above instructions).\n",
    "</div>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Automated workflow to be deployed"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "See the below code for a walkthrough of how GMOnotebook works, or view the outputs after running the workflow for help troubleshooting errors in specific steps of analysis."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<div class=\"alert alert-block alert-danger\"> <b>Danger:</b> Do not modify any below code without creating a new version of the template notebook. During routine usage, this workflow should be customized only by modifying variables above, while leaving the below code unmodified. </div>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "These internal variables are set automatically."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "datestamp=$(date +”%Y-%m-%d”)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "data_folder=$(echo $data | cut -d/ -f5-)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "timepoint=\"$(basename -- $data_folder)\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "output_directory_full=\"$output_directory_prefix$data_folder\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "dataset_name=$(echo $subdirectory | sed -e 's/\\//-/g')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### Time analysis begins:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Thu Nov 26 00:24:04 PST 2020\n"
     ]
    }
   ],
   "source": [
    "echo $(date)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 1. Quantification of fluorescent proteins by regression"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The Python package `GMOdetector` is used to quantify fluorescent proteins in each pixel of hyperspectral images via linear regression. Hyperspectral images are regressed over spectra of known components, and pixelwise maps of test-statistics are constructed for each component in the sample. This approach to quantifying components of hyperspectral images is described in-depth in the Methods section from <a href=\"https://link.springer.com/article/10.1007/s40789-019-0252-7\" target=\"_blank\">Böhme, et al. 2019</a>. Code and documentation for `GMOdetector` is on <a href=\"https://github.com/naglemi/GMOdetector_py\" target=\"_blank\">Github</a>."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "cd $gmodetector_wd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "job_list_name=\"$dataset_name.jobs\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "rm -rf $job_list_name"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "for file in $data/*.hdr\n",
    "do\n",
    " if [[ \"$file\" != *'hroma'* ]]; then\n",
    "  echo \"python wrappers/analyze_sample.py \\\n",
    "--file_path $file \\\n",
    "--fluorophores ${fluorophores[*]} \\\n",
    "--min_desired_wavelength ${desired_wavelength_range[0]} \\\n",
    "--max_desired_wavelength ${desired_wavelength_range[1]} \\\n",
    "--red_channel ${FalseColor_channels[0]} \\\n",
    "--green_channel ${FalseColor_channels[1]} \\\n",
    "--blue_channel ${FalseColor_channels[2]} \\\n",
    "--red_cap ${FalseColor_caps[0]} \\\n",
    "--green_cap ${FalseColor_caps[1]} \\\n",
    "--blue_cap ${FalseColor_caps[2]} \\\n",
    "--spectral_library_path \"$spectral_library_path\" \\\n",
    "--output_dir $output_directory_full\" >> $job_list_name\n",
    " fi\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(test-environment) (test-environment) (test-environment) (test-environment) During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA1_I5.0_F1.9_L100_132422_0_0_0_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA1_I5.0_F1.9_L100_132422_0_0_0.hdr in 85.00825623099809s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA1_I5.0_F1.9_L100_132505_1_0_1_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA1_I5.0_F1.9_L100_132505_1_0_1.hdr in 36.133327863994054s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA1_I5.0_F1.9_L100_132600_2_0_2_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA1_I5.0_F1.9_L100_132600_2_0_2.hdr in 23.846981014998164s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA1_I5.0_F1.9_L100_132750_4_0_4_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA1_I5.0_F1.9_L100_132750_4_0_4.hdr in 20.442786854007863s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA1_I5.0_F1.9_L100_132846_5_0_5_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA1_I5.0_F1.9_L100_132846_5_0_5.hdr in 19.544705205000355s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA1_I5.0_F1.9_L100_132940_6_0_6_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA1_I5.0_F1.9_L100_132940_6_0_6.hdr in 20.1419105919922s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA1_I5.0_F1.9_L100_133036_7_1_6_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA1_I5.0_F1.9_L100_133036_7_1_6.hdr in 21.226304248004453s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA1_I5.0_F1.9_L100_133121_8_1_5_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA1_I5.0_F1.9_L100_133121_8_1_5.hdr in 21.241712692994042s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA1_I5.0_F1.9_L100_133216_9_1_4_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA1_I5.0_F1.9_L100_133216_9_1_4.hdr in 21.98065230299835s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA1_I5.0_F1.9_L100_133406_11_1_2_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA1_I5.0_F1.9_L100_133406_11_1_2.hdr in 21.661330464994535s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA1_I5.0_F1.9_L100_133502_12_1_0_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA1_I5.0_F1.9_L100_133502_12_1_0.hdr in 20.015081443008967s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA1_I5.0_F1.9_L100_133559_13_2_0_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA1_I5.0_F1.9_L100_133559_13_2_0.hdr in 22.81314807200397s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA1_I5.0_F1.9_L100_133643_14_2_1_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA1_I5.0_F1.9_L100_133643_14_2_1.hdr in 21.294747193009243s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA1_I5.0_F1.9_L100_133738_15_2_2_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA1_I5.0_F1.9_L100_133738_15_2_2.hdr in 20.17567752499599s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA1_I5.0_F1.9_L100_133833_16_2_3_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA1_I5.0_F1.9_L100_133833_16_2_3.hdr in 20.30763456999557s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA1_I5.0_F1.9_L100_133928_17_2_5_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA1_I5.0_F1.9_L100_133928_17_2_5.hdr in 20.65289830400434s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA1_I5.0_F1.9_L100_134025_18_2_6_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA1_I5.0_F1.9_L100_134025_18_2_6.hdr in 21.768745930006844s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA1_I5.0_F1.9_L100_134552_0_0_3_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA1_I5.0_F1.9_L100_134552_0_0_3.hdr in 19.661084917999688s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA1_I5.0_F1.9_L100_134702_1_1_3_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA1_I5.0_F1.9_L100_134702_1_1_3.hdr in 19.619496336992597s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA2_I5.0_F1.9_L100_135339_0_0_0_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA2_I5.0_F1.9_L100_135339_0_0_0.hdr in 21.569467434994294s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA2_I5.0_F1.9_L100_135420_1_0_1_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA2_I5.0_F1.9_L100_135420_1_0_1.hdr in 19.93212291400414s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA2_I5.0_F1.9_L100_135514_2_0_2_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA2_I5.0_F1.9_L100_135514_2_0_2.hdr in 21.888597587007098s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA2_I5.0_F1.9_L100_135609_3_0_3_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA2_I5.0_F1.9_L100_135609_3_0_3.hdr in 19.71045557699108s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA2_I5.0_F1.9_L100_135704_4_0_4_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA2_I5.0_F1.9_L100_135704_4_0_4.hdr in 20.55907762100105s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA2_I5.0_F1.9_L100_135758_5_0_5_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA2_I5.0_F1.9_L100_135758_5_0_5.hdr in 21.45841382900835s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA2_I5.0_F1.9_L100_135852_6_0_6_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA2_I5.0_F1.9_L100_135852_6_0_6.hdr in 21.244131512998138s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA2_I5.0_F1.9_L100_135947_7_1_6_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA2_I5.0_F1.9_L100_135947_7_1_6.hdr in 19.861741230997723s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA2_I5.0_F1.9_L100_140033_8_1_5_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA2_I5.0_F1.9_L100_140033_8_1_5.hdr in 23.694073612990906s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA2_I5.0_F1.9_L100_140127_9_1_4_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA2_I5.0_F1.9_L100_140127_9_1_4.hdr in 21.247960399996373s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA2_I5.0_F1.9_L100_140221_10_1_3_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA2_I5.0_F1.9_L100_140221_10_1_3.hdr in 20.09097833800479s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA2_I5.0_F1.9_L100_140315_11_1_2_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA2_I5.0_F1.9_L100_140315_11_1_2.hdr in 21.30688984801236s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA2_I5.0_F1.9_L100_140410_12_1_1_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA2_I5.0_F1.9_L100_140410_12_1_1.hdr in 20.90677070300444s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA2_I5.0_F1.9_L100_140506_13_1_0_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA2_I5.0_F1.9_L100_140506_13_1_0.hdr in 20.603795759001514s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA2_I5.0_F1.9_L100_140601_14_2_0_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA2_I5.0_F1.9_L100_140601_14_2_0.hdr in 20.613186109010712s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA2_I5.0_F1.9_L100_140647_15_2_1_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA2_I5.0_F1.9_L100_140647_15_2_1.hdr in 21.362285007999162s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA2_I5.0_F1.9_L100_140742_16_2_2_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA2_I5.0_F1.9_L100_140742_16_2_2.hdr in 19.98984635200759s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA2_I5.0_F1.9_L100_140838_17_2_3_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA2_I5.0_F1.9_L100_140838_17_2_3.hdr in 19.571870348998345s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA2_I5.0_F1.9_L100_140933_18_2_4_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA2_I5.0_F1.9_L100_140933_18_2_4.hdr in 21.375812622995s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA2_I5.0_F1.9_L100_141028_19_2_5_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA2_I5.0_F1.9_L100_141028_19_2_5.hdr in 19.675318472989602s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA2_I5.0_F1.9_L100_141123_20_2_6_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA2_I5.0_F1.9_L100_141123_20_2_6.hdr in 22.413706027000444s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA3_I5.0_F1.9_L100_141906_0_0_0_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA3_I5.0_F1.9_L100_141906_0_0_0.hdr in 20.41312831999676s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA3_I5.0_F1.9_L100_141948_1_0_1_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA3_I5.0_F1.9_L100_141948_1_0_1.hdr in 19.876829770990298s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA3_I5.0_F1.9_L100_142043_2_0_2_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA3_I5.0_F1.9_L100_142043_2_0_2.hdr in 19.51012971000455s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA3_I5.0_F1.9_L100_142137_3_0_3_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA3_I5.0_F1.9_L100_142137_3_0_3.hdr in 20.291467596005532s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA3_I5.0_F1.9_L100_142237_4_0_4_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA3_I5.0_F1.9_L100_142237_4_0_4.hdr in 22.243702064995887s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA3_I5.0_F1.9_L100_142332_5_0_5_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA3_I5.0_F1.9_L100_142332_5_0_5.hdr in 21.631849486991996s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA3_I5.0_F1.9_L100_142428_6_0_6_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA3_I5.0_F1.9_L100_142428_6_0_6.hdr in 19.941054278999218s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA3_I5.0_F1.9_L100_142523_7_1_6_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA3_I5.0_F1.9_L100_142523_7_1_6.hdr in 22.174053210008424s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA3_I5.0_F1.9_L100_142608_8_1_5_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA3_I5.0_F1.9_L100_142608_8_1_5.hdr in 20.89576382699306s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA3_I5.0_F1.9_L100_142704_9_1_4_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA3_I5.0_F1.9_L100_142704_9_1_4.hdr in 21.61109536899312s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA3_I5.0_F1.9_L100_142759_10_1_3_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA3_I5.0_F1.9_L100_142759_10_1_3.hdr in 19.411715027003083s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA3_I5.0_F1.9_L100_142854_11_1_2_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA3_I5.0_F1.9_L100_142854_11_1_2.hdr in 19.630314620007994s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA3_I5.0_F1.9_L100_142948_12_1_1_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA3_I5.0_F1.9_L100_142948_12_1_1.hdr in 19.90283911601s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA3_I5.0_F1.9_L100_143043_13_1_0_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA3_I5.0_F1.9_L100_143043_13_1_0.hdr in 19.77368342000409s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA3_I5.0_F1.9_L100_143138_14_2_1_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA3_I5.0_F1.9_L100_143138_14_2_1.hdr in 20.01140182299423s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA3_I5.0_F1.9_L100_143226_15_2_2_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA3_I5.0_F1.9_L100_143226_15_2_2.hdr in 21.11960689499392s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA3_I5.0_F1.9_L100_143320_16_2_4_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA3_I5.0_F1.9_L100_143320_16_2_4.hdr in 21.636969019004027s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA3_I5.0_F1.9_L100_143417_17_2_5_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA3_I5.0_F1.9_L100_143417_17_2_5.hdr in 19.82091994599614s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA3_I5.0_F1.9_L100_143513_18_2_6_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA3_I5.0_F1.9_L100_143513_18_2_6.hdr in 20.12492955599737s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA4_I5.0_F1.9_L100_145031_0_0_0_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA4_I5.0_F1.9_L100_145031_0_0_0.hdr in 21.178948136002873s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA4_I5.0_F1.9_L100_145112_1_0_1_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA4_I5.0_F1.9_L100_145112_1_0_1.hdr in 21.276007223001216s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA4_I5.0_F1.9_L100_145206_2_0_3_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA4_I5.0_F1.9_L100_145206_2_0_3.hdr in 20.822115543996915s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA4_I5.0_F1.9_L100_145304_3_0_4_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA4_I5.0_F1.9_L100_145304_3_0_4.hdr in 22.01043310399109s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA4_I5.0_F1.9_L100_145358_4_0_5_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA4_I5.0_F1.9_L100_145358_4_0_5.hdr in 19.972360316998675s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA4_I5.0_F1.9_L100_145455_5_1_5_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA4_I5.0_F1.9_L100_145455_5_1_5.hdr in 20.45924804499373s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA4_I5.0_F1.9_L100_145543_6_1_4_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA4_I5.0_F1.9_L100_145543_6_1_4.hdr in 20.908380381006282s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA4_I5.0_F1.9_L100_145638_7_1_3_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA4_I5.0_F1.9_L100_145638_7_1_3.hdr in 21.60681144701084s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA4_I5.0_F1.9_L100_145732_8_1_2_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA4_I5.0_F1.9_L100_145732_8_1_2.hdr in 20.4666609029955s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA4_I5.0_F1.9_L100_145827_9_1_1_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA4_I5.0_F1.9_L100_145827_9_1_1.hdr in 19.55941612299648s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA4_I5.0_F1.9_L100_145922_10_1_0_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA4_I5.0_F1.9_L100_145922_10_1_0.hdr in 20.409724693003227s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA4_I5.0_F1.9_L100_150017_11_2_0_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA4_I5.0_F1.9_L100_150017_11_2_0.hdr in 20.050299272013945s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA4_I5.0_F1.9_L100_150102_12_2_1_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA4_I5.0_F1.9_L100_150102_12_2_1.hdr in 19.889506683000945s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA4_I5.0_F1.9_L100_150156_13_2_2_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA4_I5.0_F1.9_L100_150156_13_2_2.hdr in 21.7396455709968s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA4_I5.0_F1.9_L100_150253_14_2_3_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA4_I5.0_F1.9_L100_150253_14_2_3.hdr in 20.747755696007516s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA4_I5.0_F1.9_L100_150347_15_2_4_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA4_I5.0_F1.9_L100_150347_15_2_4.hdr in 20.432013932004338s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA4_I5.0_F1.9_L100_150435_16_2_6_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA4_I5.0_F1.9_L100_150435_16_2_6.hdr in 20.636712863997673s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA4_I5.0_F1.9_L100_151648_0_0_2_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA4_I5.0_F1.9_L100_151648_0_0_2.hdr in 21.09747069599689s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA4_I5.0_F1.9_L100_151748_1_1_6_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA4_I5.0_F1.9_L100_151748_1_1_6.hdr in 21.541988582001068s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA4_I5.0_F1.9_L100_151903_2_2_5_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA4_I5.0_F1.9_L100_151903_2_2_5.hdr in 20.759532529002172s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA5_I5.0_F1.9_L100_185427_0_0_1_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA5_I5.0_F1.9_L100_185427_0_0_1.hdr in 21.40941853300319s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA5_I5.0_F1.9_L100_185518_1_0_3_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA5_I5.0_F1.9_L100_185518_1_0_3.hdr in 20.714017550999415s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA5_I5.0_F1.9_L100_185615_2_0_4_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA5_I5.0_F1.9_L100_185615_2_0_4.hdr in 22.321251126995776s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA5_I5.0_F1.9_L100_185710_3_0_5_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA5_I5.0_F1.9_L100_185710_3_0_5.hdr in 20.7155113589979s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA5_I5.0_F1.9_L100_185805_4_0_6_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA5_I5.0_F1.9_L100_185805_4_0_6.hdr in 21.58377457098686s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA5_I5.0_F1.9_L100_185901_5_1_4_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA5_I5.0_F1.9_L100_185901_5_1_4.hdr in 20.715705344991875s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA5_I5.0_F1.9_L100_185956_6_1_3_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA5_I5.0_F1.9_L100_185956_6_1_3.hdr in 21.3254730799963s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA5_I5.0_F1.9_L100_190630_0_1_2_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA5_I5.0_F1.9_L100_190630_0_1_2.hdr in 23.1624109330005s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA5_I5.0_F1.9_L100_190729_1_1_1_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA5_I5.0_F1.9_L100_190729_1_1_1.hdr in 23.745735156990122s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "During startup - Warning messages:\n",
      "1: package ‘methods’ was built under R version 3.6.1 \n",
      "2: package ‘datasets’ was built under R version 3.6.1 \n",
      "3: package ‘utils’ was built under R version 3.6.1 \n",
      "4: package ‘grDevices’ was built under R version 3.6.1 \n",
      "5: package ‘graphics’ was built under R version 3.6.1 \n",
      "6: package ‘stats’ was built under R version 3.6.1 \n",
      "Running GMOdetector version 0.0.855\n",
      "wrappers/analyze_sample.py:188: UserWarning: Sample is not being normalized with a chroma standard.\n",
      "  warnings.warn('Sample is not being normalized with a chroma standard.')\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "load mode isload_full_then_crop\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/spectral/io/envi.py:175: UserWarning: Parameters with non-lowercase names encountered and converted to lowercase. To retain source file parameter name capitalization, set spectral.settings.envi_support_nonlowercase_params to True.\n",
      "  warnings.warn(msg)\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:30: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  array_in_coordinate = pd.DataFrame(np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i])),\n",
      "/home/labgroup/anaconda3/envs/test-environment/lib/python3.8/site-packages/gmodetector_py/weight_array.py:33: FutureWarning: arrays to stack must be passed as a \"sequence\" type such as list or tuple. Support for non-sequence iterables such as generators is deprecated as of NumPy 1.16 and will raise an error in the future.\n",
      "  matrix_slice_in_triplet = np.column_stack(ar.ravel() for ar in (I, J, self.weights[:, :, i]))\n",
      "Saving to /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/GFPA5_I5.0_F1.9_L100_190823_2_1_0_weights.hdf with key of weights\n",
      "\n",
      "Finished running sample /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//GFPA5_I5.0_F1.9_L100_190823_2_1_0.hdr in 19.613419859000714s\n",
      "\n",
      "R[write to console]: Warning message:\n",
      "\n",
      "R[write to console]: package ‘tools’ was built under R version 3.6.1 \n",
      "\n",
      "(test-environment) (test-environment) (base) "
     ]
    }
   ],
   "source": [
    "conda activate test-environment\n",
    "\n",
    "if [ $parallel -eq 1 ]\n",
    "then\n",
    "    parallel -a $job_list_name\n",
    "fi\n",
    "\n",
    "if [ $parallel -eq 0 ]\n",
    "then\n",
    "    bash $job_list_name\n",
    "fi\n",
    "\n",
    "conda deactivate\n",
    "conda deactivate"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### Time regression completes:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Fri Nov 27 00:00:51 PST 2020\n"
     ]
    }
   ],
   "source": [
    "echo $(date)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 2. Neural networks to segment tissues, classify missing/contaminated explants"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### 2.1. Semantic segmentation of tissues"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Images are segmented into specific plant tissues by a deep neural network of the state-of-the-art Deeplab v3 architecture <a href=\"https://arxiv.org/abs/1706.05587\" target=\"_blank\">Liang-Chieh et al., 2017</a>. The model has been trained using training sets generated with our annotation GUI Intelligent DEep Annotator for Segmentation (IDEAS, available on <a href=\"https://bitbucket.org/JialinYuan/image-annotator/src/master/\" target=\"_blank\">Bitbucket</a>, publication pending). Our branch of the Deeplab v3 repo, including a Jupyter walkthrough for training, can be found on Github."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Training is completed upstream of this notebook, which only entails analysis of test data using the latest model."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<img src=\"Figures/downsized/segmentation_composite2.png\">"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Figure: This example image was taken from an experiment on the effects of different CIMs on cottonwood regeneration. This composite image illustrates that for every sample, tissues are segmented into stem (red), callus (blue) and shoot (green). These composite images, useful for manual inspection of results, are produced when the 'composite' option is on."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### 2.1.1. Pre-processing\n",
    "##### 2.1.1.1. Crop to remove labels, and resize images to 900x600"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This script resizes images to 900x900 and then crops away top and bottom 150 pixels for a final image size of 900x600."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "cd ${cwd}/intermediates/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(base) (base) "
     ]
    }
   ],
   "source": [
    "conda activate base\n",
    "python crop.py $data\n",
    "conda deactivate"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### 2.1.1.2. prepare test.csv file for inference"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Make a list of all our image files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [],
   "source": [
    "cd $data\n",
    "ls -d $PWD/* $data | grep -i \"rgb_cropped.jpg\" > test.csv"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Remove the chroma standard from list of RGB image data to be segmented"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [],
   "source": [
    "sed -i '/hroma/d' \"${data}/test.csv\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### 2.1.2. Inference"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The trained model is deployed to perform semantic segmentation of experimental images. A list of RGB images to be segmented by the trained model is passed through the --image-list option. For each of these images, we will obtain an output mask (.png) of labeled tissues"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Dependencies include `opencv`, `scipy`, `yaml` and `tensorflow` (version 1.14)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(deeplab) /home/labgroup/anaconda3/envs/deeplab/lib/python3.7/site-packages/tensorflow/python/framework/dtypes.py:516: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.\n",
      "  _np_qint8 = np.dtype([(\"qint8\", np.int8, 1)])\n",
      "/home/labgroup/anaconda3/envs/deeplab/lib/python3.7/site-packages/tensorflow/python/framework/dtypes.py:517: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.\n",
      "  _np_quint8 = np.dtype([(\"quint8\", np.uint8, 1)])\n",
      "/home/labgroup/anaconda3/envs/deeplab/lib/python3.7/site-packages/tensorflow/python/framework/dtypes.py:518: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.\n",
      "  _np_qint16 = np.dtype([(\"qint16\", np.int16, 1)])\n",
      "/home/labgroup/anaconda3/envs/deeplab/lib/python3.7/site-packages/tensorflow/python/framework/dtypes.py:519: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.\n",
      "  _np_quint16 = np.dtype([(\"quint16\", np.uint16, 1)])\n",
      "/home/labgroup/anaconda3/envs/deeplab/lib/python3.7/site-packages/tensorflow/python/framework/dtypes.py:520: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.\n",
      "  _np_qint32 = np.dtype([(\"qint32\", np.int32, 1)])\n",
      "/home/labgroup/anaconda3/envs/deeplab/lib/python3.7/site-packages/tensorflow/python/framework/dtypes.py:525: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.\n",
      "  np_resource = np.dtype([(\"resource\", np.ubyte, 1)])\n",
      "/home/labgroup/anaconda3/envs/deeplab/lib/python3.7/site-packages/tensorboard/compat/tensorflow_stub/dtypes.py:541: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.\n",
      "  _np_qint8 = np.dtype([(\"qint8\", np.int8, 1)])\n",
      "/home/labgroup/anaconda3/envs/deeplab/lib/python3.7/site-packages/tensorboard/compat/tensorflow_stub/dtypes.py:542: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.\n",
      "  _np_quint8 = np.dtype([(\"quint8\", np.uint8, 1)])\n",
      "/home/labgroup/anaconda3/envs/deeplab/lib/python3.7/site-packages/tensorboard/compat/tensorflow_stub/dtypes.py:543: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.\n",
      "  _np_qint16 = np.dtype([(\"qint16\", np.int16, 1)])\n",
      "/home/labgroup/anaconda3/envs/deeplab/lib/python3.7/site-packages/tensorboard/compat/tensorflow_stub/dtypes.py:544: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.\n",
      "  _np_quint16 = np.dtype([(\"quint16\", np.uint16, 1)])\n",
      "/home/labgroup/anaconda3/envs/deeplab/lib/python3.7/site-packages/tensorboard/compat/tensorflow_stub/dtypes.py:545: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.\n",
      "  _np_qint32 = np.dtype([(\"qint32\", np.int32, 1)])\n",
      "/home/labgroup/anaconda3/envs/deeplab/lib/python3.7/site-packages/tensorboard/compat/tensorflow_stub/dtypes.py:550: FutureWarning: Passing (type, 1) or '1type' as a synonym of type is deprecated; in a future version of numpy, it will be understood as (type, (1,)) / '(1,)type'.\n",
      "  np_resource = np.dtype([(\"resource\", np.ubyte, 1)])\n",
      "WARNING:tensorflow:From inference.py:34: The name tf.gfile.GFile is deprecated. Please use tf.io.gfile.GFile instead.\n",
      "\n",
      "WARNING:tensorflow:From inference.py:35: The name tf.GraphDef is deprecated. Please use tf.compat.v1.GraphDef instead.\n",
      "\n",
      "WARNING:tensorflow:From inference.py:41: The name tf.Session is deprecated. Please use tf.compat.v1.Session instead.\n",
      "\n",
      "2020-11-27 02:37:42.343224: I tensorflow/core/platform/cpu_feature_guard.cc:142] Your CPU supports instructions that this TensorFlow binary was not compiled to use: SSE4.1 SSE4.2 AVX\n",
      "2020-11-27 02:37:42.558163: I tensorflow/core/platform/profile_utils/cpu_utils.cc:94] CPU Frequency: 2494555000 Hz\n",
      "2020-11-27 02:37:42.558971: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x55d5451e9860 executing computations on platform Host. Devices:\n",
      "2020-11-27 02:37:42.559054: I tensorflow/compiler/xla/service/service.cc:175]   StreamExecutor device (0): <undefined>, <undefined>\n",
      "Running inference for  89  images.\n",
      "Running deeplab on image: 1\n",
      "2020-11-27 02:37:48.493370: W tensorflow/compiler/jit/mark_for_compilation_pass.cc:1412] (One-time warning): Not using XLA:CPU for cluster because envvar TF_XLA_FLAGS=--tf_xla_cpu_global_jit was not set.  If you want XLA:CPU, either set that envvar, or use experimental_jit_scope to enable XLA:CPU.  To confirm that XLA is active, pass --vmodule=xla_compilation_cache=1 (as a proper command-line flag, not via TF_XLA_FLAGS) or set the envvar XLA_FLAGS=--xla_hlo_profile.\n",
      "Running deeplab on image: 2\n",
      "Running deeplab on image: 3\n",
      "Running deeplab on image: 4\n",
      "Running deeplab on image: 5\n",
      "Running deeplab on image: 6\n",
      "Running deeplab on image: 7\n",
      "Running deeplab on image: 8\n",
      "Running deeplab on image: 9\n",
      "Running deeplab on image: 10\n",
      "Running deeplab on image: 11\n",
      "Running deeplab on image: 12\n",
      "Running deeplab on image: 13\n",
      "Running deeplab on image: 14\n",
      "Running deeplab on image: 15\n",
      "Running deeplab on image: 16\n",
      "Running deeplab on image: 17\n",
      "Running deeplab on image: 18\n",
      "Running deeplab on image: 19\n",
      "Running deeplab on image: 20\n",
      "Running deeplab on image: 21\n",
      "Running deeplab on image: 22\n",
      "Running deeplab on image: 23\n",
      "Running deeplab on image: 24\n",
      "Running deeplab on image: 25\n",
      "Running deeplab on image: 26\n",
      "Running deeplab on image: 27\n",
      "Running deeplab on image: 28\n",
      "Running deeplab on image: 29\n",
      "Running deeplab on image: 30\n",
      "Running deeplab on image: 31\n",
      "Running deeplab on image: 32\n",
      "Running deeplab on image: 33\n",
      "Running deeplab on image: 34\n",
      "Running deeplab on image: 35\n",
      "Running deeplab on image: 36\n",
      "Running deeplab on image: 37\n",
      "Running deeplab on image: 38\n",
      "Running deeplab on image: 39\n",
      "Running deeplab on image: 40\n",
      "Running deeplab on image: 41\n",
      "Running deeplab on image: 42\n",
      "Running deeplab on image: 43\n",
      "Running deeplab on image: 44\n",
      "Running deeplab on image: 45\n",
      "Running deeplab on image: 46\n",
      "Running deeplab on image: 47\n",
      "Running deeplab on image: 48\n",
      "Running deeplab on image: 49\n",
      "Running deeplab on image: 50\n",
      "Running deeplab on image: 51\n",
      "Running deeplab on image: 52\n",
      "Running deeplab on image: 53\n",
      "Running deeplab on image: 54\n",
      "Running deeplab on image: 55\n",
      "Running deeplab on image: 56\n",
      "Running deeplab on image: 57\n",
      "Running deeplab on image: 58\n",
      "Running deeplab on image: 59\n",
      "Running deeplab on image: 60\n",
      "Running deeplab on image: 61\n",
      "Running deeplab on image: 62\n",
      "Running deeplab on image: 63\n",
      "Running deeplab on image: 64\n",
      "Running deeplab on image: 65\n",
      "Running deeplab on image: 66\n",
      "Running deeplab on image: 67\n",
      "Running deeplab on image: 68\n",
      "Running deeplab on image: 69\n",
      "Running deeplab on image: 70\n",
      "Running deeplab on image: 71\n",
      "Running deeplab on image: 72\n",
      "Running deeplab on image: 73\n",
      "Running deeplab on image: 74\n",
      "Running deeplab on image: 75\n",
      "Running deeplab on image: 76\n",
      "Running deeplab on image: 77\n",
      "Running deeplab on image: 78\n",
      "Running deeplab on image: 79\n",
      "Running deeplab on image: 80\n",
      "Running deeplab on image: 81\n",
      "Running deeplab on image: 82\n",
      "Running deeplab on image: 83\n",
      "Running deeplab on image: 84\n",
      "Running deeplab on image: 85\n",
      "Running deeplab on image: 86\n",
      "Running deeplab on image: 87\n",
      "Running deeplab on image: 88\n",
      "Running deeplab on image: 89\n",
      "(deeplab) "
     ]
    }
   ],
   "source": [
    "cd $deeplab_path\n",
    "conda activate deeplab\n",
    "python inference.py --image-list \"${data}/test.csv\"\n",
    "conda deactivate\n",
    "conda deactivate"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### 2.3. Post-processing"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Name outputs to reflect that they are segmentation results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [],
   "source": [
    "cd $data\n",
    "for file in *_rgb_cropped.png; do mv -f \"$file\" \"${file%_rgb_cropped.png}_segment_cropped.png\"; done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Re-expand segment outputs to same size as original RGB files"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Dependencies include `scikit-image`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(alignment) (alignment) Working in directory/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/\n",
      "Number of files: 89\n",
      "Reading file GFPA1_I5.0_F1.9_L100_132600_2_0_2_segment_cropped.png\n",
      "Writing file GFPA1_I5.0_F1.9_L100_132600_2_0_2_segment_uncropped.png\n",
      "Reading file GFPA1_I5.0_F1.9_L100_132750_4_0_4_segment_cropped.png\n",
      "Writing file GFPA1_I5.0_F1.9_L100_132750_4_0_4_segment_uncropped.png\n",
      "Reading file GFPA1_I5.0_F1.9_L100_133036_7_1_6_segment_cropped.png\n",
      "Writing file GFPA1_I5.0_F1.9_L100_133036_7_1_6_segment_uncropped.png\n",
      "Reading file GFPA1_I5.0_F1.9_L100_133121_8_1_5_segment_cropped.png\n",
      "Writing file GFPA1_I5.0_F1.9_L100_133121_8_1_5_segment_uncropped.png\n",
      "Reading file GFPA1_I5.0_F1.9_L100_133216_9_1_4_segment_cropped.png\n",
      "Writing file GFPA1_I5.0_F1.9_L100_133216_9_1_4_segment_uncropped.png\n",
      "Reading file GFPA1_I5.0_F1.9_L100_133559_13_2_0_segment_cropped.png\n",
      "Writing file GFPA1_I5.0_F1.9_L100_133559_13_2_0_segment_uncropped.png\n",
      "Reading file GFPA1_I5.0_F1.9_L100_133643_14_2_1_segment_cropped.png\n",
      "Writing file GFPA1_I5.0_F1.9_L100_133643_14_2_1_segment_uncropped.png\n",
      "Reading file GFPA1_I5.0_F1.9_L100_133928_17_2_5_segment_cropped.png\n",
      "Writing file GFPA1_I5.0_F1.9_L100_133928_17_2_5_segment_uncropped.png\n",
      "Reading file GFPA1_I5.0_F1.9_L100_134025_18_2_6_segment_cropped.png\n",
      "Writing file GFPA1_I5.0_F1.9_L100_134025_18_2_6_segment_uncropped.png\n",
      "Reading file GFPA2_I5.0_F1.9_L100_135339_0_0_0_segment_cropped.png\n",
      "Writing file GFPA2_I5.0_F1.9_L100_135339_0_0_0_segment_uncropped.png\n",
      "Reading file GFPA2_I5.0_F1.9_L100_135420_1_0_1_segment_cropped.png\n",
      "Writing file GFPA2_I5.0_F1.9_L100_135420_1_0_1_segment_uncropped.png\n",
      "Reading file GFPA2_I5.0_F1.9_L100_135514_2_0_2_segment_cropped.png\n",
      "Writing file GFPA2_I5.0_F1.9_L100_135514_2_0_2_segment_uncropped.png\n",
      "Reading file GFPA2_I5.0_F1.9_L100_135609_3_0_3_segment_cropped.png\n",
      "Writing file GFPA2_I5.0_F1.9_L100_135609_3_0_3_segment_uncropped.png\n",
      "Reading file GFPA2_I5.0_F1.9_L100_135852_6_0_6_segment_cropped.png\n",
      "Writing file GFPA2_I5.0_F1.9_L100_135852_6_0_6_segment_uncropped.png\n",
      "Reading file GFPA2_I5.0_F1.9_L100_135947_7_1_6_segment_cropped.png\n",
      "Writing file GFPA2_I5.0_F1.9_L100_135947_7_1_6_segment_uncropped.png\n",
      "Reading file GFPA2_I5.0_F1.9_L100_140221_10_1_3_segment_cropped.png\n",
      "Writing file GFPA2_I5.0_F1.9_L100_140221_10_1_3_segment_uncropped.png\n",
      "Reading file GFPA2_I5.0_F1.9_L100_140315_11_1_2_segment_cropped.png\n",
      "Writing file GFPA2_I5.0_F1.9_L100_140315_11_1_2_segment_uncropped.png\n",
      "Reading file GFPA2_I5.0_F1.9_L100_140410_12_1_1_segment_cropped.png\n",
      "Writing file GFPA2_I5.0_F1.9_L100_140410_12_1_1_segment_uncropped.png\n",
      "Reading file GFPA2_I5.0_F1.9_L100_140647_15_2_1_segment_cropped.png\n",
      "Writing file GFPA2_I5.0_F1.9_L100_140647_15_2_1_segment_uncropped.png\n",
      "Reading file GFPA2_I5.0_F1.9_L100_140742_16_2_2_segment_cropped.png\n",
      "Writing file GFPA2_I5.0_F1.9_L100_140742_16_2_2_segment_uncropped.png\n",
      "Reading file GFPA2_I5.0_F1.9_L100_141028_19_2_5_segment_cropped.png\n",
      "Writing file GFPA2_I5.0_F1.9_L100_141028_19_2_5_segment_uncropped.png\n",
      "Reading file GFPA2_I5.0_F1.9_L100_141123_20_2_6_segment_cropped.png\n",
      "Writing file GFPA2_I5.0_F1.9_L100_141123_20_2_6_segment_uncropped.png\n",
      "Reading file GFPA3_I5.0_F1.9_L100_141906_0_0_0_segment_cropped.png\n",
      "Writing file GFPA3_I5.0_F1.9_L100_141906_0_0_0_segment_uncropped.png\n",
      "Reading file GFPA3_I5.0_F1.9_L100_141948_1_0_1_segment_cropped.png\n",
      "Writing file GFPA3_I5.0_F1.9_L100_141948_1_0_1_segment_uncropped.png\n",
      "Reading file GFPA3_I5.0_F1.9_L100_142237_4_0_4_segment_cropped.png\n",
      "Writing file GFPA3_I5.0_F1.9_L100_142237_4_0_4_segment_uncropped.png\n",
      "Reading file GFPA3_I5.0_F1.9_L100_142332_5_0_5_segment_cropped.png\n",
      "Writing file GFPA3_I5.0_F1.9_L100_142332_5_0_5_segment_uncropped.png\n",
      "Reading file GFPA3_I5.0_F1.9_L100_142608_8_1_5_segment_cropped.png\n",
      "Writing file GFPA3_I5.0_F1.9_L100_142608_8_1_5_segment_uncropped.png\n",
      "Reading file GFPA3_I5.0_F1.9_L100_142704_9_1_4_segment_cropped.png\n",
      "Writing file GFPA3_I5.0_F1.9_L100_142704_9_1_4_segment_uncropped.png\n",
      "Reading file GFPA3_I5.0_F1.9_L100_142948_12_1_1_segment_cropped.png\n",
      "Writing file GFPA3_I5.0_F1.9_L100_142948_12_1_1_segment_uncropped.png\n",
      "Reading file GFPA3_I5.0_F1.9_L100_143043_13_1_0_segment_cropped.png\n",
      "Writing file GFPA3_I5.0_F1.9_L100_143043_13_1_0_segment_uncropped.png\n",
      "Reading file GFPA3_I5.0_F1.9_L100_143138_14_2_1_segment_cropped.png\n",
      "Writing file GFPA3_I5.0_F1.9_L100_143138_14_2_1_segment_uncropped.png\n",
      "Reading file GFPA3_I5.0_F1.9_L100_143417_17_2_5_segment_cropped.png\n",
      "Writing file GFPA3_I5.0_F1.9_L100_143417_17_2_5_segment_uncropped.png\n",
      "Reading file GFPA3_I5.0_F1.9_L100_143513_18_2_6_segment_cropped.png\n",
      "Writing file GFPA3_I5.0_F1.9_L100_143513_18_2_6_segment_uncropped.png\n",
      "Reading file GFPA4_I5.0_F1.9_L100_145031_0_0_0_segment_cropped.png\n",
      "Writing file GFPA4_I5.0_F1.9_L100_145031_0_0_0_segment_uncropped.png\n",
      "Reading file GFPA4_I5.0_F1.9_L100_145304_3_0_4_segment_cropped.png\n",
      "Writing file GFPA4_I5.0_F1.9_L100_145304_3_0_4_segment_uncropped.png\n",
      "Reading file GFPA4_I5.0_F1.9_L100_145358_4_0_5_segment_cropped.png\n",
      "Writing file GFPA4_I5.0_F1.9_L100_145358_4_0_5_segment_uncropped.png\n",
      "Reading file GFPA4_I5.0_F1.9_L100_145455_5_1_5_segment_cropped.png\n",
      "Writing file GFPA4_I5.0_F1.9_L100_145455_5_1_5_segment_uncropped.png\n",
      "Reading file GFPA4_I5.0_F1.9_L100_145732_8_1_2_segment_cropped.png\n",
      "Writing file GFPA4_I5.0_F1.9_L100_145732_8_1_2_segment_uncropped.png\n",
      "Reading file GFPA4_I5.0_F1.9_L100_145827_9_1_1_segment_cropped.png\n",
      "Writing file GFPA4_I5.0_F1.9_L100_145827_9_1_1_segment_uncropped.png\n",
      "Reading file GFPA4_I5.0_F1.9_L100_150102_12_2_1_segment_cropped.png\n",
      "Writing file GFPA4_I5.0_F1.9_L100_150102_12_2_1_segment_uncropped.png\n",
      "Reading file GFPA4_I5.0_F1.9_L100_150156_13_2_2_segment_cropped.png\n",
      "Writing file GFPA4_I5.0_F1.9_L100_150156_13_2_2_segment_uncropped.png\n",
      "Reading file GFPA1_I5.0_F1.9_L100_132422_0_0_0_segment_cropped.png\n",
      "Writing file GFPA1_I5.0_F1.9_L100_132422_0_0_0_segment_uncropped.png\n",
      "Reading file GFPA1_I5.0_F1.9_L100_132505_1_0_1_segment_cropped.png\n",
      "Writing file GFPA1_I5.0_F1.9_L100_132505_1_0_1_segment_uncropped.png\n",
      "Reading file GFPA4_I5.0_F1.9_L100_150752_0_2_4_segment_cropped.png\n",
      "Writing file GFPA4_I5.0_F1.9_L100_150752_0_2_4_segment_uncropped.png\n",
      "Reading file GFPA4_I5.0_F1.9_L100_151648_0_0_2_segment_cropped.png\n",
      "Writing file GFPA4_I5.0_F1.9_L100_151648_0_0_2_segment_uncropped.png\n",
      "Reading file GFPA5_I5.0_F1.9_L100_185427_0_0_1_segment_cropped.png\n",
      "Writing file GFPA5_I5.0_F1.9_L100_185427_0_0_1_segment_uncropped.png\n",
      "Reading file GFPA5_I5.0_F1.9_L100_185518_1_0_3_segment_cropped.png\n",
      "Writing file GFPA5_I5.0_F1.9_L100_185518_1_0_3_segment_uncropped.png\n",
      "Reading file GFPA5_I5.0_F1.9_L100_185615_2_0_4_segment_cropped.png\n",
      "Writing file GFPA5_I5.0_F1.9_L100_185615_2_0_4_segment_uncropped.png\n",
      "Reading file GFPA5_I5.0_F1.9_L100_185710_3_0_5_segment_cropped.png\n",
      "Writing file GFPA5_I5.0_F1.9_L100_185710_3_0_5_segment_uncropped.png\n",
      "Reading file GFPA5_I5.0_F1.9_L100_185956_6_1_3_segment_cropped.png\n",
      "Writing file GFPA5_I5.0_F1.9_L100_185956_6_1_3_segment_uncropped.png\n",
      "Reading file GFPA5_I5.0_F1.9_L100_190630_0_1_2_segment_cropped.png\n",
      "Writing file GFPA5_I5.0_F1.9_L100_190630_0_1_2_segment_uncropped.png\n",
      "Reading file GFPA1_I5.0_F1.9_L100_132846_5_0_5_segment_cropped.png\n",
      "Writing file GFPA1_I5.0_F1.9_L100_132846_5_0_5_segment_uncropped.png\n",
      "Reading file GFPA1_I5.0_F1.9_L100_132940_6_0_6_segment_cropped.png\n",
      "Writing file GFPA1_I5.0_F1.9_L100_132940_6_0_6_segment_uncropped.png\n",
      "Reading file GFPA1_I5.0_F1.9_L100_133406_11_1_2_segment_cropped.png\n",
      "Writing file GFPA1_I5.0_F1.9_L100_133406_11_1_2_segment_uncropped.png\n",
      "Reading file GFPA1_I5.0_F1.9_L100_133502_12_1_0_segment_cropped.png\n",
      "Writing file GFPA1_I5.0_F1.9_L100_133502_12_1_0_segment_uncropped.png\n",
      "Reading file GFPA1_I5.0_F1.9_L100_133738_15_2_2_segment_cropped.png\n",
      "Writing file GFPA1_I5.0_F1.9_L100_133738_15_2_2_segment_uncropped.png\n",
      "Reading file GFPA1_I5.0_F1.9_L100_133833_16_2_3_segment_cropped.png\n",
      "Writing file GFPA1_I5.0_F1.9_L100_133833_16_2_3_segment_uncropped.png\n",
      "Reading file GFPA1_I5.0_F1.9_L100_134552_0_0_3_segment_cropped.png\n",
      "Writing file GFPA1_I5.0_F1.9_L100_134552_0_0_3_segment_uncropped.png\n",
      "Reading file GFPA1_I5.0_F1.9_L100_134702_1_1_3_segment_cropped.png\n",
      "Writing file GFPA1_I5.0_F1.9_L100_134702_1_1_3_segment_uncropped.png\n",
      "Reading file GFPA2_I5.0_F1.9_L100_135704_4_0_4_segment_cropped.png\n",
      "Writing file GFPA2_I5.0_F1.9_L100_135704_4_0_4_segment_uncropped.png\n",
      "Reading file GFPA2_I5.0_F1.9_L100_135758_5_0_5_segment_cropped.png\n",
      "Writing file GFPA2_I5.0_F1.9_L100_135758_5_0_5_segment_uncropped.png\n",
      "Reading file GFPA2_I5.0_F1.9_L100_140033_8_1_5_segment_cropped.png\n",
      "Writing file GFPA2_I5.0_F1.9_L100_140033_8_1_5_segment_uncropped.png\n",
      "Reading file GFPA2_I5.0_F1.9_L100_140127_9_1_4_segment_cropped.png\n",
      "Writing file GFPA2_I5.0_F1.9_L100_140127_9_1_4_segment_uncropped.png\n",
      "Reading file GFPA2_I5.0_F1.9_L100_140506_13_1_0_segment_cropped.png\n",
      "Writing file GFPA2_I5.0_F1.9_L100_140506_13_1_0_segment_uncropped.png\n",
      "Reading file GFPA2_I5.0_F1.9_L100_140601_14_2_0_segment_cropped.png\n",
      "Writing file GFPA2_I5.0_F1.9_L100_140601_14_2_0_segment_uncropped.png\n",
      "Reading file GFPA2_I5.0_F1.9_L100_140838_17_2_3_segment_cropped.png\n",
      "Writing file GFPA2_I5.0_F1.9_L100_140838_17_2_3_segment_uncropped.png\n",
      "Reading file GFPA2_I5.0_F1.9_L100_140933_18_2_4_segment_cropped.png\n",
      "Writing file GFPA2_I5.0_F1.9_L100_140933_18_2_4_segment_uncropped.png\n",
      "Reading file GFPA3_I5.0_F1.9_L100_142043_2_0_2_segment_cropped.png\n",
      "Writing file GFPA3_I5.0_F1.9_L100_142043_2_0_2_segment_uncropped.png\n",
      "Reading file GFPA3_I5.0_F1.9_L100_142137_3_0_3_segment_cropped.png\n",
      "Writing file GFPA3_I5.0_F1.9_L100_142137_3_0_3_segment_uncropped.png\n",
      "Reading file GFPA3_I5.0_F1.9_L100_142428_6_0_6_segment_cropped.png\n",
      "Writing file GFPA3_I5.0_F1.9_L100_142428_6_0_6_segment_uncropped.png\n",
      "Reading file GFPA3_I5.0_F1.9_L100_142523_7_1_6_segment_cropped.png\n",
      "Writing file GFPA3_I5.0_F1.9_L100_142523_7_1_6_segment_uncropped.png\n",
      "Reading file GFPA3_I5.0_F1.9_L100_142759_10_1_3_segment_cropped.png\n",
      "Writing file GFPA3_I5.0_F1.9_L100_142759_10_1_3_segment_uncropped.png\n",
      "Reading file GFPA3_I5.0_F1.9_L100_142854_11_1_2_segment_cropped.png\n",
      "Writing file GFPA3_I5.0_F1.9_L100_142854_11_1_2_segment_uncropped.png\n",
      "Reading file GFPA3_I5.0_F1.9_L100_143226_15_2_2_segment_cropped.png\n",
      "Writing file GFPA3_I5.0_F1.9_L100_143226_15_2_2_segment_uncropped.png\n",
      "Reading file GFPA3_I5.0_F1.9_L100_143320_16_2_4_segment_cropped.png\n",
      "Writing file GFPA3_I5.0_F1.9_L100_143320_16_2_4_segment_uncropped.png\n",
      "Reading file GFPA4_I5.0_F1.9_L100_145543_6_1_4_segment_cropped.png\n",
      "Writing file GFPA4_I5.0_F1.9_L100_145543_6_1_4_segment_uncropped.png\n",
      "Reading file GFPA4_I5.0_F1.9_L100_145638_7_1_3_segment_cropped.png\n",
      "Writing file GFPA4_I5.0_F1.9_L100_145638_7_1_3_segment_uncropped.png\n",
      "Reading file GFPA4_I5.0_F1.9_L100_145922_10_1_0_segment_cropped.png\n",
      "Writing file GFPA4_I5.0_F1.9_L100_145922_10_1_0_segment_uncropped.png\n",
      "Reading file GFPA4_I5.0_F1.9_L100_150017_11_2_0_segment_cropped.png\n",
      "Writing file GFPA4_I5.0_F1.9_L100_150017_11_2_0_segment_uncropped.png\n",
      "Reading file GFPA4_I5.0_F1.9_L100_150253_14_2_3_segment_cropped.png\n",
      "Writing file GFPA4_I5.0_F1.9_L100_150253_14_2_3_segment_uncropped.png\n",
      "Reading file GFPA4_I5.0_F1.9_L100_150435_16_2_6_segment_cropped.png\n",
      "Writing file GFPA4_I5.0_F1.9_L100_150435_16_2_6_segment_uncropped.png\n",
      "Reading file GFPA4_I5.0_F1.9_L100_151748_1_1_6_segment_cropped.png\n",
      "Writing file GFPA4_I5.0_F1.9_L100_151748_1_1_6_segment_uncropped.png\n",
      "Reading file GFPA4_I5.0_F1.9_L100_151903_2_2_5_segment_cropped.png\n",
      "Writing file GFPA4_I5.0_F1.9_L100_151903_2_2_5_segment_uncropped.png\n",
      "Reading file GFPA5_I5.0_F1.9_L100_185805_4_0_6_segment_cropped.png\n",
      "Writing file GFPA5_I5.0_F1.9_L100_185805_4_0_6_segment_uncropped.png\n",
      "Reading file GFPA5_I5.0_F1.9_L100_185901_5_1_4_segment_cropped.png\n",
      "Writing file GFPA5_I5.0_F1.9_L100_185901_5_1_4_segment_uncropped.png\n",
      "Reading file GFPA4_I5.0_F1.9_L100_145112_1_0_1_segment_cropped.png\n",
      "Writing file GFPA4_I5.0_F1.9_L100_145112_1_0_1_segment_uncropped.png\n",
      "Reading file GFPA4_I5.0_F1.9_L100_145206_2_0_3_segment_cropped.png\n",
      "Writing file GFPA4_I5.0_F1.9_L100_145206_2_0_3_segment_uncropped.png\n",
      "Reading file GFPA5_I5.0_F1.9_L100_190729_1_1_1_segment_cropped.png\n",
      "Writing file GFPA5_I5.0_F1.9_L100_190729_1_1_1_segment_uncropped.png\n",
      "Reading file GFPA5_I5.0_F1.9_L100_190823_2_1_0_segment_cropped.png\n",
      "Writing file GFPA5_I5.0_F1.9_L100_190823_2_1_0_segment_uncropped.png\n",
      "(alignment) "
     ]
    }
   ],
   "source": [
    "conda activate alignment\n",
    "cd $alignment_path\n",
    "python expand.py $data\n",
    "conda deactivate"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Make composite images with side-by-side RGB, segmentation outputs and blended images"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "conda activate test-environment\n",
    "if [ $composite -eq 1 ]\n",
    "then\n",
    "    cd $gmolabeler_path\n",
    "    python image_blender.py $data 0.75 'both' 1 180\n",
    "fi\n",
    "conda deactivate"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### 2.4. Classification of contaminated/missing explants"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Plates are cropped into sub-images for each explant and each is analyzed to determine if the explant position should be excluded from analysis due to being missing or contamination. Missing and contaminated explants are recognized using a trained Densenet model (<a href=\"https://github.com/Contamination-Classification/DenseNet\" target=\"_blank\">Huang, et al. 2018</a>). Our fork of the Densenet repository is available on <a href=\"https://arxiv.org/abs/1608.06993\" target=\"_blank\">GitHub</a>."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<img src=\"Figures/Densenet.png\">\n",
    "Figure: These are four examples of contaminated explants used in the training set for this pre-trained model"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "If the mode for missing explant data is automatic, prepare input file for script to detect missing explants and run this script."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [],
   "source": [
    "img_list_path=\"${data}/rgb_list.txt\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Dependencies include `keras-preprocessing`, `termcolor`,  `protobuf` and `absl-py`"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//rgb_list.txt\n"
     ]
    }
   ],
   "source": [
    "echo $img_list_path"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [],
   "source": [
    "missing_explants=\"Automatic\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {
    "slideshow": {
     "slide_type": "slide"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Missing explants input manually by user, in file: \n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//output.csv\n"
     ]
    }
   ],
   "source": [
    "if [ $missing_explants = \"Automatic\" ]; then\n",
    "    conda activate densenet\n",
    "    echo \"Missing explants will be inferred.\"\n",
    "    cd $data\n",
    "    ls -d $PWD/* $data | grep -i \"rgb.jpg\" > rgb_list.txt\n",
    "    sed -i '/hroma/d' rgb_list.txt\n",
    "    #sed -i 's/.jpg//g' rgb_list.txt\n",
    "    #cat rgb_list.txt\n",
    "    cd $contamination_path\n",
    "    img_list_path=\"${data}/rgb_list.txt\"\n",
    "    #echo $img_list_path\n",
    "    python inference.py --img-list=$img_list_path --output_file=output.csv\n",
    "    mv -f output.csv \"${data}/output.csv\"\n",
    "    missing_explants=\"${data}/output.csv\"\n",
    "    conda deactivate\n",
    "else\n",
    "    echo \"Missing explants input manually by user, in file: \"\n",
    "    echo $missing_explants\n",
    "fi"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 3. Alignment of RGB and hyperspectral layers"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "To match the frame and angle of RGB and hyperspectral image layers, we apply a scale-invariant feature transformation (<a href=\"https://github.com/NSF-Image-alignment/ImageAlignment\" target=\"_blank\">GitHub</a>). Using a pair of standard images, a homography matrix is calculated for the necessary transformation of RGB images to align with hyperspectral images. The transformation can then be applied to large batches of images rapidly, as long as the RGB and hyperspectral cameras remain in the same positions relative to one another (as they do in the macroPhor Array platform)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<img src=\"Figures/Alignment.png\">\n",
    "Figure: To enable precise calculation of a homography matrix for transformation of RGB images to match hyperspectral images, we used images of a piece of paper with grid marks. These images are provided by the user inputs to --hyper-img and --rgb-img in the below call to the alignment script. If using a phenotyping platform other than the macroPhor Array, or using updated camera settings, these variables will need to be replaced."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### 3.1. Prepare input file for alignment"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [],
   "source": [
    "cd $data\n",
    "ls | grep -i 'rgb\\.jpg' > file_list_part1.csv\n",
    "ls | grep -i 'segment_uncropped\\.png' > file_list_part2.csv\n",
    "cat file_list_part* > file_list.csv\n",
    "sed -i '/hroma/d' file_list.csv\n",
    "cwd2=$(pwd)/\n",
    "awk -v prefix=\"$cwd2\" '{print prefix $0}' file_list.csv > temp\n",
    "mv -f temp file_list.csv\n",
    "echo 'rgb_images' | cat - file_list.csv > temp && mv -f temp file_list.csv"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### 3.2 Run alignment"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(alignment) (alignment) ------------Preprocess is saved and finished.-------------------\n",
      "No of matches are less than 4. Cannot run RANSAC\n",
      "(alignment) (base) "
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "conda activate alignment\n",
    "cd $alignment_path\n",
    "python main.py \\\n",
    "--hyper-img Grids_to_align-selected/GFPA3_I5.0_F1.9_L100_142948_12_1_1_ChlorophyllLayer.csv \\\n",
    "--rgb-img Grids_to_align-selected/GFPA3_I5.0_F1.9_L100_142948_12_1_1_rgb.jpg \\\n",
    "--mode 1 \\\n",
    "--image_thresh_high=120 \\\n",
    "--image_thresh_low=50 \\\n",
    "--distance=0.7\n",
    "conda deactivate"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(alignment) (alignment) (alignment) Traceback (most recent call last):\n",
      "  File \"main.py\", line 12, in <module>\n",
      "    import utils as utils\n",
      "  File \"/home/labgroup/code/ImageAlignment/utils.py\", line 17, in <module>\n",
      "    import cv2\n",
      "ModuleNotFoundError: No module named 'cv2'\n",
      "(alignment) "
     ]
    }
   ],
   "source": [
    "conda activate alignment\n",
    "cd $alignment_path\n",
    "file_list_input=\"${data}/file_list.csv\"\n",
    "python main.py \\\n",
    "--hyper-img Grids_to_align-selected/12itemoffcenterleftfacinga3_F1.9_I5.0_L80_cyan_172334_0_0_0_index130.csv \\\n",
    "--rgb-img Grids_to_align-selected/12itemoffcenterleftfacinga3_F1.9_I5.0_L80_cyan_172334_0_0_0_rgb.jpg \\\n",
    "--img-csv $file_list_input \\\n",
    "--mode 2 \\\n",
    "--image_thresh_high=120 \\\n",
    "--image_thresh_low=50 \\\n",
    "--distance=0.7\n",
    "conda deactivate\n",
    "conda deactivate"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 4. Cross-analyze deep segmentation and regression results"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Scripts in the <a href=\"https://github.com/naglemi/GMOlabeler\" target=\"_blank\">GMOlabeler repository</a> are used to cross-reference results from deep segmentation of RGB images and regression of hyperspectral imaging, apply thresholding parameters to classify tissues as transgenic or escapes, and produce plots."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<img src=\"Figures/GMOlabeler.png\">"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Figure: The various steps of data processing in GMOlabeler are illustrated for an example explant from an experiment on CIM optimization for cottonwood. Images of plates are cropped to a sub-image for each explant. RGB segmentation results and hyperspectral regression results are cross-referenced to calculate fluorecent proteins in specific tissues and infer whether these tissues are transgenic."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### 4.1. Prepare sample datasheet input"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Prepare input file we will use for making plots. This file contains paths to CLS results, RGB images, and hyperspectral images."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(gmolabeler) (gmolabeler) [1] \"”2020-11-26”\"\n",
      "[1] \"Looking for CLS data in: /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//\"\n",
      "[1] \"2020-11-26\"\n",
      "[1] \"Looking for CLS files in directory /home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//\"\n",
      "[1] \"How many CLS files? 89\"\n",
      "[1] \"We will only include in analysis the FINAL of multiple images for GFPA4_2_4\"\n",
      "\n",
      "[1] \"Rows missing RGB data:\"\n",
      "Empty data.table (0 rows and 3 cols): segment,CLS_data,rgb\n",
      "\n",
      "[1] \"Rows missing segment data:\"\n",
      "Empty data.table (0 rows and 3 cols): segment,CLS_data,rgb\n",
      "\n",
      "[1] \"Rows missing regressopm data:\"\n",
      "                                                                                                                                             segment\n",
      "1: /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA4_I5.0_F1.9_L100_150752_0_2_4_segment_uncropped_processed.png\n",
      "   CLS_data\n",
      "1:     <NA>\n",
      "                                                                                                                                   rgb\n",
      "1: /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA4_I5.0_F1.9_L100_150752_0_2_4_rgb_processed.jpg\n",
      "\n",
      "Warning message:\n",
      "There is an inconsistency. There are some files for which we do not have both hyperspectral and RGB data. These have been excluded. \n",
      "[1] \"Writing 88 rows to /run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///samples_pre_labeling.csv\"\n",
      "(gmolabeler) "
     ]
    }
   ],
   "source": [
    "conda activate gmolabeler\n",
    "cd \"${cwd}/intermediates/\"\n",
    "Rscript pre_label.R \\\n",
    "-r \"${data}/\" \\\n",
    "-R \"${output_directory_prefix}\" \\\n",
    "-i 1 \\\n",
    "-d $datestamp\n",
    "conda deactivate"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### 4.2. Cross-reference RGB and hyperspectral data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/labgroup/code/ImageAlignment/112itemoffcenterleftfacinga3_F1.9_I5.0_L80_cyan_172334_0_0_0_rgb_transformed_rgb.jpg\n"
     ]
    }
   ],
   "source": [
    "echo $grid_file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Error in certificate: Peer's certificate issuer is not recognized.\n",
      "smtp-server: 535-5.7.8 Username and Password not accepted. Learn more at\n",
      "smtp-server: 535 5.7.8  https://support.google.com/mail/?p=BadCredentials s10sm340065pgg.76 - gsmtp\n",
      "\"/home/labgroup/dead.letter\" 72538/5293782\n",
      ". . . message not sent.\n",
      "(gmolabeler) (gmolabeler) Grid type 12\n",
      "Loading plate 0 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA5_I5.0_F1.9_L100_190823_2_1_0_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA5_I5.0_F1.9_L100_190823_2_1_0_weights.hdf\n",
      "Loading plate 1 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA5_I5.0_F1.9_L100_190729_1_1_1_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA5_I5.0_F1.9_L100_190729_1_1_1_weights.hdf\n",
      "Loading plate 2 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA5_I5.0_F1.9_L100_190630_0_1_2_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA5_I5.0_F1.9_L100_190630_0_1_2_weights.hdf\n",
      "Loading plate 3 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA5_I5.0_F1.9_L100_185956_6_1_3_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA5_I5.0_F1.9_L100_185956_6_1_3_weights.hdf\n",
      "Loading plate 4 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA5_I5.0_F1.9_L100_185901_5_1_4_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA5_I5.0_F1.9_L100_185901_5_1_4_weights.hdf\n",
      "Loading plate 5 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA5_I5.0_F1.9_L100_185805_4_0_6_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA5_I5.0_F1.9_L100_185805_4_0_6_weights.hdf\n",
      "Loading plate 6 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA5_I5.0_F1.9_L100_185710_3_0_5_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA5_I5.0_F1.9_L100_185710_3_0_5_weights.hdf\n",
      "Loading plate 7 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA5_I5.0_F1.9_L100_185615_2_0_4_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA5_I5.0_F1.9_L100_185615_2_0_4_weights.hdf\n",
      "Loading plate 8 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA5_I5.0_F1.9_L100_185518_1_0_3_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA5_I5.0_F1.9_L100_185518_1_0_3_weights.hdf\n",
      "Loading plate 9 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA5_I5.0_F1.9_L100_185427_0_0_1_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA5_I5.0_F1.9_L100_185427_0_0_1_weights.hdf\n",
      "Loading plate 10 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA4_I5.0_F1.9_L100_151903_2_2_5_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA4_I5.0_F1.9_L100_151903_2_2_5_weights.hdf\n",
      "Loading plate 11 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA4_I5.0_F1.9_L100_151748_1_1_6_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA4_I5.0_F1.9_L100_151748_1_1_6_weights.hdf\n",
      "Loading plate 12 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA4_I5.0_F1.9_L100_151648_0_0_2_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA4_I5.0_F1.9_L100_151648_0_0_2_weights.hdf\n",
      "Loading plate 13 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA4_I5.0_F1.9_L100_150435_16_2_6_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA4_I5.0_F1.9_L100_150435_16_2_6_weights.hdf\n",
      "Loading plate 14 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA4_I5.0_F1.9_L100_150253_14_2_3_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA4_I5.0_F1.9_L100_150253_14_2_3_weights.hdf\n",
      "Loading plate 15 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA4_I5.0_F1.9_L100_150156_13_2_2_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA4_I5.0_F1.9_L100_150156_13_2_2_weights.hdf\n",
      "Loading plate 16 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA4_I5.0_F1.9_L100_150102_12_2_1_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA4_I5.0_F1.9_L100_150102_12_2_1_weights.hdf\n",
      "Loading plate 17 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA4_I5.0_F1.9_L100_150017_11_2_0_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA4_I5.0_F1.9_L100_150017_11_2_0_weights.hdf\n",
      "Loading plate 18 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA4_I5.0_F1.9_L100_145922_10_1_0_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA4_I5.0_F1.9_L100_145922_10_1_0_weights.hdf\n",
      "Loading plate 19 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA4_I5.0_F1.9_L100_145827_9_1_1_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA4_I5.0_F1.9_L100_145827_9_1_1_weights.hdf\n",
      "Loading plate 20 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA4_I5.0_F1.9_L100_145732_8_1_2_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA4_I5.0_F1.9_L100_145732_8_1_2_weights.hdf\n",
      "Loading plate 21 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA4_I5.0_F1.9_L100_145638_7_1_3_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA4_I5.0_F1.9_L100_145638_7_1_3_weights.hdf\n",
      "Loading plate 22 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA4_I5.0_F1.9_L100_145543_6_1_4_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA4_I5.0_F1.9_L100_145543_6_1_4_weights.hdf\n",
      "Loading plate 23 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA4_I5.0_F1.9_L100_145455_5_1_5_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA4_I5.0_F1.9_L100_145455_5_1_5_weights.hdf\n",
      "Loading plate 24 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA4_I5.0_F1.9_L100_145358_4_0_5_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA4_I5.0_F1.9_L100_145358_4_0_5_weights.hdf\n",
      "Loading plate 25 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA4_I5.0_F1.9_L100_145304_3_0_4_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA4_I5.0_F1.9_L100_145304_3_0_4_weights.hdf\n",
      "Loading plate 26 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA4_I5.0_F1.9_L100_145206_2_0_3_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA4_I5.0_F1.9_L100_145206_2_0_3_weights.hdf\n",
      "Loading plate 27 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA4_I5.0_F1.9_L100_145112_1_0_1_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA4_I5.0_F1.9_L100_145112_1_0_1_weights.hdf\n",
      "Loading plate 28 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA4_I5.0_F1.9_L100_145031_0_0_0_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA4_I5.0_F1.9_L100_145031_0_0_0_weights.hdf\n",
      "Loading plate 29 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA3_I5.0_F1.9_L100_143513_18_2_6_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA3_I5.0_F1.9_L100_143513_18_2_6_weights.hdf\n",
      "Loading plate 30 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA3_I5.0_F1.9_L100_143417_17_2_5_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA3_I5.0_F1.9_L100_143417_17_2_5_weights.hdf\n",
      "Loading plate 31 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA3_I5.0_F1.9_L100_143320_16_2_4_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA3_I5.0_F1.9_L100_143320_16_2_4_weights.hdf\n",
      "Loading plate 32 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA3_I5.0_F1.9_L100_143226_15_2_2_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA3_I5.0_F1.9_L100_143226_15_2_2_weights.hdf\n",
      "Loading plate 33 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA3_I5.0_F1.9_L100_143138_14_2_1_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA3_I5.0_F1.9_L100_143138_14_2_1_weights.hdf\n",
      "Loading plate 34 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA3_I5.0_F1.9_L100_143043_13_1_0_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA3_I5.0_F1.9_L100_143043_13_1_0_weights.hdf\n",
      "Loading plate 35 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA3_I5.0_F1.9_L100_142948_12_1_1_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA3_I5.0_F1.9_L100_142948_12_1_1_weights.hdf\n",
      "Loading plate 36 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA3_I5.0_F1.9_L100_142854_11_1_2_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA3_I5.0_F1.9_L100_142854_11_1_2_weights.hdf\n",
      "Loading plate 37 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA3_I5.0_F1.9_L100_142759_10_1_3_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA3_I5.0_F1.9_L100_142759_10_1_3_weights.hdf\n",
      "Loading plate 38 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA3_I5.0_F1.9_L100_142704_9_1_4_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA3_I5.0_F1.9_L100_142704_9_1_4_weights.hdf\n",
      "Loading plate 39 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA3_I5.0_F1.9_L100_142608_8_1_5_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA3_I5.0_F1.9_L100_142608_8_1_5_weights.hdf\n",
      "Loading plate 40 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA3_I5.0_F1.9_L100_142523_7_1_6_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA3_I5.0_F1.9_L100_142523_7_1_6_weights.hdf\n",
      "Loading plate 41 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA3_I5.0_F1.9_L100_142428_6_0_6_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA3_I5.0_F1.9_L100_142428_6_0_6_weights.hdf\n",
      "Loading plate 42 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA3_I5.0_F1.9_L100_142332_5_0_5_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA3_I5.0_F1.9_L100_142332_5_0_5_weights.hdf\n",
      "Loading plate 43 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA3_I5.0_F1.9_L100_142237_4_0_4_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA3_I5.0_F1.9_L100_142237_4_0_4_weights.hdf\n",
      "Loading plate 44 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA3_I5.0_F1.9_L100_142137_3_0_3_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA3_I5.0_F1.9_L100_142137_3_0_3_weights.hdf\n",
      "Loading plate 45 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA3_I5.0_F1.9_L100_142043_2_0_2_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA3_I5.0_F1.9_L100_142043_2_0_2_weights.hdf\n",
      "Loading plate 46 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA3_I5.0_F1.9_L100_141948_1_0_1_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA3_I5.0_F1.9_L100_141948_1_0_1_weights.hdf\n",
      "Loading plate 47 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA3_I5.0_F1.9_L100_141906_0_0_0_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA3_I5.0_F1.9_L100_141906_0_0_0_weights.hdf\n",
      "Loading plate 48 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA2_I5.0_F1.9_L100_141123_20_2_6_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA2_I5.0_F1.9_L100_141123_20_2_6_weights.hdf\n",
      "Loading plate 49 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA2_I5.0_F1.9_L100_141028_19_2_5_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA2_I5.0_F1.9_L100_141028_19_2_5_weights.hdf\n",
      "Loading plate 50 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA2_I5.0_F1.9_L100_140933_18_2_4_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA2_I5.0_F1.9_L100_140933_18_2_4_weights.hdf\n",
      "Loading plate 51 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA2_I5.0_F1.9_L100_140838_17_2_3_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA2_I5.0_F1.9_L100_140838_17_2_3_weights.hdf\n",
      "Loading plate 52 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA2_I5.0_F1.9_L100_140742_16_2_2_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA2_I5.0_F1.9_L100_140742_16_2_2_weights.hdf\n",
      "Loading plate 53 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA2_I5.0_F1.9_L100_140647_15_2_1_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA2_I5.0_F1.9_L100_140647_15_2_1_weights.hdf\n",
      "Loading plate 54 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA2_I5.0_F1.9_L100_140601_14_2_0_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA2_I5.0_F1.9_L100_140601_14_2_0_weights.hdf\n",
      "Loading plate 55 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA2_I5.0_F1.9_L100_140506_13_1_0_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA2_I5.0_F1.9_L100_140506_13_1_0_weights.hdf\n",
      "Loading plate 56 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA2_I5.0_F1.9_L100_140410_12_1_1_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA2_I5.0_F1.9_L100_140410_12_1_1_weights.hdf\n",
      "Loading plate 57 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA2_I5.0_F1.9_L100_140315_11_1_2_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA2_I5.0_F1.9_L100_140315_11_1_2_weights.hdf\n",
      "Loading plate 58 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA2_I5.0_F1.9_L100_140221_10_1_3_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA2_I5.0_F1.9_L100_140221_10_1_3_weights.hdf\n",
      "Loading plate 59 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA2_I5.0_F1.9_L100_140127_9_1_4_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA2_I5.0_F1.9_L100_140127_9_1_4_weights.hdf\n",
      "Loading plate 60 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA2_I5.0_F1.9_L100_140033_8_1_5_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA2_I5.0_F1.9_L100_140033_8_1_5_weights.hdf\n",
      "Loading plate 61 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA2_I5.0_F1.9_L100_135947_7_1_6_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA2_I5.0_F1.9_L100_135947_7_1_6_weights.hdf\n",
      "Loading plate 62 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA2_I5.0_F1.9_L100_135852_6_0_6_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA2_I5.0_F1.9_L100_135852_6_0_6_weights.hdf\n",
      "Loading plate 63 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA2_I5.0_F1.9_L100_135758_5_0_5_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA2_I5.0_F1.9_L100_135758_5_0_5_weights.hdf\n",
      "Loading plate 64 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA2_I5.0_F1.9_L100_135704_4_0_4_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA2_I5.0_F1.9_L100_135704_4_0_4_weights.hdf\n",
      "Loading plate 65 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA2_I5.0_F1.9_L100_135609_3_0_3_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA2_I5.0_F1.9_L100_135609_3_0_3_weights.hdf\n",
      "Loading plate 66 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA2_I5.0_F1.9_L100_135514_2_0_2_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA2_I5.0_F1.9_L100_135514_2_0_2_weights.hdf\n",
      "Loading plate 67 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA2_I5.0_F1.9_L100_135420_1_0_1_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA2_I5.0_F1.9_L100_135420_1_0_1_weights.hdf\n",
      "Loading plate 68 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA2_I5.0_F1.9_L100_135339_0_0_0_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA2_I5.0_F1.9_L100_135339_0_0_0_weights.hdf\n",
      "Loading plate 69 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA1_I5.0_F1.9_L100_134702_1_1_3_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA1_I5.0_F1.9_L100_134702_1_1_3_weights.hdf\n",
      "Loading plate 70 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA1_I5.0_F1.9_L100_134552_0_0_3_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA1_I5.0_F1.9_L100_134552_0_0_3_weights.hdf\n",
      "Loading plate 71 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA1_I5.0_F1.9_L100_134025_18_2_6_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA1_I5.0_F1.9_L100_134025_18_2_6_weights.hdf\n",
      "Loading plate 72 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA1_I5.0_F1.9_L100_133928_17_2_5_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA1_I5.0_F1.9_L100_133928_17_2_5_weights.hdf\n",
      "Loading plate 73 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA1_I5.0_F1.9_L100_133833_16_2_3_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA1_I5.0_F1.9_L100_133833_16_2_3_weights.hdf\n",
      "Loading plate 74 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA1_I5.0_F1.9_L100_133738_15_2_2_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA1_I5.0_F1.9_L100_133738_15_2_2_weights.hdf\n",
      "Loading plate 75 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA1_I5.0_F1.9_L100_133643_14_2_1_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA1_I5.0_F1.9_L100_133643_14_2_1_weights.hdf\n",
      "Loading plate 76 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA1_I5.0_F1.9_L100_133559_13_2_0_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA1_I5.0_F1.9_L100_133559_13_2_0_weights.hdf\n",
      "Loading plate 77 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA1_I5.0_F1.9_L100_133502_12_1_0_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA1_I5.0_F1.9_L100_133502_12_1_0_weights.hdf\n",
      "Loading plate 78 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA1_I5.0_F1.9_L100_133406_11_1_2_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA1_I5.0_F1.9_L100_133406_11_1_2_weights.hdf\n",
      "Loading plate 79 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA1_I5.0_F1.9_L100_133216_9_1_4_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA1_I5.0_F1.9_L100_133216_9_1_4_weights.hdf\n",
      "Loading plate 80 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA1_I5.0_F1.9_L100_133121_8_1_5_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA1_I5.0_F1.9_L100_133121_8_1_5_weights.hdf\n",
      "Loading plate 81 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA1_I5.0_F1.9_L100_133036_7_1_6_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA1_I5.0_F1.9_L100_133036_7_1_6_weights.hdf\n",
      "Loading plate 82 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA1_I5.0_F1.9_L100_132940_6_0_6_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA1_I5.0_F1.9_L100_132940_6_0_6_weights.hdf\n",
      "Loading plate 83 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA1_I5.0_F1.9_L100_132846_5_0_5_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA1_I5.0_F1.9_L100_132846_5_0_5_weights.hdf\n",
      "Loading plate 84 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA1_I5.0_F1.9_L100_132750_4_0_4_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA1_I5.0_F1.9_L100_132750_4_0_4_weights.hdf\n",
      "Loading plate 85 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA1_I5.0_F1.9_L100_132600_2_0_2_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA1_I5.0_F1.9_L100_132600_2_0_2_weights.hdf\n",
      "Loading plate 86 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA1_I5.0_F1.9_L100_132505_1_0_1_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA1_I5.0_F1.9_L100_132505_1_0_1_weights.hdf\n",
      "Loading plate 87 of 88\n",
      "/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA1_I5.0_F1.9_L100_132422_0_0_0_segment_uncropped_processed.png\n",
      "Plate segment loaded\n",
      "Plate RGB loaded\n",
      "Loading CLS data from path/home/labgroup/data/gmodetector_out/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA1_I5.0_F1.9_L100_132422_0_0_0_weights.hdf\n",
      "(gmolabeler) "
     ]
    }
   ],
   "source": [
    "conda activate gmolabeler\n",
    "cd $gmolabeler_path\n",
    "python main.py \\\n",
    "\"${data}/samples_pre_labeling.csv\" \\\n",
    "$grid_file \\\n",
    "$reporter_threshold \\\n",
    "$reporter \\\n",
    "$grid \\\n",
    "\"hdf\" \\\n",
    "$gmolabeler_path\n",
    "conda deactivate"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### 4.3. Calculate sums of statistics over combined segments"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We are interested in all regenerated tissue (callus + shoot) as well as all tissue (including stem as well). We will calculate aggregate statistics over these groups."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(gmolabeler) [1] \"Writing output with sums statistics calculated over combined tissue segments to: /home/labgroup/code/GMOlabeler//output/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//stats_with_sums_over_tissues.csv\"\n",
      "(gmolabeler) "
     ]
    }
   ],
   "source": [
    "conda activate gmolabeler\n",
    "Rscript calculate_sum_stats_over_combined_segments.R \\\n",
    "--output_dir \"${gmolabeler_path}/output/\" \\\n",
    "--datapath \"${data_folder}/\"\n",
    "conda deactivate"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### 4.4. Make plots of results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(gmolabeler) (gmolabeler) Warning message:\n",
      "In storage.mode(default) <- type : NAs introduced by coercion\n",
      "[1] \"Saving list of input arguments to : /home/labgroup/code/GMOlabeler/plots/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//gridplot_args.rds\"\n",
      "[1] \"Reading in output from GMOlabeler at path: /home/labgroup/code/GMOlabeler/output/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//stats_with_sums_over_tissues.csv\"\n",
      "\n",
      "[1] \"Rows in output from GMOlabeler: 4727\"\n",
      "\n",
      "[1] \"Max n_pixels_passing_threshold in output from GMOlabeler: 12028\"\n",
      "\n",
      "[1] \"Max total_signal in output from GMOlabeler: 11207608.3137862\"\n",
      "\n",
      "[1] \"Look at the top of output from GMOlabeler\"\n",
      "[1] \"\"                                                                                                                     \n",
      "[2] \"/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA5_I5.0_F1.9_L100_190823_2_1_0.png\"\n",
      "[3] \"/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA5_I5.0_F1.9_L100_190823_2_1_0.png\"\n",
      "[4] \"/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA5_I5.0_F1.9_L100_190823_2_1_0.png\"\n",
      "[5] \"/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA5_I5.0_F1.9_L100_190823_2_1_0.png\"\n",
      "[6] \"/run/media/labgroup/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent///GFPA5_I5.0_F1.9_L100_190823_2_1_0.png\"\n",
      "\n",
      "[1] \"Look at the top of missing explant data\"\n",
      "# A tibble: 6 x 3\n",
      "  filename                                  grid_item present\n",
      "  <chr>                                     <chr>     <chr>  \n",
      "1 GFPA1_I5.0_F1.9_L100_132422_0_0_0_rgb.jpg 1         C      \n",
      "2 GFPA1_I5.0_F1.9_L100_132422_0_0_0_rgb.jpg 2         C      \n",
      "3 GFPA1_I5.0_F1.9_L100_132422_0_0_0_rgb.jpg 3         NC     \n",
      "4 GFPA1_I5.0_F1.9_L100_132422_0_0_0_rgb.jpg 4         NC     \n",
      "5 GFPA1_I5.0_F1.9_L100_132422_0_0_0_rgb.jpg 5         NC     \n",
      "6 GFPA1_I5.0_F1.9_L100_132422_0_0_0_rgb.jpg 6         NC     \n",
      "\n",
      "[1] \"Look at the head of filenames for missing explant data\"\n",
      "[1] \"GFPA1_I5.0_F1.9_L100_132422_0_0_0_rgb.jpg\"\n",
      "[2] \"GFPA1_I5.0_F1.9_L100_132422_0_0_0_rgb.jpg\"\n",
      "[3] \"GFPA1_I5.0_F1.9_L100_132422_0_0_0_rgb.jpg\"\n",
      "[4] \"GFPA1_I5.0_F1.9_L100_132422_0_0_0_rgb.jpg\"\n",
      "[5] \"GFPA1_I5.0_F1.9_L100_132422_0_0_0_rgb.jpg\"\n",
      "[6] \"GFPA1_I5.0_F1.9_L100_132422_0_0_0_rgb.jpg\"\n",
      "\n",
      "[1] \"Look at the colnames of output from GMOlabeler\"\n",
      " [1] \"V1\"                         \"filename\"                  \n",
      " [3] \"grid_item\"                  \"segment_hex\"               \n",
      " [5] \"intensity_threshold\"        \"n_pixels_passing_threshold\"\n",
      " [7] \"mean_signal\"                \"max_signal\"                \n",
      " [9] \"total_signal\"               \"total_pixels\"              \n",
      "\n",
      "[1] \"Look at the colnames of missing explant data\"\n",
      "[1] \"filename\"  \"grid_item\" \"present\"  \n",
      "\n",
      "[1] \"Before merging the GMOlabeler and missing output data, they have rows of 4727 and 1068 respectively.\"\n",
      "\n",
      "[1] \"After merging but before removing missing/contminated explants, merged output has 4727 rows.\"\n",
      "\n",
      "[1] \"After removing the missing/contaminated explants, merged output has 4727\"\n",
      "\n",
      "[1] \"Afer processing GMOlabeler output to merge with missing explant data, Head of data (w/ first 5 col): \"\n",
      "                                filename grid_item          V1 segment_hex\n",
      "1:                                            <NA> observation            \n",
      "2: GFPA1_I5.0_F1.9_L100_132422_0_0_0.png         1 observation      00CC11\n",
      "3: GFPA1_I5.0_F1.9_L100_132422_0_0_0.png         1 observation      0006CC\n",
      "4: GFPA1_I5.0_F1.9_L100_132422_0_0_0.png         1 observation      CC0000\n",
      "5: GFPA1_I5.0_F1.9_L100_132422_0_0_0.png         1 observation  All_tissue\n",
      "   intensity_threshold\n",
      "1:                  NA\n",
      "2:                  38\n",
      "3:                  38\n",
      "4:                  38\n",
      "5:                  NA\n",
      "New names:\n",
      "* `` -> ...12\n",
      "[1] \"Upon loading randomization_datasheet, it has how many rows? 96\"\n",
      "\n",
      "[1] \"Calculating total numbers of missing explants per plate\"\n",
      "[1] \"Let's figure out why contaminated/missing plates are being said to have 0 instead of NA\"\n",
      "\n",
      "[1] \"Dimensions of missing explant data with totals\"\n",
      "[1] 89 14\n",
      "[1] \"Minimum number of explants on any plate after removing missing and contaminated explants is...\"\n",
      "[1] 2\n",
      "                                  image_name 1 2 3 4 5 6 7 8 9 10 11 12\n",
      "1: GFPA1_I5.0_F1.9_L100_132422_0_0_0_rgb.jpg 0 0 1 1 1 1 1 1 1  1  1  1\n",
      "2: GFPA1_I5.0_F1.9_L100_132505_1_0_1_rgb.jpg 1 0 1 0 1 1 0 0 1  0  1  1\n",
      "3: GFPA1_I5.0_F1.9_L100_132600_2_0_2_rgb.jpg 1 1 1 0 1 1 1 1 1  1  1  1\n",
      "4: GFPA1_I5.0_F1.9_L100_132750_4_0_4_rgb.jpg 1 1 1 0 0 0 1 1 1  0  0  0\n",
      "5: GFPA1_I5.0_F1.9_L100_132846_5_0_5_rgb.jpg 1 1 1 1 1 0 0 1 1  0  1  1\n",
      "6: GFPA1_I5.0_F1.9_L100_132940_6_0_6_rgb.jpg 1 1 1 1 1 1 1 1 0  1  0  1\n",
      "   total_explants\n",
      "1:             10\n",
      "2:              7\n",
      "3:             11\n",
      "4:              6\n",
      "5:              9\n",
      "6:             10\n",
      "\n",
      "[1] \"About to add IDs to missing explant data based on filenames. Here is head of filenames.\"\n",
      "[1] \"GFPA1_I5.0_F1.9_L100_132422_0_0_0\" \"GFPA1_I5.0_F1.9_L100_132505_1_0_1\"\n",
      "[3] \"GFPA1_I5.0_F1.9_L100_132600_2_0_2\" \"GFPA1_I5.0_F1.9_L100_132750_4_0_4\"\n",
      "[5] \"GFPA1_I5.0_F1.9_L100_132846_5_0_5\" \"GFPA1_I5.0_F1.9_L100_132940_6_0_6\"\n",
      "\n",
      "[1] \"We have added IDs to missing explant data based on filenames. Here is head of IDs.\"\n",
      "[1] \"GFPA1_1\" \"GFPA1_2\" \"GFPA1_3\" \"GFPA1_5\" \"GFPA1_6\" \"GFPA1_7\"\n",
      "\n",
      "[1] \"N rows of randomization datasheet before merging with missing explant data: \"\n",
      "[1] 96\n",
      "[1] \"Head of IDs in missing explant data\"\n",
      "[1] \"GFPA1_1\" \"GFPA1_2\" \"GFPA1_3\" \"GFPA1_5\" \"GFPA1_6\" \"GFPA1_7\"\n",
      "[1] \"Head of IDs in randomization datasheet\"\n",
      "[1] \"GFPA1_1\" \"GFPA1_2\" \"GFPA1_3\" \"GFPA1_4\" \"GFPA1_5\" \"GFPA1_6\"\n",
      "\n",
      "[1] \"Now about to merge randomization_datasheet and missing explant data.\"\n",
      "[1] \"Rows in randomization_datasheet: 96\"\n",
      "[1] \"Rows in missing explant data: 89\"\n",
      "\n",
      "[1] \"Now about to merge randomization datasheet and missing explant data, by ID. Look at head of IDs for both.\"\n",
      "[1] \"GFPA1_1\" \"GFPA1_2\" \"GFPA1_3\" \"GFPA1_4\" \"GFPA1_5\" \"GFPA1_6\"\n",
      "[1] \"GFPA1_1\" \"GFPA1_2\" \"GFPA1_3\" \"GFPA1_5\" \"GFPA1_6\" \"GFPA1_7\"\n",
      "        ID Image # Tray ID Rand # Date on Media *Plate # Index# in GH\n",
      "1  GFPA1_1       1   GFPA1      1    2020-09-17  3-2-1-2           R7\n",
      "2 GFPA1_10      10   GFPA1     10    2020-09-17  3-2-1-5           R7\n",
      "3 GFPA1_11      11   GFPA1     11    2020-09-17  1-2-1-1          140\n",
      "4 GFPA1_12      12   GFPA1     12    2020-09-17  4-2-1-3          R21\n",
      "5 GFPA1_13      13   GFPA1     13    2020-09-17  4-2-1-6          R21\n",
      "6 GFPA1_14      14   GFPA1     14    2020-09-17  3-1-2-6           R7\n",
      "  Genotype_ID Costruct Media Treatment name Replicate ...12     LableTop\n",
      "1   Nisqually      Mar   CTR        CTR Mar         2    NA   GFPA1_1_R1\n",
      "2   Nisqually      Mar   CTR        CTR Mar         5    NA GFPA1_10_R10\n",
      "3    BESC-134      Mar   CTR        CTR Mar         1    NA GFPA1_11_R11\n",
      "4   SKWB-24-2      Mar   CTR        CTR Mar         3    NA GFPA1_12_R12\n",
      "5   SKWB-24-2      Mar   CTR        CTR Mar         6    NA GFPA1_13_R13\n",
      "6   Nisqually     HSPT    AS        AS HSPT         6    NA GFPA1_14_R14\n",
      "             LableBot Plate#      Rand                           filename 1 2 3\n",
      "1 Nisqually_Mar_CTR_2     62 0.3583730  GFPA1_I5.0_F1.9_L100_132422_0_0_0 0 0 1\n",
      "2 Nisqually_Mar_CTR_5     65 0.8666917 GFPA1_I5.0_F1.9_L100_133406_11_1_2 1 1 0\n",
      "3  BESC-134_Mar_CTR_1     13 0.5296783  GFPA1_I5.0_F1.9_L100_134702_1_1_3 1 1 1\n",
      "4 SKWB-24-2_Mar_CTR_3     87 0.8703313  GFPA1_I5.0_F1.9_L100_133216_9_1_4 1 1 1\n",
      "5 SKWB-24-2_Mar_CTR_6     90 0.4970979  GFPA1_I5.0_F1.9_L100_133121_8_1_5 0 1 1\n",
      "6 Nisqually_HSPT_AS_6     60 0.6426024  GFPA1_I5.0_F1.9_L100_133036_7_1_6 0 0 1\n",
      "  4 5 6 7 8 9 10 11 12 total_explants\n",
      "1 1 1 1 1 1 1  1  1  1             10\n",
      "2 0 1 1 0 1 1  1  1  1              9\n",
      "3 1 1 1 1 1 1  1  1  1             12\n",
      "4 0 1 1 1 1 1  1  1  1             11\n",
      "5 0 1 1 1 1 1  1  1  1             10\n",
      "6 0 0 1 0 0 1  1  1  1              6\n",
      " [1] 10  9 12 11 10  6 10  9 11 10 NA  7  8  9 11 11  6  9 10 10 NA 11 10  8  7\n",
      "[26] 10 12 12  9  8  7 11  6  9  8  6  9 11  6 10 10  9  6 10  8 12 10 11 NA 11\n",
      "[51] 11 NA  9  7 11 11 10 11 11  7  7 10 11 10  4  2  9 12 11  7 10 10 11 12  7\n",
      "[76] 11 11 11 10  9  4 NA  4  8 NA  8  9  9  3 NA 12 11  9  6  7  6\n",
      "[1] \"Still figuring, now in merged rd, out why contaminated/missing plates are being said to have 0 instead of NA\"\n",
      "\n",
      "[1] \"Dimensions of missing explant data with totals\"\n",
      "[1] 89 15\n",
      "[1] \"Levels of total explants on any plate after removing missing and contaminated explants is...\"\n",
      " [1] \"2\"  \"3\"  \"4\"  \"6\"  \"7\"  \"8\"  \"9\"  \"10\" \"11\" \"12\"\n",
      "[1] \"Minimum number of explants on any plate after removing missing and contaminated explants is...\"\n",
      "[1] 2\n",
      "\n",
      "[1] \"Rows in merged output: 96\"\n",
      " [1] 10  9 12 11 10  6 10  9 11 10 NA  7  8  9 11 11  6  9 10 10 NA 11 10  8  7\n",
      "[26] 10 12 12  9  8  7 11  6  9  8  6  9 11  6 10 10  9  6 10  8 12 10 11 NA 11\n",
      "[51] 11 NA  9  7 11 11 10 11 11  7  7 10 11 10  4  2  9 12 11  7 10 10 11 12  7\n",
      "[76] 11 11 11 10  9  4 NA  4  8 NA  8  9  9  3 NA 12 11  9  6  7  6\n",
      "\n",
      "[1] \"Assuming for all explants for which we have no missing explant data that there are:\"\n",
      "[1] 12\n",
      "\n",
      "[1] \"N rows of randomization datasheet after merging with missing explant data: \"\n",
      "[1] 96\n",
      "[1] \"Column names of randomization datasheet afer merging with missing explant data: \"\n",
      " [1] \"ID\"             \"Image #\"        \"Tray ID\"        \"Rand #\"        \n",
      " [5] \"Date on Media\"  \"*Plate #\"       \"Index# in GH\"   \"Genotype_ID\"   \n",
      " [9] \"Costruct\"       \"Media\"          \"Treatment name\" \"Replicate\"     \n",
      "[13] \"...12\"          \"LableTop\"       \"LableBot\"       \"Plate#\"        \n",
      "[17] \"Rand\"           \"filename\"       \"1\"              \"2\"             \n",
      "[21] \"3\"              \"4\"              \"5\"              \"6\"             \n",
      "[25] \"7\"              \"8\"              \"9\"              \"10\"            \n",
      "[29] \"11\"             \"12\"             \"total_explants\"\n",
      "\n",
      "[1] \"In output, Maximum observations for a segment in a grid item in a filename (should be 1): 1\"\n",
      "[1] \"In randomization datasheet, Maximum observations for an ID (should be 1): 1\"\n",
      "\n",
      "[1] \"Before calculating totals, let's look at the column total_explants in randomization datasheet\"\n",
      "[1] 10  9 12 11 10  6\n",
      " [1] \"2\"  \"3\"  \"4\"  \"6\"  \"7\"  \"8\"  \"9\"  \"10\" \"11\" \"12\"\n",
      "\n",
      "\n",
      "[1] \"Nrow of output before dropping missing explants is: 4727\"\n",
      "[1] \"Head of ID_exp columns from both output data table and IDs_to_drop data tbales, respectively:\"\n",
      "[1] \"__NA\"       \"GFPA1_1_1\"  \"GFPA1_1_1\"  \"GFPA1_1_1\"  \"GFPA1_1_1\" \n",
      "[6] \"GFPA1_1_10\"\n",
      "[1] \"GFPA1_1_1\" \"GFPA1_1_2\" \"GFPA1_2_2\" \"GFPA1_2_4\" \"GFPA1_2_7\" \"GFPA1_2_8\"\n",
      "[1] \"Nrow of output AFTER dropping missing explants is: 3596\"\n",
      "[1] \"Head of output\"\n",
      "   filename grid_item          V1 segment_hex intensity_threshold\n",
      "1:               <NA> observation                              NA\n",
      "   n_pixels_passing_threshold mean_signal max_signal total_signal total_pixels\n",
      "1:                         NA          NA         NA           NA           NA\n",
      "   present ID transgenic segment_present ID_exp\n",
      "1:       Y  _          0               0   __NA\n",
      "\n",
      "# A tibble: 6 x 5\n",
      "  filename                              grid_item present ID      ID_exp   \n",
      "  <chr>                                 <chr>     <chr>   <chr>   <chr>    \n",
      "1 GFPA1_I5.0_F1.9_L100_132422_0_0_0.jpg 1         C       GFPA1_1 GFPA1_1_1\n",
      "2 GFPA1_I5.0_F1.9_L100_132422_0_0_0.jpg 2         C       GFPA1_1 GFPA1_1_2\n",
      "3 GFPA1_I5.0_F1.9_L100_132505_1_0_1.jpg 2         C       GFPA1_2 GFPA1_2_2\n",
      "4 GFPA1_I5.0_F1.9_L100_132505_1_0_1.jpg 4         C       GFPA1_2 GFPA1_2_4\n",
      "5 GFPA1_I5.0_F1.9_L100_132505_1_0_1.jpg 7         C       GFPA1_2 GFPA1_2_7\n",
      "6 GFPA1_I5.0_F1.9_L100_132505_1_0_1.jpg 8         C       GFPA1_2 GFPA1_2_8\n",
      " [1] \"filename\"                   \"grid_item\"                 \n",
      " [3] \"V1\"                         \"segment_hex\"               \n",
      " [5] \"intensity_threshold\"        \"n_pixels_passing_threshold\"\n",
      " [7] \"mean_signal\"                \"max_signal\"                \n",
      " [9] \"total_signal\"               \"total_pixels\"              \n",
      "[11] \"present\"                    \"ID\"                        \n",
      "[13] \"transgenic\"                 \"segment_present\"           \n",
      "[15] \"ID_exp\"                    \n",
      "[1] \"Maximum # grid positions with transgenic stem in any plate: 2\"\n",
      "[1] \"Maximum # grid positions with transgenic callus in any plate: 7\"\n",
      "[1] \"Maximum # grid positions with transgenic shoot in any plate: 3\"\n",
      "[1] \"Minimum # grid positions with explants in any plate: 2\"\n",
      "[1] \"Maximum # grid positions with explants in any plate: 12\"\n",
      "[1] \"Maximum # grid positions with callus in any plate: 12\"\n",
      "[1] \"Maximum # grid positions with shoot in any plate: 5\"\n",
      "[1] \"Column of interest is portion_transgenic_Shoot\"\n",
      "[1] \"For transgenic tissue, What is the range of values for portion for Shoot\"\n",
      "[1] 0\n",
      "[1] 0.375\n",
      "[1] \"What are factor levels?\"\n",
      "[1] \"0\"                  \"0.0909090909090909\" \"0.181818181818182\" \n",
      "[4] \"0.272727272727273\"  \"0.375\"             \n",
      "\n",
      "[1] \"Column of interest is portion_Shoot\"\n",
      "[1] \"For total tissue, What is the range of values for portion for Shoot\"\n",
      "[1] 0\n",
      "[1] 0.5\n",
      "[1] \"What are factor levels?\"\n",
      " [1] \"0\"                  \"0.0909090909090909\" \"0.1\"               \n",
      " [4] \"0.111111111111111\"  \"0.142857142857143\"  \"0.181818181818182\" \n",
      " [7] \"0.222222222222222\"  \"0.25\"               \"0.272727272727273\" \n",
      "[10] \"0.454545454545455\"  \"0.5\"               \n",
      "\n",
      "[1] \"Column of interest is portion_transgenic_Callus\"\n",
      "[1] \"For transgenic tissue, What is the range of values for portion for Callus\"\n",
      "[1] 0\n",
      "[1] 0.6363636\n",
      "[1] \"What are factor levels?\"\n",
      " [1] \"0\"                  \"0.0833333333333333\" \"0.0909090909090909\"\n",
      " [4] \"0.1\"                \"0.111111111111111\"  \"0.166666666666667\" \n",
      " [7] \"0.181818181818182\"  \"0.2\"                \"0.272727272727273\" \n",
      "[10] \"0.3\"                \"0.333333333333333\"  \"0.375\"             \n",
      "[13] \"0.428571428571429\"  \"0.545454545454545\"  \"0.636363636363636\" \n",
      "\n",
      "[1] \"Column of interest is portion_Callus\"\n",
      "[1] \"For total tissue, What is the range of values for portion for Callus\"\n",
      "[1] 0\n",
      "[1] 1\n",
      "[1] \"What are factor levels?\"\n",
      " [1] \"0\"                  \"0.0833333333333333\" \"0.0909090909090909\"\n",
      " [4] \"0.1\"                \"0.111111111111111\"  \"0.125\"             \n",
      " [7] \"0.142857142857143\"  \"0.166666666666667\"  \"0.181818181818182\" \n",
      "[10] \"0.2\"                \"0.222222222222222\"  \"0.285714285714286\" \n",
      "[13] \"0.333333333333333\"  \"0.375\"              \"0.555555555555556\" \n",
      "[16] \"0.571428571428571\"  \"0.636363636363636\"  \"0.666666666666667\" \n",
      "[19] \"0.7\"                \"0.714285714285714\"  \"0.75\"              \n",
      "[22] \"0.777777777777778\"  \"0.8\"                \"0.818181818181818\" \n",
      "[25] \"0.833333333333333\"  \"0.857142857142857\"  \"0.875\"             \n",
      "[28] \"0.888888888888889\"  \"0.9\"                \"0.909090909090909\" \n",
      "[31] \"1\"                 \n",
      "\n",
      "[1] \"Column of interest is portion_transgenic_Stem\"\n",
      "[1] \"For transgenic tissue, What is the range of values for portion for Stem\"\n",
      "[1] 0\n",
      "[1] 0.25\n",
      "[1] \"What are factor levels?\"\n",
      "[1] \"0\"                  \"0.0909090909090909\" \"0.1\"               \n",
      "[4] \"0.111111111111111\"  \"0.125\"              \"0.142857142857143\" \n",
      "[7] \"0.181818181818182\"  \"0.222222222222222\"  \"0.25\"              \n",
      "\n",
      "[1] \"Column of interest is portion_Stem\"\n",
      "[1] \"For total tissue, What is the range of values for portion for Stem\"\n",
      "[1] 0\n",
      "[1] 1\n",
      "[1] \"What are factor levels?\"\n",
      " [1] \"0\"                 \"0.181818181818182\" \"0.25\"             \n",
      " [4] \"0.272727272727273\" \"0.3\"               \"0.333333333333333\"\n",
      " [7] \"0.428571428571429\" \"0.454545454545455\" \"0.545454545454545\"\n",
      "[10] \"0.555555555555556\" \"0.636363636363636\" \"0.666666666666667\"\n",
      "[13] \"0.7\"               \"0.714285714285714\" \"0.727272727272727\"\n",
      "[16] \"0.777777777777778\" \"0.8\"               \"0.818181818181818\"\n",
      "[19] \"0.833333333333333\" \"0.857142857142857\" \"0.875\"            \n",
      "[22] \"0.888888888888889\" \"0.9\"               \"0.909090909090909\"\n",
      "[25] \"1\"                \n",
      "\n",
      "[1] \"Column of interest is portion_transgenic_All_tissue\"\n",
      "[1] \"For transgenic tissue, What is the range of values for portion for All_tissue\"\n",
      "[1] 0\n",
      "[1] 0.6363636\n",
      "[1] \"What are factor levels?\"\n",
      " [1] \"0\"                  \"0.0833333333333333\" \"0.0909090909090909\"\n",
      " [4] \"0.1\"                \"0.111111111111111\"  \"0.125\"             \n",
      " [7] \"0.142857142857143\"  \"0.166666666666667\"  \"0.181818181818182\" \n",
      "[10] \"0.2\"                \"0.222222222222222\"  \"0.272727272727273\" \n",
      "[13] \"0.3\"                \"0.333333333333333\"  \"0.363636363636364\" \n",
      "[16] \"0.428571428571429\"  \"0.545454545454545\"  \"0.625\"             \n",
      "[19] \"0.636363636363636\" \n",
      "\n",
      "[1] \"Column of interest is portion_All_tissue\"\n",
      "[1] \"For total tissue, What is the range of values for portion for All_tissue\"\n",
      "[1] 0\n",
      "[1] 1\n",
      "[1] \"What are factor levels?\"\n",
      "[1] \"0\" \"1\"\n",
      "\n",
      "[1] \"Column of interest is portion_transgenic_All_regenerated_tissue\"\n",
      "[1] \"For transgenic tissue, What is the range of values for portion for All_regenerated_tissue\"\n",
      "[1] 0\n",
      "[1] 0.6363636\n",
      "[1] \"What are factor levels?\"\n",
      " [1] \"0\"                  \"0.0833333333333333\" \"0.0909090909090909\"\n",
      " [4] \"0.1\"                \"0.111111111111111\"  \"0.166666666666667\" \n",
      " [7] \"0.181818181818182\"  \"0.2\"                \"0.272727272727273\" \n",
      "[10] \"0.3\"                \"0.333333333333333\"  \"0.363636363636364\" \n",
      "[13] \"0.428571428571429\"  \"0.5\"                \"0.545454545454545\" \n",
      "[16] \"0.636363636363636\" \n",
      "\n",
      "[1] \"Column of interest is portion_All_regenerated_tissue\"\n",
      "[1] \"For total tissue, What is the range of values for portion for All_regenerated_tissue\"\n",
      "[1] 0\n",
      "[1] 1\n",
      "[1] \"What are factor levels?\"\n",
      " [1] \"0\"                  \"0.0833333333333333\" \"0.0909090909090909\"\n",
      " [4] \"0.1\"                \"0.111111111111111\"  \"0.125\"             \n",
      " [7] \"0.142857142857143\"  \"0.166666666666667\"  \"0.181818181818182\" \n",
      "[10] \"0.2\"                \"0.222222222222222\"  \"0.285714285714286\" \n",
      "[13] \"0.333333333333333\"  \"0.375\"              \"0.555555555555556\" \n",
      "[16] \"0.571428571428571\"  \"0.636363636363636\"  \"0.666666666666667\" \n",
      "[19] \"0.7\"                \"0.714285714285714\"  \"0.75\"              \n",
      "[22] \"0.777777777777778\"  \"0.8\"                \"0.818181818181818\" \n",
      "[25] \"0.833333333333333\"  \"0.857142857142857\"  \"0.875\"             \n",
      "[28] \"0.888888888888889\"  \"0.9\"                \"0.909090909090909\" \n",
      "[31] \"1\"                 \n",
      "\n",
      "[1] \"Column of interest is portion_transgenic_Background\"\n",
      "[1] \"For transgenic tissue, What is the range of values for portion for Background\"\n",
      "[1] 0\n",
      "[1] 0\n",
      "[1] \"What are factor levels?\"\n",
      "[1] \"0\"\n",
      "\n",
      "[1] \"Column of interest is portion_Background\"\n",
      "[1] \"For total tissue, What is the range of values for portion for Background\"\n",
      "[1] 0\n",
      "[1] 0\n",
      "[1] \"What are factor levels?\"\n",
      "[1] \"0\"\n",
      "\n",
      "[1] \"N rows of randomization datasheet: \"\n",
      "[1] 96\n",
      "[1] \"N rows of randomization datasheet: \"\n",
      "[1] 96\n",
      "\n",
      "[1] \"Plots will be saved to: \"\n",
      "[1] \"/home/labgroup/code/GMOlabeler/plots/Elements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent//\"\n",
      "\n",
      "\n",
      "[1] \"How many groups per treatment?\"\n",
      "\n",
      " AS HSPT   AS Mar CTR HSPT  CTR Mar \n",
      "      24       24       24       24 \n",
      "[1] \"How many rows in randomization datasheet?\"\n",
      "[1] 96\n",
      "[1] \"How many rows in randomization datasheet with portion_Callus data?\"\n",
      "[1] 96\n",
      "Warning message:\n",
      "`fun.y` is deprecated. Use `fun` instead. \n",
      "Warning message:\n",
      "`fun.y` is deprecated. Use `fun` instead. \n",
      "Warning message:\n",
      "`fun.y` is deprecated. Use `fun` instead. \n",
      "Warning message:\n",
      "`fun.y` is deprecated. Use `fun` instead. \n",
      "Warning message:\n",
      "`fun.y` is deprecated. Use `fun` instead. \n",
      "Warning message:\n",
      "`fun.y` is deprecated. Use `fun` instead. \n",
      "Warning message:\n",
      "`fun.y` is deprecated. Use `fun` instead. \n",
      "Warning message:\n",
      "`fun.y` is deprecated. Use `fun` instead. \n",
      "\n",
      "[1] \"How many groups per treatment?\"\n",
      "\n",
      " AS HSPT   AS Mar CTR HSPT  CTR Mar \n",
      "      24       24       24       24 \n",
      "[1] \"How many rows in randomization datasheet?\"\n",
      "[1] 96\n",
      "[1] \"How many rows in randomization datasheet with portion_Callus data?\"\n",
      "[1] 96\n",
      "Warning messages:\n",
      "1: Ignoring unknown parameters: outlier.shape \n",
      "2: `fun.y` is deprecated. Use `fun` instead. \n",
      "Warning messages:\n",
      "1: Ignoring unknown parameters: outlier.shape \n",
      "2: `fun.y` is deprecated. Use `fun` instead. \n",
      "Warning messages:\n",
      "1: Ignoring unknown parameters: outlier.shape \n",
      "2: `fun.y` is deprecated. Use `fun` instead. \n",
      "Warning messages:\n",
      "1: Ignoring unknown parameters: outlier.shape \n",
      "2: `fun.y` is deprecated. Use `fun` instead. \n",
      "Warning messages:\n",
      "1: Ignoring unknown parameters: outlier.shape \n",
      "2: `fun.y` is deprecated. Use `fun` instead. \n",
      "Warning messages:\n",
      "1: Ignoring unknown parameters: outlier.shape \n",
      "2: `fun.y` is deprecated. Use `fun` instead. \n",
      "Warning messages:\n",
      "1: Ignoring unknown parameters: outlier.shape \n",
      "2: `fun.y` is deprecated. Use `fun` instead. \n",
      "Warning messages:\n",
      "1: Ignoring unknown parameters: outlier.shape \n",
      "2: `fun.y` is deprecated. Use `fun` instead. \n",
      "\n",
      "[1] \"How many groups per treatment?\"\n",
      "\n",
      " AS HSPT   AS Mar CTR HSPT  CTR Mar \n",
      "      24       24       24       24 \n",
      "[1] \"How many rows in randomization datasheet?\"\n",
      "[1] 96\n",
      "[1] \"How many rows in randomization datasheet with n_Callus data?\"\n",
      "[1] 96\n",
      "[1] \"Head:\"\n",
      "        ID Image # Tray ID Rand # Date on Media *Plate # Index# in GH\n",
      "1  GFPA1_1       1   GFPA1      1    2020-09-17  3-2-1-2           R7\n",
      "2 GFPA1_10      10   GFPA1     10    2020-09-17  3-2-1-5           R7\n",
      "3 GFPA1_11      11   GFPA1     11    2020-09-17  1-2-1-1          140\n",
      "4 GFPA1_12      12   GFPA1     12    2020-09-17  4-2-1-3          R21\n",
      "5 GFPA1_13      13   GFPA1     13    2020-09-17  4-2-1-6          R21\n",
      "6 GFPA1_14      14   GFPA1     14    2020-09-17  3-1-2-6           R7\n",
      "  Genotype_ID Costruct Media Treatment name Replicate ...12     LableTop\n",
      "1   Nisqually      Mar   CTR        CTR Mar         2    NA   GFPA1_1_R1\n",
      "2   Nisqually      Mar   CTR        CTR Mar         5    NA GFPA1_10_R10\n",
      "3    BESC-134      Mar   CTR        CTR Mar         1    NA GFPA1_11_R11\n",
      "4   SKWB-24-2      Mar   CTR        CTR Mar         3    NA GFPA1_12_R12\n",
      "5   SKWB-24-2      Mar   CTR        CTR Mar         6    NA GFPA1_13_R13\n",
      "6   Nisqually     HSPT    AS        AS HSPT         6    NA GFPA1_14_R14\n",
      "             LableBot Plate#      Rand                           filename 1 2 3\n",
      "1 Nisqually_Mar_CTR_2     62 0.3583730  GFPA1_I5.0_F1.9_L100_132422_0_0_0 0 0 1\n",
      "2 Nisqually_Mar_CTR_5     65 0.8666917 GFPA1_I5.0_F1.9_L100_133406_11_1_2 1 1 0\n",
      "3  BESC-134_Mar_CTR_1     13 0.5296783  GFPA1_I5.0_F1.9_L100_134702_1_1_3 1 1 1\n",
      "4 SKWB-24-2_Mar_CTR_3     87 0.8703313  GFPA1_I5.0_F1.9_L100_133216_9_1_4 1 1 1\n",
      "5 SKWB-24-2_Mar_CTR_6     90 0.4970979  GFPA1_I5.0_F1.9_L100_133121_8_1_5 0 1 1\n",
      "6 Nisqually_HSPT_AS_6     60 0.6426024  GFPA1_I5.0_F1.9_L100_133036_7_1_6 0 0 1\n",
      "  4 5 6 7 8 9 10 11 12 total_explants n_transgenic_Shoot\n",
      "1 1 1 1 1 1 1  1  1  1             10                  0\n",
      "2 0 1 1 0 1 1  1  1  1              9                  0\n",
      "3 1 1 1 1 1 1  1  1  1             12                  0\n",
      "4 0 1 1 1 1 1  1  1  1             11                  0\n",
      "5 0 1 1 1 1 1  1  1  1             10                  0\n",
      "6 0 0 1 0 0 1  1  1  1              6                  0\n",
      "  portion_transgenic_Shoot n_transgenic_Callus portion_transgenic_Callus\n",
      "1                        0                   0                         0\n",
      "2                        0                   0                         0\n",
      "3                        0                   0                         0\n",
      "4                        0                   0                         0\n",
      "5                        0                   0                         0\n",
      "6                        0                   0                         0\n",
      "  n_transgenic_Stem portion_transgenic_Stem n_transgenic_All_tissue\n",
      "1                 0                     0.0                       0\n",
      "2                 0                     0.0                       0\n",
      "3                 0                     0.0                       0\n",
      "4                 0                     0.0                       0\n",
      "5                 1                     0.1                       1\n",
      "6                 0                     0.0                       0\n",
      "  portion_transgenic_All_tissue n_transgenic_All_regenerated_tissue\n",
      "1                           0.0                                   0\n",
      "2                           0.0                                   0\n",
      "3                           0.0                                   0\n",
      "4                           0.0                                   0\n",
      "5                           0.1                                   0\n",
      "6                           0.0                                   0\n",
      "  portion_transgenic_All_regenerated_tissue n_transgenic_Background\n",
      "1                                         0                       0\n",
      "2                                         0                       0\n",
      "3                                         0                       0\n",
      "4                                         0                       0\n",
      "5                                         0                       0\n",
      "6                                         0                       0\n",
      "  portion_transgenic_Background n_Shoot portion_Shoot n_Callus portion_Callus\n",
      "1                             0       0             0        0      0.0000000\n",
      "2                             0       0             0        0      0.0000000\n",
      "3                             0       0             0        4      0.3333333\n",
      "4                             0       0             0       11      1.0000000\n",
      "5                             0       0             0        9      0.9000000\n",
      "6                             0       0             0        0      0.0000000\n",
      "  n_Stem portion_Stem n_All_tissue portion_All_tissue n_All_regenerated_tissue\n",
      "1     10    1.0000000           10                  1                        0\n",
      "2      9    1.0000000            9                  1                        0\n",
      "3     12    1.0000000           12                  1                        4\n",
      "4     10    0.9090909           11                  1                       11\n",
      "5      8    0.8000000           10                  1                        9\n",
      "6      6    1.0000000            6                  1                        0\n",
      "  portion_All_regenerated_tissue n_Background portion_Background\n",
      "1                      0.0000000            0                  0\n",
      "2                      0.0000000            0                  0\n",
      "3                      0.3333333            0                  0\n",
      "4                      1.0000000            0                  0\n",
      "5                      0.9000000            0                  0\n",
      "6                      0.0000000            0                  0\n",
      "Warning message:\n",
      "`fun.y` is deprecated. Use `fun` instead. \n",
      "Warning message:\n",
      "`fun.y` is deprecated. Use `fun` instead. \n",
      "Warning message:\n",
      "`fun.y` is deprecated. Use `fun` instead. \n",
      "Warning message:\n",
      "`fun.y` is deprecated. Use `fun` instead. \n",
      "Warning message:\n",
      "`fun.y` is deprecated. Use `fun` instead. \n",
      "Warning message:\n",
      "`fun.y` is deprecated. Use `fun` instead. \n",
      "Warning message:\n",
      "`fun.y` is deprecated. Use `fun` instead. \n",
      "Warning message:\n",
      "`fun.y` is deprecated. Use `fun` instead. \n",
      "\n",
      "[1] \"List of treatments in this randomization_datasheet: \"\n",
      "[1] \"AS HSPT\"  \"AS Mar\"   \"CTR HSPT\" \"CTR Mar\" \n",
      "[1] \"Now about to merge randomization_datasheet and GMOlabeler output.\"\n",
      "[1] \"Rows in randomization_datasheet: 96\"\n",
      "[1] \"Rows in GMOlabeler output: 3596\"\n",
      "[1] \"Before merging, look at table of ID in both. \"\n",
      "\n",
      " GFPA1_1 GFPA1_10 GFPA1_11 GFPA1_12 GFPA1_13 GFPA1_14 GFPA1_15 GFPA1_16 \n",
      "       1        1        1        1        1        1        1        1 \n",
      "GFPA1_17 GFPA1_18 GFPA1_19  GFPA1_2 GFPA1_20 GFPA1_21  GFPA1_3  GFPA1_4 \n",
      "       1        1        1        1        1        1        1        1 \n",
      " GFPA1_5  GFPA1_6  GFPA1_7  GFPA1_8  GFPA1_9  GFPA2_1 GFPA2_10 GFPA2_11 \n",
      "       1        1        1        1        1        1        1        1 \n",
      "GFPA2_12 GFPA2_13 GFPA2_14 GFPA2_15 GFPA2_16 GFPA2_17 GFPA2_18 GFPA2_19 \n",
      "       1        1        1        1        1        1        1        1 \n",
      " GFPA2_2 GFPA2_20 GFPA2_21  GFPA2_3  GFPA2_4  GFPA2_5  GFPA2_6  GFPA2_7 \n",
      "       1        1        1        1        1        1        1        1 \n",
      " GFPA2_8  GFPA2_9  GFPA3_1 GFPA3_10 GFPA3_11 GFPA3_12 GFPA3_13 GFPA3_14 \n",
      "       1        1        1        1        1        1        1        1 \n",
      "GFPA3_15 GFPA3_16 GFPA3_17 GFPA3_18 GFPA3_19  GFPA3_2 GFPA3_20 GFPA3_21 \n",
      "       1        1        1        1        1        1        1        1 \n",
      " GFPA3_3  GFPA3_4  GFPA3_5  GFPA3_6  GFPA3_7  GFPA3_8  GFPA3_9  GFPA4_1 \n",
      "       1        1        1        1        1        1        1        1 \n",
      "GFPA4_10 GFPA4_11 GFPA4_12 GFPA4_13 GFPA4_14 GFPA4_15 GFPA4_16 GFPA4_17 \n",
      "       1        1        1        1        1        1        1        1 \n",
      "GFPA4_18 GFPA4_19  GFPA4_2 GFPA4_20 GFPA4_21  GFPA4_3  GFPA4_4  GFPA4_5 \n",
      "       1        1        1        1        1        1        1        1 \n",
      " GFPA4_6  GFPA4_7  GFPA4_8  GFPA4_9  GFPA5_1 GFPA5_10 GFPA5_11 GFPA5_12 \n",
      "       1        1        1        1        1        1        1        1 \n",
      " GFPA5_2  GFPA5_3  GFPA5_4  GFPA5_5  GFPA5_6  GFPA5_7  GFPA5_8  GFPA5_9 \n",
      "       1        1        1        1        1        1        1        1 \n",
      "\n",
      "       _  GFPA1_1 GFPA1_10 GFPA1_11 GFPA1_12 GFPA1_13 GFPA1_14 GFPA1_15 \n",
      "       1       40       36       52       55       49       24       40 \n",
      "GFPA1_16 GFPA1_17 GFPA1_18  GFPA1_2 GFPA1_20 GFPA1_21  GFPA1_3  GFPA1_4 \n",
      "      41       53       49       30       35       43       55       55 \n",
      " GFPA1_5  GFPA1_6  GFPA1_7  GFPA1_8  GFPA2_1 GFPA2_10 GFPA2_11 GFPA2_12 \n",
      "      29       44       50       48       44       48       33       33 \n",
      "GFPA2_13 GFPA2_14 GFPA2_15 GFPA2_16 GFPA2_17 GFPA2_18 GFPA2_19  GFPA2_2 \n",
      "      42       49       48       36       39       34       54       28 \n",
      "GFPA2_20 GFPA2_21  GFPA2_3  GFPA2_4  GFPA2_5  GFPA2_6  GFPA2_7  GFPA2_8 \n",
      "      38       40       30       38       51       24       47       40 \n",
      " GFPA2_9  GFPA3_1 GFPA3_10 GFPA3_11 GFPA3_12 GFPA3_13 GFPA3_14 GFPA3_16 \n",
      "      42       24       49       32       57       40       44       55 \n",
      "GFPA3_17 GFPA3_19  GFPA3_2 GFPA3_20 GFPA3_21  GFPA3_3  GFPA3_4  GFPA3_5 \n",
      "      45       45       34       51       54       40       53       55 \n",
      " GFPA3_6  GFPA3_7  GFPA3_8  GFPA3_9  GFPA4_1 GFPA4_10 GFPA4_11 GFPA4_12 \n",
      "      35       28       41       54       40       16        8       45 \n",
      "GFPA4_13 GFPA4_14 GFPA4_15 GFPA4_16 GFPA4_17 GFPA4_18  GFPA4_2 GFPA4_20 \n",
      "      49       55       32       41       40       46       29       55 \n",
      "GFPA4_21  GFPA4_3  GFPA4_4  GFPA4_5  GFPA4_6  GFPA4_8  GFPA4_9 GFPA5_10 \n",
      "      55       55       49       45       16       16       32       40 \n",
      "GFPA5_11 GFPA5_12  GFPA5_2  GFPA5_4  GFPA5_5  GFPA5_6  GFPA5_7  GFPA5_8 \n",
      "      38       37       13       60       54       44       25       28 \n",
      " GFPA5_9 \n",
      "      30 \n",
      "[1] \"Rows in merged output: 3596\"\n",
      "[1] \"List of treatments in this combined data: \"\n",
      "[1] \"AS HSPT\"  \"AS Mar\"   \"CTR HSPT\" \"CTR Mar\" \n",
      "\n",
      "[1] \"Making final plots for tissue: \"\n",
      "  Tissue hex_code color\n",
      "1  Shoot   00CC11 green\n",
      "[1] \"Dim before and after subsetting combined data to this tissue only\"\n",
      "[1] 3596   69\n",
      "[1] 793  69\n",
      "[1] \"Max mean_signal\"\n",
      "[1] 654.3476\n",
      "[1] \"Mean signal plots\"\n",
      "[1] \"Finished making mean signal plot... Now to save and make the rest.\"\n",
      "[1] \"Max signal plots\"\n",
      "[1] \"Total signal plots\"\n",
      "[1] \"Making final plots for tissue: \"\n",
      "  Tissue hex_code color\n",
      "2 Callus   0006CC  blue\n",
      "[1] \"Dim before and after subsetting combined data to this tissue only\"\n",
      "[1] 3596   69\n",
      "[1] 793  69\n",
      "[1] \"Max mean_signal\"\n",
      "[1] 496.792\n",
      "[1] \"Mean signal plots\"\n",
      "[1] \"Finished making mean signal plot... Now to save and make the rest.\"\n",
      "[1] \"Max signal plots\"\n",
      "[1] \"Total signal plots\"\n",
      "[1] \"Making final plots for tissue: \"\n",
      "  Tissue hex_code color\n",
      "3   Stem   CC0000   red\n",
      "[1] \"Dim before and after subsetting combined data to this tissue only\"\n",
      "[1] 3596   69\n",
      "[1] 793  69\n",
      "[1] \"Max mean_signal\"\n",
      "[1] 593.8481\n",
      "[1] \"Mean signal plots\"\n",
      "[1] \"Finished making mean signal plot... Now to save and make the rest.\"\n",
      "[1] \"Max signal plots\"\n",
      "[1] \"Total signal plots\"\n",
      "[1] \"Making final plots for tissue: \"\n",
      "      Tissue   hex_code color\n",
      "4 All_tissue All_tissue black\n",
      "[1] \"Dim before and after subsetting combined data to this tissue only\"\n",
      "[1] 3596   69\n",
      "[1] 793  69\n",
      "[1] \"Max mean_signal\"\n",
      "[1] -Inf\n",
      "[1] \"Making final plots for tissue: \"\n",
      "                  Tissue               hex_code color\n",
      "5 All_regenerated_tissue All_regenerated_tissue  <NA>\n",
      "[1] \"Dim before and after subsetting combined data to this tissue only\"\n",
      "[1] 3596   69\n",
      "[1] 423  69\n",
      "[1] \"Max mean_signal\"\n",
      "[1] -Inf\n",
      "There were 50 or more warnings (use warnings() to see the first 50)\n",
      "(gmolabeler) "
     ]
    }
   ],
   "source": [
    "conda activate gmolabeler\n",
    "cd $gmolabeler_path\n",
    "Rscript grid_item_plots.R \\\n",
    "-d \"${data_folder}/\" \\\n",
    "-r \"$randomization_datasheet\" \\\n",
    "-p $pixel_threshold \\\n",
    "-v categorical \\\n",
    "-m 1 \\\n",
    "-M $missing_explants \\\n",
    "-g $grid \\\n",
    "--sort 1 \\\n",
    "--height $height \\\n",
    "--width $width\n",
    "conda deactivate"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 5. Email plots to user"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### 5.1. ZIP results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/labgroup/code/GMOlabeler//plotsElements_7//GFP_versions/GFPA_cottonwoods/wk7/Fluorescent/\n"
     ]
    }
   ],
   "source": [
    "echo \"${gmolabeler_path}/plots${data_folder}\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [],
   "source": [
    "cd \"${gmolabeler_path}/plots/${data_folder}\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [],
   "source": [
    "rm -f ./plants_over_plates.csv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [],
   "source": [
    "cp \"${gmolabeler_path}/output/${data_folder}/plants_over_plates.csv\" ./"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [],
   "source": [
    "rm -f Rplots.pdf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [],
   "source": [
    "cd ../"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This messy substitution is explained here: https://superuser.com/questions/1068031/replace-backslash-with-forward-slash-in-a-variable-in-bash"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [],
   "source": [
    "data_folder_Compress=\"${data_folder////-}.zip\"\n",
    "data_folder_Compress=${data_folder_Compress#?};"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Fluorescent\n"
     ]
    }
   ],
   "source": [
    "ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "  adding: 2A1c_portion_Callus_total.png (deflated 7%)\n",
      "  adding: 2A1_portion_Callus_total.png (deflated 12%)\n",
      "  adding: 2A2c_portion_Callus_transgenic.png (deflated 11%)\n",
      "  adding: 2A2_portion_Callus_transgenic.png (deflated 12%)\n",
      "  adding: 2B1c_portion_Shoot_total.png (deflated 11%)\n",
      "  adding: 2B1_portion_Shoot_total.png (deflated 13%)\n",
      "  adding: 2B2c_portion_Shoot_transgenic.png (deflated 13%)\n",
      "  adding: 2B2_portion_Shoot_transgenic.png (deflated 14%)\n",
      "  adding: 2C1c_portion_All_regenerated_tissue_total.png (deflated 7%)\n",
      "  adding: 2C1_portion_All_regenerated_tissue_total.png (deflated 12%)\n",
      "  adding: 2C2c_portion_All_regenerated_tissue_transgenic.png (deflated 11%)\n",
      "  adding: 2C2_portion_All_regenerated_tissue_transgenic.png (deflated 12%)\n",
      "  adding: 2D1c_portion_All_tissue_total.png (deflated 10%)\n",
      "  adding: 2D1_portion_All_tissue_total.png (deflated 15%)\n",
      "  adding: 2D2c_portion_All_tissue_transgenic.png (deflated 9%)\n",
      "  adding: 2D2_portion_All_tissue_transgenic.png (deflated 12%)\n",
      "  adding: 3A1_Callus_total.png (deflated 13%)\n",
      "  adding: 3A2_Callus_transgenic.png (deflated 14%)\n",
      "  adding: 3B1_Shoot_total.png (deflated 14%)\n",
      "  adding: 3B2_Shoot_transgenic.png (deflated 16%)\n",
      "  adding: 3D1_All_regenerated_tissue_total.png (deflated 13%)\n",
      "  adding: 3D1_All_tissue_total.png (deflated 12%)\n",
      "  adding: 3D2_All_regenerated_tissue_transgenic.png (deflated 13%)\n",
      "  adding: 3D2_All_tissue_transgenic.png (deflated 13%)\n",
      "  adding: A1_Shoot_Total_signal.png (deflated 17%)\n",
      "  adding: A2_Callus_Total_signal.png (deflated 17%)\n",
      "  adding: A3_Stem_Total_signal.png (deflated 31%)\n",
      "  adding: B1_Shoot_Max_signal.png (deflated 15%)\n",
      "  adding: B2_Callus_Max_signal.png (deflated 16%)\n",
      "  adding: B3_Stem_Max_signal.png (deflated 33%)\n",
      "  adding: C1_Shoot_Mean_signal.png (deflated 17%)\n",
      "  adding: C2_Callus_Mean_signal.png (deflated 31%)\n",
      "  adding: C3_Stem_Mean_signal.png (deflated 35%)\n",
      "  adding: gridplot_args.rds (stored 0%)\n",
      "  adding: plants_over_plates.csv (deflated 79%)\n"
     ]
    }
   ],
   "source": [
    "cd $timepoint\n",
    "zip -r $data_folder_Compress ./*"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### 5.2. Write email"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [],
   "source": [
    "duration=$(( SECONDS - start ))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "https://unix.stackexchange.com/questions/53841/how-to-use-a-timer-in-bash"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [],
   "source": [
    "rm -f \"${gmolabeler_path}/email_to_send.txt\"\n",
    "cp \"${gmolabeler_path}/email_to_send_template.txt\" \"${gmolabeler_path}/email_to_send.txt\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [],
   "source": [
    "echo \"\" >> \"${gmolabeler_path}/email_to_send.txt\"\n",
    "echo \"Number of samples run: \" >> \"${gmolabeler_path}/email_to_send.txt\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [],
   "source": [
    "cat \"${data}/test.csv\" | wc -l >> \"${gmolabeler_path}/email_to_send.txt\"\n",
    "echo \"\" >> \"${gmolabeler_path}/email_to_send.txt\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [],
   "source": [
    "if (( $SECONDS > 3600 )) ; then\n",
    "    let \"hours=SECONDS/3600\"\n",
    "    let \"minutes=(SECONDS%3600)/60\"\n",
    "    let \"seconds=(SECONDS%3600)%60\"\n",
    "    echo \"Completed in $hours hour(s), $minutes minute(s) and $seconds second(s)\" >> \"${gmolabeler_path}/email_to_send.txt\"\n",
    "elif (( $SECONDS > 60 )) ; then\n",
    "    let \"minutes=(SECONDS%3600)/60\"\n",
    "    let \"seconds=(SECONDS%3600)%60\"\n",
    "    echo \"Completed in $minutes minute(s) and $seconds second(s)\" >> \"${gmolabeler_path}/email_to_send.txt\"\n",
    "else\n",
    "    echo \"Completed in $SECONDS seconds\" >> \"${gmolabeler_path}/email_to_send.txt\"\n",
    "fi"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {},
   "outputs": [],
   "source": [
    "echo \"\" >> \"${gmolabeler_path}/email_to_send.txt\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### 5.3. Send email with results to user"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/labgroup/code/GMOlabeler/plots/Elements_7/GFP_versions/GFPA_cottonwoods/wk7/Fluorescent\n"
     ]
    }
   ],
   "source": [
    "pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {},
   "outputs": [],
   "source": [
    "mail -a $data_folder_Compress -s $data_folder $email < \"${gmolabeler_path}/email_to_send.txt\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3.8M\tlements_7--GFP_versions-GFPA_cottonwoods-wk7-Fluorescent-.zip\n"
     ]
    }
   ],
   "source": [
    "du -sh $data_folder_Compress"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 90,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Your job is complete!\n",
      "\n",
      "Number of samples run: \n",
      "89\n",
      "\n",
      "Completed in 113 hour(s), 48 minute(s) and 29 second(s)\n",
      "\n"
     ]
    }
   ],
   "source": [
    "cat \"${gmolabeler_path}/email_to_send.txt\""
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "##### Time analysis ends"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Mon Nov 30 18:12:18 PST 2020\n"
     ]
    }
   ],
   "source": [
    "echo $(date)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
